{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 389,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "from NBT import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 377,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gene SYMBOL 2 ENSG\n",
    "Symbol2ENSG, Uniprot2ENSG, ENSP2ENSG = LoadIDMapping()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### SCZ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sczgenenames</th>\n",
       "      <th>ensembl_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KMT2E</td>\n",
       "      <td>ENSG00000005483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MVP</td>\n",
       "      <td>ENSG00000013364</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sczgenenames    ensembl_names\n",
       "0        KMT2E  ENSG00000005483\n",
       "1          MVP  ENSG00000013364"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "SCZ_high_conf_gene = pd.read_csv(\"../dat/network/psychencode/INT-17_SCZ_High_Confidence_Gene_List.csv\")\n",
    "display(SCZ_high_conf_gene.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>STOP</th>\n",
       "      <th>NSNPS</th>\n",
       "      <th>NPARAM</th>\n",
       "      <th>N</th>\n",
       "      <th>ZSTAT</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>1</td>\n",
       "      <td>854993</td>\n",
       "      <td>884961</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "      <td>150065</td>\n",
       "      <td>1.14730</td>\n",
       "      <td>0.125630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>1</td>\n",
       "      <td>874583</td>\n",
       "      <td>899679</td>\n",
       "      <td>40</td>\n",
       "      <td>3</td>\n",
       "      <td>150065</td>\n",
       "      <td>1.55460</td>\n",
       "      <td>0.060021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KLHL17</td>\n",
       "      <td>1</td>\n",
       "      <td>890967</td>\n",
       "      <td>906099</td>\n",
       "      <td>26</td>\n",
       "      <td>3</td>\n",
       "      <td>150065</td>\n",
       "      <td>1.33840</td>\n",
       "      <td>0.090389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>1</td>\n",
       "      <td>896872</td>\n",
       "      <td>915488</td>\n",
       "      <td>29</td>\n",
       "      <td>7</td>\n",
       "      <td>150065</td>\n",
       "      <td>0.43019</td>\n",
       "      <td>0.333530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PERM1</td>\n",
       "      <td>1</td>\n",
       "      <td>905579</td>\n",
       "      <td>922473</td>\n",
       "      <td>40</td>\n",
       "      <td>7</td>\n",
       "      <td>150065</td>\n",
       "      <td>-0.45048</td>\n",
       "      <td>0.673820</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      GENE CHR   START    STOP  NSNPS  NPARAM       N    ZSTAT         P\n",
       "0   SAMD11   1  854993  884961     30       6  150065  1.14730  0.125630\n",
       "1    NOC2L   1  874583  899679     40       3  150065  1.55460  0.060021\n",
       "2   KLHL17   1  890967  906099     26       3  150065  1.33840  0.090389\n",
       "3  PLEKHN1   1  896872  915488     29       7  150065  0.43019  0.333530\n",
       "4    PERM1   1  905579  922473     40       7  150065 -0.45048  0.673820"
      ]
     },
     "execution_count": 383,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GeneTest = pd.read_csv(\"/Users/jiayao/Work/NB_proposal/dat/GWAS/GeneTest/scz.genetest.tsv\", delimiter=\"\\t\")\n",
    "GeneTest.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEWCAYAAABrDZDcAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3Xd4VHXaxvHvAyQhoYTeS+gdBEO1sYorqCsiuuraVlfRfd36ulLsXSxbfN21oGt311WCigXsqGsHS0LonVATAiGkJ/N7/5hBQwwtzMyZcn+uy2vOnDmZ8xxJ5p7fKc8x5xwiIhK/6nldgIiIeEtBICIS5xQEIiJxTkEgIhLnFAQiInFOQSAiEucUBCIhZGZPmdkdXtchciAKAokbZvZLM8sys2Iz22pmD5lZao1l+pvZXDMrMLNCM3vfzEaFqT5nZj3DsS6R6hQEEhfM7BrgHuBaIBUYBaQBb5tZQmCZHsAnQBbQDegAvAK8Y2YjPChbJCwUBBLzzKwpcCvwW+fcfOdchXNuHfBz/B/4vwgsegvwmXPueudcvnOu0Dn3f8Bz+EOktvcea2Y5ZnadmeWZ2Tozu+AAtVxhZqvMLD8w8ugQmP9RYJHvzGyPmZ0bjG0XORQKAokHY4CGwJzqM51ze4B5wE8Ds04GXqrl518EjjOzhvt5/3ZAK6AjcAkwy8z61FzIzE4E7sYfQO2B9cALgVqODyw2xDnX2Dn3n0PeOpEjpCCQeNAKyHPOVdby2hagdbXltuxnmfpAiwOs40bnXJlz7kPgDfwf9jVdADzhnPvaOVcGzABGm1naIW2FSIgoCCQe5AGtzKxBLa+1B3KrLdd+P8u4wOu12emcK6r2fD3+4ws1dQi8Bnw/ItmBfyQh4hkFgcSDz4Ay4KzqM82sETAB+DAw613gnFp+/ufA58658v28f/PAe+3VBdhcy3Kbga411t8S2HQI2yASMgoCiXnOuQL8B4sfNLPxZpYQ2B3zEv5v+c8HFr0VGGNmd5pZCzNrYma/BS4Fbj7Iam41s0QzOw44ndqPNfwLuNTMjjKzJOAu4IvAgWuAbUD3Om+oSB3VNlQWiTnOuXvNbAdwP9ATSMI/Ehi3d7eOc26lmR0LzATWAY2AQmCSc+6dA7z9VmAn/m/8xcBVzrlltdTwnpndCGQAzYFPgfOqLXIL8LSZJQNTnHMv1n2LRQ6d6cY0Eo/M7DL8I4BjnHMb9rNMJ+Bz4Gbn3D/3s8xY4DnnXKdQ1SoSahoRSFxyzj1hZhX4Ty2tNQicczlmNgGYaGaNAwd3RWKOgkDilnPu2UNYJgv/lcYiMUu7hkRE4pzOGhIRiXNRsWuoVatWLi0tzesyRESiyqJFi/Kcc60PtlxUBEFaWhoLFy70ugwRkahiZusPvpR2DYmIxD0FgYhInFMQiIjEOQWBiEicUxCIiMQ5BYGISJxTEIiIxDkFgYhIBNq0q4RbX8umssoX8nVFxQVlIiLxwudzPP/FembOW4bPwaShHRncqVlI16kgEBGJEGty9zA9I4sv1+VzXK9W3DVpEJ1bpIR8vQoCERGPVVb5eOzjtfz13RU0bFCP+84ezNlHd8LMwrJ+BYGIiIeyNxcwLSOTxZt2M35AO247cwBtmjQMaw0KAhERD5RWVPHg+yt55MM1NE9J5OELhjFhUHtPalEQiIiE2aL1+Uydncnq3CImD+vEjaf3o1lKomf1KAhERMKkqKyS+95aztOfraNDajJPXzaCE3of9HYBIacgEBEJg49W5DJjThabC0q4ZHQa157Sh0ZJkfERHBlViIjEqF3F5dzxxlJmL8qhe+tGvHTlaNLTWnhd1j4UBCIiITIvaws3vprNzuJyrv5JD357Yi8aJtT3uqwfURCIiATZ9sJSbn41m3mLtzKgQ1Oevmw4Azqkel3WfikIRESCxDnH7EU53PHGUkoqqpg6vg9XHNedhPqR3dZNQSAiEgQb84u57uUsPl6Zx/C05sycPJgerRt7XdYhURCIiBwBn8/xzGfruPet5Rhw+8QBXDCyK/Xqhac9RDAoCERE6mjV9kKmZWSxaP1OTujdmjsnDaRT89A3iQu2kAWBmT0BnA5sd84NDMxrAfwHSAPWAT93zu0MVQ0iIqFQUeVj1kdreODdlaQk1ecvPx/CpKEdw9YkLthCeQTjKWB8jXnTgfecc72A9wLPRUSixuJNBUz8+yfc99ZyTu7flnf+eAJnDQtfp9BQCNmIwDn3kZml1Zg9ERgbmH4aWABMC1UNse6bDTu56PHP2FPuDvln6gF7f10NSEyApAYNqKzykZRQn6QG9WjcsAEl5VWkJieQ0KAeSQ3q0yE1mc27SsCgV5smNG1Y+69O0+QEhndrwZdrd2AYw7u1YMW2Qnq3bcKKbYUc18t/Of3HK3N/NN2hWTIAm3eV/GhedQd7XSQUSiuqeOC9lcz6aA0tGiXyyIVHM35gO6/LCopwHyNo65zbAuCc22Jmbfa3oJlNAaYAdOnSJUzlRZf75i87rBAAqHnTu8oKKK6oBKCwPPBqQZn/cWcpAPUMEuoZFVUOzP+NKDmx9l+d1OQEVmwv5NsNuzAzVmwvZGtBGe1S89m6932B177b8qPpc4f7/50/Xpn7o3nVHex1kWD7cm0+0zMyWZNXxLnpnbnu1H6kpiR4XVbQROzBYufcLGAWQHp6+uF92sWJrbtLD/tnwjUi6NWm8QFHBMBBp6vPq+5gr4sEy56ySu6Zt4xnP19Pp+bJPPerkRzbq5XXZQWdORe6z9jArqHXqx0sXg6MDYwG2gMLnHN9DvY+6enpbuHChSGrM1rd9EoWc77ZxFlDO3LbmYO8LkckpnywfDvXz8liy+5SLh3TjT+d0puU/YyEI5WZLXLOpR9suXBv1VzgEmBm4PHVMK8/pkwa1oniiiomDevkdSkiMWNnUTm3v76EOd9somebxsy+agxHd23udVkhFcrTR/+N/8BwKzPLAW7GHwAvmtmvgA3AOaFafzz4am0+320soHebfIZ2ie1fVJFQc87xZtZWbp67mF3FFfzuxJ5cfWJPkhpEXpO4YAvlWUPn7+elk0K1zniTs7OErQUl5Ows8boUkai2fXcpN7yymLeXbGNQx1Se/dVI+rVv6nVZYRNdO7xkH3lFpVT4fOQVHf5BYxHxjwJeWpjD7W8sobzSx4wJffnVsd1oEOFN4oJNQRDFJh3ViW27y5h0lI4RiByuDTuKmfFyJp+s2sGIbi24Z/JgurVq5HVZnlAQRLG1eUXsLqlkbV6R16WIRI0qn+OpT9dx/1vLqV/PuOPMgfxiRJeoahIXbAqCKLZiWyEb8otYsa3Q61JEosLKbYVMzcjkmw27+Emf1tw5aZCuTkdBELU27yrhs9V5lFc6lisIRA6ovNLHIx+u5u/vr6JRUn3+du5RTDyqQ1T3BwomBUGUevTD1eQVl9GqUQK/O7GX1+WIRKzMnF1MnZ3Jsq2F/GxIB27+WX9aNU7yuqyIoiCIUguWbaOsAlIa1+fkAbHR+EokmErKq/jbuyt47OM1tG6SxGMXp3Ny/7ZelxWRFARRandp5T6PIvKDz9fsYHpGJut2FHP+iM7MOLUfTRvGTpO4YFMQRKm9HUP3PooIFJZWMHPeMp7/YgNdWqTwr8tHMqZn7DWJCzYFQZTq2qIRK7YX0bVFfJ73LFLT+8u2cf3Li9m2u5TLj+3GNT/tQ3Ji7LeHCAYFQRTavKuElo0SaZRYwuju+rYj8S2/qJzbXsvmlW8307ttYx66YIx6bx0mBUEUevTD1SzasJPkhAZ0aN7Q63JEPOGc47XMLdwyN5vC0gr+MK4X/zO2J4kN4qs9RDAoCKLQgmXbKK+CxonwsyEdvS5HJOy2FpRywytZvLt0O0M6N+PeyYPp066J12VFLQVBFMovKgegosqnqyIlrjjneOGrjdz1xlIqfD5uOK0flx7Tjfpx3B4iGBQEUWjvvYW/v8ewSBxYv6OI6RlZfLZmB6O7t2Tm5EF0bamTJYJBQRCFurVKZm1eCd1aaTQgsa/K53jyk7Xc//ZyEurV4+6zBnHe8M5qDxFECoIodNXxPXlowSquOr6n16WIhNTyrf4mcd9t3MW4fm2448xBtEvVCRLBpiCIMvfNW8rDH64hob7x1fp8zh3RxeuSRIKuvNLHPz5YxUMLVtG0YQIPnj+U0we31yggRBQEUWbWx2vwAWVVjgtGdvW6HJGg+3bjLqbO/o4V2/Zw5lEduOlnA2jRKNHrsmKagiDKNKhnVPgcyQ1MF81ITCkpr+LPby/niU/W0rZpQ574ZTon9lWTuHBQEESZ1k2S2LCzlNZN1EZXYsenq/OYnpHFhvxiLhjZhekT+tJETeLCRkEQZQpLq/Z5FIlmu0sruPvNpfz7y42ktUzhhSmjGNW9pddlxR0FQRTZvKuE5ERjZwl0bKYzJyS6vbNkGze8kkVuYRlXHt+dP4zrrSZxHlEQRJHr52SyuaCchHpwQp/WXpcjUid5e8q4ZW42r2duoW+7Jjx2cTqDOzXzuqy4piCIIp+sygPA54MLRqV5W4zIYXLO8eq3m7n1tWyKyqq45uTeXHlCDzWJiwAKgiiSnFCf8rIqGifVV48hiSqbd5VwwyuLeX/ZdoZ28TeJ69VWTeIihYIgiqSmJFBQVkVqis6mkOjg8zn+9eUGZs5bRpXPcdPp/blkTJqaxEUYT4LAzP4IXA44IAu41DlX6kUt0WRAh1Q27SplQIdUr0sROai1eUVMy8jky7X5HNOzJXdPGkyXlilelyW1CHsQmFlH4HdAf+dciZm9CJwHPBXuWqLN9sJSfM7/KBKpKqt8PP7ftfz1nRUkNqjHvZMHc056J7WHiGBe7RpqACSbWQWQAmz2qI6o8c2GnSzeXIDDf1MOkUi0ZPNupmVkkrWpgJ/2b8vtZw6kbVOd6hzpwh4EzrlNZnY/sAEoAd52zr1dczkzmwJMAejSRY3Vpmd8R1mlfzopQedaS2Qpq6zi7++v4uEFq2mWksA/fjGMUwe10yggSnixa6g5MBHoBuwCXjKzC51zz1Vfzjk3C5gFkJ6e7sJdZ6RZs73o++kZE/p5WInIvhat38m0jExWbd/DWcM6cuNp/WmuJnFRxYtdQ+OAtc65XAAzmwOMAZ474E/FuYYJ9ago99EksR4nD2jndTkiFJdXct9by3nq03W0b9qQJy8dzk/6tPG6LKkDL4JgAzDKzFLw7xo6CVjoQR1Rpdzn2+dRxEv/XZnH9DmZ5Ows4eLRXZk6vi+Nk3Q2erTy4hjBF2Y2G/gaqAS+IbALSGr3zYad3x8fqKz0thaJbwXFFdz55hJeXJhD91aNePHK0Yzo1sLrsuQIeRLhzrmbgZu9WHc0um/+su+ne7TVzbrFG/MXb+XGVxeTX1TOr8f24Pcn9aKhTlyICRrLRYGcncUAJCcY90we4nE1Em9yC/1N4t7I2kL/9k158pfDGdhRFzXGEgVBFGjV2H8zmn7tmuquZBI2zjnmfL2J215fQkl5Fdee0ocpx3cnob6axMUaBUEUSAp0Z0xSl0YJk5ydxVz/8mI+XJHL0V2bc8/kwfRs09jrsiREFARR4Kt1O/d5FAkVn8/x3BfruWfeMhxw6xkDuGhUV+qpSVxMUxBEuM27SqgMXE5XGfeX1Ukorc7dw/SMTL5at5PjerXirkmD6NxCTeLigYIgwv32+UXfTzdP1j+XBF9FlY/HPl7D395dSXJCfe4/ZwiTh3VUe4g4ok+WCPf1xoLvp+89W2cMSXAt3lTAtIxMsjfvZsLAdtw6cQBtmqhJXLxREES4tJbJrN1RQreWyWotIUFTWlHFg++v5JEP19A8JZGHLxjGhEHtvS5LPKIgiHDllb59HkWO1MJ1+UzNyGRNbhFnH92JG07rR7MUNYmLZwqCCJdXVLbPo0hd7Smr5L75y3jm8/V0SE3mmctGcHzv1l6XJRFAQRDhKtVjSILgwxW5XDcni80FJVwyOo1rT+lDIzWJkwD9JkSw/3y5garAdPNGumG9HL5dxeXc/vpSMr7OoUfrRrx05WjS09QkTvalIIhgt72e/f30qTqQJ4dpXtYWbnw1m53F5fzmJz35zYk91SROaqUgiGBF5T8cIL5qbE8PK5Fosn13KTe9ms387K0M6NCUpy8bzoAOahIn+6cgiBIdmiV7XYJEOOccLy3K4Y7Xl1Ba6WPa+L5ccVw3GqhJnByEgkAkBmzML+a6l7P4eGUeI9JaMHPyILq3VpM4OTQKggj162d+uHtnor7QyX5U+RzPfLaO+95ajgG3TxzABSPVJE4Oj4IgQs1bsu376ZP6tfWwEolUq7YXMi0ji0Xrd3JC79bcddYgOmoXotSBgiAK3PizAV6XIBGkosrHox+u5v/eW0VKUn3+8vMhTBqqJnFSdwqCKKADxbJXVk4BUzMyWbplN6cNbs8tPxtA6yZJXpclUU5BEIG+2aAb0Mi+Siuq+Nu7K3ns4zW0bJTIoxcdzSlqQihBoiCIQPfNX/b9dMtG+ieKd1+s2cH0OVmszSvi3PTOXHdaP1KTdaW5BI8+ZSLQ6tw9ACTVNx6/ZITH1YhXCksruHf+cp79fD2dWyTz/OUjOaZnK6/LkhikIIhA+XvKAf/9Y4d2ae5xNeKFD5Zv5/o5WWzZXcplx3TjT6f0JiVRf64SGvrNikAVbt9HiR87i8q5/fUlzPlmE73aNCbj12MYpi8DEmIKggiUWA/KfbqQLJ4453gjaws3v5pNQUkFvzuxJ1ef2JOkBmoSJ6F3wCAws2TgAqB/YNZCYLZzrjzUhcWzCt++jxLbtu0u5YZXFvPOkm0M7pTKc5ePpF/7pl6XJXFkv985zWwQsBQ4DlgHrAdOAT4xs2ZmdkddVxr4+dlmtszMlprZ6Lq+VyxyNR4lNjnn+M9XGxj3lw/5aEUu153alzm/HqMQkLA70Ijg/4ArnHPvVJ9pZuOAxUB2rT91aB4A5jvnzjazRCDlCN4rpryTvZXUpPoUlFUxob9aS8SqDTuKmT4nk09X72BktxbcM3kwaa0aeV2WxKkDBUH7miEA4Jx718wqgEl1WaGZNQWOB34ZeL9yQLuaAv700rcUlFWR2rA+D1+c7nU5EmRVPsdTn67j/reWU7+eceekgZw/vIuaxImnDhQE9cwsyTm3z13TzawhUOGcK67jOrsDucCTZjYEWAT83jlXVGM9U4ApAF26dKnjqqJPQWnVPo8SO1ZsK2Tq7Ey+3biLE/u24c5JA2mfqvYh4r0DnZfyDJBhZml7ZwSmXwSePYJ1NgCGAQ8754YCRcD0mgs552Y559Kdc+mtW7c+gtWJeKu80scD767ktP/7mPU7injgvKP45yXpCgGJGPsdETjn7jCz3wAfmdneffhFwP3OuQePYJ05QI5z7ovA89nUEgTx6L55S70uQYLsu427mJaRybKthZwxpAM3/6w/LRurSZxElgOePuqc+zvwdzNrEnheeKQrdM5tNbONZtbHObccOAlYcqTvGwv+8eGa76d1gUd0Kymv4q/vruDxj9fQpklDHr84nXE6+C8R6pA+b4IRADX8Fng+cMbQGuDSIL9/1LvzrEFelyB19NnqHcyYk8m6HcWcP6ILM07tS9OGahInkcuTL57OuW8BnRJTTc3W0+eOiJ8D5LFid2kFM+ct419fbKBryxT+dcVIxvRQkziJfNoDESGufn6R1yXIEXhv6Tauf3kx2wtLueK4bvzvyX1ITlR7CIkOBw0CMzsH/8VfhWZ2A/4zfu5wzn0d8uriyOaCH87SvXikRgPRYseeMm59bQlzv9tMn7ZNeOSiozmqczOvyxI5LIcyIrjROfeSmR2Lv8XE/cDDwMiQVhbHbpuk4wORzjnH3O82c+trSygsreCP43rz67E9SGygToESfQ4lCPZe2XQa/nP/XzWzW0JXUvzRaaPRZUtBCTe8vJj3lm1nSOdm3Dt5MH3aNfG6LJE6O5Qg2GRmjwLjgHvMLIkDX4gmh+mhaqeNSuTy+RwvfLWRu99cSoXPxw2n9ePSY7pRX+0hJModShD8HBiP/0KyXWbWHrg2tGXFl+pdRtVoLjKtyyti+pxMPl+Tz+juLZk5eRBdW6pJnMSGgwaBc67YzNYBE8xsPPCJc+7tkFcWp9RoLrJUVvl44pO1/PntFSTWr8fMswZx7vDOmGkUILHjUM4augk4B5gTmPWkmb3knKvz/QjkB+9kb/W6BNmPZVt3M212Jt/lFDCuX1vuOHMg7VIbel2WSNAdyq6h84GhzrlSADObCXwNKAiC4H9f/NbrEqSGssoq/vHBah76YBWpyQk8eP5QTh/cXqMAiVmHEgTrgIZAaeB5ErA6VAXFm8KyH9pN6/iA977ZsJNpGZms2LaHSUM7cuPp/WnRKNHrskRCar9BYGYP4j+OWQZkm9k7gecnA/8NT3nxRccHvFNcXsmf317BE5+spV3Thjzxy3RO7KtglvhwoBHBwsDjIuDlavMXhKyaOPPoglVelyDAp6vymD4niw35xVw4qgvTxveliZrESRw50P0Inobv70jWE/9oYPXeYwVy5O59a7nXJcS1gpIK7n5zKS98tZG0lim8MGUUo7q39LoskbA70K6hBsBdwGXAevwXkXUysyeB651zFeEpMXZVVbuAQP2Fwuvt7K3c8Mpi8vaUceUJ3fnjuN40TFCTOIlPB9o1dB/QBOi2934EgRvP3x/47/ehLy9+qL9QeOTtKeOWudm8nrmFvu2a8Pgl6QzupCZxEt8OFASnA72dc99/b3XO7TazXwPLUBBIFHHO8cq3m7j1tSUUl1Vxzcm9uWpsDxLqq1uKyIGCwFUPgWozq8zsR/Pl8Pz0zwu8LiFubN5VwvUvZ/HB8lyGdvE3ievVVk3iRPY6UBAsMbOLnXPPVJ9pZhfiHxHIEViRW/T9dONEXagUCj6f4/kvNzDzzaX4HNx0en8uGZOmJnEiNRwoCK4G5pjZZfhPIXXAcCAZmBSG2uLGs5eP9rqEmLMmdw/TM7L4cl0+x/Zsxd1nDaJzixSvyxKJSAc6fXQTMNLMTgQGAAbMc869F67iYlXN/kJDuzT3qJLYU1nl4/H/ruWv76wgqUE97j17MOcc3UntIUQO4FC6j74PvB+GWuLGlGd1f+JQWLJ5N1MzvmPxpt2cMqAtt08cSJumahIncjC6eb0Hqh9pb99EfWyOVFllFX9/fxUPL1hNs5QEHrpgGBMGttMoQOQQKQg89tn1J3tdQlRbtD6faRlZrNq+h7OGdeTG0/rTXE3iRA6LgiDMdNpocBSVVXLfW8t5+rN1dEhN5qlLhzO2TxuvyxKJSgqCMKt+2qjUzccrc5kxJ4ucnSVcMror147vS+Mk/SqL1JX+ejx09QndvS4hqhQUV3DHG0t4aVEO3Vs34qWrRjM8rYXXZYlEPQVBGJ33yKf7PL92Qj+PKok+8xdv5cZXF5NfVM7/jO3B707qpSZxIkHiWRCYWX389zzY5Jw73as6wunzdTu9LiHqbC8s5Za52byZtZX+7Zvy5C+HM7BjqtdlicQUL0cEvweWAk09rMEzvVs38rqEiOacI+PrTdz++hJKKqq49pQ+TDm+u5rEiYSAJ0FgZp2A04A7gf/1ogavvX3NWK9LiFg5O4u57uXFfLQil6O7NueeyYPp2aax12WJxCyvRgR/A6biv99BrcxsCjAFoEuX6L9py1E3z/e6hIjn8zme/Xw998z39zS89YwBXDSqK/XUJE4kpMIeBGZ2OrDdObfIzMbubznn3CxgFkB6enrUt73eVVbldQkRbXXuHqbNzmTh+p0c37s1d00aSKfmahInEg5ejAiOAc4ws1OBhkBTM3vOOXehB7V44rGLjva6hIhRUeVj1kdreOC9lSQn1Of+c4YweVhHtYcQCaOwB4FzbgYwAyAwIvhTrIdAzauJTx7QzptCIsziTQVMnZ3Jki27OXVQO245YwBtmqhJnEi46TqCMNDVxPsqrajigfdWMuujNTRPSeSRC4cxfmB7r8sSiVueBoFzbgGwwMsawq1hnF8D9dW6fKbNzmRNXhHnHN2JG07rT2pKgtdlicQ1jQhC7Cf37nsrh2V3nuZRJd7aU1bJvfOX8cxn6+nUPJlnfzWC43q19rosEUFBEHJr80u8LsFzH67I5bo5WWwuKOGXY9K49pQ+NFKTOJGIob/GMEqKs4tidxaVc/sbS5jz9SZ6tG7E7KtGc3RXNYkTiTQKghDqPeONfZ4vvys+dgs555i3eCs3vbqYXcUV/OYnPfnNiT3VJE4kQikIQqg86i+DO3zbd5dy46uLeSt7GwM7NuXpy0YwoIOaxIlEMgVBmIzt1crrEkLKOcdLi3K44/UllFX6mD6hL5cf240GahInEvEUBCFy3N3v7vP8qV+N9KiS0NuYX8yMOVn8d1UeI9JaMHPyILq3VpM4kWihIAiRjQVlXpcQclU+xzOfrePe+cupZ3D7mQO5YEQXNYkTiTIKgjDo1iLZ6xKCbuW2QqZlZPL1hl2M7dOaOycNomOz2NtOkXigIAiBEbe/vc/zD6ae6FElwVdR5eORBat58P1VpCTV56/nDuHMo9QkTiSaKQhCYHtRhdclhERWTgHXzv6OZVsLOW1we249YwCtGid5XZaIHCEFQYjFQhed0ooq/vruCh77aA2tGifx6EVHc4o6qIrEDAVBkKVN3/cispUzo/sisi/W7GD6nCzW5hVx3vDOzDi1H6nJsRBvIrKXgkBqVVhawT3zl/Hc5xvo3CKZ5y8fyTE9Y/taCJF4pSAIoVFpzb0uoU4+WLad617OYuvuUn51bDeu+WlvUhL1qyISq/TXHUR9rtt3t9ALV43xqJK6yS8q57bXsnnl2830atOYjF+PYViX6AwzETl0CoIgKvN5XUHdOOd4PXMLt8zNpqCkgt+d1Iurf9KDpAZqEicSDxQEIZISJf9nt+0u5fqXF/Pu0m0M7pTK81eMpG+7pl6XJSJhFCUfV5Gv5tlCS+6I7LOFnHP856uN3PnmUsorfVx/aj8uPSZNTeJE4pCCIA6t31HEjDlZfLp6ByO7teCeyYNJa9XI67JExCMKgiCo2VKicUJktluo8jme/GQt97+9nAb16nHXpEGcN7zoT2tVAAAMDUlEQVSzmsSJxDkFQRDUbCmx+PZTPapk/5ZvLWRqRibfbdzFiX3bcOekgbRPVZM4EVEQxLzySh8PLVjFPz5YRZOGCTxw3lGcMaSDmsSJyPcUBEeo5kHidRHUUuK7jbuYOjuT5dsKmXhUB246vT8t1SRORGpQEMSgkvIq/vLOcv7537W0adKQxy9OZ1z/tl6XJSIRSkFwBGqOBiYf1cGjSn7w6eo8pmdksSG/mF+M7ML0CX1p2lBN4kRk/xQEQfTn84Z6tu7dpRXc/eYy/v3lBrq2TOFfV4xkTA81iRORgwt7EJhZZ+AZoB3gA2Y55x4Idx1H6poXvvG6hO+9u2Qb17+SRW5hGVOO784fx/UmOVHtIUTk0HgxIqgErnHOfW1mTYBFZvaOc26JB7XUWca3m/d57sVB4h17yrj1tSXM/W4zfds1YdZF6Qzp3CzsdYhIdAt7EDjntgBbAtOFZrYU6AhEVRB4yTnH3O82c8vcbPaUVfLHcb359dgeJDZQewgROXyeHiMwszRgKPBFLa9NAaYAdOnSJax1HYyXp4xuKSjhhpcX896y7RzVuRn3nj2Y3m2bhG39IhJ7PAsCM2sMZAB/cM7trvm6c24WMAsgPT3dhbm8iOPzOf791QbufnMZlT4fN5zWj0uP6UZ9tYcQkSPkSRCYWQL+EHjeOTfHixrqquZooE2j0J+auTaviOkZmXyxNp8xPVoy86zBdGmZEvL1ikh88OKsIQP+CSx1zv0l3OsPti9v/GnI3ruyyscTn6zlz2+vILFBPe6ZPIifp3dWewgRCSovRgTHABcBWWb2bWDedc65Nz2o5bDUHA2E0tItu5mWkUlmTgEn92/LHWcOpG3ThmFbv4jEDy/OGvovEBNfaUNxkLissop/vL+KhxasJjU5gb//YiinDWqvUYCIhIyuLD5E4RgNfL1hJ9NmZ7Jy+x4mDe3ITaf3p3mjxJCvV0Tim4KgjoI5Gigur+T+t1bw5Kdrade0IU/+cjg/6dsmaO8vInIgCoJDEMrRwCer8pg+J5ON+SVcOKoL08b3pYmaxIlIGCkI6iAYo4GCkgruemMp/1m4kW6tGvGfKaMY2b1lEKoTETk8CoKDCMVo4O3srdzwymJ2FJVz1Qk9+MO4XjRMUJM4EfGGguAwHcloILewjFtey+aNzC30a9+Uf14ynEGdUoNYnYjI4VMQHECwRgPOOV7+ZhO3vb6E4rIq/vTT3lx5Qg8S6qtJnIh4T0GwH7WFQF1GA5t2lXD9y1ksWJ7LsC7+JnE926hJnIhEDgXBITrc/1E+n+P5L9Yzc94yfA5u/ll/Lh6dpiZxIhJxFAS1qG00sOowRgNrcvcwPSOLL9flc1yvVtw1aRCdW6hJnIhEJgXBITjUXUKVVT4e+3gtf313BQ0b1OO+swdz9tGd1B5CRCKagqCGuh4gzt5cwLSMTBZv2s0pA9py+8SBtFGTOBGJAgqCaupygLi0oooH31/JIx+uoXlKIg9fMIwJg9qHqkQRkaBTEByBRevzmTo7k9W5RUwe1okbT+9HsxQ1iROR6KIgCDic0UBRWSX3vbWcpz9bR4fUZJ6+bAQn9G4d4gpFREJDQQCMvvOdH83bXwh8tCKXGXOy2FxQwsWjunLt+L40TtL/RhGJXvoEA7YUlh90mV3F5dzxxlJmL8qhe+tGvHjlaIantQhDdSIioRX3QXAou4TmZW3hxlez2Vlczv+M7cHvTlKTOBGJHXEdBAcLge2Fpdz8ajbzFm+lf/umPHXpcAZ2VJM4EYktcR0E++OcY/aiHO54YyklFVVce0ofphzfXU3iRCQmxW0Q7G80sDG/mOtezuLjlXmkd23OzMmD6dmmsQcVioiER1wGQW0hsOauU3nqk7Xc+9ZyDLht4gAuHNmVemoSJyIxLu6CYH8tJM559DMWrd/J8b1bc9ekgXRqriZxIhIf4ioI9hcCifXrsWr7Hv58zhDOGtZRTeJEJK7EVRDsz7j+bbj1jIG0bpLkdSkiImEXN0FQ22igdZMkbp84gPED1SROROJXXARBbSHw8/ROXH9qf1JTEjyoSEQkcnhyYryZjTez5Wa2ysymh3Jd+zsucO/ZQxQCIiJ4EARmVh/4BzAB6A+cb2b9Q7Gu/YVAXW5CLyISq7wYEYwAVjnn1jjnyoEXgInhWrlCQERkX14EQUdgY7XnOYF5+zCzKWa20MwW5ubmBmXFCgERkR/zIghqO0nf/WiGc7Occ+nOufTWrY/8pi8KARGR2nlx1lAO0Lna807A5lCsSB/+IiIH58WI4Cugl5l1M7NE4Dxgrgd1iIgIHowInHOVZvYb4C2gPvCEcy473HWIiIifJxeUOefeBN70Yt0iIrIv3WlFRCTOKQhEROKcgkBEJM4pCERE4pw596NruSKOmeUC6+v4462AvCCWE0liedsgtrdP2xa9omn7ujrnDnpFblQEwZEws4XOuXSv6wiFWN42iO3t07ZFr1jcPu0aEhGJcwoCEZE4Fw9BMMvrAkIolrcNYnv7tG3RK+a2L+aPEYiIyIHFw4hAREQOQEEgIhLnYjoIzGy8mS03s1VmNt3reoLFzDqb2QdmttTMss3s917XFGxmVt/MvjGz172uJdjMrJmZzTazZYF/w9Fe1xQsZvbHwO/kYjP7t5k19LqmI2FmT5jZdjNbXG1eCzN7x8xWBh6be1ljMMRsEJhZfeAfwASgP3C+mfX3tqqgqQSucc71A0YBV8fQtu31e2Cp10WEyAPAfOdcX2AIMbKdZtYR+B2Q7pwbiL/N/HneVnXEngLG15g3HXjPOdcLeC/wPKrFbBAAI4BVzrk1zrly4AVgosc1BYVzbotz7uvAdCH+D5If3fc5WplZJ+A04HGvawk2M2sKHA/8E8A5V+6c2+VtVUHVAEg2swZACiG6+2C4OOc+AvJrzJ4IPB2Yfho4M6xFhUAsB0FHYGO15znE0IflXmaWBgwFvvC2kqD6GzAV8HldSAh0B3KBJwO7vh43s0ZeFxUMzrlNwP3ABmALUOCce9vbqkKirXNuC/i/lAFtPK7niMVyEFgt82LqXFkzawxkAH9wzu32up5gMLPTge3OuUVe1xIiDYBhwMPOuaFAETGwawEgsK98ItAN6AA0MrMLva1KDkUsB0EO0Lna805E+TC1OjNLwB8Czzvn5nhdTxAdA5xhZuvw78470cye87akoMoBcpxze0dws/EHQywYB6x1zuU65yqAOcAYj2sKhW1m1h4g8Ljd43qOWCwHwVdALzPrZmaJ+A9azfW4pqAwM8O/j3mpc+4vXtcTTM65Gc65Ts65NPz/Zu8752LmW6Vzbiuw0cz6BGadBCzxsKRg2gCMMrOUwO/oScTIgfAa5gKXBKYvAV71sJag8OSexeHgnKs0s98Ab+E/e+EJ51y2x2UFyzHARUCWmX0bmHdd4F7QEvl+Czwf+IKyBrjU43qCwjn3hZnNBr7Gf2bbN0R5OwYz+zcwFmhlZjnAzcBM4EUz+xX+8DvHuwqDQy0mRETiXCzvGhIRkUOgIBARiXMKAhGROKcgEBGJcwoCEZE4pyAQCTCzKjP7ttp/Qbni18ymBDqNLjOzhWY2NhjvKxIsMXsdgUgdlDjnjgrmGwZaZlwJHOucyzOzYcBcMxsZ6M0j4jmNCEQOwMxSA/e06BN4/m8zuyIwvcfM/mxmX5vZe2bWupa3mAZc65zLAwh0jX0SuDpc2yByMAoCkR8k19g1dK5zrgD4DfCUmZ0HNHfOPRZYvhHwtXNuGPAh/qtOaxoA1GygtxD/PTJEIoJ2DYn8oNZdQ865d8zsHPw3OhpS7SUf8J/A9HP4m6wdito644p4RiMCkYMws3pAP6AEaHGARWvr17IEOLrGvGH4RwUiEUFBIHJwf8TfRfN84IlAC3Dw//2cHZj+BfDfWn72XuAeM2sJYGZHAZOAR0Naschh0K4hkR8kV+vmCjAfeAK4HBjhnCs0s4+AG/AfDygCBpjZIqAAOLfmGzrn5ppZB+CTwO0b2wFDnHO5Id4WkUOm7qMidWRme5xzjQ9j+Qb4zxiqB1zo9McnEUJBIFJHhxsEIpFKQSAiEud0sFhEJM4pCERE4pyCQEQkzikIRETinIJARCTO/T+QVT37cHFZowAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYAAAAD8CAYAAAB+UHOxAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAEtxJREFUeJzt3X+MXeV95/H3pzgkTdrUJgyItZ01Va20abUBdgR0kardODUGopg/ikTUFi9rybsSbZNVpcZ0/7AWGsnRrpoE7RYJgbumpaEsTYQV2JCpk6jaPyCYH0sCDmuXUDxriqe1IW1RE5F+9495nFycsefe8Xiu7ef9kq7OOd/znHueR5j7mfPr3lQVkqT+/Ni4OyBJGg8DQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktSpZePuwImcf/75tWbNmnF3Q5LOKE8++eTfVNXEfO1O6wBYs2YNe/bsGXc3JOmMkuSvhmnnKSBJ6pQBIEmdmjcAkrwvyTMDr+8k+XiS85JMJdnXpita+yS5I8n+JM8muWzgvTa19vuSbDqVA5Mkndi8AVBVL1TVJVV1CfAvgTeALwBbgd1VtRbY3ZYBrgHWttcW4E6AJOcB24ArgMuBbUdDQ5K09EY9BbQO+Muq+itgI7Cz1XcC17f5jcC9NesxYHmSi4CrgamqOlxVR4ApYMNJj0CStCCjBsCNwOfa/IVV9QpAm17Q6iuBAwPbTLfa8eqSpDEYOgCSnAt8BPif8zWdo1YnqB+7ny1J9iTZMzMzM2z3JEkjGuUI4Brgqap6tS2/2k7t0KaHWn0aWD2w3Srg4Anqb1FVd1XVZFVNTkzM+xyDJGmBRgmAj/LD0z8Au4Cjd/JsAh4aqN/U7ga6Eni9nSJ6FFifZEW7+Lu+1SRJYzDUk8BJ3gn8MvDvB8rbgQeSbAZeBm5o9UeAa4H9zN4xdDNAVR1OcjvwRGt3W1UdPukR6AfWbH14bPt+aft1Y9u3pIUZKgCq6g3gPcfU/pbZu4KObVvALcd5nx3AjtG7KUlabD4JLEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktSpoQIgyfIkDyb5VpK9SX4xyXlJppLsa9MVrW2S3JFkf5Jnk1w28D6bWvt9STadqkFJkuY37BHAZ4EvVdXPAh8A9gJbgd1VtRbY3ZYBrgHWttcW4E6AJOcB24ArgMuBbUdDQ5K09OYNgCTvBn4JuAegqr5XVa8BG4GdrdlO4Po2vxG4t2Y9BixPchFwNTBVVYer6ggwBWxY1NFIkoY2zBHATwMzwB8meTrJ3UneBVxYVa8AtOkFrf1K4MDA9tOtdry6JGkMhgmAZcBlwJ1VdSnwD/zwdM9cMketTlB/68bJliR7kuyZmZkZonuSpIUYJgCmgemqerwtP8hsILzaTu3QpocG2q8e2H4VcPAE9beoqruqarKqJicmJkYZiyRpBPMGQFX9NXAgyftaaR3wPLALOHonzybgoTa/C7ip3Q10JfB6O0X0KLA+yYp28Xd9q0mSxmDZkO1+E7gvybnAi8DNzIbHA0k2Ay8DN7S2jwDXAvuBN1pbqupwktuBJ1q726rq8KKMQpI0sqECoKqeASbnWLVujrYF3HKc99kB7Bilg5KkU8MngSWpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6NVQAJHkpyTeSPJNkT6udl2Qqyb42XdHqSXJHkv1Jnk1y2cD7bGrt9yXZdGqGJEkaxihHAP+mqi6pqsm2vBXYXVVrgd1tGeAaYG17bQHuhNnAALYBVwCXA9uOhoYkaemdzCmgjcDONr8TuH6gfm/NegxYnuQi4GpgqqoOV9URYArYcBL7lySdhGEDoIAvJ3kyyZZWu7CqXgFo0wtafSVwYGDb6VY7Xl2SNAbLhmx3VVUdTHIBMJXkWydomzlqdYL6WzeeDZgtAO9973uH7J4kaVRDHQFU1cE2PQR8gdlz+K+2Uzu06aHWfBpYPbD5KuDgCerH7uuuqpqsqsmJiYnRRiNJGtq8AZDkXUl+8ug8sB74JrALOHonzybgoTa/C7ip3Q10JfB6O0X0KLA+yYp28Xd9q0mSxmCYU0AXAl9IcrT9n1TVl5I8ATyQZDPwMnBDa/8IcC2wH3gDuBmgqg4nuR14orW7raoOL9pINFZrtj48lv2+tP26sexXOhvMGwBV9SLwgTnqfwusm6NewC3Hea8dwI7RuylJWmw+CSxJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUqaEDIMk5SZ5O8sW2fHGSx5PsS/KnSc5t9be35f1t/ZqB97i11V9IcvViD0aSNLxRjgA+BuwdWP4U8OmqWgscATa3+mbgSFX9DPDp1o4k7wduBH4e2AD8QZJzTq77kqSFGioAkqwCrgPubssBPgg82JrsBK5v8xvbMm39utZ+I3B/VX23qr4N7AcuX4xBSJJGN+wRwGeA3wH+qS2/B3itqt5sy9PAyja/EjgA0Na/3tr/oD7HNpKkJTZvACT5MHCoqp4cLM/RtOZZd6JtBve3JcmeJHtmZmbm654kaYGGOQK4CvhIkpeA+5k99fMZYHmSZa3NKuBgm58GVgO09T8FHB6sz7HND1TVXVU1WVWTExMTIw9IkjSceQOgqm6tqlVVtYbZi7hfqapfBb4K/Eprtgl4qM3vasu09V+pqmr1G9tdQhcDa4GvL9pIJEkjWTZ/k+P6BHB/kt8DngbuafV7gD9Ksp/Zv/xvBKiq55I8ADwPvAncUlXfP4n9S5JOwkgBUFVfA77W5l9kjrt4quofgRuOs/0ngU+O2klJ0uLzSWBJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTqZ7wLScazZ+vC4uyBJ8/IIQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKn5g2AJO9I8vUk/yfJc0n+c6tfnOTxJPuS/GmSc1v97W15f1u/ZuC9bm31F5JcfaoGJUma3zBHAN8FPlhVHwAuATYkuRL4FPDpqloLHAE2t/abgSNV9TPAp1s7krwfuBH4eWAD8AdJzlnMwUiShjdvANSsv2+Lb2uvAj4IPNjqO4Hr2/zGtkxbvy5JWv3+qvpuVX0b2A9cviijkCSNbKhrAEnOSfIMcAiYAv4SeK2q3mxNpoGVbX4lcACgrX8deM9gfY5tJElLbKgAqKrvV9UlwCpm/2r/ubmatWmOs+549bdIsiXJniR7ZmZmhumeJGkBRroLqKpeA74GXAksT3L09wRWAQfb/DSwGqCt/yng8GB9jm0G93FXVU1W1eTExMQo3ZMkjWCYu4Amkixv8z8OfAjYC3wV+JXWbBPwUJvf1ZZp679SVdXqN7a7hC4G1gJfX6yBSJJGM8wvgl0E7Gx37PwY8EBVfTHJ88D9SX4PeBq4p7W/B/ijJPuZ/cv/RoCqei7JA8DzwJvALVX1/cUdjiRpWPMGQFU9C1w6R/1F5riLp6r+EbjhOO/1SeCTo3dTkrTYfBJYkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmdMgAkqVPzBkCS1Um+mmRvkueSfKzVz0sylWRfm65o9SS5I8n+JM8muWzgvTa19vuSbDp1w5IkzWfZEG3eBH67qp5K8pPAk0mmgH8L7K6q7Um2AluBTwDXAGvb6wrgTuCKJOcB24BJoNr77KqqI4s9KPVjzdaHx7bvl7ZfN7Z9S4th3iOAqnqlqp5q838H7AVWAhuBna3ZTuD6Nr8RuLdmPQYsT3IRcDUwVVWH24f+FLBhUUcjSRraSNcAkqwBLgUeBy6sqldgNiSAC1qzlcCBgc2mW+14dUnSGAwdAEl+Avgz4ONV9Z0TNZ2jVieoH7ufLUn2JNkzMzMzbPckSSMaKgCSvI3ZD//7qurzrfxqO7VDmx5q9Wlg9cDmq4CDJ6i/RVXdVVWTVTU5MTExylgkSSMY5i6gAPcAe6vq9wdW7QKO3smzCXhooH5TuxvoSuD1doroUWB9khXtjqH1rSZJGoNh7gK6Cvh14BtJnmm13wW2Aw8k2Qy8DNzQ1j0CXAvsB94AbgaoqsNJbgeeaO1uq6rDizIKSdLI5g2AqvrfzH3+HmDdHO0LuOU477UD2DFKByVJp4ZPAktSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmdMgAkqVMGgCR1at4ASLIjyaEk3xyonZdkKsm+Nl3R6klyR5L9SZ5NctnANpta+31JNp2a4UiShjXMEcD/ADYcU9sK7K6qtcDutgxwDbC2vbYAd8JsYADbgCuAy4FtR0NDkjQe8wZAVf0FcPiY8kZgZ5vfCVw/UL+3Zj0GLE9yEXA1MFVVh6vqCDDFj4aKJGkJLfQawIVV9QpAm17Q6iuBAwPtplvteHVJ0pgs9kXgzFGrE9R/9A2SLUn2JNkzMzOzqJ2TJP3QQgPg1XZqhzY91OrTwOqBdquAgyeo/4iququqJqtqcmJiYoHdkyTNZ6EBsAs4eifPJuChgfpN7W6gK4HX2ymiR4H1SVa0i7/rW02SNCbL5muQ5HPAvwbOTzLN7N0824EHkmwGXgZuaM0fAa4F9gNvADcDVNXhJLcDT7R2t1XVsReWJUlLaN4AqKqPHmfVujnaFnDLcd5nB7BjpN5Jkk6ZeQNA0tzWbH14LPt9aft1Y9mvzj5+FYQkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmd8vcApDPMuH6HAPwtgrPNWR0A4/wfRZJOd54CkqROGQCS1CkDQJI6teTXAJJsAD4LnAPcXVXbl7oPkhZmXNfVvPh8aixpACQ5B/jvwC8D08ATSXZV1fNL2Q9JZxaD59RY6lNAlwP7q+rFqvoecD+wcYn7IEli6U8BrQQODCxPA1cscR8kaShn+zMXSx0AmaNWb2mQbAG2tMW/T/LCKe/V8M4H/mbcnVgEjuP04jhOL6fFOPKpk9r8nw/TaKkDYBpYPbC8Cjg42KCq7gLuWspODSvJnqqaHHc/TpbjOL04jtPL2TKOYSz1NYAngLVJLk5yLnAjsGuJ+yBJYomPAKrqzSS/ATzK7G2gO6rquaXsgyRp1pI/B1BVjwCPLPV+F8lpeWpqARzH6cVxnF7OlnHMK1U1fytJ0lnHr4KQpE4ZACNK8l+SfCvJs0m+kGT5uPs0rCQbkryQZH+SrePuz0IkWZ3kq0n2JnkuycfG3aeTkeScJE8n+eK4+3IykixP8mD7f2Nvkl8cd59GleQ/tn9T30zyuSTvGHefTjUDYHRTwC9U1b8A/i9w65j7M5SBr+G4Bng/8NEk7x9vrxbkTeC3q+rngCuBW87QcRz1MWDvuDuxCD4LfKmqfhb4AGfYmJKsBH4LmKyqX2D2JpUbx9urU88AGFFVfbmq3myLjzH7LMOZ4Kz4Go6qeqWqnmrzf8fsB83K8fZqYZKsAq4D7h53X05GkncDvwTcA1BV36uq18bbqwVZBvx4kmXAOznmGaWzkQFwcv4d8L/G3YkhzfU1HGfkB+dRSdYAlwKPj7cnC/YZ4HeAfxp3R07STwMzwB+201l3J3nXuDs1iqr6f8B/BV4GXgFer6ovj7dXp54BMIckf97OAx772jjQ5j8xezrivvH1dCTzfg3HmSTJTwB/Bny8qr4z7v6MKsmHgUNV9eS4+7IIlgGXAXdW1aXAPwBn1DWmJCuYPSK+GPhnwLuS/Np4e3XqndW/CbxQVfWhE61Psgn4MLCuzpz7aOf9Go4zRZK3Mfvhf19VfX7c/Vmgq4CPJLkWeAfw7iR/XFVn4ofONDBdVUePxB7kDAsA4EPAt6tqBiDJ54F/BfzxWHt1inkEMKL2gzafAD5SVW+Muz8jOCu+hiNJmD3XvLeqfn/c/Vmoqrq1qlZV1Rpm/1t85Qz98Keq/ho4kOR9rbQOONN+4+Nl4Mok72z/xtZxhl3IXgiPAEb334C3A1Oz/054rKr+w3i7NL+z6Gs4rgJ+HfhGkmda7XfbE+Yan98E7mt/XLwI3Dzm/oykqh5P8iDwFLOndp+mgyeCfRJYkjrlKSBJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSp/4/YSNgWRZZsPgAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#volcano(GeneTest[\"P\"].values, GeneTest[\"ZSTAT\"])\n",
    "QQplot(GeneTest[\"P\"].values)\n",
    "plt.hist(GeneTest[\"ZSTAT\"].values)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "109\n"
     ]
    }
   ],
   "source": [
    "Core_Gene = GeneTest[GeneTest[\"ZSTAT\"]>7][\"GENE\"].values\n",
    "print(len(Core_Gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "GeneGTF, GeneGTF2 = ReadGeneGTF(\"../dat/gencode.v19.gene.gtf\", withCHR=True)\n",
    "GeneDict = SNP2Gene(GeneGTF, \"../dat/GWAS/PGC/ckqny.scz2snpres.INFO0.9.P0.05.bed.gz\",\n",
    "                   CHROM=\"hg19chrc\", POS=\"bp\", EFFECT=\"or\", SE=\"se\", P=\"p\", )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {},
   "outputs": [],
   "source": [
    "#RGN = ig.read(\"../dat/network/saved/psychencode.rgn2.gml\")\n",
    "#RGN = LoadPsychencodeRGN(Symbol2ENSG)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### PrePPI + SCZ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/jiayao/anaconda3/lib/python3.7/site-packages/igraph/__init__.py:2280: RuntimeWarning: A non-numeric, non-string, non-boolean edge attribute was ignored at foreign.c:2648\n",
      "  return writer(f, *args, **kwds)\n"
     ]
    }
   ],
   "source": [
    "PrePPI, unmappedUniprot = LoadNetworkPrePPI(Uniprot2ENSG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "metadata": {},
   "outputs": [],
   "source": [
    "subPrePPI = PrePPI.subgraph_edges(PrePPI.es.select(weight_ge=1e4))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAY0AAAD8CAYAAACLrvgBAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAGRtJREFUeJzt3X2UVdWd5vHvI2AAXwbU0lEKAsmUDgZaghXEcekQHbV8WUJmxRkY0zJp1iIv2tM9GWxhMom0rVlm8uK06QQXRgacsUFH40A65Quhk7EnS9RCEVCjlMaWKwwgGPNC0KC/+ePuMofiVtXm3gu3gOez1l333N/Z+5x9QH08e59bpYjAzMwsx1GNHoCZmR06HBpmZpbNoWFmZtkcGmZmls2hYWZm2RwaZmaWzaFhZmbZHBpmZpbNoWFmZtkGNnoA9XbSSSfF6NGjGz0MM7NDypo1a96MiKa+2h12oTF69Gg6OjoaPQwzs0OKpH/MaefpKTMzy+bQMDOzbA4NMzPLdtitaZjZoe/3v/89pVKJ3bt3N3ooh53BgwfT3NzMoEGDqurv0DCzfqdUKnHccccxevRoJDV6OIeNiGDHjh2USiXGjBlT1TE8PWVm/c7u3bs58cQTHRh1JokTTzyxpjs4h4aZ9UsOjAOj1j9Xh4aZmWXzmoaZ9Xvz5/f3483n2GOPZc6cOb22GzBgAOPHj2fPnj2MHTuWJUuWMHTo0OzzTJkyhS1btjB48GCOPfZYFi1axBlnnNFr/Zvf/Catra21XuIHHBoF9f4H6VA5t5kdHEOGDGHt2rUAXHPNNdx555186Utf+mB/RBARHHVUz5NA9957L62trSxcuJAbbriBFStW9Fqvtz6npyQtkrRN0oZC7T5Ja9PrNUlrU320pN8V9t1Z6HO2pPWSOiXdoTSxJukESSslbUzvw1NdqV2npHWSJtb/8s3M9vXb3/6WK664grPOOotx48Zx3333AeUfU/Tmm28C0NHRwZQpUz7o89xzz3HhhRfS0tLCXXfd1ec5zj//fDo7O3nttdcYO3YsX/ziF5k4cSKbNm3iscce49xzz2XixIlcffXV/OY3v9mn/wUXXEBnZ2d2vV5y1jQWA23FQkT824iYEBETgAeBHxR2v9K1LyI+X6gvAGYDLenVdcy5wKqIaAFWpc8AlxXazk79zcwOuEceeYTTTjuN5557jg0bNtDW1tZnn3Xr1vGjH/2IJ554gptvvpnNmzf32HbPnj08/PDDjB8/HoCXXnqJa6+9lmeffZZjjjmGW265hR//+Mc888wztLa28u1vf3ufY/zwhz/8oH9OvV76nJ6KiMclja60L90t/Bvgwt6OIelU4PiIeCJ9vgeYBjwMTAWmpKZLgJ8CN6b6PRERwGpJwySdGhFb+rwqM7MajB8/njlz5nDjjTdy5ZVXcv755/fZZ+rUqQwZMoQhQ4bwyU9+kqeeeopp06bt1eZ3v/sdEyZMAMp3GrNmzWLz5s18+MMfZvLkyQCsXr2aF154gfPOOw+Ad999l3PPPfeDY1xzzTUMGTKE0aNH853vfKfPer3VuqZxPrA1IjYWamMkPQv8CvgvEfEPwAigVGhTSjWAU7qCICK2SDo51UcAmyr02Sc0JM2mfDfCqFGjarwkMzvSnX766axZs4b29nbmzZvHJZdcwle/+lUGDhzI+++/D7DPdx26P8pa6dHW4ppG0THHHPPBdkRw8cUXs3Tp0opj61q7yK3XW62P3M4Aile2BRgVER8HvgT8raTjgUoPBkcfx87uExELI6I1Ilqbmvr8cfBmZr3avHkzQ4cO5TOf+Qxz5szhmWeeAcprGmvWrAHgwQcf3KvP8uXL2b17Nzt27OCnP/0pn/jEJ6o69+TJk/nZz372wbrErl27ePnll2u4mvqq+k5D0kDgXwNnd9Ui4h3gnbS9RtIrwOmU7xKaC92bga4Jv61d005pGmtbqpeAkT30MbMjyMF+unD9+vXccMMNHHXUUQwaNIgFC8pLqjfddBOzZs3ia1/7Guecc85efSZNmsQVV1zB66+/zle+8hVOO+20qs7d1NTE4sWLmTFjBu+88w4At9xyC6effnptF1UnKi8Z9NGovKbxdxExrlBrA+ZFxL8s1JqAnRHxnqSPAP8AjI+InZKeBv4UeBJoB74TEe2SvgHsiIjbJM0FToiIv5B0BXA9cDlwDnBHREzqa6ytra1R7S9h8iO3Zv3Diy++yNixYxs9jMNWpT9fSWsios/5rZxHbpcCTwBnSCpJmpV2TWfvqSmAC4B1kp4DHgA+HxE7074vAN8HOoFXKC+CA9wGXCxpI3Bx+gzlYHk1tb8L+GJfYzUzswMr5+mpGT3U/32F2oOUH8Gt1L4DGFehvgO4qEI9gOv6Gp+ZmR08/tlTZtYv5Uyd2/6r9c/VoWFm/c7gwYPZsWOHg6POun6fxuDBg6s+hn/2lJn1O83NzZRKJbZv397ooRx2un5zX7UcGmbW7wwaNKjq3yxnB5anp8zMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsW5+hIWmRpG2SNhRq8yW9IWltel1e2DdPUqeklyRdWqi3pVqnpLmF+hhJT0raKOk+SUen+ofS5860f3S9LtrMzKqTc6exGGirUL89IiakVzuApDOB6cDHUp/vSRogaQDwXeAy4ExgRmoL8PV0rBbgLWBWqs8C3oqIfwbcntqZmVkD9RkaEfE4sDPzeFOBZRHxTkT8AugEJqVXZ0S8GhHvAsuAqZIEXAg8kPovAaYVjrUkbT8AXJTam5lZg9SypnG9pHVp+mp4qo0ANhXalFKtp/qJwC8jYk+3+l7HSvvfTu3NzKxBqg2NBcBHgQnAFuBbqV7pTiCqqPd2rH1Imi2pQ1KHfxG9mdmBU1VoRMTWiHgvIt4H7qI8/QTlO4WRhabNwOZe6m8CwyQN7Fbf61hp/z+hh2myiFgYEa0R0drU1FTNJZmZWYaqQkPSqYWPnwK6nqxaAUxPTz6NAVqAp4CngZb0pNTRlBfLV0READ8BPp36zwSWF441M21/Gvj71N7MzBpkYF8NJC0FpgAnSSoBNwFTJE2gPF30GvA5gIh4XtL9wAvAHuC6iHgvHed64FFgALAoIp5Pp7gRWCbpFuBZ4O5Uvxv4H5I6Kd9hTK/5as3MrCZ9hkZEzKhQvrtCrav9rcCtFertQHuF+qv8YXqrWN8NXN3X+MzM7ODxN8LNzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLL1GRqSFknaJmlDofYNST+XtE7SQ5KGpfpoSb+TtDa97iz0OVvSekmdku6QpFQ/QdJKSRvT+/BUV2rXmc4zsf6Xb2Zm+yPnTmMx0NatthIYFxF/BLwMzCvseyUiJqTX5wv1BcBsoCW9uo45F1gVES3AqvQZ4LJC29mpv5mZNVCfoRERjwM7u9Uei4g96eNqoLm3Y0g6FTg+Ip6IiADuAaal3VOBJWl7Sbf6PVG2GhiWjmNmZg1SjzWNPwEeLnweI+lZSf9H0vmpNgIoFdqUUg3glIjYApDeTy702dRDHzMza4CBtXSW9GVgD3BvKm0BRkXEDklnA/9b0scAVegefR0+t4+k2ZSnsBg1alTO0M3MrApV32lImglcCVyTppyIiHciYkfaXgO8ApxO+S6hOIXVDGxO21u7pp3S+7ZULwEje+izl4hYGBGtEdHa1NRU7SWZmVkfqgoNSW3AjcBVEbGrUG+SNCBtf4TyIvaradrp15Imp6emrgWWp24rgJlpe2a3+rXpKarJwNtd01hmZtYYfU5PSVoKTAFOklQCbqL8tNSHgJXpydnV6UmpC4CbJe0B3gM+HxFdi+hfoPwk1hDKayBd6yC3AfdLmgW8Dlyd6u3A5UAnsAv4bC0XamZmteszNCJiRoXy3T20fRB4sId9HcC4CvUdwEUV6gFc19f4zMzs4PE3ws3MLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsmWFhqRFkrZJ2lConSBppaSN6X14qkvSHZI6Ja2TNLHQZ2Zqv1HSzEL9bEnrU587JKm3c5iZWWPk3mksBtq61eYCqyKiBViVPgNcBrSk12xgAZQDALgJOAeYBNxUCIEFqW1Xv7Y+zmFmZg2QFRoR8Tiws1t5KrAkbS8BphXq90TZamCYpFOBS4GVEbEzIt4CVgJtad/xEfFERARwT7djVTqHmZk1QC1rGqdExBaA9H5yqo8ANhXalVKtt3qpQr23c5iZWQMciIVwVahFFfX8E0qzJXVI6ti+ffv+dDUzs/1QS2hsTVNLpPdtqV4CRhbaNQOb+6g3V6j3do69RMTCiGiNiNampqYaLsnMzHpTS2isALqegJoJLC/Ur01PUU0G3k5TS48Cl0ganhbALwEeTft+LWlyemrq2m7HqnQOMzNrgIE5jSQtBaYAJ0kqUX4K6jbgfkmzgNeBq1PzduByoBPYBXwWICJ2Svor4OnU7uaI6Fpc/wLlJ7SGAA+nF72cw8zMGiArNCJiRg+7LqrQNoDrejjOImBRhXoHMK5CfUelc5iZWWP4G+FmZpbNoWFmZtkcGmZmls2hYWZm2RwaZmaWzaFhZmbZHBpmZpbNoWFmZtkcGmZmls2hYWZm2RwaZmaWzaFhZmbZHBpmZpbNoWFmZtkcGmZmls2hYWZm2RwaZmaWzaFhZmbZHBpmZpbNoWFmZtmqDg1JZ0haW3j9StKfS5ov6Y1C/fJCn3mSOiW9JOnSQr0t1TolzS3Ux0h6UtJGSfdJOrr6SzUzs1pVHRoR8VJETIiICcDZwC7gobT79q59EdEOIOlMYDrwMaAN+J6kAZIGAN8FLgPOBGaktgBfT8dqAd4CZlU7XjMzq129pqcuAl6JiH/spc1UYFlEvBMRvwA6gUnp1RkRr0bEu8AyYKokARcCD6T+S4BpdRqvmZlVoV6hMR1YWvh8vaR1khZJGp5qI4BNhTalVOupfiLwy4jY061uZmYNUnNopHWGq4D/lUoLgI8CE4AtwLe6mlboHlXUK41htqQOSR3bt2/fj9Gbmdn+qMedxmXAMxGxFSAitkbEexHxPnAX5eknKN8pjCz0awY291J/ExgmaWC3+j4iYmFEtEZEa1NTUx0uyczMKqlHaMygMDUl6dTCvk8BG9L2CmC6pA9JGgO0AE8BTwMt6UmpoylPda2IiAB+Anw69Z8JLK/DeM3MrEoD+27SM0lDgYuBzxXK/1XSBMpTSa917YuI5yXdD7wA7AGui4j30nGuBx4FBgCLIuL5dKwbgWWSbgGeBe6uZbxmZlabmkIjInZRXrAu1v64l/a3ArdWqLcD7RXqr/KH6S0zM2swfyPczMyyOTTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybQ8PMzLI5NMzMLFtNP0bE6mf+/CPrvGZ2aPKdhpmZZXNomJlZNoeGmZllc2iYmVk2h4aZmWVzaJiZWTaHhpmZZXNomJlZNoeGmZllqzk0JL0mab2ktZI6Uu0ESSslbUzvw1Ndku6Q1ClpnaSJhePMTO03SppZqJ+djt+Z+qrWMZuZWXXqdafxyYiYEBGt6fNcYFVEtACr0meAy4CW9JoNLIByyAA3AecAk4CbuoImtZld6NdWpzGbmdl+OlDTU1OBJWl7CTCtUL8nylYDwySdClwKrIyInRHxFrASaEv7jo+IJyIigHsKxzIzs4OsHqERwGOS1kianWqnRMQWgPR+cqqPADYV+pZSrbd6qULdzMwaoB4/5fa8iNgs6WRgpaSf99K20npEVFHf+6DlsJoNMGrUqL5HbGZmVan5TiMiNqf3bcBDlNcktqapJdL7ttS8BIwsdG8GNvdRb65Q7z6GhRHRGhGtTU1NtV6SmZn1oKbQkHSMpOO6toFLgA3ACqDrCaiZwPK0vQK4Nj1FNRl4O01fPQpcIml4WgC/BHg07fu1pMnpqalrC8cyM7ODrNbpqVOAh9JTsAOBv42IRyQ9DdwvaRbwOnB1at8OXA50AruAzwJExE5JfwU8ndrdHBE70/YXgMXAEODh9DIzswaoKTQi4lXgrAr1HcBFFeoBXNfDsRYBiyrUO4BxtYzTzMzqw98INzOzbA4NMzPL5tAwM7NsDg0zM8vm0DAzs2wODTMzy+bQMDOzbA4NMzPL5tAwM7NsDg0zM8vm0DAzs2wODTMzy+bQMDOzbA4NMzPL5tAwM7NsDg0zM8vm0DAzs2wODTMzy+bQMDOzbA4NMzPLVnVoSBop6SeSXpT0vKQ/S/X5kt6QtDa9Li/0mSepU9JLki4t1NtSrVPS3EJ9jKQnJW2UdJ+ko6sdr5mZ1a6WO409wH+KiLHAZOA6SWemfbdHxIT0agdI+6YDHwPagO9JGiBpAPBd4DLgTGBG4ThfT8dqAd4CZtUwXjMzq9HAajtGxBZgS9r+taQXgRG9dJkKLIuId4BfSOoEJqV9nRHxKoCkZcDUdLwLgX+X2iwB5gMLqh2z7Wv+/CPz3GZWnbqsaUgaDXwceDKVrpe0TtIiScNTbQSwqdCtlGo91U8EfhkRe7rVzcysQWoODUnHAg8Cfx4Rv6J8J/BRYALlO5FvdTWt0D2qqFcaw2xJHZI6tm/fvp9XYGZmuWoKDUmDKAfGvRHxA4CI2BoR70XE+8Bd/GEKqgSMLHRvBjb3Un8TGCZpYLf6PiJiYUS0RkRrU1NTLZdkZma9qOXpKQF3Ay9GxLcL9VMLzT4FbEjbK4Dpkj4kaQzQAjwFPA20pCeljqa8WL4iIgL4CfDp1H8msLza8ZqZWe2qXggHzgP+GFgvaW2q/WfKTz9NoDyV9BrwOYCIeF7S/cALlJ+8ui4i3gOQdD3wKDAAWBQRz6fj3Qgsk3QL8CzlkDIzswap5emp/0vldYf2XvrcCtxaod5eqV96ompS97qZmTWGvxFuZmbZHBpmZpbNoWFmZtkcGmZmls2hYWZm2RwaZmaWzaFhZmbZHBpmZpbNoWFmZtlq+TEiZjVp1O/T8O/xMKue7zTMzCybQ8PMzLI5NMzMLJtDw8zMsjk0zMwsm0PDzMyyOTTMzCybv6dhRxx/P8Sser7TMDOzbA4NMzPL1u9DQ1KbpJckdUqa2+jxmJkdyfr1moakAcB3gYuBEvC0pBUR8UJjR2a2/xq5puH1FKuX/n6nMQnojIhXI+JdYBkwtcFjMjM7YvXrOw1gBLCp8LkEnNOgsZgdso7EO40j8ZoPhv4eGqpQi30aSbOB2enjbyS9VOX5TgLerLLvocLXeHjwNfbhL/+yjiM5cPrT3+OHcxr199AoASMLn5uBzd0bRcRCYGGtJ5PUERGttR6nP/M1Hh58jYeHQ/Ea+/uaxtNAi6Qxko4GpgMrGjwmM7MjVr++04iIPZKuBx4FBgCLIuL5Bg/LzOyI1a9DAyAi2oH2g3S6mqe4DgG+xsODr/HwcMhdoyL2WVc2MzOrqL+vaZiZWT/i0CiQNEDSs5L+rtFjORAkDZP0gKSfS3pR0rmNHlO9SfqPkp6XtEHSUkmDGz2mWklaJGmbpA2F2gmSVkramN6HN3KMterhGr+R/lldJ+khScMaOcZaVbrGwr45kkLSSY0Y2/5waOztz4AXGz2IA+ivgUci4p8DZ3GYXaukEcB/AFojYhzlhyemN3ZUdbEYaOtWmwusiogWYFX6fChbzL7XuBIYFxF/BLwMzDvYg6qzxex7jUgaSflHJb1+sAdUDYdGIqkZuAL4fqPHciBIOh64ALgbICLejYhfNnZUB8RAYIikgcBQKnyv51ATEY8DO7uVpwJL0vYSYNpBHVSdVbrGiHgsIvakj6spf0/rkNXD3yPA7cBfUOGLy/2RQ+MP/hvlv7j3Gz2QA+QjwHbgv6cpuO9LOqbRg6qniHgD+Cbl/2PbArwdEY81dlQHzCkRsQUgvZ/c4PEcaH8CPNzoQdSbpKuANyLiuUaPJZdDA5B0JbAtItY0eiwH0EBgIrAgIj4O/JZDf0pjL2lefyowBjgNOEbSZxo7KquVpC8De4B7Gz2WepI0FPgy8NVGj2V/ODTKzgOukvQa5Z+ke6Gk/9nYIdVdCShFxJPp8wOUQ+Rw8q+AX0TE9oj4PfAD4F80eEwHylZJpwKk920NHs8BIWkmcCVwTRx+3w/4KOX/wXku/benGXhG0j9t6Kj64NAAImJeRDRHxGjKC6d/HxGH1f+hRsT/AzZJOiOVLgIOt99L8jowWdJQSaJ8jYfVYn/BCmBm2p4JLG/gWA4ISW3AjcBVEbGr0eOpt4hYHxEnR8To9N+eEjAx/bvabzk0jix/CtwraR0wAfhag8dTV+ku6gHgGWA95X++D7lv3HYnaSnwBHCGpJKkWcBtwMWSNlJ+8ua2Ro6xVj1c498AxwErJa2VdGdDB1mjHq7xkONvhJuZWTbfaZiZWTaHhpmZZXNomJlZNoeGmZllc2iYmVk2h4aZmWVzaJiZWTaHhpmZZfv/0wjp0UgDptQAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYAAAAD8CAYAAAB+UHOxAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAEohJREFUeJzt3XGMpPV93/H3JxyQOG59GNYOvbvmSHNxS6ykpitMaimyQmwDjTikGokoMmeX6NTGSRy7VYwTKajOP4lahZQ2dXQJ1EdlYSPilquF616xLatSIV6IjcHYsCUpbCBmk8M4LU3cS7/9Y35nxntzu8PM3swcv/dLGu3z/J7vzPOdR6vns8/zzDObqkKS1J/vmHcDkqT5MAAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJndox7wY2c8EFF9TevXvn3YYknVEeeOCBP62qpa3qFjoA9u7dy8rKyrzbkKQzSpL/OU6dp4AkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqe2DIAktyV5NsnDI5b9sySV5II2nyS3JFlN8lCSS4ZqDyR5vD0ObO/bkCS9VOMcAXwYuGLjYJI9wFuAJ4eGrwT2tcdB4EOt9tXATcAbgUuBm5KcN03jkqTpbBkAVfU54NiIRTcDvwjU0Nh+4PYauA/YmeRC4G3A0ao6VlXPAUcZESqSpNmZ6BpAkquBP66qL25YtAt4amh+rY2danzUax9MspJkZX19fZL2JEljeMkBkOQVwC8DvzJq8Yix2mT85MGqQ1W1XFXLS0tb/ktLSdKEJjkC+FvARcAXk/wRsBt4MMn3MPjLfs9Q7W7g6U3GJUlz8pIDoKq+VFWvqaq9VbWXwc79kqr6E+AIcH37NNBlwPNV9QzwKeCtSc5rF3/f2sYkSXMyzsdA7wD+O/C6JGtJbtik/B7gCWAV+B3gZwCq6hjwq8Dn2+ODbUySNCepGnkqfiEsLy/XysrKvNuQpDNKkgeqanmrOu8ElqROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjo1zj+Fvy3Js0keHhr7F0m+kuShJP8hyc6hZR9Isprkq0neNjR+RRtbTXLj9r8VSdJLMc4RwIeBKzaMHQVeX1U/BDwGfAAgycXAdcAPtuf82yRnJTkL+C3gSuBi4CdbrSRpTrYMgKr6HHBsw9h/qarjbfY+YHeb3g98tKr+sqr+EFgFLm2P1ap6oqq+CXy01UqS5mQ7rgH8I+CTbXoX8NTQsrU2dqpxSdKcTBUASX4ZOA585MTQiLLaZHzUax5MspJkZX19fZr2JEmbmDgAkhwAfgL4qao6sTNfA/YMle0Gnt5k/CRVdaiqlqtqeWlpadL2JElbmCgAklwBvB+4uqpeGFp0BLguyblJLgL2Ab8PfB7Yl+SiJOcwuFB8ZLrWJUnT2LFVQZI7gDcDFyRZA25i8Kmfc4GjSQDuq6p/XFWPJLkT+DKDU0Pvrqq/aq/zs8CngLOA26rqkdPwfiRJY8qLZ28Wz/Lycq2srMy7DUk6oyR5oKqWt6rzTmBJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpU1sGQJLbkjyb5OGhsVcnOZrk8fbzvDaeJLckWU3yUJJLhp5zoNU/nuTA6Xk7kqRxjXME8GHgig1jNwL3VtU+4N42D3AlsK89DgIfgkFgADcBbwQuBW46ERqSpPnYMgCq6nPAsQ3D+4HDbfowcM3Q+O01cB+wM8mFwNuAo1V1rKqeA45ycqhIkmZo0msAr62qZwDaz9e08V3AU0N1a23sVOOSpDnZ7ovAGTFWm4yf/ALJwSQrSVbW19e3tTlJ0osmDYCvtVM7tJ/PtvE1YM9Q3W7g6U3GT1JVh6pquaqWl5aWJmxPkrSVSQPgCHDikzwHgLuHxq9vnwa6DHi+nSL6FPDWJOe1i79vbWOSpDnZsVVBkjuANwMXJFlj8GmeXwPuTHID8CRwbSu/B7gKWAVeAN4FUFXHkvwq8PlW98Gq2nhhWZI0Q6kaeSp+ISwvL9fKysq825CkM0qSB6pqeas67wSWpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA2CObj762LxbkNQxA0CSOjVVACR5b5JHkjyc5I4k35nkoiT3J3k8yceSnNNqz23zq2353u14A5KkyUwcAEl2AT8PLFfV64GzgOuAXwdurqp9wHPADe0pNwDPVdX3Aze3OknSnEx7CmgH8F1JdgCvAJ4Bfgy4qy0/DFzTpve3edryy5NkyvVLkiY0cQBU1R8D/xJ4ksGO/3ngAeDrVXW8la0Bu9r0LuCp9tzjrf78ja+b5GCSlSQr6+vrk7YnSdrCNKeAzmPwV/1FwN8Avhu4ckRpnXjKJsteHKg6VFXLVbW8tLQ0aXuSpC1Mcwrox4E/rKr1qvq/wMeBvw/sbKeEAHYDT7fpNWAPQFv+KuDYFOuXJE1hmgB4ErgsySvaufzLgS8DnwHe3moOAHe36SNtnrb801V10hGAJGk2prkGcD+Di7kPAl9qr3UIeD/wviSrDM7x39qecitwfht/H3DjFH1Lkqa0Y+uSU6uqm4CbNgw/AVw6ovYvgGunWZ8kaft4J7AkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUASFKnDABJ6pQBIEmdMgAkqVMGgCR1ygA4Q9x89LF5tyDpZcYAkKROTRUASXYmuSvJV5I8muRHkrw6ydEkj7ef57XaJLklyWqSh5Jcsj1vQZI0iWmPAP4V8J+r6m8DPww8CtwI3FtV+4B72zzAlcC+9jgIfGjKdUuSpjBxACT568CPArcCVNU3q+rrwH7gcCs7DFzTpvcDt9fAfcDOJBdO3LkkaSrTHAF8H7AO/Lskf5Dkd5N8N/DaqnoGoP18TavfBTw19Py1NiZJmoNpAmAHcAnwoap6A/C/efF0zygZMVYnFSUHk6wkWVlfX5+iPUnSZqYJgDVgrarub/N3MQiEr504tdN+PjtUv2fo+buBpze+aFUdqqrlqlpeWlqaoj1J0mYmDoCq+hPgqSSva0OXA18GjgAH2tgB4O42fQS4vn0a6DLg+ROniiRJs7djyuf/HPCRJOcATwDvYhAqdya5AXgSuLbV3gNcBawCL7RaSdKcTBUAVfUFYHnEostH1Bbw7mnWJ0naPt4JLEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUAdAB/52kpFEMAEnqlAEgSZ0yACSpUwaAJHXKAJCkThkAktQpA0CSOmUA6Nt4z4DUDwNAkjplAEhSp6YOgCRnJfmDJJ9o8xcluT/J40k+1v5hPEnObfOrbfneadctSZrcdhwBvAd4dGj+14Gbq2of8BxwQxu/AXiuqr4fuLnVSZLmZKoASLIb+AfA77b5AD8G3NVKDgPXtOn9bZ62/PJWL0mag2mPAH4T+EXg/7X584GvV9XxNr8G7GrTu4CnANry51u9JGkOJg6AJD8BPFtVDwwPjyitMZYNv+7BJCtJVtbX1ydtT5K0hWmOAN4EXJ3kj4CPMjj185vAziQ7Ws1u4Ok2vQbsAWjLXwUc2/iiVXWoqparanlpaWmK9iRJm5k4AKrqA1W1u6r2AtcBn66qnwI+A7y9lR0A7m7TR9o8bfmnq+qkIwBJ0mycjvsA3g+8L8kqg3P8t7bxW4Hz2/j7gBtPw7olSWPasXXJ1qrqs8Bn2/QTwKUjav4CuHY71idJmp53Ausl8buCpJcPA0CSOmUASFKnDABJ6pQBIEmdMgAkqVMGgCR1ygCQpE4ZAJLUKQNAkjplAEhSpwwASeqUASBJnTIAJKlTBoAkdcoAkKROGQA6Lfy/AdLiMwAkqVMGgCR1ygCQpE5NHABJ9iT5TJJHkzyS5D1t/NVJjiZ5vP08r40nyS1JVpM8lOSS7XoTOnN5rUCan2mOAI4D/7Sq/g5wGfDuJBcDNwL3VtU+4N42D3AlsK89DgIfmmLdkqQpTRwAVfVMVT3Ypv8ceBTYBewHDreyw8A1bXo/cHsN3AfsTHLhxJ3rZcUjAWn2tuUaQJK9wBuA+4HXVtUzMAgJ4DWtbBfw1NDT1trYxtc6mGQlycr6+vp2tCdJGmHqAEjySuD3gF+oqm9sVjpirE4aqDpUVctVtby0tDRte5KkU5gqAJKczWDn/5Gq+ngb/tqJUzvt57NtfA3YM/T03cDT06xfkjS5aT4FFOBW4NGq+o2hRUeAA236AHD30Pj17dNAlwHPnzhVJJ2K1wak02fHFM99E/AO4EtJvtDGfgn4NeDOJDcATwLXtmX3AFcBq8ALwLumWLckaUoTB0BV/TdGn9cHuHxEfQHvnnR96svNRx/jvW/5gXm3Ib2seSewJHXKAJCkThkAktQpA0CSOmUASFKnDAC9rHjfgDQ+A0CSOmUASFKnDABJ6pQBoO4MXyfwmoF6ZgCoG+7spW9nAKgrJ0Lg5qOPbRoIhoV6YABIUqcMAGkD//pXLwwAaROGgV7ODABpC6NCwE8S6eXAAJBeInf4erkwAKQJbQwCg0FnGgNAOo0MBS2ymQdAkiuSfDXJapIbZ71+6XTb7Mhg+D4Ead5mGgBJzgJ+C7gSuBj4ySQXz7IHaVGcKgw23qQ2zk1rBoomMesjgEuB1ap6oqq+CXwU2D/jHqSFsdmdyVuNjXOksdlrSTtmvL5dwFND82vAG2fcg3RGGudU0qj5977lB06a3yoMTjxnuG7j807Mj/q52XvYbPmktZpMqmp2K0uuBd5WVT/d5t8BXFpVPzdUcxA42GZfB3x1ilVeAPzpFM+fJXs9Pez19LDX02O7ev3eqlraqmjWRwBrwJ6h+d3A08MFVXUIOLQdK0uyUlXL2/Fap5u9nh72enrY6+kx615nfQ3g88C+JBclOQe4Djgy4x4kScz4CKCqjif5WeBTwFnAbVX1yCx7kCQNzPoUEFV1D3DPjFa3LaeSZsReTw97PT3s9fSYaa8zvQgsSVocfhWEJHXqjA+Arb5aIsm5ST7Wlt+fZO/su/xWL1v1+s4k60m+0B4/PY8+Wy+3JXk2ycOnWJ4kt7T38lCSS2bd41AvW/X65iTPD23XX5l1j62PPUk+k+TRJI8kec+ImoXYrmP2uijb9TuT/H6SL7Ze//mImoXYD4zZ6+z2A1V1xj4YXEj+H8D3AecAXwQu3lDzM8Bvt+nrgI8tcK/vBP7NvLdr6+VHgUuAh0+x/Crgk0CAy4D7F7jXNwOfWIBteiFwSZv+a8BjI34HFmK7jtnromzXAK9s02cD9wOXbahZlP3AOL3ObD9wph8BjPPVEvuBw236LuDyJJlhjyecUV+DUVWfA45tUrIfuL0G7gN2JrlwNt19uzF6XQhV9UxVPdim/xx4lMHd8cMWYruO2etCaNvqf7XZs9tj48XNhdgPjNnrzJzpATDqqyU2/pJ+q6aqjgPPA+fPpLtT9NGM6hXgH7ZD/7uS7BmxfFGM+34WxY+0w+5PJvnBeTfTTkG8gcFfgMMWbrtu0issyHZNclaSLwDPAker6pTbdc77gXF6hRntB870ABiV4BvTdJyaWRinj/8E7K2qHwL+Ky/+xbKIFmW7juNBBrfG/zDwr4H/OM9mkrwS+D3gF6rqGxsXj3jK3LbrFr0uzHatqr+qqr/L4NsFLk3y+g0lC7Ndx+h1ZvuBMz0AtvxqieGaJDuAVzGf0wXjfA3Gn1XVX7bZ3wH+3ox6m8Q4234hVNU3Thx21+A+lLOTXDCPXpKczWCH+pGq+viIkoXZrlv1ukjbdainrwOfBa7YsGhR9gPfcqpeZ7kfONMDYJyvljgCHGjTbwc+Xe1Ky4xt2euGc71XMzjvuqiOANe3T61cBjxfVc/Mu6lRknzPifO9SS5l8Hv/Z3PoI8CtwKNV9RunKFuI7TpOrwu0XZeS7GzT3wX8OPCVDWULsR8Yp9dZ7gdmfifwdqpTfLVEkg8CK1V1hMEv8b9Pssog8a9b4F5/PsnVwPHW6zvn0StAkjsYfMrjgiRrwE0MLlhRVb/N4G7uq4BV4AXgXfPpdKxe3w78kyTHgf8DXDenPwLeBLwD+FI7BwzwS8DfHOp1UbbrOL0uyna9EDicwT+c+g7gzqr6xCLuB8bsdWb7Ae8ElqROnemngCRJEzIAJKlTBoAkdcoAkKROGQCS1CkDQJI6ZQBIUqcMAEnq1P8HXyzo683gke0AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#plt.hist(RGN.degree(), bins=1000, alpha = 0.5, label = \"RGN\")\n",
    "\n",
    "plt.hist(np.log10(subPrePPI.es[\"weight\"]), alpha = 0.5, label = \"sub PrePPI\", color=\"blue\")\n",
    "#plt.hist(np.log10(PrePPI.es[\"weight\"]), alpha = 0.5, label = \"PrePPI\", color=\"red\")\n",
    "plt.legend()\n",
    "plt.show()\n",
    "plt.hist(np.log10(subPrePPI.degree()), bins=1000, alpha = 0.5, label = \"PrePPI\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13464\n",
      "13329\n",
      "20345\n"
     ]
    }
   ],
   "source": [
    "print(len(set(Brain_Reg_Net2_csv_trim[\"Transcription_Factor\"].values).union(\n",
    "    set(Brain_Reg_Net2_csv_trim[\"Target_Gene\"].values))))\n",
    "print(len(RGN.vs[\"ENSGID\"]))\n",
    "print(len(GeneDict.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<igraph.EdgeSeq at 0x1a4c3ec7d8>"
      ]
     },
     "execution_count": 280,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RGN.es(weight_lt=0.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "277\n"
     ]
    }
   ],
   "source": [
    "degreesEff = {}\n",
    "degreesEff[0] = []\n",
    "degreesNode = {}\n",
    "degreesNode[0] = []\n",
    "SCZ_Lost_Genes = []\n",
    "for i, row in SCZ_high_conf_gene.iterrows():\n",
    "    ensid = row[\"ensembl_names\"]\n",
    "    try:\n",
    "        gene = GeneDict[ensid]\n",
    "    except KeyError:\n",
    "        #print (row[\"sczgenenames\"], row[\"ensembl_names\"])\n",
    "        SCZ_Lost_Genes.append(row[\"sczgenenames\"])\n",
    "        continue\n",
    "    if len(gene.SNPs) > 0:\n",
    "        degreesEff[0].append(gene.max_eff)\n",
    "        degreesNode[0].append(gene)\n",
    "        #RGN.vs[.index(ensid)]\n",
    "print(len(degreesNode[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247\n"
     ]
    }
   ],
   "source": [
    "CoreGeneIDs = []\n",
    "for gene in degreesNode[0]:\n",
    "    #CoreGeneIDs.gene.GeneID\n",
    "    if gene.GeneID in ENSID2IDX:\n",
    "        CoreGeneIDs.append(ENSID2IDX[gene.GeneID])\n",
    "print(len(CoreGeneIDs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-375-0445f2f6d2f1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0me\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mRGN\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"weight\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mCutoff\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m         \u001b[0mRGN2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete_edges\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/igraph/__init__.py\u001b[0m in \u001b[0;36mdelete_edges\u001b[0;34m(self, *args, **kwds)\u001b[0m\n\u001b[1;32m    359\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    360\u001b[0m             \u001b[0medge_seq\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 361\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mGraphBase\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete_edges\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0medge_seq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    362\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    363\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# Filter edges\n",
    "Cutoff = 0.5\n",
    "RGN2 = RGN.copy()\n",
    "\n",
    "for e in RGN.es:\n",
    "    if e[\"weight\"] < Cutoff:\n",
    "        RGN2.delete_edges(e.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [],
   "source": [
    "Nedges = []\n",
    "for i in range(1000):\n",
    "    sample = [x.index for x in np.random.choice(RGN.vs, size=247, replace=False)]\n",
    "    sampleGraph = RGN.subgraph(sample)\n",
    "    #len(sampleGraph.es)\n",
    "    Nedges.append(sampleGraph.ecount())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAADv1JREFUeJzt3W2MpWV9x/HvrysPRm0RGcgG3C40m1TSVCRTQkJjWjAWoenSBBpM0+4Lkk1aTTRpU9eatJrYZGlSbZuYmq1Qt60VKEog1raSFWL6ougiDy5dEcStpWxYrOLDG1v03xfnWhmXmT1nZs7jtd9PMjnn3OeePT8u5vzmmmvu+55UFZKkxfcTsw4gSRoPC12SOmGhS1InLHRJ6oSFLkmdsNAlqRMWuiR1wkKXpE5Y6JLUiZdN88XOOeec2r59+zRfUpIW3oMPPviNqloatt9UC3379u0cPHhwmi8pSQsvyX+Osp9LLpLUCQtdkjphoUtSJyx0SeqEhS5JnbDQJakTFrokdcJCl6ROWOiS1Impnimq+bJ9zz+tuv3I3munnETSODhDl6ROWOiS1AkLXZI6YaFLUicsdEnqhIUuSZ2w0CWpExa6JHXCQpekTljoktSJkU79T3IE+C7wA+CFqlpOcjZwO7AdOAL8RlV9azIxtRlrneIvqS/rmaH/clVdUlXL7fEe4EBV7QAOtMeSpBnZzJLLTmB/u78fuG7zcSRJGzVqoRfwmSQPJtndtp1XVUcB2u25kwgoSRrNqJfPvaKqnklyLnBvki+P+gLtG8BugG3btm0gohaBl+KVZm+kGXpVPdNujwF3AZcBzybZCtBuj63xufuqarmqlpeWlsaTWpL0EkMLPckrkrzq+H3gzcAh4B5gV9ttF3D3pEJKkoYbZcnlPOCuJMf3/4eq+pckXwDuSHIT8HXghsnFlCQNM7TQq+op4PWrbP8f4KpJhNLGeLy5dGrzTFFJ6oSFLkmdsNAlqRMWuiR1wkKXpE5Y6JLUCQtdkjphoUtSJyx0SeqEhS5JnRj18rmSlxaQ5pwzdEnqhIUuSZ2w0CWpE66hLyDXsiWtxhm6JHXCQpekTljoktQJ19DnwFpr4kf2XjvlJNNzKv43S5PmDF2SOmGhS1InLHRJ6oSFLkmdsNAlqRMWuiR1wkKXpE54HPoEeIy1pFlwhi5JnbDQJakTFrokdWLkQk+yJclDST7VHl+Y5IEkTyS5Pcnpk4spSRpmPTP0dwCHVzy+GfhgVe0AvgXcNM5gkqT1GanQk1wAXAt8pD0OcCVwZ9tlP3DdJAJKkkYz6gz9z4E/AH7YHr8GeL6qXmiPnwbOX+0Tk+xOcjDJweeee25TYSVJaxta6El+FThWVQ+u3LzKrrXa51fVvqparqrlpaWlDcaUJA0zyolFVwC/luQa4EzgJxnM2M9K8rI2S78AeGZyMSVJwwydoVfVu6vqgqraDtwIfLaqfhO4D7i+7bYLuHtiKSVJQ23m1P93AbcleT/wEHDLeCKpJ2tdBkHS+K2r0KvqfuD+dv8p4LLxR5IkbYRnikpSJyx0SeqEl8+dIteTJU2SM3RJ6oSFLkmdsNAlqROuoeslXOuXFpMzdEnqhIUuSZ2w0CWpE66ha66stX5/ZO+1U04iLR5n6JLUCQtdkjphoUtSJ1xDn2MeDy5pPZyhS1InLHRJ6oRLLuqShz/qVOQMXZI6YaFLUicsdEnqhIUuSZ2w0CWpExa6JHXCQpekTngc+iZ4ar6keeIMXZI6YaFLUicsdEnqxNBCT3Jmks8neSTJY0ne17ZfmOSBJE8kuT3J6ZOPK0layygz9O8DV1bV64FLgKuTXA7cDHywqnYA3wJumlxMSdIwQwu9Br7XHp7WPgq4Erizbd8PXDeRhJKkkYy0hp5kS5KHgWPAvcBXgeer6oW2y9PA+ZOJKEkaxUjHoVfVD4BLkpwF3AW8brXdVvvcJLuB3QDbtm3bYEyd6ry+uTTcuo5yqarngfuBy4Gzkhz/hnAB8Mwan7OvqparanlpaWkzWSVJJzHKUS5LbWZOkpcDbwIOA/cB17fddgF3TyqkJGm4UZZctgL7k2xh8A3gjqr6VJL/AG5L8n7gIeCWCeaUJA0xtNCr6lHgDatsfwq4bBKhJEnr55miktQJC12SOuHlcyU8LFJ9cIYuSZ2w0CWpExa6JHXCNXQtNP8MoPQiZ+iS1AkLXZI6YaFLUicsdEnqhIUuSZ2w0CWpExa6JHXCQpekTljoktQJC12SOmGhS1InLHRJ6oSFLkmdsNAlqRMWuiR1wkKXpE5Y6JLUCQtdkjrhn6Abwj9xJmlROEOXpE5Y6JLUCQtdkjoxtNCTvDbJfUkOJ3ksyTva9rOT3JvkiXb76snHlSStZZQZ+gvA71XV64DLgbcluRjYAxyoqh3AgfZYkjQjQwu9qo5W1Rfb/e8Ch4HzgZ3A/rbbfuC6SYWUJA23rjX0JNuBNwAPAOdV1VEYlD5w7rjDSZJGN3KhJ3kl8AngnVX1nXV83u4kB5McfO655zaSUZI0gpEKPclpDMr8Y1X1ybb52SRb2/NbgWOrfW5V7auq5apaXlpaGkdmSdIqRjnKJcAtwOGq+sCKp+4BdrX7u4C7xx9PkjSqUU79vwL4LeBLSR5u2/4Q2AvckeQm4OvADZOJKEkaxdBCr6p/A7LG01eNN87krXVtliN7r51yEi0Cv160SDxTVJI6YaFLUicsdEnqhNdDb7zuuaRF5wxdkjphoUtSJyx0SeqEhS5JnbDQJakTFrokdcLDFqUp8TICmjRn6JLUCQtdkjphoUtSJ1xD1ynFSzyoZ87QJakTFrokdcJCl6ROdLuG7lqpJsmvL80jZ+iS1AkLXZI6YaFLUicsdEnqhIUuSZ2w0CWpExa6JHXCQpekTljoktQJC12SOmGhS1InhhZ6kluTHEtyaMW2s5Pcm+SJdvvqycaUJA0zygz9o8DVJ2zbAxyoqh3AgfZYkjRDQwu9qj4HfPOEzTuB/e3+fuC6MeeSJK3TRi+fe15VHQWoqqNJzl1rxyS7gd0A27Zt2+DLSaeetS7Re2TvtVNOokUx8V+KVtW+qlququWlpaVJv5wknbI2WujPJtkK0G6PjS+SJGkjNlro9wC72v1dwN3jiSNJ2qiha+hJPg78EnBOkqeBPwb2AnckuQn4OnDDJENKepFr61rL0EKvqreu8dRVY84iSdoEzxSVpE5Y6JLUiY0ehy5JgGv688QZuiR1wkKXpE5Y6JLUiYVfQ19r/U7SePlem3/O0CWpExa6JHXCQpekTiz8GrqkgXEdD+5a+eJyhi5JnbDQJakTFrokdcI1dGnGJr1m7bVWTh3O0CWpExa6JHViYZZcPJRKGi/fU/1xhi5JnbDQJakTFrokdWJh1tAlLZaTrdGP65BJD8n8cc7QJakTFrokdcJCl6ROuIYuaerWewz8uC4BPOm19Vmv6TtDl6ROWOiS1AkLXZI6sak19CRXA38BbAE+UlV7x5JKklYY13Vnev8zfRueoSfZAnwIeAtwMfDWJBePK5gkaX02s+RyGfBkVT1VVf8L3AbsHE8sSdJ6babQzwf+a8Xjp9s2SdIMbGYNPatsq5fslOwGdreH30vy+Dpe4xzgGxvINk2LkBEWI+ciZITFyLkIGWFOcubmkz696YxD/v1R/PQoO22m0J8GXrvi8QXAMyfuVFX7gH0beYEkB6tqeWPxpmMRMsJi5FyEjLAYORchIyxGzkXIeNxmlly+AOxIcmGS04EbgXvGE0uStF4bnqFX1QtJ3g78K4PDFm+tqsfGlkyStC6bOg69qj4NfHpMWVazoaWaKVuEjLAYORchIyxGzkXICIuRcxEyApCql/weU5K0gDz1X5I6MZeFnuTqJI8neTLJnlnnWSnJkSRfSvJwkoNt29lJ7k3yRLt99ZQz3ZrkWJJDK7atmikDf9nG9tEkl84453uT/Hcbz4eTXLPiuXe3nI8n+ZUpZXxtkvuSHE7yWJJ3tO1zM54nyThvY3lmks8neaTlfF/bfmGSB9pY3t4OqiDJGe3xk+357TPM+NEkX1sxlpe07TN7/4ykqubqg8EvWL8KXAScDjwCXDzrXCvyHQHOOWHbnwJ72v09wM1TzvRG4FLg0LBMwDXAPzM4j+By4IEZ53wv8Pur7Htx+39/BnBh+5rYMoWMW4FL2/1XAV9pWeZmPE+Scd7GMsAr2/3TgAfaGN0B3Ni2fxj4nXb/d4EPt/s3ArfPMONHgetX2X9m759RPuZxhr6IlxTYCexv9/cD103zxavqc8A3R8y0E/jbGvh34KwkW2eYcy07gduq6vtV9TXgSQZfGxNVVUer6ovt/neBwwzOgJ6b8TxJxrXMaiyrqr7XHp7WPgq4ErizbT9xLI+P8Z3AVUlWO4FxGhnXMrP3zyjmsdDn/ZICBXwmyYPtLFiA86rqKAzebMC5M0v3orUyzeP4vr39+HrriuWqmedsP/K/gcGsbS7H84SMMGdjmWRLkoeBY8C9DH46eL6qXlgly49ytue/Dbxm2hmr6vhY/kkbyw8mOePEjKvkn7l5LPSRLikwQ1dU1aUMrjL5tiRvnHWgdZq38f0r4GeAS4CjwJ+17TPNmeSVwCeAd1bVd0626yrbppJzlYxzN5ZV9YOquoTBmeSXAa87SZaZ5DwxY5KfA94N/CzwC8DZwLtmmXFU81joI11SYFaq6pl2ewy4i8EX6bPHf+xqt8dml/BH1so0V+NbVc+2N9QPgb/mxaWAmeVMchqDovxYVX2ybZ6r8Vwt4zyO5XFV9TxwP4N157OSHD8HZmWWH+Vsz/8Uoy/RjTPj1W1Zq6rq+8DfMEdjeTLzWOhze0mBJK9I8qrj94E3A4cY5NvVdtsF3D2bhD9mrUz3AL/dflt/OfDt40sJs3DC+uOvMxhPGOS8sR35cCGwA/j8FPIEuAU4XFUfWPHU3IznWhnncCyXkpzV7r8ceBOD9f77gOvbbieO5fExvh74bLXfRE4545dXfPMOgzX+lWM5N++fl5j1b2VX+2Dwm+SvMFhve8+s86zIdRGDowUeAR47no3BOt8B4Il2e/aUc32cwY/Y/8dgBnHTWpkY/Mj4oTa2XwKWZ5zz71qORxm8Wbau2P89LefjwFumlPEXGfwI/SjwcPu4Zp7G8yQZ520sfx54qOU5BPxR234Rg28oTwL/CJzRtp/ZHj/Znr9ohhk/28byEPD3vHgkzMzeP6N8eKaoJHViHpdcJEkbYKFLUicsdEnqhIUuSZ2w0CWpExa6JHXCQpekTljoktSJ/wdDrtJPDNBrKwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.hist(Nedges, bins=50)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {},
   "outputs": [],
   "source": [
    "CoreGeneNet = RGN.subgraph(CoreGeneIDs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "56"
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CoreGeneNet.ecount()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[8005, 12575]"
      ]
     },
     "execution_count": 358,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CoreGeneIDs[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[8005, 7, 12575]]"
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RGN.get_shortest_paths(8005, 12575)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathDist = []\n",
    "for v1, v2 in itertools.combinations(CoreGeneIDs, r=2):\n",
    "    #print(v1, v2)\n",
    "    path = len(RGN.get_shortest_paths(v1, v2)[0])-1\n",
    "    pathDist.append(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathDist2 = []\n",
    "sample = [x.index for x in np.random.choice(RGN.vs, size=247, replace=False)]\n",
    "for v1, v2 in itertools.combinations(sample, r=2):\n",
    "    #print(v1, v2)\n",
    "    path = len(RGN.get_shortest_paths(v1, v2)[0])-1\n",
    "    pathDist2.append(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXcAAAD8CAYAAACMwORRAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAADwRJREFUeJzt3W+IXXedx/H3xyT+gYoFM2A2fxzBsqCitjvESmEp6rK1K80DK0SwWlECrl1bVlisD2r0mU90cSuWuC22rquVKhJLilRU1AdGp9m0No0uWdmlQwsdW00t/lnifvfBPd0drndyz8zcyZ376/sFl55zz2/O+f7yaz73N+eec5KqQpLUludNuwBJ0uQZ7pLUIMNdkhpkuEtSgwx3SWqQ4S5JDTLcJalBhrskNchwl6QGbZ/WgXfu3Fnz8/PTOrwkzaQHHnjgl1U1N67d1MJ9fn6excXFaR1ekmZSkv/q087TMpLUIMNdkhpkuEtSgwx3SWqQ4S5JDTLcJalBhrskNchwl6QGGe6S1KCp3aGqGbKwMJ3jegeztG7O3CWpQYa7JDXIcJekBo0N9yQvTPLjJA8mOZXk4yPavCDJ3UnOJDmeZH4zipUk9dNn5v4H4E1V9Trg9cBVSS4favM+4FdV9Urg08AnJ1umJGktxoZ7DTzTre7oXjXU7ABwZ7d8D/DmJJlYlZKkNel1zj3JtiQngSeA+6vq+FCT3cCjAFV1DjgLvHTEfg4lWUyyuLy8vLHKJUmr6hXuVfXHqno9sAfYn+Q1Q01GzdKHZ/dU1ZGqWqiqhbm5sf9KlCRpndZ0tUxV/Rr4HnDV0KYlYC9Aku3AS4CnJlCfJGkd+lwtM5fk4m75RcBbgJ8NNTsKvKdbvhb4TlX9ycxdknRh9Hn8wC7gziTbGHwYfLWq7k3yCWCxqo4CtwNfTHKGwYz94KZVLEkaa2y4V9VDwKUj3r9lxfLvgXdMtjRJ0np5h6okNchwl6QGGe6S1CDDXZIaZLhLUoMMd0lqkOEuSQ0y3CWpQYa7JDXIcJekBhnuktQgw12SGtTnqZB6jjv82KHpHHcqR5Xa4MxdkhpkuEtSgwx3SWqQ4S5JDTLcJalBhrskNchwl6QGGe6S1CDDXZIaZLhLUoMMd0lq0NhwT7I3yXeTnE5yKsmNI9pcmeRskpPd65bNKVeS1EefB4edAz5cVSeSvBh4IMn9VfXIULsfVNXbJl+iJGmtxs7cq+rxqjrRLf8GOA3s3uzCJEnrt6Zz7knmgUuB4yM2vzHJg0nuS/LqCdQmSVqn3s9zT3IR8DXgpqp6emjzCeDlVfVMkquBbwCXjNjHIeAQwL59+9ZdtCTp/HrN3JPsYBDsX6qqrw9vr6qnq+qZbvkYsCPJzhHtjlTVQlUtzM3NbbB0SdJq+lwtE+B24HRVfWqVNi/r2pFkf7ffJydZqCSpvz6nZa4ArgN+muRk995HgX0AVXUbcC3wgSTngN8BB6uqNqFeSVIPY8O9qn4IZEybW4FbJ1WUJGljvENVkhpkuEtSgwx3SWqQ4S5JDTLcJalBhrskNchwl6QGGe6S1CDDXZIaZLhLUoMMd0lqkOEuSQ0y3CWpQYa7JDXIcJekBhnuktQgw12SGmS4S1KDDHdJapDhLkkNMtwlqUGGuyQ1yHCXpAYZ7pLUoLHhnmRvku8mOZ3kVJIbR7RJks8kOZPkoSSXbU65kqQ+tvdocw74cFWdSPJi4IEk91fVIyvavBW4pHu9Afhc919J0hSMnblX1eNVdaJb/g1wGtg91OwAcFcN/Ai4OMmuiVcrSeplTefck8wDlwLHhzbtBh5dsb7En34ASJIukN7hnuQi4GvATVX19PDmET9SI/ZxKMliksXl5eW1VSpJ6q1XuCfZwSDYv1RVXx/RZAnYu2J9D/DYcKOqOlJVC1W1MDc3t556JUk99LlaJsDtwOmq+tQqzY4C7+6umrkcOFtVj0+wTknSGvS5WuYK4Drgp0lOdu99FNgHUFW3AceAq4EzwG+B906+VElSX2PDvap+yOhz6ivbFPDBSRUlSdoY71CVpAYZ7pLUIMNdkhpkuEtSgwx3SWqQ4S5JDTLcJalBhrskNchwl6QGGe6S1CDDXZIaZLhLUoMMd0lqkOEuSQ0y3CWpQYa7JDXIcJekBhnuktQgw12SGmS4S1KDDHdJapDhLkkNMtwlqUGGuyQ1aGy4J7kjyRNJHl5l+5VJziY52b1umXyZkqS12N6jzReAW4G7ztPmB1X1tolUJEnasLEz96r6PvDUBahFkjQhkzrn/sYkDya5L8mrJ7RPSdI69TktM84J4OVV9UySq4FvAJeMapjkEHAIYN++fRM4tCRplA3P3Kvq6ap6pls+BuxIsnOVtkeqaqGqFubm5jZ6aEnSKjYc7kleliTd8v5un09udL+SpPUbe1omyZeBK4GdSZaAjwE7AKrqNuBa4ANJzgG/Aw5WVW1axZKkscaGe1W9c8z2WxlcKilJ2iK8Q1WSGmS4S1KDDHdJapDhLkkNMtwlqUGGuyQ1yHCXpAYZ7pLUIMNdkhpkuEtSgwx3SWqQ4S5JDTLcJalBhrskNchwl6QGGe6S1CDDXZIaZLhLUoMMd0lqkOEuSQ0y3CWpQYa7JDXIcJekBhnuktQgw12SGjQ23JPckeSJJA+vsj1JPpPkTJKHklw2+TIlSWvRZ+b+BeCq82x/K3BJ9zoEfG7jZUmSNmJsuFfV94GnztPkAHBXDfwIuDjJrkkVKElau0mcc98NPLpifal7708kOZRkMcni8vLyBA4tSRplEuGeEe/VqIZVdaSqFqpqYW5ubgKHliSNMolwXwL2rljfAzw2gf1KktZpEuF+FHh3d9XM5cDZqnp8AvuVJK3T9nENknwZuBLYmWQJ+BiwA6CqbgOOAVcDZ4DfAu/drGIlSf2MDfeqeueY7QV8cGIVSZI2zDtUJalBhrskNchwl6QGGe6S1CDDXZIaZLhLUoMMd0lqkOEuSQ0y3CWpQYa7JDXIcJekBhnuktQgw12SGmS4S1KDDHdJapDhLkkNMtwlqUGGuyQ1yHCXpAYZ7pLUIMNdkhpkuEtSgwx3SWqQ4S5JDeoV7kmuSvLzJGeSfGTE9uuTLCc52b3eP/lSJUl9bR/XIMk24LPAXwFLwE+SHK2qR4aa3l1VN2xCjZKkNeozc98PnKmqX1TVfwNfAQ5sblmSpI3oE+67gUdXrC917w17e5KHktyTZO9EqpMkrUufcM+I92po/ZvAfFW9Fvg2cOfIHSWHkiwmWVxeXl5bpZKk3saec2cwU185E98DPLayQVU9uWL188AnR+2oqo4ARwAWFhaGPyCk57zDh59bx9Xm6TNz/wlwSZJXJHk+cBA4urJBkl0rVq8BTk+uREnSWo2duVfVuSQ3AN8CtgF3VNWpJJ8AFqvqKPChJNcA54CngOs3sWZJ0hh9TstQVceAY0Pv3bJi+Wbg5smWJklaL+9QlaQGGe6S1CDDXZIaZLhLUoMMd0lqkOEuSQ0y3CWpQYa7JDXIcJekBhnuktQgw12SGmS4S1KDDHdJapDhLkkNMtwlqUG9nucu6QI5cmQ6xz18aDrH1aZx5i5JDTLcJalBhrskNchwl6QGGe6S1CCvlpH0nHT4cNvHduYuSQ0y3CWpQb3CPclVSX6e5EySj4zY/oIkd3fbjyeZn3ShkqT+xoZ7km3AZ4G3Aq8C3pnkVUPN3gf8qqpeCXwa+OSkC5Uk9dfnC9X9wJmq+gVAkq8AB4BHVrQ5ABzulu8Bbk2SqqoJ1ipJkzOtRz3ABXncQ59w3w08umJ9CXjDam2q6lySs8BLgV9Ooshhh/9seoNy+DGfwSFp68u4yXWSdwB/XVXv79avA/ZX1d+taHOqa7PUrf9H1+bJoX0dAp5Nxz8Hfr7OuneySR8cU2BftqZW+tJKP8C+POvlVTU3rlGfmfsSsHfF+h7gsVXaLCXZDrwEeGp4R1V1BNjwtDvJYlUtbHQ/W4F92Zpa6Usr/QD7slZ9rpb5CXBJklckeT5wEDg61OYo8J5u+VrgO55vl6TpGTtz786h3wB8C9gG3FFVp5J8AlisqqPA7cAXk5xhMGM/uJlFS5LOr9fjB6rqGHBs6L1bViz/HnjHZEs7ryl+zT1x9mVraqUvrfQD7MuajP1CVZI0e3z8gCQ1aEuHe5I7kjyR5OFVtifJZ7rHHjyU5LILXWMfPfpxZZKzSU52r1tGtdsKkuxN8t0kp5OcSnLjiDZbflx69mMmxiXJC5P8OMmDXV8+PqLNTDwipGdfrk+yvGJc3j+NWvtIsi3JvyW5d8S2zR2TqtqyL+AvgcuAh1fZfjVwHxDgcuD4tGteZz+uBO6ddp09+7ILuKxbfjHw78CrZm1cevZjJsal+3O+qFveARwHLh9q87fAbd3yQeDuade9gb5cD9w67Vp79ufvgX8d9f/RZo/Jlp65V9X3GXG9/AoHgLtq4EfAxUl2XZjq+uvRj5lRVY9X1Ylu+TfAaQZ3KK+05celZz9mQvfn/Ey3uqN7DX+ZdgC4s1u+B3hzklygEnvr2ZeZkGQP8DfAP6/SZFPHZEuHew+jHo0wk39BgTd2v4rel+TV0y6mj+7XyEsZzK5WmqlxOU8/YEbGpfv1/yTwBHB/Va06JlV1Dnj2ESFbTo++ALy9O+V3T5K9I7ZvBf8I/APwP6ts39QxmfVwH/UpN4uf8icY3FL8OuCfgG9MuZ6xklwEfA24qaqeHt484ke25LiM6cfMjEtV/bGqXs/gDvL9SV4z1GRmxqRHX74JzFfVa4Fv8/+z3y0jyduAJ6rqgfM1G/HexMZk1sO9z6MRtryqevrZX0VrcE/BjiQ7p1zWqpLsYBCIX6qqr49oMhPjMq4fszYuAFX1a+B7wFVDm/5vTM73iJCtZLW+VNWTVfWHbvXzwF9c4NL6uAK4Jsl/Al8B3pTkX4babOqYzHq4HwXe3V2dcTlwtqoen3ZRa5XkZc+ea0uyn8G4PHn+n5qOrs7bgdNV9alVmm35cenTj1kZlyRzSS7ull8EvAX42VCzmXhESJ++DH1/cw2D70u2lKq6uar2VNU8gy9Lv1NV7xpqtqljsqX/gewkX2ZwxcLOJEvAxxh8wUJV3cbgrtmrgTPAb4H3TqfS8+vRj2uBDyQ5B/wOOLgV/+J1rgCuA37anRcF+CiwD2ZqXPr0Y1bGZRdwZwb/sM7zgK9W1b2ZzUeE9OnLh5JcA5xj0Jfrp1btGl3IMfEOVUlq0KyflpEkjWC4S1KDDHdJapDhLkkNMtwlqUGGuyQ1yHCXpAYZ7pLUoP8FiqgvY+smwREAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.067904282281689 2.201474605839176\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=410390403.5, pvalue=0.0)"
      ]
     },
     "execution_count": 373,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plt.hist(pathDist, color=\"red\", alpha=0.8, density=1)\n",
    "plt.hist(pathDist2, color = 'blue', alpha=.5, density=1)\n",
    "plt.show()\n",
    "print(np.mean(pathDist), np.mean(pathDist2))\n",
    "mannwhitneyu(pathDist, pathDist2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [],
   "source": [
    "CoreGeneeCcentricity = RGN.eccentricity(CoreGeneIDs)\n",
    "AllGeneeCcentricity = RGN.eccentricity()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[4.0, 4.0, 4.0, 4.0, 4.0, 3.0, 4.0, 4.0, 4.0, 3.0]"
      ]
     },
     "execution_count": 355,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CoreGeneeCcentricity[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[3.0, 4.0, 4.0, 3.0, 4.0, 3.0, 4.0, 3.0, 3.0, 3.0]"
      ]
     },
     "execution_count": 356,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AllGeneeCcentricity[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAADWhJREFUeJzt3X+MZfVdxvH30x0KBaos7tgg1A6ahliJEZwQlIRUKAGBgEZNtoYGmjYbFWm1JpXGP1rrP8QaRaOxrrSG2hZa1zYiqbVoS9Qo284C/QErgoB0Bd2pa6vYpAh8/OOepZdhd+fee87c2X55v5LJnHvPOfc8HL7z7Jlz7rmTqkKS9K3vJZsdQJI0DAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGrFvoSd6fZH+SL409d3KSO5I82H3furExJUnryXp3iiY5H3gS+EBVndk99xvAgaq6Icn1wNaq+pX1NrZt27ZaWlrqn1qSXkT27NnzlapaXG+5hfUWqKq/TbK05ukrgdd20zcDdwLrFvrS0hIrKyvrLSZJGpPkXydZbtZz6K+oqicAuu/fOePrSJIGsuEXRZPsSLKSZGV1dXWjNydJL1qzFvp/JDkFoPu+/3ALVtXOqlququXFxXVPAUmSZjRrod8GXN1NXw38+TBxJEmzmuRti7cA/wickWRfkjcBNwAXJXkQuKh7LEnaRJO8y+X1h5l14cBZJEk9eKeoJDXCQpekRljoktSIdc+hS98ylpdnX9c7mNUAj9AlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSI3oVepJfSnJfki8luSXJcUMFkyRNZ+ZCT3Iq8BZguarOBLYA24cKJkmaTt9TLgvAy5IsAMcDj/ePJEmaxcyFXlX/Bvwm8BjwBPC1qvrUUMEkSdPpc8plK3AlcDrwXcAJSa46xHI7kqwkWVldXZ09qSTpiPqccnkd8EhVrVbV/wEfA35k7UJVtbOqlqtqeXFxscfmJElH0qfQHwPOTXJ8kgAXAnuHiSVJmlafc+i7gV3A3cAXu9faOVAuSdKUFvqsXFXvBN45UBZJUg/eKSpJjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpEb0KPclJSXYl+acke5P88FDBJEnTWei5/u8An6yqn0ryUuD4ATJJkmYwc6En+TbgfOAagKp6CnhqmFiSpGn1OeXyPcAq8MdJ7klyU5ITBsolSZpSn0JfAM4G/qCqzgL+F7h+7UJJdiRZSbKyurraY3OSpCPpU+j7gH1Vtbt7vItRwT9PVe2squWqWl5cXOyxOUnSkcxc6FX178CXk5zRPXUhcP8gqSRJU+v7LpfrgA9173B5GHhj/0iSpFn0KvSquhdYHiiLJKkH7xSVpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1Ijev2RaKkV73rXi3PbaotH6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY3oXehJtiS5J8ntQwSSJM1miCP0twJ7B3gdSVIPvQo9yWnAZcBNw8SRJM2q7xH6jcDbgWcHyCJJ6mHmQk9yObC/qvass9yOJCtJVlZXV2fdnCRpHX2O0M8DrkjyKHArcEGSD65dqKp2VtVyVS0vLi722Jwk6UhmLvSqekdVnVZVS8B24NNVddVgySRJU/F96JLUiIUhXqSq7gTuHOK1JEmz8QhdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiNmLvQkr0zymSR7k9yX5K1DBpMkTWehx7pPA79cVXcneTmwJ8kdVXX/QNkkSVOY+Qi9qp6oqru76f8B9gKnDhVMkjSdQc6hJ1kCzgJ2D/F6kqTp9S70JCcCfwb8YlX99yHm70iykmRldXW17+YkSYfRq9CTHMOozD9UVR871DJVtbOqlqtqeXFxsc/mJElH0OddLgHeB+ytqt8aLpIkaRZ9jtDPA94AXJDk3u7r0oFySZKmNPPbFqvq74EMmEWS1IN3ikpSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGLGx2gIktL8++7srKcDkk6SjlEbokNcJCl6RGWOiS1IhehZ7kkiQPJHkoyfVDhZIkTW/mi6JJtgC/D1wE7AM+l+S2qrp/qHDj3vX4jh4r91i1x7qSNE99jtDPAR6qqoer6ingVuDKYWJJkqbVp9BPBb489nhf95wkaROkqmZbMflp4OKqenP3+A3AOVV13ZrldgAHz5ecATwwY9ZtwFdmXHcjmWs65pqOuabTaq5XVdXiegv1ubFoH/DKscenAY+vXaiqdgI7e2wHgCQrVdXj7qKNYa7pmGs65prOiz1Xn1MunwNeneT0JC8FtgO3DRNLkjStmY/Qq+rpJL8A/BWwBXh/Vd03WDJJ0lR6fZZLVX0C+MRAWdbT+7TNBjHXdMw1HXNN50Wda+aLopKko4u3/ktSIza90JMcl+SzST6f5L4kv3aIZY5N8pHuIwZ2J1kam/eO7vkHklw851xvS3J/ki8k+Zskrxqb90ySe7uvwS4WT5jrmiSrY9t/89i8q5M82H1dPedcvz2W6Z+TfHVs3obsr7HX35LkniS3H2Le3MfXhLnmPr4mzDX38TVhrk0ZX0keTfLF7rVf8FndGfndbhx9IcnZY/OG3V9VtalfQIATu+ljgN3AuWuW+Xngvd30duAj3fRrgM8DxwKnA/8CbJljrh8Fju+mf+5gru7xk5u4v64Bfu8Q654MPNx939pNb51XrjXLX8foQvqG7q+x138b8GHg9kPMm/v4mjDX3MfXhLnmPr4mybVZ4wt4FNh2hPmXAn/Z/YycC+zeqP216UfoNfJk9/CY7mvtif0rgZu76V3AhUnSPX9rVX2jqh4BHmL0kQRzyVVVn6mqr3cP72L0XvwNNeH+OpyLgTuq6kBV/RdwB3DJJuV6PXDLENteT5LTgMuAmw6zyNzH1yS5NmN8TZLrCDZsfM2Qa27jawJXAh/ofkbuAk5KcgobsL82vdDhuV+j7gX2M/oP3L1mkec+ZqCqnga+BnwHG/zxAxPkGvcmRv8KH3RckpUkdyX58aEyTZHrJ7tf73YlOXgD2FGxv7pTB6cDnx57esP2F3Aj8Hbg2cPM35TxNUGucXMbXxPmmvv4mjDXZoyvAj6VZE9Gd8avdbj9Mvj+OioKvaqeqaofZHQEck6SM9cskkOtdoTn55VrFC65ClgG3jP29HfX6M6wnwFuTPK9c8z1F8BSVf0A8Nd88+jzqNhfjE5r7KqqZ8ae25D9leRyYH9V7TnSYod4bkPH14S5Di47t/E1Ya65j69p9hdzHF+d86rqbODHgGuTnL9m/tzG11FR6AdV1VeBO3nhrx3PfcxAkgXg24EDTPjxAxuYiySvA34VuKKqvjG2zuPd94e7dc+aV66q+s+xLH8E/FA3ven7q7OdNb8Ob+D+Og+4IsmjjD4R9IIkH1yzzGaMr0lybcb4WjfXJo2vifZXZ57ja/y19wMf54Wn5Q63X4bfX0NdGJj1C1gETuqmXwb8HXD5mmWu5fkXrT7aTX8/z79o9TDDXRSdJNdZjC6UvXrN81uBY7vpbcCDwGvmmOuUsemfAO6qb16EeaTLt7WbPnleubp5ZzC6iJR57K81234th77IN/fxNWGuuY+vCXPNfXxNkmszxhdwAvDysel/AC5Zs8xlPP+i6Gc3an/1ulN0IKcAN2f0BzNewuiH6fYk7wZWquo24H3AnyR5iNGR03aAqrovyUeB+4GngWvr+b9mbXSu9wAnAn86uobGY1V1BfB9wB8mebZb94Ya7g9/TJLrLUmuYLRPDjB6VwJVdSDJrzP6HB6Ad1fVgTnmgtHFqlurG9Gdjdxfh3QUjK9Jcm3G+Jok12aMr0lywfzH1yuAj3f/fxaAD1fVJ5P8LEBVvZfR3fSXMrqo/nXgjd28wfeXd4pKUiOOqnPokqTZWeiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXi/wFCyNvr/AxShQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.9635627530364372 3.9717158076374823\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=1632850.0, pvalue=0.2409400693121318)"
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plt.hist(CoreGeneeCcentricity, color=\"red\", alpha=0.8, density=1)\n",
    "plt.hist(AllGeneeCcentricity, color = 'blue', alpha=.5, density=1)\n",
    "plt.show()\n",
    "print(np.mean(CoreGeneeCcentricity), np.mean(AllGeneeCcentricity))\n",
    "mannwhitneyu(CoreGeneeCcentricity, AllGeneeCcentricity)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [],
   "source": [
    "CoreGeneCloseness = RGN.closeness(CoreGeneIDs)\n",
    "AllGeneCloseness = RGN.closeness()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAADUJJREFUeJzt3W+sZPVdx/H3B2hrbGmym72QFaEXCVapUTZOsJGYtKkVStICaZuIkWyVZqsp2qY8EFsTN/WBjdqiiU3NEkh50BYb26ZoaHXdgIREGu/iloWuCGxRYTdwKSZg1Bro1wf3LJ0u9+7Mnf/3x/uVTObMmd+Z88lh98PZ82cmVYUkaes7bd4BJEmTYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGnHGLFe2Y8eOWl5enuUqJWnLO3jw4DNVtTRo3EwLfXl5mZWVlVmuUpK2vCT/Nsw4D7lIUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjZnqnqKQ56fUm+3ne8b2Q3EOXpEYMLPQk5ya5K8mRJA8l+VA3f2+SJ5Mc6h5XTD+uJGkjwxxyeQG4oaruT3ImcDDJ/u69m6rqT6YXT5I0rIGFXlXHgePd9PNJjgDnTDuYJGlzNnUMPckysAv4Rjfr+iQPJLk1ybYNltmTZCXJyurq6lhhJUkbG7rQk7wO+BLw4ap6DvgMcAFwMWt78J9cb7mq2ldVvarqLS0N/H52SdKIhir0JK9ircw/V1VfBqiqp6rqxar6HnAzcMn0YkqSBhnmKpcAtwBHqupTffN39g27Gnhw8vEkScMa5iqXS4FrgcNJDnXzPgpck+RioIDHgQ9MJaEkaSjDXOVyL5B13rpz8nEkSaPyTlFJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhoxsNCTnJvkriRHkjyU5EPd/O1J9id5pHveNv24kqSNDLOH/gJwQ1X9JPBm4INJLgJuBA5U1YXAge61JGlOBhZ6VR2vqvu76eeBI8A5wJXAbd2w24CrphVSkjTYpo6hJ1kGdgHfAM6uquOwVvrAWZMOJ0ka3tCFnuR1wJeAD1fVc5tYbk+SlSQrq6uro2SUJA1hqEJP8irWyvxzVfXlbvZTSXZ27+8Enl5v2araV1W9quotLS1NIrMkaR3DXOUS4BbgSFV9qu+tO4Dd3fRu4KuTjydJGtYZQ4y5FLgWOJzkUDfvo8AngC8muQ74d+C904koSRrGwEKvqnuBbPD22yYbR5I0Ku8UlaRGWOiS1AgLXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqxBnzDiC90u3dO4OVHNvz8vX+yL4ZrFiz5B66JDViYKEnuTXJ00ke7Ju3N8mTSQ51jyumG1OSNMgwe+ifBS5fZ/5NVXVx97hzsrEkSZs1sNCr6h7g2RlkkSSNYZxj6NcneaA7JLNtYokkSSMZtdA/A1wAXAwcBz650cAke5KsJFlZXV0dcXWSpEFGKvSqeqqqXqyq7wE3A5ecYuy+qupVVW9paWnUnJKkAUYq9CQ7+15eDTy40VhJ0mwMvLEoyReAtwA7kjwB/D7wliQXAwU8DnxgihklSUMYWOhVdc06s2+ZQhZJ0hi8U1SSGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhrhb4pKi2ifv/epzXMPXZIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNWJgoSe5NcnTSR7sm7c9yf4kj3TP26YbU5I0yDB76J8FLj9p3o3Agaq6EDjQvZYkzdHAQq+qe4BnT5p9JXBbN30bcNWEc0mSNmnUY+hnV9VxgO75rI0GJtmTZCXJyurq6oirkyQNMvWTolW1r6p6VdVbWlqa9uok6RVr1EJ/KslOgO756clFkiSNYtRCvwPY3U3vBr46mTiSpFENc9niF4B/BN6Y5Ikk1wGfAN6e5BHg7d1rSdIcnTFoQFVds8Fbb5twFknSGLxTVJIaYaFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRljoktQIC12SGmGhS1IjLHRJaoSFLkmNsNAlqREWuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcJCl6RGWOiS1Igz5h1A0nzsPbZnjIXHXPeYy2t97qFLUiMsdElqhIUuSY2w0CWpERa6JDXCQpekRox12WKSx4HngReBF6qqN4lQkqTNm8R16G+tqmcm8DmSpDF4yEWSGjFuoRfwd0kOJhnjtjNJ0rjGPeRyaVUdS3IWsD/Jv1TVPf0DuqLfA3DeeeeNuTpJ0kbG2kOvqmPd89PAV4BL1hmzr6p6VdVbWloaZ3WSpFMYeQ89yWuB06rq+W76l4CPTyyZNEN+WZRaMM4hl7OBryQ58Tmfr6qvTySVJGnTRi70qjoK/MwEs0iSxuBli5LUCAtdkhphoUtSIyx0SWqEhS5JjfBHoqVJ2bdv3gn0CuceuiQ1wkKXpEZY6JLUCAtdkhphoUtSIyx0SWqEhS5JjbDQJakRFrokNcI7RfXK1et9f/qYv3Gurc89dElqhIUuSY3wkIsWykx/rNnDLGqMe+iS1AgLXZIaYaFLUiMsdElqhIUuSY3wKhdtHf7Em3RK7qFLUiMsdElqhIdc9DIzvblH0sS4hy5JjbDQJakRHnJZYFv+0IdXpUgzNdYeepLLkzyc5NEkN04qlCRp80Yu9CSnA58G3gFcBFyT5KJJBZMkbc44h1wuAR6tqqMASW4HrgS+NYlgC6PXY+8if83qngXOJmmmxin0c4D/6Hv9BPBz48XZ2NyOJy9ymUtSn1TVaAsm7wUuq6r3d6+vBS6pqt86adwe4EQrvhF4+BQfuwN4ZqRA82f2+TD7fJh9tt5QVUuDBo2zh/4EcG7f6x8Fjp08qKr2AUNd7pBkpap6g0cuHrPPh9nnw+yLaZyrXP4JuDDJ+UleDfwycMdkYkmSNmvkPfSqeiHJ9cDfAqcDt1bVQxNLJknalLFuLKqqO4E7J5QFhjw0s6DMPh9mnw+zL6CRT4pKkhaL3+UiSY2YWaEP+pqAJB9J8q0kDyQ5kOQNfe+9mORQ95j5idchsv9GksNdvnv775hN8rvdcg8nuWy2yUfPnmQ5yf/0bfe/WLTsfePek6SS9PrmLfR27xv3A9m3wnZP8r4kq30Z39/33u4kj3SP3bNNPnb2ufbMRFTV1B+snTR9DPgx4NXAN4GLThrzVuCHu+nfBP6y773/mkXOMbK/vm/6XcDXu+mLuvGvAc7vPuf0LZJ9GXhwkbd7N+5M4B7gPqC3Vbb7KbIv/HYH3gf8+TrLbgeOds/buultWyF7997cemZSj1ntob/0NQFV9X/Aia8JeElV3VVV/929vI+169oXwTDZn+t7+VrgxImJK4Hbq+q7VfVt4NHu82ZlnOzzNjB75w+APwL+t2/ewm/3znrZ523Y7Ou5DNhfVc9W1X8C+4HLp5RzPeNkb8KsCn29rwk45xTjrwO+1vf6h5KsJLkvyVXTCHgKQ2VP8sEkj7H2F/S3N7PsFI2THeD8JP+c5B+S/MJ0o77MwOxJdgHnVtXfbHbZKRsnOyz4du+8uzs8+ldJTtxguPDbvbNedphvz0zErAo968xbd08wya8CPeCP+2afV2t3dv0K8KdJLph8xA0Nlb2qPl1VFwC/A/zeZpadonGyH2dtu+8CPgJ8Psnrp5b05U6ZPclpwE3ADZtddgbGyb7Q273z18ByVf008PfAbZtYdprGyQ7z7ZmJmFWhD/U1AUl+EfgY8K6q+u6J+VV1rHs+CtwN7Jpm2JMMlb3P7cCJ/7tvdtlJGzl7d7jiO930QdaOTf74lHKuZ1D2M4GfAu5O8jjwZuCO7uTiom/3DbNvge1OVX2n7+/nzcDPDrvslI2Tfd49MxmzOFDP2g1MR1k7QXXiZMWbThqzi7U/vBeeNH8b8JpuegfwCOucYJpz9gv7pt8JrHTTb+IHT84dZbYn58bJvnQiK2snmZ4Eti9S9pPG3833Tywu/HY/RfaF3+7Azr7pq4H7uuntwLe7v7Pbuumtkn2uPTOxbTDDjX0F8K9daX+sm/dx1vbGYe2fP08Bh7rHHd38nwcOd/9xDgPXzXwjDc7+Z8BDXe67+v8QsfYvjsdY+5bJd2yV7MC7u/nfBO4H3rlo2U8a+1IpboXtvlH2rbDdgT/sy3gX8BN9y/46ayehHwV+batkX4SemcTDO0UlqRHeKSpJjbDQJakRFrokNcJCl6RGWOiS1AgLXZIaYaFLUiMsdElqxP8DwTWBPltK9zoAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.47115834567716425 0.45774358496013257\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "MannwhitneyuResult(statistic=1209420.0, pvalue=4.1706111331002457e-13)"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plt.hist(CoreGeneCloseness, color=\"red\", alpha=0.8, density=1)\n",
    "plt.hist(AllGeneCloseness, color = 'blue', alpha=.5, density=1)\n",
    "plt.show()\n",
    "print(np.mean(CoreGeneCloseness), np.mean(AllGeneCloseness))\n",
    "mannwhitneyu(CoreGeneCloseness, AllGeneCloseness)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [],
   "source": [
    "#GetIndexOfAllNodes\n",
    "ENSID2IDX = {}\n",
    "for idx, v in enumerate(RGN.vs):\n",
    "    ENSID2IDX[v[\"ENSGID\"]] = idx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'ENSG00000230124'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-295-7873f33617e2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0mdegreesNode\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdegreesNode\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mensgid\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdegreesNode\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 26\u001b[0;31m         \u001b[0mdegreesEff\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mGeneDict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mensgid\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax_eff\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     27\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdegreesNode\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m     \u001b[0mPreNodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPreNodes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munion\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdegreesNode\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ENSG00000230124'"
     ]
    }
   ],
   "source": [
    "PreNodes = set(degreesNode[0])\n",
    "Not_In_Network = []\n",
    "for i in range(1,11,1):\n",
    "    degreesEff[i] = []\n",
    "    degreesNode[i] = []\n",
    "    for gene in degreesNode[i-1]:\n",
    "        ENSGID = gene.GeneID\n",
    "        try:\n",
    "            NodeID = ENSID2IDX[ENSGID]\n",
    "        except:\n",
    "            Not_In_Network.append(ENSGID)\n",
    "            continue\n",
    "        if ENSGID in ENSID2IDX:\n",
    "            for edge in RGN.es.select(_source = NodeID):\n",
    "                #print(NodeID, edge.tuple)\n",
    "                target = edge.tuple[1]\n",
    "                if target not in PreNodes:\n",
    "                    degreesNode[i].append(RGN.vs[target][\"ENSGID\"])\n",
    "            for edge in RGN.es.select(_target = NodeID):\n",
    "                #print(NodeID, edge.tuple)\n",
    "                source = edge.tuple[0]\n",
    "                if source not in PreNodes:\n",
    "                    degreesNode[i].append(RGN.vs[source][\"ENSGID\"])\n",
    "    degreesNode[i] = list(set(degreesNode[i]))\n",
    "    for ensgid in degreesNode[i]:\n",
    "        degreesEff[i].append(GeneDict[ensgid].max_eff)\n",
    "    print(len(degreesNode[i]))\n",
    "    PreNodes = PreNodes.union(set(degreesNode[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "igraph.Vertex(<igraph.Graph object at 0x1a7cf3ce58>, 273, {'ENSGID': 'ENSG00000025434'}) 273 igraph.Edge(<igraph.Graph object at 0x1a7cf3ce58>, 1142, {'weight': 0.11449726357037401, 'region': 'chr19:45414026-45415025', 'TF': 'NR1H3', 'Taget': 'APOE'}) (273, 390)\n"
     ]
    }
   ],
   "source": [
    "print(RGN.vs[273], RGN.vs[273].index, RGN.es[1142], RGN.es[1142].tuple)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.09101 1.062365 1.06332\n",
      "1.0945847653429601 1.0697580970504281 1.0694282787526546\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAi0AAAF3CAYAAABpFHt+AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3X98lvV97/H35w4xgQAhEXAKBizV00hqu5VtZ9bTSs8goi3raVdtxuY5koMnVFJ3rAfnSXfs1oHVTXt2cJip2LVbe+9XO2tVBjpSWE6nbdAVibjKo6KkSqUmQYgkhOSzP3LnbhIguTC57uu6cr+ej0ce931d1/3jjb1755PvT3N3AQAAxF0q6gAAAABBULQAAIBEoGgBAACJQNECAAASgaIFAAAkAkULAABIBIoWAACQCBQtAAAgEShaAABAIlC0AACARJgSdYCzNXv2bF+4cGHUMQAAwATYvXv3z9x9TpDHJq5oWbhwoVpaWqKOAQAAJoCZvRL0sXQPAQCARKBoAQAAiUDRAgAAEoGiBQAAJAJFCwAASASKFgAAkAgULQAAIBEoWgAAQCJQtAAAgESgaAEAIKB0Oq2qqioVFBSoqqpK6XQ66kh5JXHL+AMAEIV0Oq2GhgZt2bJFV1xxhZqbm1VbWytJqqmpiThdfjB3jzrDWVmyZImz9xAAINeqqqq0adMmLV26NHuuqalJ9fX12rt3b4TJks3Mdrv7kkCPpWgBAGBsBQUF6u7uVmFhYfZcb2+viouL1dfXF2GyZDubooUxLQAABFBZWanm5uZh55qbm1VZWRlRovxD0QIAQAANDQ2qra1VU1OTent71dTUpNraWjU0NEQdLW8wEBcAgAAGB9vW19dr3759qqys1IYNGxiEm0OMaQEAAJFhTAsAAJh0KFoAAAiIxeWiRdECAEAA6XRaN998s7q6uuTu6urq0s0330zhkkMULQAABLB+/XoVFBTo4YcfVk9Pjx5++GEVFBRo/fr1UUfLGxQtAAAE0NbWphtuuEH19fUqLi5WfX29brjhBrW1tUUdLW8w5RkAgIC+8pWv6Bvf+EZ276Hf+q3fijpSXqFoAQAggClTpujo0aNavXq1Xn31VVVUVOjo0aOaMoVfpbnCf2kAAAI4efKkTp48qa6uLknSgQMHog2UhxjTAiBWqqurlUqlZGZKpVKqrq6OOhKAmKBoARAb1dXV2r59u+rq6tTZ2am6ujpt376dwgWxUlZWplQqpbKysqij5B26hwDExpNPPqm1a9dq8+bNkpS9bWxsjDIWkHXOOeeotLRUnZ2dKi0tVVdXl06cOBF1rLxBSwuA2HB3vfjii8O6h1588UUlbY80TF4nTpzQgQMH5O46cOAABUuOUbQAiJWmpqZh3UNNTU1RRwIQE+zyDCA2zOyM15L2XYXJh89nONjlGQAATDoULQBi5/LLL9drr72myy+/POoowDBmlm1xGXofuUHRAiB2nn76aV1wwQV6+umno44CDOPumjt3rvbt26e5c+fSLZRjTHkGEDv9/f3DboE4+elPf6rKysqoY+QlWloAxM7atWvV2dmptWvXRh0FOMXQ7iHkFrOHAMQGszMQZ4NjWIZ+FgeP+Xy+c8weApBIRUVFKi0tHXautLRURUVFESUChjMz3XPPPerq6tI999xDa0uOUbQAiI01a9boyJEjw84dOXJEa9asiSgRMFxFRYU+97nPqaSkRJ/73OdUUVERdaS8QtECIDZ27twp6dQxA4PngSiVlJTowIEDw84dOHBAJSUl0QTKQxQtAGLj+eef18qVK9Xf3y93V39/v1auXKnnn38+6miA3n777bM6j4lH0QIgVrZs2TLqMRCVwcG2qVRq2C2DcHOHogVArCxfvlxVVVUqKChQVVWVli9fHnUkIKuwsFBPPfWUTpw4oaeeekqFhYVRR8orFC0AYuPCCy/Uc889p9LSUrW1tam0tFTPPfecLrzwwqijAZKk3t5e1dTU6JxzzlFNTY16e3ujjpRXWBEXQGzMnDlTF110kb73ve/pggsukCRddNFFmjZtWsTJgJ87fPjwsFvkDkULgNjYt2+furu7hzW59/b2qri4OMJUwIDBheRGbjPBWi25Q/cQgNiorKzUtddeq+LiYpmZiouLde2117LPC2LhTANuGYibOxQtAGJj3rx5euSRR7R69Wp1dnZq9erVeuSRRzRv3ryoowGSpIKCgmxLYGFhoQoKCiJOlF8oWvJUfX39sL9m6+vro44EaOfOnVq1apV27dql8vJy7dq1S6tWrWJxOcRGX1/fsO6hvr6+iBPlF8a05KH6+no1NjbqrrvuUl1dnRobG3XbbbdJkjZt2hRxOuSznp4ePfDAA8MG3r799tv6+te/HmEqAHFBS0seevDBB3XXXXfplltu0bRp03TLLbforrvu0oMPPhh1NOS5oqIiNTY2DjvX2NjIhomIlWuuuUaHDx/WNddcE3WUvGNJG0C0ZMkSb2lpiTpGopmZurq6TvlrtqSkhAFliNSZWgHr6upoBUTkzCw7g2joOXfnu3MczGy3uy8J8lhaWvIQf80irjZt2qTzzz9/2C66559/PgULYuW8884bdovcoWjJQ2vWrNFtt92me++9V2+//bbuvfde3XbbbVqzZk3U0ZDnqqurdfDgQa1du1adnZ1au3atDh48qOrq6qijAdmZQmamVCqVXZ+FGUS5Q/dQnqqvr9eDDz6onp4eFRUVac2aNfw1i8ilUinNmzdPP/nJT+TuMrPs8eCMDSAqg4XK0M9iKpUatuAczl4suofM7GEze8PM9p7h+ioz25P5+Z6ZvS+sLDjVpk2b1N3dLXdXd3c3BQtiwd3V1tamuro6dXZ2qq6uTm1tbYwXQCxMmzZN/f39w3Z57u/vZ5uJHAqze+gvJF01yvWXJX3Y3S+T9EVJD4SYBUBCLF68WJs3b1Zpaak2b96sxYsXRx0JkCR1dXVJkubOnatUKqW5c+cOO4/whVa0uPsuSe2jXP+eu3dkDp+WND+sLACSo7W1VZ/5zGd05MgRfeYzn1Fra2vUkYAsM9OhQ4fU39+vQ4cOse9QjsVlIG6tpK1RhwAQLTPT/Pnz1djYqFmzZqmxsVHz58/nFwNiw921cuVKHT58WCtXrqTrMsciL1rMbKkGipbbRnnMjWbWYmYtbAUOTF7Lli077ZiWZcuWRR0NyJo3b54KCwvZEysCoc4eMrOFkh5z96ozXL9M0j9IWuHuPwrymsweAia3yy67TM8//3z2+L3vfa/27NkTYSJgwGgtfrS4vHOxmD00FjOrkPQtSb8TtGABMLml02kdO3ZMO3bs0IkTJ7Rjxw4dO3ZM6XQ66mgAYiC0lhYzS0u6UtJsST+VdIekQkly90Yze0jSJyW9knnKySCVFi0twORVVVWliy++WFu3bs2uIbRixQq99NJL2rv3tKsnADlDS0s4zqalhcXlAMRGKpVSKpXS3Xffnd17aP369erv72fxLkSOoiUcFC0AEimVSunSSy/V/v37sy0t7373u/XCCy9QtCByFC3hSMSYFgAYyd3V2tqqadOmycw0bdo0tba28gsBgCSKFgAxY2bq6OiQu6ujo4M1WgBkUbRMImaWkx8gTCzeBeBMpkQdABPnnXy5mxm/FBArCxYs0LZt2zRnzhwVFRVpwYIFeuWVV8Z+IoBJj5YWALFy8OBBbdy4UV1dXdq4caMOHjwYdSQAMcHsoTxHSwvipLCwUAUFBerv71dvb68KCwuVSqXU19en3t7eqOMhzzF7KBxnM3uI7iEAORF0PNTJkyez94cWKkGfzy8PYPKiewhATrh7oJ9169apqKhIklRUVKR169YFfi4FC96JiZiEwCSG3KB7KM/RPYS44rOJuEmlUqf9TJoZix+OA4vLAQAwwW666aazOo+Jx5gWAAAC2LRpkyTpwQcfzG4zsWbNmux5hI/uoTxHEzziis8m4ozP58ShewgAAEw6FC0AACARKFoAAEAiULQAAIBEoGgBAACJQNECAAASgaIFAAAkAkULAABIBIoWAACQCBQtAAAgEShaAABAIlC0AACARKBoAQAAiUDRAgAAEoGiBQAAJAJFCwAASASKFgAAkAgULQAAIBEoWgAAQCJQtAAAgESgaAEAAIlA0QIAABKBogUAACQCRQsAAEgEihYAAJAIFC0AACARpgR9oJmVSbpA0nFJB9y9P7RUAAAAI4xatJhZqaSbJNVIOkfSYUnFks4zs6clbXb3ptBTAgCAvDdWS8vfS/qapP/k7p1DL5jZByT9jpm9y923hBUQAABAGqNocfdlo1zbLWn3hCcCAAA4jbMZ07JS0ocyhzvd/TvhRAIAADhVoNlDZvYlSTdLeiHz81kzuzPMYAAAAEMFbWm5WtL7B2cMmdlXJT0n6fawggEAAAx1Nuu0zBpyv3SigwAAAIwmaEvLnZKeM7MmSaaBsS20soSsvLxcHR0dob+PmYX6+mVlZWpvbw/1PQAAk1+gosXd02b2XUm/rIGi5TZ3PxRmMEgdHR1y96hjjFvYRREAID+M2j1kZu/J3P6SpPMltUk6KOmCzDkAAICcGKul5RZJN0q65zTXXNJHJjwRAADAaYy1uNyNmbsr3L176DUzKw4tFQAAwAhBZw99L+A5AACAUIy1YeIvSJonaaqZ/aIGBuFK0kxJ00LOBgAAkDXWmJZqSf9N0nxJ9w45f1TS/w4pEwAAwCnGGtPyVUlfNbNPuvs3c5QJAADgFEHXafmmmV0jabGk4iHn/zCsYAAAAEMF3TCxUdJ1kuo1MK7lU5IWhJgLAABgmKCzhy539+sldbj7H0j6NUkXhhcLAABguKBFy/HM7dtmdoGkXkkXjfYEM3vYzN4ws71nuP4eM/sXM+sxs1uDRwYAAPkoaNHymJnNkvTHkp6VdEDSX4/xnL+QdNUo19slfVbSnwTMAAAA8ljQgbhfzNz9ppk9JqnY3Y+M8ZxdZrZwlOtvSHojM8AXAABgVEFbWrLcvUfSr5jZkyHkAQAAOK2xdnn+iJn9yMyOmdlfmdmlZtYi6UuS7s9NRMnMbjSzFjNrOXz4cK7eFgAAxMhYLS33aGCX53Ml/b2kpyX9pbt/wN2/FXa4Qe7+gLsvcfclc+bMydXbAgCAGBlrTIu7+3cz9x8xs8Pu/qchZwIQc+Xl5ero6Aj9fcxs7AeNQ1lZmdrb20N9DwATZ6yiZZaZfWLIsQ09Hq21xczSkq6UNNvM2iTdIakw87zGzGaMLRrYfLHfzH5X0qXu/tY7+pcAyJmOjg65e9Qxxi3sogjRoKievMYqWnZK+tgZjl3SGYsWd68Z7YXd/ZAGNmIEAGDCUFRPXmNtmHhDroIAAACM5qynPAMAAEQh0OJyiIbfMVP6QmnUMcbN75gZdQQAwCRA0RJj9gdvTZp+Wf9C1CkAAEkXqHvIzD5lZjMy9z9vZt8ys18KNxoAAMDPBR3T8vvuftTMrpBULemryuGKuAAAAEGLlr7M7TWS7nf3b0s6J5xIAAAApwpatPzEzP5c0rWSnjCzorN4LgAAwLgFLTyulbRN0lXu3impXNL/Ci0VAADACKPOHjKz8iGH3x1yrkcDS/ADAADkxFhTnndrYLl+k1QhqSNzf5akVyVdFGo6AACAjFG7h9z9Ind/lwa6hj7m7rPd/VxJH9Uo+w4BAABMtKBjWn7Z3Z8YPHD3rZI+HE4kAACAUwVdEfdnZvZ5SX+lge6i35b0ZmipAAAARgja0lIjaY6kf5D0iKS5mXMAAAA5Eailxd3bJd0cchachplFHWHcysrKoo4AAJgExpry/B0NdAedlruvnPBEyMrFZolmNik2ZQQATH5jtbT8Seb2E5J+QQNjWqSBrqEDIWUCAAA4xahFi7vvlCQz+6K7f2jIpe+Y2a5QkwEAAAwRdCDuHDN71+CBmV2kgYG5AAAAORF0yvP/lPRdM/tx5nihpBtDSQQAwDj4HTOlL5RGHWPc/I6ZUUeIHQs6CDOzs/N7MocvuntPaKlGsWTJEm9pYdujicJAXLwjk+AXQtYXjkSdABNssnyvTZZ/x1jMbLe7Lwny2DFbWsxsrqSbJC3WwEyiFyT9maQ3xhMSQHLZH7w1Kb5MzUz+hahTAAhq1DEtZvZBST/IHH5NP5899P3MNQAAgJwYq6XlHkkfd/fnhpz7tpn9g6Q/l/SroSUDAAAYYqzZQzNHFCySJHf/V0kzwokEAABwqrGKFjOzU9ZgN7PyAM8FAACYMGMVHl+WtN3MPmxmMzI/V0ramrkGAACQE2OtiPuAmb0m6YsaPnvoj9z9OznIBwAAICnAlGd3f0zSYznIAgAAcEZjTXn+fGb8ypmuf8TMPjrxsQAAAIYbq6XleQ1sjtgt6VlJhyUVS7pY0vslPSVpY6gJAQAANPaYlm9rYF2WiyV9UNL5kt7SwCJzN7r78fAjAgAABNww0d1fkvRSyFkAAADOiLVWAABAIlC0AACARAjUPQQAQJKYWdQRxq2s7JQF6fNeoKLFzC6RdL+k89y9yswuk7TS3f8o1HQAYotfCogrdw/9PcwsJ++D4YJ2Dz0o6XZJvZLk7nskfTqsUADizd1D/8nF+7S3t0f8XxLA2QhatExz9++POHdyosMAAACcSdCi5WdmtkgDew/JzH5T0uuhpQIAABgh6EDcmyQ9IOk9ZvYTSS9L+u3QUgEAAIwQdHG5H0v6dTMrkZRy96PhxgIAABguUPeQmfWZ2ZckvT1YsJjZs6EmAwAAGCLomJbWzGO3D9n1OfnzHQEAQGIELVpOuvt6DUx9/mcz+4Ayg3IBAAByIehAXJMkd/9bM2uVlJZUEVoqAACAEYIWLf998I67t5rZFZI+Hk4kAACAU41atJjZR9x9h6QFZrZgxOVj4cXCO/FOl1U/2+exdDUAIApjtbR8WNIOSR87zTWX9K0JT4R3jGICADCZjVq0uPsdmdsbchMHAADg9IKu03Kzmc20AQ+Z2bNmtjzscAAAAIOCTnle7e5vSVouaa6kGyR9KbRUAAAAIwQtWgZHal4t6Svu/kOxuBwAAMihoEXLbjPbroGiZZuZzZDUH14sAACA4YKu01Ir6f2Sfuzub5vZuRroIgIAAMiJoLs890t6dsjxm5LeDCsUAADASEG7hwAAACJF0QIAABIh6DotfxnkHAAAQFiCtrQsHnpgZgWSPjDaE8zsYTN7w8z2nuG6mdn/M7P9ZrbHzH4pYBYAAJCHRi1azOx2Mzsq6TIzeyvzc1TSG5K+PcZr/4Wkq0a5vkLSxZmfGyXdHzg1AADIO6MWLe5+p7vPkPTH7j4z8zPD3c9199vHeO4uSe2jPOQ3JH3NBzwtaZaZnX/W/wIAAJAXgnYPfd/MSgcPzGyWmX18nO89T9LBIcdtmXMAAACnCFq03OHuRwYP3L1T0h3jfO/TbQPgp32g2Y1m1mJmLYcPHx7n2wIAgCQKWrSc7nFBV9M9kzZJFw45ni/ptdM90N0fcPcl7r5kzpw543xbAACQREGLlhYzu9fMFpnZu8zsy5J2j/O9H5V0fWYW0X+UdMTdXx/nawIAgEkqaGtJvaTfl/Q3mePtkj4/2hPMLC3pSkmzzaxNA91JhZLk7o2SntDABoz7Jb0t9jICAACjCLr3UJek3zOz6e5+LOBzasa47pJuCvJaAAAAQVfEvdzMXpD0Qub4fWa2OdRkAAAAQwQd0/JlSdXK7Ozs7j+U9KGwQgEAAIwUeMNEdz844lTfBGcBAAA4o6ADcQ+a2eWS3MzOkfRZSfvCiwUAADBc0JaWOg0Mmp2ngfVV3i8G0QIAgBwataXFzO5y99skLXX3VTnKBAAAcIqxWlquNrNCSaNujggAABC2sca0/KOkn0kqMbO3NLBfkA/euvvMkPMBAABIGrul5fPuXirpcXef6e4zht7mIiAAAIA0dtHyL5nbt8IOAgAAMJqxuofOMbP/KulyM/vEyIvu/q1wYgEAAAw3VtFSJ2mVpFmSPjbimkuiaAEAADkxatHi7s2Sms2sxd235CgTAADAKUYd02Jm6yXJ3beY2adGXNsYZjAAAIChxhqI++kh90eu1XLVBGcBAAA4o7GKFjvD/dMdAwAAhGasosXPcP90xwAAAKEZq2h5n5m9ZWZHJV2WuT94/N4c5ENI0um0qqqqVFBQoKqqKqXT6agjAQAwqrFmDxXkKghyJ51Oq6GhQVu2bNEVV1yh5uZm1dbWSpJqamoiTgcAwOmN1dKCSWjDhg3asmWLli5dqsLCQi1dulRbtmzRhg0boo4GAMAZmXuyhqYsWbLEW1paoo6RaAUFBeru7lZhYWH2XG9vr4qLi9XX1xdhMuDnzExJ+35C/uDzOXHMbLe7LwnyWFpa8lBlZaWam5uHnWtublZlZWVEiQAAGBtFSx5qaGhQbW2tmpqa1Nvbq6amJtXW1qqhoSHqaAAQaxUVFTIbWPHDzFRRURFxovwy1t5DmIQGB9vW19dr3759qqys1IYNGxiECwCjqKio0MGDB4edO3jwoCoqKvTqq69GlCq/MKYFQKxUV1frySeflLvLzLRs2TJt27Yt6lhAtoXldJL2uzROGNMCIJGqq6u1ffv27C8Ad9f27dtVXV0dcTJMZmYW6Ge8r4Hxo2gBEBvbt2+XJJWVlQ27HTwPhMHdA/2M9zUwfoxpAZATZ/OXZkdHx7Dbs3k+vxyAyYuiBUBOBCkmzEyXXXaZ+vr6soPECwoKtGfPHooRAHQPAYiXPXv2aNGiRfrpT3+qRYsWac+ePVFHAhATtLQAiI3BVUYfffRRzZkzZ9h5AKClBUCsTJ06ddRjAPmLlhYAsXHppZdq6tSp2r17d3adlsWLF+v48eNRRwMQA7S0AIiNefPmqaWlRXV1ders7FRdXZ1aWlo0b968qKMBiAGKFgCxsXPnTq1atUq7du1SeXm5du3apVWrVmnnzp1RRwOyFi9erFdeeUWLFy+OOkreoWgBEBs9PT0qKSnR/v371d/fr/3796ukpEQ9PT1RRwOyWltbtWDBArW2tkYdJe9QtACIjVQqpYceekgbN25UV1eXNm7cqIceekipFF9VAChaAMTImaY2M+UZgETRAiBG+vr6dOWVV+rWW29VSUmJbr31Vl155ZXq6+uLOhqAGKBoARAbU6ZM0TPPPKMFCxYolUppwYIFeuaZZzRlCqszAKBoARAjRUVF6urq0ooVK9Te3q4VK1aoq6tLRUVFUUcDssrKyrRnz57sLuTIHUvaJmRLlizxlpaWqGMACIGZaeXKldq2bZt6enpUVFSk6upqPfroo2yYiMiNNraKz+c7Z2a73X1JkMfS0gIgVtatW6fu7m65u7q7u7Vu3bqoIwGICYoWALExf/58XX/99WpqalJvb6+ampp0/fXXa/78+VFHAxADFC0AYuPuu+9WX1+fVq9eraKiIq1evVp9fX26++67o44GIAYoWgDERk1Nja677jq9/vrrcne9/vrruu6661RTUxN1NAAxQNECIDbS6bQef/xxbd26VSdOnNDWrVv1+OOPK51ORx0NyJo6darMTFOnTo06St5h9hCA2KiqqtLFF1+srVu3ZmcPrVixQi+99JL27t0bdTzkucHZQ6lUSv39/dlbidlD43E2s4coWgDERiqVkruroKBAfX192Vszy/5yAKLClOdwMOUZQCINfvHPnj1bZqbZs2cPOw8gv1G0AIiVVCql9evX69ixY1q/fj07PAPIonsIQGyYmWbNmqXOzs7sucHjpH1XYfKheygcdA8BSKzOzk4VFxdLkoqLi4cVMADyG0ULgNjp7u4edgvEycKFC7V//34tXLgw6ih5h/3eAQA4CwcOHNC73/3uqGPkJVpaAMTKlClTtHDhQpmZFi5cqClT+NsKwACKFgCxcvLkSa1YsUIdHR1asWKFTp48GXUkADHB7CEAsWFm2QXlBg0eJ+27CpMPs4fCwewhAIlUXl6u/v5+nXfeeTIznXfeeerv71d5eXnU0QBJUlFR0ajHCFeoRYuZXWVm/2Zm+83s905zfYGZ/ZOZ7TGz75rZ/DDzAIi3++67TzNmzFB7e7vcXe3t7ZoxY4buu+++qKMBmjJlinp6eoZtmNjT08O4qxwKrWgxswJJfyZphaRLJdWY2aUjHvYnkr7m7pdJ+kNJd4aVB0D81dTUqLGxUZdccolSqZQuueQSNTY2qqamJupogGbOnClJOn78uNxdx48fH3Ye4QttTIuZ/ZqkL7h7deb4dkly9zuHPKZVUrW7t9lAZ+ERdx/1f33GtAAAomBmKiwsVG9vb/bc4DFjWt65uIxpmSfp4JDjtsy5oX4o6ZOZ+/9F0gwzOzfETAAAvGMnT54cNuaK2W25FWbRcrph1iNL0VslfdjMnpP0YUk/kXTKJ8DMbjSzFjNrOXz48MQnBQAggMEWlZG3yI0wi5Y2SRcOOZ4v6bWhD3D319z9E+7+i5IaMueOjHwhd3/A3Ze4+5I5c+aEGBkAgNGZmVKp1KhToBGOMIuWH0i62MwuMrNzJH1a0qNDH2Bms81sMMPtkh4OMQ+ABEin06qqqlJBQYGqqqqUTqejjgRkpVIpHTp0SP39/Tp06JBSKVYOyaXQ5mm5+0kzWydpm6QCSQ+7e6uZ/aGkFnd/VNKVku40M5e0S9JNYeUBEH/pdFq1tbXZWRmtra2qra2VJGYQIRb6+/uVSqWG3SJ3WBEXQGxMnz5dXV1dWrt2re68807dfvvtuv/++1VSUqJjx45FHQ95jhVxw3E2s4dYEQdAbHR1denGG2/U5s2bJUmbN29WX1+fHnjggYiTAT83Y8YMdXV1qaSkREePHo06Tl6hMw5ArHz0ox8d9RiI0qJFi1RRUSFJqqio0KJFiyJOlF8oWgDEyqpVq9TU1KTe3l41NTVp1apVUUcCsl5++WWtXr1aR48e1erVq/Xyyy9HHSmvMKYFQGxUV1dr+/btKisr05EjR1RaWqqOjg4tX75c27Ztizoe8tzgiriS1NvbO+x+0n6XxklcVsQFgLOybds2LV++XJ2dnerv71dnZycFC2Jj+fLl6u3t1fTp05VKpTR9+nT19vZq+fLlUUfLG7S0AAAQUHV1tZ588km5u8xMy5Yto6geJ1paAADBunJzAAAG70lEQVQIQUtLy7Al/PkjOrcoWgAACODcc89Ve3u7Fi9erFdeeUWLFy9We3u7zj2XfX5zhXVaAAAIYLBg2bt3ryRp7969qqqqUmtra8TJ8gctLQAABPTEE0+MeoxwUbQAABDQ1VdfPeoxwkXRAgBAAOXl5WptbVVVVZVeffXVbNdQeXl51NHyBmNaAAAI4M0339T06dPV2tqqBQsWSJJKSkr05ptvRpwsf9DSAgBAAOl0WnPnztWOHTt04sQJ7dixQ3PnzlU6nY46Wt5gcTkAAAKoqqrSpk2btHTp0uy5pqYm1dfXZ2cU4eydzeJyFC0AAARQUFCg7u7u7J5D0sC+Q8XFxerr64swWbKxIi6AxEqn06qqqlJBQYGqqqpoekdsVFZWqrm5edi55uZmVVZWRpQo/1C0AIiNdDqthoYGbdq0Sd3d3dq0aZMaGhooXBALDQ0Nqq2tVVNTk3p7e9XU1KTa2lo1NDREHS1v0D0EIDYYM4C4S6fT2rBhg/bt26fKyko1NDSopqYm6liJxpgWAInEmAEg/zCmBUAiMWYAwGgoWgDEBmMGAIyGFXEBxMbg2ID6+vrsmIENGzYwZgCAJMa0AACACDGmBUBisU4LgDOhewhAbAyu07JlyxZdccUVam5uVm1trSTRRQSA7iEA8cE6LUD+YZ0WAInEOi1A/mFMC4BEYp0WAKOhaAEQG6zTAmA0DMQFEBus0wJgNIxpAQAAkWFMCwAAmHQoWgAAQCJQtAAAgESgaAEAAIlA0QIAABKBogUAgIDY0DNarNMCAEAAbOgZPdZpAQAgADb0DAcbJgIAMMHY0DMcLC4HAMAEY0PP6FG0AAAQABt6Ro+BuAAABMCGntFjTAsAAIgMY1oAAMCkQ9ECAAASgaIFAAAkAkULAABIBIoWAACQCBQtAAAgEShaAABAIlC0AACARKBoAQAAiUDRAgAAEiFxy/ib2WFJr0SdYxKZLelnUYcAToPPJuKMz+fEWeDuc4I8MHFFCyaWmbUE3fMByCU+m4gzPp/RoHsIAAAkAkULAABIBIoWPBB1AOAM+Gwizvh8RoAxLQAAIBFoaQEAAIlA0ZKnzOwqM/s3M9tvZr8XdR5gkJk9bGZvmNneqLMAQ5nZhWbWZGb7zKzVzG6OOlO+oXsoD5lZgaQfSVomqU3SDyTVuPsLkQYDJJnZhyQdk/Q1d6+KOg8wyMzOl3S+uz9rZjMk7Zb0cb47c4eWlvz0K5L2u/uP3f2EpL+W9BsRZwIkSe6+S1J71DmAkdz9dXd/NnP/qKR9kuZFmyq/ULTkp3mSDg45bhP/xwOAwMxsoaRflPRMtEnyC0VLfrLTnKOfEAACMLPpkr4p6Xfd/a2o8+QTipb81CbpwiHH8yW9FlEWAEgMMyvUQMHydXf/VtR58g1FS376gaSLzewiMztH0qclPRpxJgCINTMzSVsk7XP3e6POk48oWvKQu5+UtE7SNg0MJPtbd2+NNhUwwMzSkv5F0n8wszYzq406E5DxQUm/I+kjZvavmZ+row6VT5jyDAAAEoGWFgAAkAgULQAAIBEoWgAAQCJQtAAAgESgaAEAAIkwJeoAACYfM+uT9LykQkknJX1V0v919/5IgwFINIoWAGE47u7vlyQzmyvpG5JKJd0x3hc2swJ37xvv6wBIHrqHAITK3d+QdKOkdTagwMz+2Mx+YGZ7zOx/SJKZpcxss5m1mtljZvaEmf1m5toBM/s/ZtYs6VNmtsjM/tHMdpvZP5vZezKPm2Nm38y89g/M7IOR/cMBTDhaWgCEzt1/bGYpSXMl/YakI+7+y2ZWJOn/m9l2SR+QtFDSezOP2yfp4SEv0+3uV0iSmf2TpDp3f8nMflXSZkkfkfSnkr7s7s1mVqGBVZ8rc/KPBBA6ihYAuTK4u/hySZcNtqJooNvoYklXSPq7zLiXQ2bWNOL5fyNld9i9XNLfDWwFI0kqytz+uqRLh5yfaWYz3P3oRP9jAOQeRQuA0JnZuyT1SXpDA8VLvbtvG/GYa8Z4ma7MbUpS5+CYmRFSkn7N3Y+PMzKAGGJMC4BQmdkcSY2S7vOBzc62SVprZoWZ65eYWYmkZkmfzIxtOU/Slad7PXd/S9LLZvapzPPNzN6XubxdA5uBDr736QobAAlFSwuAMEw1s3/Vz6c8/6WkezPXHtLA2JVnbaAf57Ckj0v6pqT/LGmvpB9JekbSkTO8/ipJ95vZ5zPv8deSfijps5L+zMz2aOD7bZekuon+xwGIBrs8A4gNM5vu7sfM7FxJ35f0QXc/FHUuAPFASwuAOHnMzGZJOkfSFylYAAxFSwsAAEgEBuICAIBEoGgBAACJQNECAAASgaIFAAAkAkULAABIBIoWAACQCP8OBgoZC7IaOOMAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 648x432 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_to_plot = [degreesEff[0], degreesEff[1], degreesEff[2]]\n",
    "print(np.median(degreesEff[0]),np.median(degreesEff[1]),np.median(degreesEff[2]))\n",
    "print(np.mean(degreesEff[0]),np.mean(degreesEff[1]),np.mean(degreesEff[2]))\n",
    "fig = plt.figure(1, figsize=(9, 6))\n",
    "ax = fig.add_subplot(111)\n",
    "bp = ax.boxplot(data_to_plot)\n",
    "ax.set_xticklabels(['0', '1', '2'])\n",
    "ax.set_xlabel(\"Degree\")\n",
    "ax.set_ylabel(\"Effect size (Odds Ratio)\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "277\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = pd.read_csv(\"/Users/jiayao/Work/NB_proposal/dat/network/9606.protein.links.v11.0.txt\", delimiter=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcription_Factor</th>\n",
       "      <th>Target_Gene</th>\n",
       "      <th>Enhancer_Region</th>\n",
       "      <th>Edge_Weight</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BARHL2</td>\n",
       "      <td>SHC1</td>\n",
       "      <td>chr1:154869072-154870071</td>\n",
       "      <td>0.284806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MNT</td>\n",
       "      <td>SHC1</td>\n",
       "      <td>chr1:154716711-154717710</td>\n",
       "      <td>0.312268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NR2F2</td>\n",
       "      <td>SHC1</td>\n",
       "      <td>chr1:154722868-154723867</td>\n",
       "      <td>0.122871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>PLAGL1</td>\n",
       "      <td>SHC1</td>\n",
       "      <td>chr1:154548318-154549317</td>\n",
       "      <td>0.114818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>HMX3</td>\n",
       "      <td>CDCA2</td>\n",
       "      <td>chr8:25045544-25046543</td>\n",
       "      <td>0.289291</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Transcription_Factor Target_Gene           Enhancer_Region  Edge_Weight\n",
       "0                 BARHL2        SHC1  chr1:154869072-154870071     0.284806\n",
       "2                    MNT        SHC1  chr1:154716711-154717710     0.312268\n",
       "6                  NR2F2        SHC1  chr1:154722868-154723867     0.122871\n",
       "68                PLAGL1        SHC1  chr1:154548318-154549317     0.114818\n",
       "194                 HMX3       CDCA2    chr8:25045544-25046543     0.289291"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Brain_Reg_Net2_csv_trim.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "#edges1 = edges[edges[\"combined_score\"]>900]\n",
    "#edges1 = edges1[(edges1[\"protein1\"].isin(set(GeneDict.keys())))&(edges[\"protein2\"].isin(set(GeneDict.keys())))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'edges1' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-107-6af1505d50f6>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0medges\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0medges1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'edges1' is not defined"
     ]
    }
   ],
   "source": [
    "print(edges.shape, edges1.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.hist(Brain_Reg_Net2_csv[\"Edge_Weight\"].values, bins=500)\n",
    "plt.xlim((0,2))\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16437"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for K in list(GeneDict.keys()):\n",
    "    if len(GeneDict[K].SNPs) == 0:\n",
    "        GeneDict.pop(K)\n",
    "len(GeneDict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "SymbolGeneDict = {}\n",
    "for k,v in GeneDict.items():\n",
    "    SymbolGeneDict[v.GeneName] = v"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "278000"
     ]
    }
   ],
   "source": [
    "graph = {}\n",
    "count = 0\n",
    "for i, row in Brain_Reg_Net2_csv_trim.iterrows():\n",
    "    g1, g2 = row[\"Transcription_Factor\"], row[\"Target_Gene\"]\n",
    "    #print(g1, g2)\n",
    "    if g1 not in SymbolGeneDict or g2 not in SymbolGeneDict:\n",
    "        continue\n",
    "    if g1 not in graph:\n",
    "        graph[g1] = []\n",
    "    if g2 not in graph:\n",
    "        graph[g2] = []\n",
    "    graph[g1].append(g2)\n",
    "    graph[g2].append(g1)\n",
    "    if count % 1000 == 0:\n",
    "        sys.stdout.write(\"\\r%d\"%count)\n",
    "    count += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(372661, 4)"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Brain_Reg_Net2_csv_trim.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11297\n"
     ]
    }
   ],
   "source": [
    "print(len(graph.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "277\n",
      "2102\n",
      "8947\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "degreesNode[0] = [gene.GeneName for gene in degreesNode[0]]\n",
    "PreNodes = set(degreesNode[0])\n",
    "print(len(PreNodes))\n",
    "for i in range(1,11,1):\n",
    "    degreesEff[i] = []\n",
    "    degreesNode[i] = []\n",
    "    for gene in degreesNode[i-1]:\n",
    "        symbol = gene\n",
    "        if symbol in graph:\n",
    "            for node in graph[symbol]:\n",
    "                if node not in PreNodes:\n",
    "                    degreesNode[i].append(node)\n",
    "    degreesNode[i] = list(set(degreesNode[i]))\n",
    "    for symbol in degreesNode[i]:\n",
    "        degreesEff[i].append(SymbolGeneDict[symbol].max_eff)\n",
    "    print(len(degreesNode[i]))\n",
    "    PreNodes = PreNodes.union(set(degreesNode[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3644"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "2878 + 538 + 227 + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.085345 1.062365 1.06332\n",
      "1.0926595222929938 1.0697580970504281 1.0694282787526546\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAi0AAAF3CAYAAABpFHt+AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3X98lvV97/H35w4xgQAhEXAKBizV00hKu5VtZ9bTSs8goi3raVdtxuY5koMnVFJ3rAfnSXd068Dqpj07OMxQ7NqtvfernbUqAx0pLKfTNuiKRFzlUVFSRalJEAIJIfnsj9xJkwDJhcl1X9eV+/V8PPK47+u67h9v7N07n3x/mrsLAAAg7lJRBwAAAAiCogUAACQCRQsAAEgEihYAAJAIFC0AACARKFoAAEAiULQAAIBEoGgBAACJQNECAAASgaIFAAAkwqSoA5yrmTNn+vz586OOAQAAxsHu3bt/5u6zgjw2cUXL/Pnz1dTUFHUMAAAwDszs1aCPpXsIAAAkAkULAABIBIoWAACQCBQtAAAgEShaAABAIlC0AACARKBoAQAAiUDRAgAAEoGiBQAAJAJFCwAAAaXTaVVUVCgvL08VFRVKp9NRR8opiVvGHwCAKKTTadXV1WnLli268sor1djYqOrqaklSVVVVxOlyg7l71BnOyeLFi529hwAA2VZRUaGNGzdqyZIlA+caGhpUW1urvXv3Rpgs2cxst7svDvRYihYAAEaXl5enzs5O5efnD5zr7u5WYWGhenp6IkyWbOdStDCmBQCAAMrLy9XY2DjkXGNjo8rLyyNKlHsoWgAACKCurk7V1dVqaGhQd3e3GhoaVF1drbq6uqij5QwG4gIAEED/YNva2lrt27dP5eXlWr9+PYNws4gxLQAAIDKMaQEAABMORQsAAAGxuFy0KFoAAAggnU7rlltuUUdHh9xdHR0duuWWWyhcsoiiBQCAANatW6e8vDw98sgj6urq0iOPPKK8vDytW7cu6mg5g6IFAIAAWlpadOONN6q2tlaFhYWqra3VjTfeqJaWlqij5QymPAMAENBXv/pVffOb3xzYe+i3fuu3oo6UUyhaAAAIYNKkSTp69KhWrVql1157TWVlZTp69KgmTeJXabbwXxoAgABOnTqlU6dOqaOjQ5J04MCBaAPlIMa0AIiVyspKpVIpmZlSqZQqKyujjgQgJihaAMRGZWWltm/frpqaGrW3t6umpkbbt2+ncEGslJSUKJVKqaSkJOooOYfuIQCx8dRTT2nNmjXatGmTJA3c1tfXRxkLGHDeeeepuLhY7e3tKi4uVkdHh06ePBl1rJxBSwuA2HB3vfTSS0O6h1566SUlbY80TFwnT57UgQMH5O46cOAABUuWUbQAiJWGhoYh3UMNDQ1RRwIQE+zyDCA2zOys15L2XYWJh89nONjlGQAATDgULQBi54orrtDrr7+uK664IuoowBBmNtDiMvg+soOiBUDsPPPMM7rooov0zDPPRB0FGMLdNXv2bO3bt0+zZ8+mWyjLmPIMIHZ6e3uH3AJx8uabb6q8vDzqGDmJlhYAsbNmzRq1t7drzZo1UUcBTjO4ewjZxewhALHB7AzEWf8YlsGfxf5jPp/vHrOHACRSQUGBiouLh5wrLi5WQUFBRImAocxM9913nzo6OnTffffR2pJlFC0AYmP16tU6cuTIkHNHjhzR6tWrI0oEDFVWVqYvfOELKioq0he+8AWVlZVFHSmnULQAiI2dO3dKOn3MQP95IEpFRUU6cODAkHMHDhxQUVFRNIFyEEULgNh44YUXtGLFCvX29srd1dvbqxUrVuiFF16IOhqg48ePn9N5jD+KFgCxsmXLlhGPgaj0D7ZNpVJDbhmEmz0ULQBiZdmyZaqoqFBeXp4qKiq0bNmyqCMBA/Lz8/X000/r5MmTevrpp5Wfnx91pJxC0QIgNi6++GI9//zzKi4uVktLi4qLi/X888/r4osvjjoaIEnq7u5WVVWVzjvvPFVVVam7uzvqSDmFFXEBxMb06dN1ySWX6Pvf/74uuugiSdIll1yiKVOmRJwM+LnDhw8PuUX2ULQAiI19+/aps7NzSJN7d3e3CgsLI0wF9OlfSG74NhOs1ZI9dA8BiI3y8nJdd911KiwslJmpsLBQ1113Hfu8IBbONuCWgbjZQ9ECIDbmzJmjRx99VKtWrVJ7e7tWrVqlRx99VHPmzIk6GiBJysvLG2gJzM/PV15eXsSJcgtFS46qra0d8tdsbW1t1JEA7dy5UytXrtSuXbtUWlqqXbt2aeXKlSwuh9jo6ekZ0j3U09MTcaLcwpiWHFRbW6v6+nrdc889qqmpUX19vW6//XZJ0saNGyNOh1zW1dWlzZs3Dxl4e/z4cX3jG9+IMBWAuKClJQc99NBDuueee3TrrbdqypQpuvXWW3XPPffooYceijoaclxBQYHq6+uHnKuvr2fDRMTKtddeq8OHD+vaa6+NOkrOsaQNIFq8eLE3NTVFHSPRzEwdHR2n/TVbVFTEgDJE6mytgDU1NbQCInJmNjCDaPA5d+e7cwzMbLe7Lw7yWFpachB/zSKuNm7cqAsvvHDILroXXnghBQti5YILLhhyi+yhaMlBq1ev1u233677779fx48f1/3336/bb79dq1evjjoaclxlZaUOHjyoNWvWqL29XWvWrNHBgwdVWVkZdTRgYKaQmSmVSg2sz8IMouyheyhH1dbW6qGHHlJXV5cKCgq0evVq/ppF5FKplObMmaOf/vSncneZ2cBx/4wNICr9hcrgz2IqlRqy4BzOXSy6h8zsETN7y8z2nuX6SjPbk/n5vpl9IKwsON3GjRvV2dkpd1dnZycFC2LB3dXS0qKamhq1t7erpqZGLS0tjBdALEyZMkW9vb1Ddnnu7e1lm4ksCrN76C8kXT3C9VckfdTdF0n6kqTNIWYBkBALFy7Upk2bVFxcrE2bNmnhwoVRRwIkSR0dHZKk2bNnK5VKafbs2UPOI3yhFS3uvktS6wjXv+/ubZnDZyTNDSsLgORobm7W5z73OR05ckSf+9zn1NzcHHUkYICZ6dChQ+rt7dWhQ4fYdyjL4jIQt1rS1qhDAIiWmWnu3Lmqr6/XjBkzVF9fr7lz5/KLAbHh7lqxYoUOHz6sFStW0HWZZZEXLWa2RH1Fy+0jPOYmM2sysya2AgcmrqVLl55xTMvSpUujjgYMmDNnjvLz89kTKwKhzh4ys/mSHnf3irNcXyTpHyQtd/cfB3lNZg8BE9uiRYv0wgsvDBy///3v1549eyJMBPQZqcWPFpd3Lxazh0ZjZmWSvi3pd4IWLAAmtnQ6rWPHjmnHjh06efKkduzYoWPHjimdTkcdDUAMhNbSYmZpSVdJminpTUl3SsqXJHevN7OHJX1a0quZp5wKUmnR0gJMXBUVFbr00ku1devWgTWEli9frpdffll7955x9QQga2hpCce5tLSwuByA2EilUkqlUrr33nsH9h5at26dent7WbwLkaNoCQdFC4BESqVSuvzyy7V///6Blpb3vve9evHFFylaEDmKlnAkYkwLAAzn7mpubtaUKVNkZpoyZYqam5v5hQBAEkULgJgxM7W1tcnd1dbWxhotAAZQtEwgZpaVHyBMLN4F4GwmRR0A4+fdfLmbGb8UECvz5s3Ttm3bNGvWLBUUFGjevHl69dVXR38igAmPlhYAsXLw4EFt2LBBHR0d2rBhgw4ePBh1JAAxweyhHEdLC+IkPz9feXl56u3tVXd3t/Lz85VKpdTT06Pu7u6o4yHHMXsoHOcye4juIQBZEXQ81KlTpwbuDy5Ugj6fXx7AxEX3EICscPdAP2vXrlVBQYEkqaCgQGvXrg38XAoWvBvjMQmBSQzZQfdQjqN7CHHFZxNxk0qlzviZNDMWPxwDFpcDAGCc3Xzzzed0HuOPMS0AAASwceNGSdJDDz00sM3E6tWrB84jfHQP5Tia4BFXfDYRZ3w+xw/dQwAAYMKhaAEAAIlA0QIAABKBogUAACQCRQsAAEgEihYAAJAIFC0AACARKFoAAEAiULQAAIBEoGgBAACJQNECAAASgaIFAAAkAkULAABIBIoWAACQCBQtAAAgEShaAABAIlC0AACARKBoAQAAiUDRAgAAEoGiBQAAJAJFCwAASASKFgAAkAgULQAAIBEoWgAAQCJQtAAAgESgaAEAAIkwKegDzaxE0kWSTkg64O69oaUCAAAYZsSixcyKJd0sqUrSeZIOSyqUdIGZPSNpk7s3hJ4SAADkvNFaWv5e0tcl/Sd3bx98wcw+JOl3zOw97r4lrIAAAADSKEWLuy8d4dpuSbvHPREAAMAZnMuYlhWSPpI53Onu3w0nEgAAwOkCzR4ysy9LukXSi5mfz5vZ3WEGAwAAGCxoS8s1kj7YP2PIzL4m6XlJd4QVDAAAYLBzWadlxqD7xeMdBAAAYCRBW1rulvS8mTVIMvWNbaGVBQAAZE2gosXd02b2PUm/rL6i5XZ3PxRmMEilpaVqa2sL/X3MLNTXLykpUWtra6jvAQCY+EZbXO597v6Smf1S5lRL5vYiM7vI3Z8LN15ua2trk7tHHWPMwi6KAAC5YbSWllsl3STpvjNcc0kfG/dEAAAAZzDa4nI3Ze4ud/fOwdfMrDC0VAAAAMMEnT30/YDnAAAAQjHamJZfkDRH0mQz+0X1DcKVpOmSpoScDQAAYMBoY1oqJf03SXMl3T/o/FFJ/zukTAAAAKcZbUzL1yR9zcw+7e7fylImAACA0wRdp+VbZnatpIWSCged/8OwggEAAAwWdMPEeknXS6pV37iWz0iaF2IuAACAIYLOHrrC3W+Q1ObufyDp1yRdHF4sAACAoYIWLScyt8fN7CJJ3ZIuGekJZvaImb1lZnvPcv19ZvYvZtZlZrcFjwwAAHJR0KLlcTObIemPJT0n6YCkvx7lOX8h6eoRrrdK+rykPwmYAQAA5LCgA3G/lLn7LTN7XFKhux8Z5Tm7zGz+CNffkvRWZoAvAADAiIK2tAxw9y5Jv2JmT4WQBwAA4IxGLFrM7GNm9mMzO2Zmf2Vml5tZk6QvS3owOxElM7vJzJrMrOnw4cPZelsAABAjo7W03Ke+XZ7Pl/T3kp6R9Jfu/iF3/3bY4fq5+2Z3X+zui2fNmpWttwUAADEy2pgWd/fvZe4/amaH3f1PQ84EIOZKS0vV1tYW+vuY2egPGoOSkhK1traG+h4Axs9oRcsMM/vUoGMbfDxSa4uZpSVdJWmmmbVIulNSfuZ59ZnNGJvUt/lir5n9rqTL3f2dd/UvAZA1bW1tcveoY4xZ2EURokFRPXGNVrTslPSJsxy7pLMWLe5eNdILu/sh9W3ECADAuKGonrhG2zDxxmwFAQAAGMk5T3kGAACIAkULAABIBIoWAACQCIGKFjP7jJlNy9z/opl928x+KdxoAAAAPxdo7yFJv+/uf2dmV0qqVN8mhw9K+tXQkkF+53TpruKoY4yZ3zk96ggAgAkgaNHSk7m9VtKD7v4dM7srnEjoZ3/wzoSZtud3RZ0CAJB0Qce0/NTM/lzSdZKeNLOCc3guAADAmAUtPK6TtE3S1e7eLqlU0v8KLRUAAMAwI3YPmVnpoMPvDTrXpb4l+AEAALJitDEtu9W3XL9JKpPUlrk/Q9Jrki4JNR0AAEDGiN1D7n6Ju79HfV1Dn3D3me5+vqSPa4R9hwAAAMZb0DEtv+zuT/YfuPtWSR8NJxIAAMDpgk55/pmZfVHSX6mvu+i3Jb0dWioAAIBhgra0VEmaJekfJD0qaXbmHAAAQFYEamlx91ZJt4ScBQAA4KxGm/L8XfV1B52Ru68Y90QYwsyijjBmJSUlUUcAAEwAo7W0/Enm9lOSfkF9Y1qkvq6hAyFlQkY2lvA3swmxVQAAYOIbsWhx952SZGZfcvePDLr0XTPbFWoyAACAQYIOxJ1lZu/pPzCzS9Q3MBcAACArgk55/p+SvmdmP8kcz5d0UyiJAAAYA79zunRXcdQxxszvnB51hNixoOMZMjs7vy9z+JK7d4WWagSLFy/2pia2PRovjGnBuzIBfiEMuOtI1AkwzibK99pE+XeMxsx2u/viII8dtaXFzGZLulnSQvXNJHpR0p9JemssIQEkl/3BOxPiy9TM5HdFnQJAUCOOaTGzD0v6Yebw6/r57KEfZK4BAABkxWgtLfdJ+qS7Pz/o3HfM7B8k/bmkXw0tGQAAwCCjzR6aPqxgkSS5+79KmhZOJAAAgNONVrSYmZ22nKmZlQZ4LgAAwLgZrfD4iqTtZvZRM5uW+blK0tbMNQAAgKwYbUXczWb2uqQvaejsoT9y9+9mIR8AAICkAFOe3f1xSY9nIQsAAMBZjTbl+YuZ8Stnu/4xM/v4+McCAAAYarSWlhfUtzlip6TnJB2WVCjpUkkflPS0pA2hJgQAANDoY1q+o751WS6V9GFJF0p6R32LzN3k7ifCjwgAABBww0R3f1nSyyFnAQAAOCvWWgEAAIlA0QIAABIhUPcQAABJYmZRRxizkpLTFqTPeYGKFjO7TNKDki5w9wozWyRphbv/UajpAMQWvxQQV+4e+nuYWVbeB0MF7R56SNIdkrolyd33SPpsWKEAxJu7h/6TjfdpbW2N+L8kgHMRtGiZ4u4/GHbu1HiHAQAAOJugRcvPzGyB+vYekpn9pqQ3QksFAAAwTNCBuDdL2izpfWb2U0mvSPrt0FIBAAAME3RxuZ9I+nUzK5KUcvej4cYCAAAYKlD3kJn1mNmXJR3vL1jM7LlQkwEAAAwSdExLc+ax2wft+pz8+Y4AACAxghYtp9x9nfqmPv+zmX1ImUG5AAAA2RB0IK5Jkrv/rZk1S0pLKgstFQAAwDBBi5b/3n/H3ZvN7EpJnwwnEgAAwOlGLFrM7GPuvkPSPDObN+zysfBi4d14t8uqn+vzWLoaABCF0VpaPipph6RPnOGaS/r2uCfCu0YxAQCYyEYsWtz9zsztjdmJAwAAcGZB12m5xcymW5+Hzew5M1sWdjgAAIB+Qac8r3L3dyQtkzRb0o2SvhxaKgAAgGGCFi39IzWvkfRVd/+RWFwOAABkUdCiZbeZbVdf0bLNzKZJ6g0vFgAAwFBB12mplvRBST9x9+Nmdr76uogAAACyIuguz72Snht0/Lakt8MKBQAAMFzQ7iEAAIBIUbQAAIBECLpOy18GOQcAABCWoC0tCwcfmFmepA+N9AQze8TM3jKzvWe5bmb2/8xsv5ntMbNfCpgFAADkoBGLFjO7w8yOSlpkZu9kfo5KekvSd0Z57b+QdPUI15dLujTzc5OkBwOnBgAAOWfEosXd73b3aZL+2N2nZ36mufv57n7HKM/dJal1hIf8hqSve59nJM0wswvP+V8AAAByQtDuoR+YWXH/gZnNMLNPjvG950g6OOi4JXMOAADgNEGLljvd/Uj/gbu3S7pzjO99pm0A/IwPNLvJzJrMrOnw4cNjfFsAAJBEQYuWMz0u6Gq6Z9Mi6eJBx3MlvX6mB7r7Zndf7O6LZ82aNca3BQAASRS0aGkys/vNbIGZvcfMviJp9xjf+zFJN2RmEf1HSUfc/Y0xviYAAJiggraW1Er6fUl/kzneLumLIz3BzNKSrpI008xa1NedlC9J7l4v6Un1bcC4X9JxsZcRAAAYQdC9hzok/Z6ZTXX3YwGfUzXKdZd0c5DXAgAACLoi7hVm9qKkFzPHHzCzTaEmAwAAGCTomJavSKpUZmdnd/+RpI+EFQoAAGC4wBsmuvvBYad6xjkLAADAWQUdiHvQzK6Q5GZ2nqTPS9oXXiwAAIChgra01Khv0Owc9a2v8kExiBYAAGTRiC0tZnaPu98uaYm7r8xSJgAAgNOM1tJyjZnlSxpxc0QAAICwjTam5R8l/UxSkZm9o779grz/1t2nh5wPAABA0ugtLV9092JJT7j7dHefNvg2GwEBAACk0YuWf8ncvhN2EAAAgJGM1j10npn9V0lXmNmnhl9092+HEwsAAGCo0YqWGkkrJc2Q9Ilh11wSRQsAAMiKEYsWd2+U1GhmTe6+JUuZAAAATjPimBYzWydJ7r7FzD4z7NqGMIMBAAAMNtpA3M8Ouj98rZarxzkLAADAWY1WtNhZ7p/pGAAAIDSjFS1+lvtnOgYAAAjNaEXLB8zsHTM7KmlR5n7/8fuzkA8hSafTqqioUF5enioqKpROp6OOBADAiEabPZSXrSDInnQ6rbq6Om3ZskVXXnmlGhsbVV1dLUmqqqqKOB0AAGc2WksLJqD169dry5YtWrJkifLz87VkyRJt2bJF69evjzoaAABnZe7JGpqyePFib2pqijpGouXl5amzs1P5+fkD57q7u1VYWKienp4IkwE/Z2ZK2vcTcgefz/FjZrvdfXGQx9LSkoPKy8vV2Ng45FxjY6PKy8sjSgQAwOgoWnJQXV2dqqur1dDQoO7ubjU0NKi6ulp1dXVRRwOAWCsrK5NZ34ofZqaysrKIE+WW0fYewgTUP9i2trZW+/btU3l5udavX88gXAAYQVlZmQ4ePDjk3MGDB1VWVqbXXnstolS5hTEtAGKlsrJSTz31lNxdZqalS5dq27ZtUccCBlpYziRpv0vjhDEtABKpsrJS27dvH/gF4O7avn27KisrI06GiczMAv2M9TUwdhQtAGJj+/btkqSSkpIht/3ngTC4e6Cfsb4Gxo4xLQCy4lz+0mxraxtyey7P55cDMHFRtADIiiDFhJlp0aJF6unpGRgknpeXpz179lCMAKB7CEC87NmzRwsWLNCbb76pBQsWaM+ePVFHAhATtLQAiI3+VUYfe+wxzZo1a8h5AKClBUCsTJ48ecRjALmLlhYAsXH55Zdr8uTJ2r1798A6LQsXLtSJEyeijgYgBmhpARAbc+bMUVNTk2pqatTe3q6amho1NTVpzpw5UUcDEAMULQBiY+fOnVq5cqV27dql0tJS7dq1SytXrtTOnTujjgYMWLhwoV599VUtXLgw6ig5h6IFQGx0dXWpqKhI+/fvV29vr/bv36+ioiJ1dXVFHQ0Y0NzcrHnz5qm5uTnqKDmHogVAbKRSKT388MPasGGDOjo6tGHDBj388MNKpfiqAkDRAiBGzja1mSnPACSKFgAx0tPTo6uuukq33XabioqKdNttt+mqq65ST09P1NEAxABFC4DYmDRpkp599lnNmzdPqVRK8+bN07PPPqtJk1idAQBFC4AYKSgoUEdHh5YvX67W1lYtX75cHR0dKigoiDoaMKCkpER79uwZ2IUc2WNJ24Rs8eLF3tTUFHUMACEwM61YsULbtm1TV1eXCgoKVFlZqccee4wNExG5kcZW8fl898xst7svDvJYWloAxMratWvV2dkpd1dnZ6fWrl0bdSQAMUHRAiA25s6dqxtuuEENDQ3q7u5WQ0ODbrjhBs2dOzfqaABigKIFQGzce++96unp0apVq1RQUKBVq1app6dH9957b9TRAMQARQuA2KiqqtL111+vN954Q+6uN954Q9dff72qqqqijgYgBihaAMRGOp3WE088oa1bt+rkyZPaunWrnnjiCaXT6aijAQMmT54sM9PkyZOjjpJzmD0EIDYqKip06aWXauvWrQOzh5YvX66XX35Ze/fujToeclz/7KFUKqXe3t6BW4nZQ2NxLrOHKFoAxEYqlZK7Ky8vTz09PQO3ZjbwywGIClOew8GUZwCJ1P/FP3PmTJmZZs6cOeQ8gNxG0QIgVlKplNatW6djx45p3bp17PAMYADdQwBiw8w0Y8YMtbe3D5zrP07adxUmHrqHwkH3EIDEam9vV2FhoSSpsLBwSAEDILdRtACInc7OziG3QJzMnz9f+/fv1/z586OOknPY7x0AgHNw4MABvfe97406Rk6ipQVArEyaNEnz58+XmWn+/PmaNIm/rQD0oWgBECunTp3S8uXL1dbWpuXLl+vUqVNRRwIQE8weAhAbZjawoFy//uOkfVdh4mH2UDiYPQQgkUpLS9Xb26sLLrhAZqYLLrhAvb29Ki0tjToaIEkqKCgY8RjhCrVoMbOrzezfzGy/mf3eGa7PM7N/MrM9ZvY9M5sbZh4A8fbAAw9o2rRpam1tlburtbVV06ZN0wMPPBB1NECTJk1SV1fXkA0Tu7q6GHeVRaEVLWaWJ+nPJC2XdLmkKjO7fNjD/kTS1919kaQ/lHR3WHkAxF9VVZXq6+t12WWXKZVK6bLLLlN9fb2qqqqijgZo+vTpkqQTJ07I3XXixIkh5xG+0Ma0mNmvSbrL3Sszx3dIkrvfPegxzZIq3b3F+joLj7j7iP/rM6YFABAFM1N+fr66u7sHzvUfM6bl3YvLmJY5kg4OOm7JnBvsR5I+nbn/XyRNM7PzQ8wEAMC7durUqSFjrpjdll1hFi1nGmY9vBS9TdJHzex5SR+V9FNJp30CzOwmM2sys6bDhw+Pf1IAAALob1EZfovsCLNoaZF08aDjuZJeH/wAd3/d3T/l7r8oqS5z7sjwF3L3ze6+2N0Xz5o1K8TIAACMzMyUSqVGnAKNcIRZtPxQ0qVmdomZnSfps5IeG/wAM5tpZv0Z7pD0SIh5ACRAOp1WRUWF8vLyVFFRoXQ6HXUkYEAqldKhQ4fU29urQ4cOKZVi5ZBsCm2elrufMrO1krZJypP0iLs3m9kfSmpy98ckXSXpbjNzSbsk3RxWHgDxl06nVV1dPTAro7m5WdXV1ZLEDCLEQm9vr1Kp1JBbZA8r4gKIjalTp6qjo0Nr1qzR3XffrTvuuEMPPvigioqKdOzYsajjIcexIm44zmX2ECviAIiNjo4O3XTTTdq0aZMkadOmTerp6dHmzZsjTgb83LRp09TR0aGioiIdPXo06jg5hc44ALHy8Y9/fMRjIEoLFixQWVmZJKmsrEwLFiyIOFFuoWgBECsrV65UQ0ODuru71dDQoJUrV0YdCRjwyiuvaNWqVTp69KhWrVqlV155JepIOYUxLQBio7KyUtu3b1dJSYmOHDmi4uJitbW1admyZdq2bVvU8ZDj+lfElaTu7u4h95P2uzRO4rIiLgCck23btmnZsmVqb29Xb2+v2tvbKVgQG8uWLVN3d7emTp2qVCqlqVOnqru7W8uWLYs6Ws6gpQUAgIAqKyv11FNPyd1lZlq6dCk5iSREAAAG90lEQVRF9RjR0gIAQAiampqGLOHPH9HZRdECAEAA559/vlpbW7Vw4UK9+uqrWrhwoVpbW3X++ezzmy2s0wIAQAD9BcvevXslSXv37lVFRYWam5sjTpY7aGkBACCgJ598csRjhIuiBQCAgK655poRjxEuihYAAAIoLS1Vc3OzKioq9Nprrw10DZWWlkYdLWcwpgUAgADefvttTZ06Vc3NzZo3b54kqaioSG+//XbEyXIHLS0AAASQTqc1e/Zs7dixQydPntSOHTs0e/ZspdPpqKPlDBaXAwAggIqKCm3cuFFLliwZONfQ0KDa2tqBGUU4d+eyuBxFCwAAAeTl5amzs3NgzyGpb9+hwsJC9fT0RJgs2VgRF0BipdNpVVRUKC8vTxUVFTS9IzbKy8vV2Ng45FxjY6PKy8sjSpR7KFoAxEY6nVZdXZ02btyozs5Obdy4UXV1dRQuiIW6ujpVV1eroaFB3d3damhoUHV1terq6qKOljPoHgIQG4wZQNyl02mtX79e+/btU3l5uerq6lRVVRV1rERjTAuARGLMAJB7GNMCIJEYMwBgJBQtAGKDMQMARsKKuABio39sQG1t7cCYgfXr1zNmAIAkxrQAAIAIMaYFQGKxTguAs6F7CEBs9K/TsmXLFl155ZVqbGxUdXW1JNFFBIDuIQDxwTotQO5hnRYAicQ6LUDuYUwLgERinRYAI6FoARAbrNMCYCQMxAUQG6zTAmAkjGkBAACRYUwLAACYcChaAABAIlC0AACARKBoAQAAiUDRAgAAEoGiBQCAgNjQM1qs0wIAQABs6Bk91mkBACAANvQMBxsmAgAwztjQMxwsLgcAwDhjQ8/oUbQAABAAG3pGj4G4AAAEwIae0WNMCwAAiAxjWgAAwIRD0QIAABKBogUAACQCRQsAAEgEihYAAJAIFC0AACARKFoAAEAiULQAAIBEoGgBAACJQNECAAASIXHL+JvZYUmvRp1jApkp6WdRhwDOgM8m4ozP5/iZ5+6zgjwwcUULxpeZNQXd8wHIJj6biDM+n9GgewgAACQCRQsAAEgEihZsjjoAcBZ8NhFnfD4jwJgWAACQCLS0AACARKBoyVFmdrWZ/ZuZ7Tez34s6D9DPzB4xs7fMbG/UWYDBzOxiM2sws31m1mxmt0SdKdfQPZSDzCxP0o8lLZXUIumHkqrc/cVIgwGSzOwjko5J+rq7V0SdB+hnZhdKutDdnzOzaZJ2S/ok353ZQ0tLbvoVSfvd/SfuflLSX0v6jYgzAZIkd98lqTXqHMBw7v6Guz+XuX9U0j5Jc6JNlVsoWnLTHEkHBx23iP/jAUBgZjZf0i9KejbaJLmFoiU32RnO0U8IAAGY2VRJ35L0u+7+TtR5cglFS25qkXTxoOO5kl6PKAsAJIaZ5auvYPmGu3876jy5hqIlN/1Q0qVmdomZnSfps5IeizgTAMSamZmkLZL2ufv9UefJRRQtOcjdT0laK2mb+gaS/a27N0ebCuhjZmlJ/yLpP5hZi5lVR50JyPiwpN+R9DEz+9fMzzVRh8olTHkGAACJQEsLAABIBIoWAACQCBQtAAAgEShaAABAIlC0AACARJgUdQAAE4+Z9Uh6QVK+pFOSvibp/7p7b6TBACQaRQuAMJxw9w9KkpnNlvRNScWS7hzrC5tZnrv3jPV1ACQP3UMAQuXub0m6SdJa65NnZn9sZj80sz1m9j8kycxSZrbJzJrN7HEze9LMfjNz7YCZ/R8za5T0GTNbYGb/aGa7zeyfzex9mcfNMrNvZV77h2b24cj+4QDGHS0tAELn7j8xs5Sk2ZJ+Q9IRd/9lMyuQ9P/NbLukD0maL+n9mcftk/TIoJfpdPcrJcnM/klSjbu/bGa/KmmTpI9J+lNJX3H3RjMrU9+qz+VZ+UcCCB1FC4Bs6d9dfJmkRf2tKOrrNrpU0pWS/i4z7uWQmTUMe/7fSAM77F4h6e/6toKRJBVkbn9d0uWDzk83s2nufnS8/zEAso+iBUDozOw9knokvaW+4qXW3bcNe8y1o7xMR+Y2Jam9f8zMMClJv+buJ8YYGUAMMaYFQKjMbJakekkPeN9mZ9skrTGz/Mz1y8ysSFKjpE9nxrZcIOmqM72eu78j6RUz+0zm+WZmH8hc3q6+zUD73/tMhQ2AhKKlBUAYJpvZv+rnU57/UtL9mWsPq2/synPW149zWNInJX1L0n+WtFfSjyU9K+nIWV5/paQHzeyLmff4a0k/kvR5SX9mZnvU9/22S1LNeP/jAESDXZ4BxIaZTXX3Y2Z2vqQfSPqwux+KOheAeKClBUCcPG5mMySdJ+lLFCwABqOlBQAAJAIDcQEAQCJQtAAAgESgaAEAAIlA0QIAABKBogUAACQCRQsAAEiEfwdlmR/wr0/piQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 648x432 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_to_plot = [degreesEff[0], degreesEff[1], degreesEff[2]]\n",
    "print(np.median(degreesEff[0]),np.median(degreesEff[1]),np.median(degreesEff[2]))\n",
    "print(np.mean(degreesEff[0]),np.mean(degreesEff[1]),np.mean(degreesEff[2]))\n",
    "fig = plt.figure(1, figsize=(9, 6))\n",
    "ax = fig.add_subplot(111)\n",
    "bp = ax.boxplot(data_to_plot)\n",
    "ax.set_xticklabels(['0', '1', '2'])\n",
    "ax.set_xlabel(\"Degree\")\n",
    "ax.set_ylabel(\"Effect size (Odds Ratio)\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8.212674941550832 3.4826177291998446e-16\n",
      "0.2986077445284085 0.7652449721087354\n"
     ]
    }
   ],
   "source": [
    "t1, p1 = stats.ttest_ind(degreesEff[0],degreesEff[1])\n",
    "print(t1, p1)\n",
    "t2, p2 = stats.ttest_ind(degreesEff[1],degreesEff[2])\n",
    "print(t2, p2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### SCZ PrePPI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "PrePPI = pd.read_csv(\"../dat/network/preppi_final600.txt\", delimiter=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>prot1</th>\n",
       "      <th>prot2</th>\n",
       "      <th>str_score</th>\n",
       "      <th>protpep_score</th>\n",
       "      <th>str_max_score</th>\n",
       "      <th>red_score</th>\n",
       "      <th>ort_score</th>\n",
       "      <th>phy_score</th>\n",
       "      <th>coexp_score</th>\n",
       "      <th>go_score</th>\n",
       "      <th>total_score</th>\n",
       "      <th>dbs</th>\n",
       "      <th>pubs</th>\n",
       "      <th>exp_score</th>\n",
       "      <th>final_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q13131</td>\n",
       "      <td>P14625</td>\n",
       "      <td>18.5900</td>\n",
       "      <td>6.44772</td>\n",
       "      <td>18.5900</td>\n",
       "      <td>4.2492</td>\n",
       "      <td>0.6153</td>\n",
       "      <td>2.4160</td>\n",
       "      <td>9.4687</td>\n",
       "      <td>10.80</td>\n",
       "      <td>12008.400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12008.400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P06400</td>\n",
       "      <td>Q96N96</td>\n",
       "      <td>1.8315</td>\n",
       "      <td>14.32220</td>\n",
       "      <td>14.3222</td>\n",
       "      <td>4.2492</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.4160</td>\n",
       "      <td>2.1077</td>\n",
       "      <td>10.80</td>\n",
       "      <td>3346.930</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3346.930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q7Z6V5</td>\n",
       "      <td>Q8NCE0</td>\n",
       "      <td>4.5712</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>4.5712</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.5978</td>\n",
       "      <td>9.4687</td>\n",
       "      <td>24.11</td>\n",
       "      <td>1667.400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1667.400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P37058</td>\n",
       "      <td>P15428</td>\n",
       "      <td>484.5640</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>484.5640</td>\n",
       "      <td>1.7740</td>\n",
       "      <td>0.6153</td>\n",
       "      <td>2.4160</td>\n",
       "      <td>1.1757</td>\n",
       "      <td>2.07</td>\n",
       "      <td>3109.970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3109.970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O00468</td>\n",
       "      <td>Q9NR23</td>\n",
       "      <td>34.0309</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>34.0309</td>\n",
       "      <td>0.5125</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.4160</td>\n",
       "      <td>2.7714</td>\n",
       "      <td>5.86</td>\n",
       "      <td>684.323</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>684.323</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    prot1   prot2  str_score  protpep_score  str_max_score  red_score  \\\n",
       "0  Q13131  P14625    18.5900        6.44772        18.5900     4.2492   \n",
       "1  P06400  Q96N96     1.8315       14.32220        14.3222     4.2492   \n",
       "2  Q7Z6V5  Q8NCE0     4.5712        0.00000         4.5712     0.0000   \n",
       "3  P37058  P15428   484.5640        0.00000       484.5640     1.7740   \n",
       "4  O00468  Q9NR23    34.0309        0.00000        34.0309     0.5125   \n",
       "\n",
       "   ort_score  phy_score  coexp_score  go_score  total_score  dbs pubs  \\\n",
       "0     0.6153     2.4160       9.4687     10.80    12008.400  NaN  NaN   \n",
       "1     0.0000     2.4160       2.1077     10.80     3346.930  NaN  NaN   \n",
       "2     0.0000     1.5978       9.4687     24.11     1667.400  NaN  NaN   \n",
       "3     0.6153     2.4160       1.1757      2.07     3109.970  NaN  NaN   \n",
       "4     0.0000     2.4160       2.7714      5.86      684.323  NaN  NaN   \n",
       "\n",
       "   exp_score  final_score  \n",
       "0        NaN    12008.400  \n",
       "1        NaN     3346.930  \n",
       "2        NaN     1667.400  \n",
       "3        NaN     3109.970  \n",
       "4        NaN      684.323  "
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PrePPI.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "degreesEff = {}\n",
    "degreesEff[0] = []\n",
    "degreesNode = {}\n",
    "degreesNode[0] = []\n",
    "for i, row in SCZ_high_conf_gene.iterrows():\n",
    "    ensid = row[\"ensembl_names\"]\n",
    "    try:\n",
    "        gene = GeneDict[ensid]\n",
    "    except KeyError:\n",
    "        #print (row[\"sczgenenames\"], row[\"ensembl_names\"])\n",
    "        pass\n",
    "    if len(gene.SNPs) > 0:\n",
    "        degreesEff[0].append(gene.max_eff)\n",
    "        degreesNode[0].append(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1544550"
     ]
    }
   ],
   "source": [
    "IDGeneDict = {}\n",
    "for k,v in GeneDict.items():\n",
    "    IDGeneDict[v.GeneID] = v\n",
    "\n",
    "graph2 = {}\n",
    "count = 0\n",
    "for i, row in PrePPI.iterrows():\n",
    "    prot1, prot2 = row[\"prot1\"], row[\"prot2\"]\n",
    "    try:\n",
    "        g1, g2 = Uniprot2ENSG[prot1], Uniprot2ENSG[prot2]\n",
    "    except KeyError:\n",
    "        continue\n",
    "    #print(g1, g2)\n",
    "    if g1 not in IDGeneDict or g2 not in IDGeneDict:\n",
    "        continue\n",
    "    if g1 not in graph2:\n",
    "        graph2[g1] = []\n",
    "    if g2 not in graph2:\n",
    "        graph2[g2] = []\n",
    "    graph2[g1].append(g2)\n",
    "    graph2[g2].append(g1)\n",
    "    if count % 1000 == 0:\n",
    "        sys.stdout.write(\"\\r%d\"%i)\n",
    "    count += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['KMT2E',\n",
       " 'MVP',\n",
       " 'MTMR11',\n",
       " 'PLEKHO1',\n",
       " 'HSD17B6',\n",
       " 'NR1H3',\n",
       " 'SPA17',\n",
       " 'KDM4A',\n",
       " 'NGEF',\n",
       " 'SREBF1',\n",
       " 'NT5C2',\n",
       " 'KIF22',\n",
       " 'DDHD2',\n",
       " 'TMED2',\n",
       " 'SF3B2',\n",
       " 'PITPNM2',\n",
       " 'PUS7',\n",
       " 'CACNA1I',\n",
       " 'GRAP2',\n",
       " 'RANGAP1',\n",
       " 'PPP1R16B',\n",
       " 'NDFIP2',\n",
       " 'MAZ',\n",
       " 'CDIPT',\n",
       " 'TRIM35',\n",
       " 'RAI1',\n",
       " 'SIAE',\n",
       " 'MDK',\n",
       " 'AMBRA1',\n",
       " 'MADD',\n",
       " 'OGFOD2',\n",
       " 'GTF2H3',\n",
       " 'DDX55',\n",
       " 'EFHD1',\n",
       " 'NEU2',\n",
       " 'PRPF3',\n",
       " 'PTK2B',\n",
       " 'CHRNA2',\n",
       " 'EPHX2',\n",
       " 'CRY2',\n",
       " 'ARHGAP9',\n",
       " 'LRP1',\n",
       " 'KLC1',\n",
       " 'ZSWIM6',\n",
       " 'ARHGEF16',\n",
       " 'HIP1R',\n",
       " 'FAM83F',\n",
       " 'DDB2',\n",
       " 'ACP2',\n",
       " 'SRPK2',\n",
       " 'PHF21A',\n",
       " 'OS9',\n",
       " 'VPS45',\n",
       " 'SBNO1',\n",
       " 'VPS37B',\n",
       " 'RBM26',\n",
       " 'GID4',\n",
       " 'RERE',\n",
       " 'PRDM16',\n",
       " 'PTPRF',\n",
       " 'SF3B4',\n",
       " 'ECM1',\n",
       " 'TARS2',\n",
       " 'ANP32E',\n",
       " 'STAR',\n",
       " 'CNNM2',\n",
       " 'DGKZ',\n",
       " 'CELF1',\n",
       " 'DOC2A',\n",
       " 'HIRIP3',\n",
       " 'TAOK2',\n",
       " 'ABCB9',\n",
       " 'CACNA1C',\n",
       " 'RABGAP1L',\n",
       " 'PCGF6',\n",
       " 'TDRD9',\n",
       " 'BAG4',\n",
       " 'CACNA1D',\n",
       " 'PEX10',\n",
       " 'SV2A',\n",
       " 'CTSS',\n",
       " 'ASB5',\n",
       " 'LETM2',\n",
       " 'BAG5',\n",
       " 'STAT6',\n",
       " 'MARS',\n",
       " 'C16orf92',\n",
       " 'PRRT2',\n",
       " 'ZEB2',\n",
       " 'INO80E',\n",
       " 'ESCO2',\n",
       " 'ATPAF2',\n",
       " 'EFEMP2',\n",
       " 'MUS81',\n",
       " 'CFL1',\n",
       " 'SNX32',\n",
       " 'OVOL1',\n",
       " 'RNASEH2C',\n",
       " 'KAT5',\n",
       " 'RELA',\n",
       " 'LRRN3',\n",
       " 'SEZ6L2',\n",
       " 'ASPHD1',\n",
       " 'PACS1',\n",
       " 'ZNF408',\n",
       " 'CKAP5',\n",
       " 'ARHGAP1',\n",
       " 'ATG13',\n",
       " 'GAL3ST3',\n",
       " 'CST6',\n",
       " 'LSM1',\n",
       " 'BANF1',\n",
       " 'EIF1AD',\n",
       " 'SART1',\n",
       " 'DRAP1',\n",
       " 'C11orf68',\n",
       " 'FOSL1',\n",
       " 'TOM1L2',\n",
       " 'BOLA1',\n",
       " 'HYI',\n",
       " 'ZBTB18',\n",
       " 'R3HDM2',\n",
       " 'HARBI1',\n",
       " 'CHRM4',\n",
       " 'ARL6IP4',\n",
       " 'NXPH4',\n",
       " 'C2orf82',\n",
       " 'C2orf82',\n",
       " 'C2orf82',\n",
       " 'HIST2H2AC',\n",
       " 'HIST2H2AB',\n",
       " 'HIST2H2BE',\n",
       " 'NDUFA4L2',\n",
       " 'NDUFA4L2',\n",
       " 'SMIM15',\n",
       " 'NACA',\n",
       " 'PRIM1',\n",
       " 'SZT2',\n",
       " 'SZT2',\n",
       " 'HIST2H3A',\n",
       " 'NFAM1',\n",
       " 'AP5B1',\n",
       " 'APOPT1',\n",
       " 'MAD1L1',\n",
       " 'NISCH',\n",
       " 'GLT8D1',\n",
       " 'CUL3',\n",
       " 'CTNNA1',\n",
       " 'ITIH4',\n",
       " 'BTBD1',\n",
       " 'ANKRD44',\n",
       " 'MPPED2',\n",
       " 'SMG6',\n",
       " 'AP3B2',\n",
       " 'DPH1',\n",
       " 'INPP4B',\n",
       " 'CLCN3',\n",
       " 'FXR1',\n",
       " 'SF3B1',\n",
       " 'PLCL1',\n",
       " 'ALMS1',\n",
       " 'AKT3',\n",
       " 'STAG1',\n",
       " 'MAU2',\n",
       " 'CCDC62',\n",
       " 'NEK1',\n",
       " 'FURIN',\n",
       " 'GFOD2',\n",
       " 'TMEM219',\n",
       " 'TMEM219',\n",
       " 'KCTD18',\n",
       " 'ADAMTSL3',\n",
       " 'SLC45A1',\n",
       " 'RFTN2',\n",
       " 'DCLK3',\n",
       " 'SFMBT1',\n",
       " 'WBP1L',\n",
       " 'NT5DC2',\n",
       " 'SLC35G2',\n",
       " 'CPT1C',\n",
       " 'EMB',\n",
       " 'TSNARE1',\n",
       " 'LRRC48',\n",
       " 'ZSCAN2',\n",
       " 'ZSCAN2',\n",
       " 'F2',\n",
       " 'RPS17L',\n",
       " 'RPS17L',\n",
       " 'GOLGA6L4',\n",
       " 'GOLGA6L4',\n",
       " 'GOLGA6L4',\n",
       " 'GOLGA6L4',\n",
       " 'TMEM110',\n",
       " 'CPEB1',\n",
       " 'CPEB1',\n",
       " 'CPEB1',\n",
       " 'CPEB1',\n",
       " 'CPEB1',\n",
       " 'PROB1',\n",
       " 'PROB1',\n",
       " 'PROB1',\n",
       " 'PROB1',\n",
       " 'RP5-966M1.6',\n",
       " 'MARS2',\n",
       " 'MARS2',\n",
       " 'MARS2',\n",
       " 'MARS2',\n",
       " 'MARS2',\n",
       " 'MARS2',\n",
       " 'RP11-182J1.16',\n",
       " 'RP11-182J1.16',\n",
       " 'RP11-182J1.16',\n",
       " 'RP11-182J1.16',\n",
       " 'RP11-182J1.16',\n",
       " 'RP11-182J1.16',\n",
       " 'RP11-182J1.16',\n",
       " 'EMX1',\n",
       " 'KCTD13',\n",
       " 'HORMAD1',\n",
       " 'CTSK',\n",
       " 'ARNT',\n",
       " 'GPM6A',\n",
       " 'WDR17',\n",
       " 'BNIPL',\n",
       " 'GIGYF2',\n",
       " 'CALHM2',\n",
       " 'MPHOSPH9',\n",
       " 'POLR3H',\n",
       " 'CSDC2',\n",
       " 'DESI1',\n",
       " 'C12orf65',\n",
       " 'CDK2AP1',\n",
       " 'MEI1',\n",
       " 'SNRNP35',\n",
       " 'RILPL1',\n",
       " 'WBP2NL',\n",
       " 'NAGA',\n",
       " 'FAM109B',\n",
       " 'VSIG2',\n",
       " 'MSANTD2',\n",
       " 'TKT',\n",
       " 'PPP1R13B',\n",
       " 'CCDC15',\n",
       " 'C11orf49',\n",
       " 'CHEK1',\n",
       " 'PACSIN3',\n",
       " 'MYBPC3',\n",
       " 'KBTBD4',\n",
       " 'ELOVL7',\n",
       " 'DRG2',\n",
       " 'SLC9C2',\n",
       " 'ANKRD45',\n",
       " 'DARS2',\n",
       " 'SERPINC1',\n",
       " 'RC3H1',\n",
       " 'FANCL',\n",
       " 'EPN2',\n",
       " 'NFATC3',\n",
       " 'HSPA9',\n",
       " 'HSPE1',\n",
       " 'SEC11A',\n",
       " 'HSPD1',\n",
       " 'WHSC1L1',\n",
       " 'PLCH2',\n",
       " 'DGKI',\n",
       " 'THOC7',\n",
       " 'ATXN7',\n",
       " 'ZNF592',\n",
       " 'DPEP2',\n",
       " 'DUS2',\n",
       " 'MAP3K11',\n",
       " 'WDR73',\n",
       " 'WDR73',\n",
       " 'PRKD1',\n",
       " 'PRKD1',\n",
       " 'PCNXL3',\n",
       " 'NMB',\n",
       " 'NMB',\n",
       " 'NMB',\n",
       " 'RP11-691N7.6',\n",
       " 'RP11-691N7.6',\n",
       " 'RP11-133K1.2',\n",
       " 'RP11-133K1.2',\n",
       " 'RP11-133K1.2',\n",
       " 'RP11-133K1.2',\n",
       " 'ZFYVE21',\n",
       " 'FOXP1',\n",
       " 'SHMT2',\n",
       " 'SIPA1',\n",
       " 'VRK2',\n",
       " 'AKAP6',\n",
       " 'BRINP2',\n",
       " 'SETD6',\n",
       " 'ARL14EP',\n",
       " 'TYW5',\n",
       " 'GATAD2A',\n",
       " 'SRR',\n",
       " 'TRANK1',\n",
       " 'CHRNA5',\n",
       " 'CALB2',\n",
       " 'FES',\n",
       " 'FES',\n",
       " 'FTCDNL1',\n",
       " 'FTCDNL1',\n",
       " 'FTCDNL1',\n",
       " 'FTCDNL1',\n",
       " 'FTCDNL1',\n",
       " 'FTCDNL1',\n",
       " 'SLC39A8',\n",
       " 'SLC39A8',\n",
       " 'SNAP91',\n",
       " 'CNTN4',\n",
       " 'DNAJC19',\n",
       " 'NDUFA13',\n",
       " 'TSSK6',\n",
       " 'IRF3',\n",
       " 'RGS6',\n",
       " 'NDRG4',\n",
       " 'NDRG4',\n",
       " 'PLCB2',\n",
       " 'PTPRK']"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "degreesNode[0] = [gene.GeneName for gene in degreesNode[0]]\n",
    "degreesNode[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "277\n",
      "9061\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "#degreesNode[0] = [gene.GeneName for gene in degreesNode[0]]\n",
    "PreNodes = set(degreesNode[0])\n",
    "print(len(PreNodes))\n",
    "for i in range(1,11,1):\n",
    "    degreesEff[i] = []\n",
    "    degreesNode[i] = []\n",
    "    for gene in degreesNode[i-1]:\n",
    "        try:\n",
    "            symbol = Symbol2ENSG[gene]\n",
    "        except:\n",
    "            continue\n",
    "        if symbol in graph2:\n",
    "            for node in graph2[symbol]:\n",
    "                if node not in PreNodes:\n",
    "                    degreesNode[i].append(node)\n",
    "    degreesNode[i] = list(set(degreesNode[i]))\n",
    "    for symbol in degreesNode[i]:\n",
    "        try:\n",
    "            degreesEff[i].append(SymbolGeneDict[symbol].max_eff)\n",
    "        except:\n",
    "            continue\n",
    "    print(len(degreesNode[i]))\n",
    "    PreNodes = PreNodes.union(set(degreesNode[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.08513 nan nan\n",
      "1.0926142679127726 nan nan\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjMAAAF3CAYAAABQc8olAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAGsZJREFUeJzt3X2QZXWd3/H3xwFmFEEeZvAJRsRiI0gUtdVdIQpkVcRViK4biauuIU4mpaNGzaJglkQimlIxLj6MRIgPmx3XXXFFCgWjg+xEWelBRZBVWTRhRGtQEJAH5eGbP+7tSdNM9z3jzOnbv+n3q+pW33N+597+dDFFfeqc3zm/VBWSJEmtetC4A0iSJG0Py4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJatou4w6wIy1fvrwOPPDAcceQJEk7wMaNG39eVStGHbdTlZkDDzyQycnJcceQJEk7QJL/0+U4LzNJkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZnQ/a9asYdmyZSRh2bJlrFmzZtyRJEmak2VGW6xZs4a1a9dyxhlncPvtt3PGGWewdu1aC40kaUFLVY07ww4zMTFRrs3021u2bBlnnHEGb3rTm7bsO/PMMznllFO46667xphMkrQYJdlYVRMjj7PMaEoSbr/9dh7ykIds2XfHHXew++67szP9O5EktaFrmfEyk7ZYunQpa9euvd++tWvXsnTp0jElkiRptF3GHUALx2te8xpOPvlkAFavXs3atWs5+eSTWb169ZiTSZI0O8uMtjjrrLMAOOWUU3jzm9/M0qVLWb169Zb9kiQtRM6ZkSRJC5JzZiRJ0qJgmZEkSU2zzEiSpKZZZiRJUtN6KzNJzk2yOclVs4y/PMmVw9fXkzxp2tixSb6f5Nokb+0royRJal+fZ2Y+Dhw7x/iPgGdX1ROB04GzAZIsAT4EPB84FDgxyaE95pQkSQ3rrcxU1aXATXOMf72qbh5uXgbsP3z/dODaqrquqn4DfBo4vq+ckiSpbQtlzsxJwBeH7x8NXD9tbNNwnyRJ0gOM/QnASY5mUGaOnNq1lcNmfbJfklXAKoCVK1fu8HySJGlhG+uZmSRPBD4GHF9Vvxju3gQcMO2w/YEbZvuOqjq7qiaqamLFihX9hZUkSQvS2MpMkpXAecArquoH04YuBw5O8tgkuwEvA84fR0ZJkrTw9XaZKck64ChgeZJNwGnArgBVtRb4M2Bf4MNJAO4ZnmG5J8nrgIuAJcC5VXV1XzklSVLbXGhSkiQtSC40KUmSFgXLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUtLGvmq35MVwyonc70xOlJUltsMwsEttaMpJYTCRJTfAykyRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktS03spMknOTbE5y1Szjj0/yjSS/TvKWGWM/TvLdJN9OMtlXRkmS1L4+z8x8HDh2jvGbgNcD751l/OiqOryqJnZ0MEmStPPorcxU1aUMCsts45ur6nLg7r4ySJKknd9CnTNTwMVJNiZZNe4wkiRp4dpl3AFmcURV3ZBkP+DLSf5heKbnAYZlZxXAypUr5zOjJElaABbkmZmqumH4czPwOeDpcxx7dlVNVNXEihUr5iuiJElaIBZcmUmye5I9pt4DzwW2ekeUJElSb5eZkqwDjgKWJ9kEnAbsClBVa5M8ApgE9gTuS/JG4FBgOfC5JFP5/rKqvtRXTkmS1LbeykxVnThi/GfA/lsZuhV4Ui+hJEnSTmfBXWaSJEnaFpYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkprWedXsJHsDjwLuBH5cVff1lkqSJKmjOctMkocBrwVOBHYDbgSWAQ9Pchnw4apa33tKSZKkWYw6M/M3wCeBf1ZVv5w+kOSpwCuSHFRV5/QVUJIkaS5zlpmqes4cYxuBjTs8kSRJ0jbYljkzLwKeNdz8WlV9oZ9IkiRJ3XW6mynJu4E3AN8bvl6f5F19BpMkSeqi65mZ44DDp+5gSvIJ4FvA2/oKJkmS1MW2PGdmr2nvH7ajg0iSJP02up6ZeRfwrSTrgTCYO+NZGUmSNHadykxVrUtyCfA0BmXm5Kr6WZ/BJEmSupjzMlOSxw9/PgV4JLAJuB541HCfJEnSWI06M/MmYBXwvq2MFXDMDk8kSZK0DUY9NG/V8O3zq+qu6WNJlvWWSpIkqaOudzN9veM+SZKkeTVqoclHAI8GHpzkyQwm/wLsCTyk52ySJEkjjZoz8zzgT4D9gTOn7b8NOKWnTJIkSZ2NmjPzCeATSV5SVZ+dp0waYZ999uHmm2/u/fckGX3Qdth777256aabev0dkqSdX9fnzHw2yQuAJwDLpu1/R1/BNLubb76Zqhp3jO3Wd1mSJC0OXReaXAv8S2ANg3kzLwUe02MuSZKkTrrezfTMqnolcHNV/Wfg94AD+oslSZLUTdcyc+fw5x1JHgXcDTy2n0iSJEnddV1o8oIkewHvAa5g8PTfj/WWSpIkqaOuE4BPH779bJILgGVVdUt/sSRJkrrpeplpi6r6NfD0JF/uIY8kSdI2GbVq9jFJfpDkV0n+IsmhSSaBdwMfmZ+IkiRJsxt1ZuZ9DFbN3hf4G+Ay4FNV9dSqOq/vcJIkSaOMmjNTVXXJ8P3fJrmxqj7QcyZJkqTORpWZvZK8eNp2pm97dkaSJI3bqDLzNeCFs2wXYJmRJEljNWqhyVfPVxBJkqTfxjbfmi1JkrSQWGYkSVLTLDOSJKlpncpMkpcm2WP4/u1JzkvylH6jSZIkjdb1zMx/rKrbkhwJPA/4BD4BWJIkLQBdy8y9w58vAD5SVZ8HdusnkiRJUnddy8xPknwU+CPgwiRLt+GzkiRJvelaSP4IuAg4tqp+CewD/IfeUkmSJHU0atXsfZLsAywDLgF+Mdz+NTA54rPnJtmc5KpZxh+f5BtJfp3kLTPGjk3y/STXJnnrtvxBkiRpcRm1nMFGBssWBFgJ3Dx8vxfwf4HHzvHZjwMfBD45y/hNwOuBE6bvTLIE+BDwHGATcHmS86vqeyOySpKkRWjOMzNV9diqOojBJaYXVtXyqtoX+ANGrMtUVZcyKCyzjW+uqsuBu2cMPR24tqquq6rfAJ8Gjh/9p0iSpMWo65yZp1XVhVMbVfVF4Nn9ROLRwPXTtjcN90mSJD1A1zLz8+HD8g5M8pgkpwK/6ClTtrKvZj04WZVkMsnkjTfe2FMkSZK0UHUtMycCK4DPAX8L7Dfc14dNwAHTtvcHbpjt4Ko6u6omqmpixYoVPUWSJEkL1agJwABU1U3AG3rOMuVy4OAkjwV+ArwM+Ffz9LslSVJj5iwzSb7AHJd4qupFc3x2HXAUsDzJJuA0YNfh59YmeQSD27v3BO5L8kbg0Kq6NcnrGEw6XgKcW1VXb9NfJUmSFo1RZ2beO/z5YuARwF8Mt08EfjzXB6tqzstQVfUzBpeQtjZ2IXDh1sYkSZKmm7PMVNXXAJKcXlXPmjb0hSSX9ppMkiSpg64TgFckOWhqYzifxdm2kiRp7DpNAAb+PXBJkuuG2wcCq3pJJEmStA263s30pSQHA48f7vqHqvp1f7EkSZK6GVlmkuwHvBZ4AoM7m77HYO2kzf1GkyRJGm3UqtlHMHjuCwwWjJy6m+mbwzFJkqSxGnVm5n3ACVX1rWn7Pp/kc8BHgWf0lkySJKmDUXcz7TmjyABQVd8G9ugnkiRJUnejykyS7L2Vnft0+KwkSVLvRhWS9wMXJ3l2kj2Gr6OALw7HJEmSxmrUE4DPTnIDcDr3v5vpv1TVF+YhnyRJ0pxG3ppdVRcAF8xDFkmSpG026tbstw/nx8w2fkySP9jxsSRJkroZdWbmuwwWlbwLuAK4EVgGHAwcDvwv4IxeE+oB6rQ94T89bNwxtludtue4I0iSdgKpqtEHDZYyOAJ4JHAncA1waVXd2W+8bTMxMVGTk5PjjtG7JHT577bQ7Sx/hySpH0k2VtXEqOO6rs30Q+CH251KkiRpB/NZMZIkqWmWGUmS1DTLjCRJalqnMpPkd5J8JclVw+0nJnl7v9EkSZJG63pm5r8DbwPuBqiqK4GX9RVKkiSpq65l5iFV9c0Z++7Z0WEkSZK2Vdcy8/Mkj2OwNhNJ/hD4aW+pJEmSOur0nBngtcDZwOOT/AT4EfDHvaWSJEnqqOtD864Dfj/J7sCDquq2fmNJkiR10/VupnuTvBu4Y6rIJLmi12SSJEkddJ0zc/Xw2IunraKdfiJJkiR117XM3FNVf8rgFu2/S/JUhpOBJUmSxqnrBOAAVNVnklwNrANW9pZKkiSpo65l5t9Mvamqq5McCZzQTyRJkqTu5iwzSY6pqq8Cj0nymBnDv+ovlkZJ2p+ytPfee487giRpJzDqzMyzga8CL9zKWAHn7fBEGqmq/+lKSebl90iStL3mLDNVddrw56vnJ44kSdK26fqcmTck2TMDH0tyRZLn9h1OkiRplK63Zv/rqroVeC6wH/Bq4N29pZIkSeqoa5mZmm16HPA/quo7+NA8SZK0AHQtMxuTXMygzFyUZA/gvv5iSZIkddP1OTMnAYcD11XVHUn2ZXCpSZIkaay6rpp9H3DFtO1fAL/oK5QkSVJXXS8zSZIkLUiWGUmS1LSuz5n5VJd9kiRJ863rmZknTN9IsgR46o6PI0mStG3mLDNJ3pbkNuCJSW4dvm4DNgOfn5eEkiRJc5izzFTVu6pqD+A9VbXn8LVHVe1bVW+bp4ySJEmz6nqZ6ZtJHja1kWSvJCf0lEmSJKmzrmXmtKq6ZWqjqn4JnNZPJEmSpO66lpmtHdf16cGSJEm96VpmJpOcmeRxSQ5K8n5gY5/BJEmSuuhaZtYAvwH+CvgMcCfw2r5CSZIkddV1babbgbcmeWhV/arnTJIkSZ11fQLwM5N8D/jecPtJST484jPnJtmc5KpZxpPkz5Ncm+TKJE+ZNnZvkm8PX+dvw98jSZIWma6Xmd4PPI/hStlV9R3gWSM+83Hg2DnGnw8cPHytAj4ybezOqjp8+HpRx4ySJGkR6rzQZFVdP2PXvSOOvxS4aY5Djgc+WQOXAXsleWTXPJIkSdC9zFyf5JlAJdktyVuAa7bzdz8amF6QNg33ASxLMpnkMh/OJ0mS5tL1WTGrgQ8wKBubgIvZ/ruZspV9Nfy5sqpuSHIQ8NUk362qf9zqlySrGFymYuXKldsZSZIktWbUQpP/dfj26Kp6eVU9vKr2q6o/rqpfbOfv3gQcMG17f+AGgKqa+nkdcAnw5Nm+pKrOrqqJqppYsWLFdkaSJEmtGXWZ6bgkuwJ9LCp5PvDK4V1NvwvcUlU/TbJ3kqUASZYDRzC8i0qSJGmmUZeZvgT8HNg9ya0MLg3V1M+q2nO2DyZZBxwFLE+yicFaTrsy+OBa4ELgOOBa4A7g1cOPHgJ8NMl9DMrWu6vKMiNJkrYqVTX7YLK0qn6d5PNVdfw85vqtTExM1OTk5Lhj7BSSMNe/DUmS+pZkY1VNjDpu1GWmbwx/3rr9kSRJkna8UZeZdkvyKuCZSV48c7CqzusnliRJUjejysxq4OXAXsALZ4wVYJmRJEljNWeZqaoNwIYkk1V1zjxlkiRJ6mzUc2b+FKCqzkny0hljZ/QZTJIkqYtRE4BfNu39zGfNzLWIpCRJ0rwYVWYyy/utbUuSJM27UWWmZnm/tW1JkqR5N+pupidNe/Lvg4fvGW4v6zWZJElSB6PuZloyX0EkSZJ+G6MuM0mSJC1olhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpabuMO4DmR5J5+UxVbfNnJEnaHpaZRcKSIUnaWXmZSZIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWm9lpkk5ybZnOSqWcaT5M+TXJvkyiRPmTb2qiQ/HL5e1WdOSZLUrr7PzHwcOHaO8ecDBw9fq4CPACTZBzgNeAbwdOC0JHv3mlSSJDWp1zJTVZcCN81xyPHAJ2vgMmCvJI8Engd8uapuqqqbgS8zdymSJEmL1LjnzDwauH7a9qbhvtn2S5Ik3c+4y0y2sq/m2P/AL0hWJZlMMnnjjTfu0HCSJGnhG3eZ2QQcMG17f+CGOfY/QFWdXVUTVTWxYsWK3oJKkqSFadxl5nzglcO7mn4XuKWqfgpcBDw3yd7Dib/PHe6TJEm6n136/PIk64CjgOVJNjG4Q2lXgKpaC1wIHAdcC9wBvHo4dlOS04HLh1/1jqqaayKxJElapHotM1V14ojxAl47y9i5wLl95JIkSTuPcV9mkiRJ2i6WGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTKj+1m3bh2HHXYYS5Ys4bDDDmPdunXjjiRJ0px2GXcALRzr1q3j1FNP5ZxzzuHII49kw4YNnHTSSQCceOKJY04nSdLWparGnWGHmZiYqMnJyXHHaNZhhx3GWWedxdFHH71l3/r161mzZg1XXXXVGJNJkhajJBuramLkcZYZTVmyZAl33XUXu+6665Z9d999N8uWLePee+8dYzJJ0mLUtcw4Z0ZbHHLIIWzYsOF++zZs2MAhhxwypkSSJI1mmdEWp556KieddBLr16/n7rvvZv369Zx00kmceuqp444mSdKsnACsLaYm+a5Zs4ZrrrmGQw45hHe+851O/pUkLWjOmZEkSQuSc2YkSdKiYJmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktS0XstMkmOTfD/JtUneupXxxyT5SpIrk1ySZP9pY/cm+fbwdX6fOSVJUrt6e2hekiXAh4DnAJuAy5OcX1Xfm3bYe4FPVtUnkhwDvAt4xXDszqo6vK98kiRp59DnmZmnA9dW1XVV9Rvg08DxM445FPjK8P36rYxLkiTNqc8y82jg+mnbm4b7pvsO8JLh+38B7JFk3+H2siSTSS5LckKPOSVJUsP6XJspW9k3c+2EtwAfTPInwKXAT4B7hmMrq+qGJAcBX03y3ar6xwf8kmQVsGq4+ask398h6bUc+Pm4Q0iz8N+nFir/be5Yj+lyUJ9lZhNwwLTt/YEbph9QVTcALwZI8lDgJVV1y7Qxquq6JJcATwYeUGaq6mzg7B7yL2pJJrushyGNg/8+tVD5b3M8+rzMdDlwcJLHJtkNeBlwv7uSkixPMpXhbcC5w/17J1k6dQxwBDB94rAkSRLQY5mpqnuA1wEXAdcAn6mqq5O8I8mLhocdBXw/yQ+AhwPvHO4/BJhM8h0GE4PfPeMuKEmSJABSNXMaizSYizS8hCctOP771ELlv83xsMxIkqSmuZyBJElqmmVGDzBqGQppXJKcm2RzkqvGnUWaLskBSdYnuSbJ1UneMO5Mi4mXmXQ/w2UofsC0ZSiAE52ArYUgybOAXzFYBuWwceeRpiR5JPDIqroiyR7ARuAE/985Pzwzo5m6LEMhjUVVXQrcNO4c0kxV9dOqumL4/jYGd/HOfOq9emKZ0UxdlqGQJM0iyYEMHvT69+NNsnhYZjRTl2UoJElbMXya/WeBN1bVrePOs1hYZjTTyGUoJEkPlGRXBkXmf1bVeePOs5hYZjTTyGUoJEn3lyTAOcA1VXXmuPMsNpYZ3c9sy1CMN5U0kGQd8A3gnyTZlOSkcWeSho4AXgEck+Tbw9dx4w61WHhrtiRJappnZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNW2XcQeQtHgkuRf4LrArcA/wCeC/VdV9Yw0mqWmWGUnz6c6qOhwgyX7AXwIPA07b3i9OsqSq7t3e75HUHi8zSRqLqtoMrAJel4ElSd6T5PIkVyb5twBJHpTkw0muTnJBkguT/OFw7MdJ/izJBuClSR6X5EtJNib5uySPHx63Islnh999eZIjxvaHS9rhPDMjaWyq6rokDwL2A44HbqmqpyVZCvzvJBcDTwUOBP7p8LhrgHOnfc1dVXUkQJKvAKur6odJngF8GDgG+ADw/qrakGQlgydcHzIvf6Sk3llmJI3b1ErtzwWeOHXWhcHlp4OBI4G/Hs6r+VmS9TM+/1ewZbXiZwJ/PVgmB4Clw5+/Dxw6bf+eSfaoqtt29B8jaf5ZZiSNTZKDgHuBzQxKzZqqumjGMS8Y8TW3D38+CPjl1JycGR4E/F5V3bmdkSUtQM6ZkTQWSVYAa4EP1mCRuIuAf5dk1+H47yTZHdgAvGQ4d+bhwFFb+76quhX4UZKXDj+fJE8aDl/MYAHVqd+9tcIjqVGemZE0nx6c5Nv8/1uzPwWcORz7GIO5MVdkcD3oRuAE4LPAPweuAn4A/D1wyyzf/3LgI0nePvwdnwa+A7we+FCSKxn8f+9SYPWO/uMkjYerZkta8JI8tKp+lWRf4JvAEVX1s3HnkrQweGZGUgsuSLIXsBtwukVG0nSemZEkSU1zArAkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtP+H5nwj1i6D1zqAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 648x432 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_to_plot = [degreesEff[0], degreesEff[1], degreesEff[2]]\n",
    "print(np.median(degreesEff[0]),np.median(degreesEff[1]),np.median(degreesEff[2]))\n",
    "print(np.mean(degreesEff[0]),np.mean(degreesEff[1]),np.mean(degreesEff[2]))\n",
    "fig = plt.figure(1, figsize=(9, 6))\n",
    "ax = fig.add_subplot(111)\n",
    "bp = ax.boxplot(data_to_plot)\n",
    "ax.set_xticklabels(['0', '1', '2'])\n",
    "ax.set_xlabel(\"Degree\")\n",
    "ax.set_ylabel(\"Effect size (Odds Ratio)\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ASD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"/Users/jiayao/Work/spark/dat/SFARI-Gene_genes_05-06-2019release_05-22-2019export.csv\")\n",
    "ASD_Core = df[df[\"gene-score\"].isin([1.0, 2.0])][\"gene-symbol\"].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "GeneGTF = ReadGeneGTF(\"../dat/gencode.v19.gene.gtf\")\n",
    "GeneDict = SNP2Gene(GeneGTF, \"../dat/GWAS/PGC/iPSYCH-PGC_ASD_Nov2017.INFO0.9.P0.05.sort.bed.gz\",\n",
    "                   CHROM=\"CHR\", POS=\"BP\", EFFECT=\"OR\", SE=\"SE\", P=\"P\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAD8CAYAAAB6paOMAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAEIlJREFUeJzt3X+MZeVdx/H3V7aA0AoLO1Tc3XaWdEMFYwNOCKVNbdkmBWq7GEtCU+2CazYorVSMdiuJNf5jmxipRFOzQpslIfxwi7IWqiI/YpTs1oHye0tZFoR1EablV5GUlvr1j/tM9zA7u3OWOffHzPN+JZM553mec+/3Pnu4nznn3HuIzESSVK+fGnYBkqThMggkqXIGgSRVziCQpMoZBJJUOYNAkipnEEhS5QwCSaqcQSBJlVsy7AIAli1bluPj48MuQ5IWlLvvvvu7mTk238cZiSAYHx9ncnJy2GVI0oISEf/VxeN4akiSKmcQSFLlDAJJqpxBIEmVMwgkqXIGgSRVziCQpMoZBJJUOYNAkipnEEhvwPjGm4ddgtQZg0CSKmcQSFLlDAJJqpxBIEmVMwgkqXIGgSRVziCQpMoZBJJUOYNAkipnEEhS5QwCSaqcQSBJlTMIJKlyBoEkVc4gkKTKGQSSVDmDQJIqZxBIUuVaBUFE/F5EPBQRD0bEtRFxeESsiojtEfFoRFwfEYeWsYeV9Z2lf7yfL0CSND9zBkFELAd+F5jIzF8ADgHOB74IXJ6Zq4HngfVlk/XA85n5DuDyMk6SNKLanhpaAvx0RCwBjgCeBs4EtpT+zcC5ZXltWaf0r4mI6KZcSVLX5gyCzPxv4M+BJ+kFwIvA3cALmflaGbYbWF6WlwNPlW1fK+OPnfm4EbEhIiYjYnJqamq+r0OS9Aa1OTW0lN5f+auAnwOOBM6eZWhOb3KAvr0NmZsycyIzJ8bGxtpXLI2I8Y03D7sEqRNtTg19EHg8M6cy80fAjcAZwNHlVBHACmBPWd4NrAQo/UcBz3VatSSpM22C4Eng9Ig4opzrXwM8DNwBfKyMWQfcVJa3lnVK/+2Zuc8RgSRpNLS5RrCd3kXfe4AHyjabgM8Cl0bETnrXAK4qm1wFHFvaLwU29qFuSVJHlsw9BDLz88DnZzTvAk6bZewPgPPmX5okaRD8ZrF0kLxIrMXGIJCkyhkEklQ5g0CSKmcQSFLlDAJJqpxBIEmVMwgkqXIGgSRVziCQpMoZBJJUOYNAkipnEEhS5QwCSaqcQSDNg3ci1WJgEEhS5QwCSaqcQSBJlTMIJKlyBoEkVc4gkKTKGQSSVDmDQJIqZxBIUuUMAkmqnEEgSZUzCKR58n5DWugMAkmqnEEgSZUzCCSpcgaBJFXOIJCkyhkEklQ5g0CSKmcQSFLlDAJJqpxBIEmVaxUEEXF0RGyJiG9HxI6IeHdEHBMRt0bEo+X30jI2IuKKiNgZEfdHxKn9fQnS8HmbCS1kbY8I/hL4p8x8J/AuYAewEbgtM1cDt5V1gLOB1eVnA/DlTiuWJHVqziCIiJ8B3gdcBZCZP8zMF4C1wOYybDNwblleC1ydPduAoyPi+M4rlyR1os0RwQnAFPDViPhWRFwZEUcCb83MpwHK7+PK+OXAU43td5e214mIDRExGRGTU1NT83oRkqQ3rk0QLAFOBb6cmacA/8ve00CziVnacp+GzE2ZOZGZE2NjY62KlSR1r00Q7AZ2Z+b2sr6FXjA8M33Kp/x+tjF+ZWP7FcCebsqVJHVtziDIzP8BnoqIE0vTGuBhYCuwrrStA24qy1uBT5ZPD50OvDh9CkmSNHqWtBz3aeCaiDgU2AVcSC9EboiI9cCTwHll7C3AOcBO4JUyVpI0oloFQWbeC0zM0rVmlrEJXDzPuiRJA+I3iyWpcgaBJFXOIJA64m0mtFAZBJJUOYNA6phHBlpoDAJJqpxBIHXIowEtRAaB1CeGghYKg0CSKmcQSFLlDAJJqpxBIEmVMwgkqXIGgdQHfmJIC4lBIPWZoaBRZxBIUuUMAkmqnEEgSZUzCCSpcgaBJFXOIJCkyhkEklQ5g0CSKmcQSFLlDAJJqpxBIEmVMwgkqXIGgdRH3nBOC4FBIEmVMwgkqXIGgSRVziCQBsTrBRpVBoEkVc4gkAbIowKNIoNAkipnEEhS5QwCSapc6yCIiEMi4lsR8fWyvioitkfEoxFxfUQcWtoPK+s7S/94f0qXJHXhYI4ILgF2NNa/CFyemauB54H1pX098HxmvgO4vIyTJI2oVkEQESuADwNXlvUAzgS2lCGbgXPL8tqyTulfU8ZLkkZQ2yOCLwF/CPxfWT8WeCEzXyvru4HlZXk58BRA6X+xjJckjaA5gyAifgV4NjPvbjbPMjRb9DUfd0NETEbE5NTUVKtiJUnda3NE8B7goxHxBHAdvVNCXwKOjoglZcwKYE9Z3g2sBCj9RwHPzXzQzNyUmROZOTE2NjavFyGNuuYXyfxSmUbNnEGQmZ/LzBWZOQ6cD9yemZ8A7gA+VoatA24qy1vLOqX/9szc54hAkjQa5vM9gs8Cl0bETnrXAK4q7VcBx5b2S4GN8ytRktRPS+Yesldm3gncWZZ3AafNMuYHwHkd1CZJGgC/WSxJlTMIJKlyBoEkVc4gkKTKGQSSVDmDQJIqZxBIUuUMAkmqnEEgDYH3G9IoMQgkqXIGgSRVziCQpMoZBJJUOYNAGhIvGGtUGASSVDmDQJIqZxBIUuUMAmmIxjfe7LUCDZ1BII0IA0HDYhBIUuUMAkmqnEEgSZUzCCSpcgaBJFXOIJCkyhkEklQ5g0CSKmcQSFLlDAJJqpxBII0Aby+hYTIIJKlyBoEkVc4gkKTKGQSSVDmDQJIqZxBIUuUMAkmq3JxBEBErI+KOiNgREQ9FxCWl/ZiIuDUiHi2/l5b2iIgrImJnRNwfEaf2+0VIi4nfKdCgtTkieA34/cz8eeB04OKIOAnYCNyWmauB28o6wNnA6vKzAfhy51VLkjozZxBk5tOZeU9Z/j6wA1gOrAU2l2GbgXPL8lrg6uzZBhwdEcd3XrkkqRMHdY0gIsaBU4DtwFsz82nohQVwXBm2HHiqsdnu0iZJGkGtgyAi3gx8DfhMZr50oKGztOUsj7chIiYjYnJqaqptGdKi5vUBDUOrIIiIN9ELgWsy88bS/Mz0KZ/y+9nSvhtY2dh8BbBn5mNm5qbMnMjMibGxsTdav7QoGQgapDafGgrgKmBHZv5Fo2srsK4srwNuarR/snx66HTgxelTSJKk0bOkxZj3AL8BPBAR95a2PwK+ANwQEeuBJ4HzSt8twDnATuAV4MJOK5YkdWrOIMjMf2f28/4Aa2YZn8DF86xLkjQgfrNYkipnEEhS5QwCSaqcQSCNKD9CqkExCCSpcgaBVPgXuGplEEhS5QwCSaqcQSBJlTMIJLw+oLoZBJJUOYNAkipnEEhS5QwCSaqcQSBJlTMIpBHnJ5rUbwaBJFXOIJBGmEcDGgSDQJIqZxBIC4BHBuong0CSKmcQSFLlDAJJqpxBoOp5/l21MwikBcLAUr8YBKqWb6xSj0GgKhkC0l4GgapiAEj7MghUnYUcBgu5do0ug0BV841VMghUEd/0pdkZBFq0fOOX2jEItCgZAlJ7BoG0wBhy6ppBoEVhf2+O4xtvXrRvnIv1dWnwDAItKgcKBEmzMwi06NTwpj/9Gmt4req/vgRBRJwVEY9ExM6I2NiP55Cm1f6m2Hz9tc6B5qfzIIiIQ4C/Bs4GTgI+HhEndf08qlPzjc43vr1mmwfnRm3144jgNGBnZu7KzB8C1wFr+/A8qox/+R682o+W1E4/gmA58FRjfXdpk94Q38QOzsyg3F8YOK+atqQPjxmztOU+gyI2ABvK6qsR8WAfaunaMuC7wy6ihUVXZ3yxz5Xs/3kW1VzOfH2DmteGRTWfI+DELh6kH0GwG1jZWF8B7Jk5KDM3AZsAImIyMyf6UEunrLNbC6HOhVAjWGfXFlKdXTxOP04N/SewOiJWRcShwPnA1j48jySpA50fEWTmaxHxKeCfgUOAr2TmQ10/jySpG/04NURm3gLcchCbbOpHHX1gnd1aCHUuhBrBOrtWVZ2Ruc91XElSRbzFhCRVru9BMNftJiLisIi4vvRvj4jxRt/nSvsjEfGhIdZ4aUQ8HBH3R8RtEfH2Rt+PI+Le8tPXi+It6rwgIqYa9fxWo29dRDxaftYNuc7LGzV+JyJeaPQNcj6/EhHP7u+jy9FzRXkd90fEqY2+gcxnixo/UWq7PyLuioh3NfqeiIgHylx28umSedT5/oh4sfFv+8eNvoHdkqZFnX/QqPHBsj8eU/oGOZ8rI+KOiNgREQ9FxCWzjOlu/8zMvv3Qu1j8GHACcChwH3DSjDG/A/xNWT4fuL4sn1TGHwasKo9zyJBq/ABwRFn+7ekay/rL/ZzDg6zzAuCvZtn2GGBX+b20LC8dVp0zxn+a3gcKBjqf5bneB5wKPLif/nOAb9D7bszpwPYhzOdcNZ4x/dz0buuyvdH3BLBsROby/cDX57u/9LvOGWM/Atw+pPk8Hji1LL8F+M4s/713tn/2+4igze0m1gKby/IWYE1ERGm/LjNfzczHgZ3l8QZeY2bekZmvlNVt9L4bMWjzuXXHh4BbM/O5zHweuBU4a0Tq/DhwbZ9qOaDM/DfguQMMWQtcnT3bgKMj4ngGOJ9z1ZiZd5UaYHj7Zpu53J+B3pLmIOsc5r75dGbeU5a/D+xg3zs0dLZ/9jsI2txu4idjMvM14EXg2JbbDqrGpvX0Unja4RExGRHbIuLcPtQ3rW2dv1YOE7dExPQX+wZ524/Wz1VOsa0Cbm80D2o+29jfaxnV26jM3DcT+JeIuDt63+QftndHxH0R8Y2IOLm0jeRcRsQR9N48v9ZoHsp8Ru90+SnA9hldne2fffn4aEOb203sb0yrW1V0oPXzRMSvAxPALzea35aZeyLiBOD2iHggMx8bUp3/CFybma9GxEX0jrTObLltVw7muc4HtmTmjxttg5rPNoa9b7YWER+gFwTvbTS/p8zlccCtEfHt8hfxMNwDvD0zX46Ic4B/AFYzgnNZfAT4j8xsHj0MfD4j4s30wugzmfnSzO5ZNnlD+2e/jwja3G7iJ2MiYglwFL1Dt1a3qhhQjUTEB4HLgI9m5qvT7Zm5p/zeBdxJL7n7Yc46M/N7jdr+FvilttsOss6G85lx6D3A+Wxjf69lkPM5p4j4ReBKYG1mfm+6vTGXzwJ/T39OrbaSmS9l5stl+RbgTRGxjBGby4YD7ZsDmc+IeBO9ELgmM2+cZUh3+2efL3gsoXehYhV7LwSdPGPMxbz+YvENZflkXn+xeBf9uVjcpsZT6F3QWj2jfSlwWFleBjxKny50tazz+MbyrwLbcu/Fo8dLvUvL8jHDqrOMO5HexbcYxnw2nnOc/V/g/DCvvxj3zUHPZ4sa30bv+tkZM9qPBN7SWL4LOGuIc/mz0//W9N5Anyzz2mp/GVSdpX/6j9EjhzWfZW6uBr50gDGd7Z99m+xGsefQu+L9GHBZaftTen9ZAxwO/F3Zmb8JnNDY9rKy3SPA2UOs8V+BZ4B7y8/W0n4G8EDZeR8A1g95Lv8MeKjUcwfwzsa2v1nmeCdw4TDrLOt/AnxhxnaDns9rgaeBH9H7K2o9cBFwUekPev+TpcdKPRODns8WNV4JPN/YNydL+wllHu8r+8RlQ57LTzX2zW00gmu2/WVYdZYxF9D7oEpzu0HP53vpnc65v/Fve06/9k+/WSxJlfObxZJUOYNAkipnEEhS5QwCSaqcQSBJlTMIJKlyBoEkVc4gkKTK/T8MNEN/f1xuugAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "Effects = []\n",
    "for k, v in GeneDict.items():\n",
    "    if len(v.SNPs) == 0:\n",
    "        continue\n",
    "    Effects.append(v.max_eff)\n",
    "plt.hist(Effects, bins=100)\n",
    "plt.xlim([0,2])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MSNP1AS\n"
     ]
    }
   ],
   "source": [
    "degreesEff = {}\n",
    "degreesEff[0] = []\n",
    "degreesNode = {}\n",
    "degreesNode[0] = []\n",
    "for gene_symbol in ASD_Core:\n",
    "    try:\n",
    "        ensg = symbol2ensg[gene_symbol]\n",
    "        gene = GeneDict[ensid]\n",
    "    except KeyError:\n",
    "        print (gene_symbol)\n",
    "        continue\n",
    "    if len(gene.SNPs) > 0:\n",
    "        degreesEff[0].append(gene.max_eff)\n",
    "        degreesNode[0].append(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "Brain_Reg_Net2_csv_trim = Brain_Reg_Net2_csv.drop_duplicates(\n",
    "    subset=[\"Transcription_Factor\", \"Target_Gene\"], keep=\"first\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "SymbolGeneDict = {}\n",
    "for k,v in GeneDict.items():\n",
    "    SymbolGeneDict[v.GeneName] = v"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4148287"
     ]
    }
   ],
   "source": [
    "graph = {}\n",
    "count = 0\n",
    "for i, row in Brain_Reg_Net2_csv_trim.iterrows():\n",
    "    g1, g2 = row[\"Transcription_Factor\"], row[\"Target_Gene\"]\n",
    "    #print(g1, g2)\n",
    "    if g1 not in SymbolGeneDict or g2 not in SymbolGeneDict:\n",
    "        continue\n",
    "    if g1 not in graph:\n",
    "        graph[g1] = []\n",
    "    if g2 not in graph:\n",
    "        graph[g2] = []\n",
    "    graph[g1].append(g2)\n",
    "    graph[g2].append(g1)\n",
    "    if count % 1000 == 0:\n",
    "        sys.stdout.write(\"\\r%d\"%i)\n",
    "    count += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "degreesNode[0] = [gene.GeneName for gene in degreesNode[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "PreNodes = set(degreesNode[0])\n",
    "print(len(PreNodes))\n",
    "for i in range(1,11,1):\n",
    "    degreesEff[i] = []\n",
    "    degreesNode[i] = []\n",
    "    for gene in degreesNode[i-1]:\n",
    "        symbol = gene\n",
    "        if symbol in graph:\n",
    "            for node in graph[symbol]:\n",
    "                if node not in PreNodes:\n",
    "                    degreesNode[i].append(node)\n",
    "    degreesNode[i] = list(set(degreesNode[i]))\n",
    "    for symbol in degreesNode[i]:\n",
    "        degreesEff[i].append(SymbolGeneDict[symbol].max_eff)\n",
    "    print(len(degreesNode[i]))\n",
    "    PreNodes = PreNodes.union(set(degreesNode[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.14214 nan nan\n",
      "1.1421399999999995 nan nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/jiayao/anaconda3/lib/python3.7/site-packages/numpy/core/fromnumeric.py:2920: RuntimeWarning: Mean of empty slice.\n",
      "  out=out, **kwargs)\n",
      "/Users/jiayao/anaconda3/lib/python3.7/site-packages/numpy/core/_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjMAAAF3CAYAAABQc8olAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAGaJJREFUeJzt3X20ZXV93/H3h2EQHxhQZhQFxgFDgqPFpysqWCFoFPEBajSFpba1thO7UMDWGExNJilZ0dbnLgWcIOJTIUZIVIpCW1FCRZw7+MQ4UVloZAQ6gygPUXn89o9zxlyv996zB2bfc37M+7XWXefs/dvn3M9da9asz9r7t/cvVYUkSVKrdhl3AEmSpPvDMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmrbruAPsSMuXL69Vq1aNO4YkSdoBNmzYcFNVrRh13AOqzKxatYrp6elxx5AkSTtAkn/ocpyXmSRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTeutzCQ5O8mWJFfPM/7KJN8c/nw5yZNmjB2d5DtJrklyal8ZJUlS+/o8M3MOcPQC498HjqiqQ4DTgHUASZYAHwBeCKwGTkiyuseckiSpYb2Vmaq6DLh5gfEvV9VPhptfAfYbvj8UuKaqrq2qO4HzgGP7yilJkto2KXNmXgt8bvh+X+C6GWObh/skSZJ+zdjXZkry2wzKzLO37ZrjsFrg82uANQArV67c4fkkSdJkG+uZmSSHAGcBx1bVj4e7NwP7zzhsP+D6+b6jqtZV1VRVTa1YMXJhTUmS9AAztjKTZCVwAfDqqvrujKH1wEFJDkiyG3A88JlxZJQkSZOvt8tMSc4FjgSWJ9kMrAWWAlTVmcCfAHsDpycBuHt4huXuJK8HLgaWAGdX1ca+ckqSpLalat7pKM2Zmpqq6enpcceQJEk7QJINVTU16rhJuZtJkiTpPrHMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKb1VmaSnJ1kS5Kr5xk/OMkVSe5I8qZZY29MsjHJ1UnOTbJ7XzklSVLb+jwzcw5w9ALjNwMnAe+cuTPJvsP9U1X1RGAJcHxPGSVJUuN6KzNVdRmDwjLf+JaqWg/cNcfwrsCDk+wKPAS4vp+UkiSpdRM3Z6aqfsTgbM0PgRuAW6rqkvGmkiRJk2riykyShwPHAgcAjwEemuRVCxy/Jsl0kumtW7cuVkxJkjQhJq7MAM8Dvl9VW6vqLuAC4LD5Dq6qdVU1VVVTK1asWLSQkiRpMkximfkh8MwkD0kS4LnApjFnkiRJE2rXvr44ybnAkcDyJJuBtcBSgKo6M8k+wDSwDLg3ySnA6qq6MsmngKuAu4GvAev6yilJktrWW5mpqhNGjN8I7DfP2FoG5UeSJGlBk3iZSZIkqTPLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKatmvXA5M8HHgM8HPgB1V1b2+pJEmSOlqwzCTZEzgROAHYDdgK7A48KslXgNOr6tLeU0qSJM1j1JmZTwEfBf55Vf105kCSpwGvTnJgVX2or4CSJEkLWbDMVNXvLDC2AdiwwxNJkiRth+2ZM/NS4DnDzS9V1Wf7iSRJktRdp7uZkrwdOBn49vDnpCRv6zOYJElSF13PzBwDPHnbHUxJPgJ8DXhLX8EkSZK62J7nzOw14/2eow5OcnaSLUmunmf84CRXJLkjyZtmje2V5FNJ/j7JpiTP2o6ckiRpJ9L1zMzbgK8luRQIg7kzo87KnAO8n8HdUHO5GTgJOG6OsfcBn6+qlyfZDXhIx5ySJGkn06nMVNW5Sb4IPJ1BmfnDqrpxxGcuS7JqgfEtwJYkL5q5P8kyBmXp3wyPuxO4s0tOSZK081nwMlOSg4evTwUeDWwGrgMeM9zXhwMZPJzvw0m+luSsJA/t6XdJkqTGjToz8x+BNcC75hgr4KgdnmiQ6anAG6rqyiTvA04F/niug5OsGWZk5cqVPcSRJEmTbNRD89YM376wqn4xcyzJ7j1l2gxsrqorh9ufYlBm5lRV64B1AFNTU9VTJkmSNKG63s305Y777rfhXJzrkvzWcNdzGTzbRpIk6deMWmhyH2Bf4MFJnsJg8i/AMkbcYZTkXOBIYHmSzcBaYClAVZ05/O7p4Xfdm+QUYHVV3Qq8AfjE8E6ma4HX3Lc/T5IkPdCNmjPzAgZ3Fe0HvHvG/tuAP1rog1V1wojxG4ffO9fY14GpEdkkSZJGzpn5CPCRJL9bVecvUiZJkqTOuj5n5vzh82CeAOw+Y/9/6SuYJElSF10XmjwT+JcM5rIEeAXw2B5zSZIkddL1bqbDqupfAT+pqj8DngXs318sSZKkbrqWmZ8PX3+W5DHAXcAB/USSJEnqrutCkxcm2Qt4B3AVg6f/ntVbKkmSpI66TgA+bfj2/CQXArtX1S39xZIkSeqm62WmX6qqO4BDk/yvHvJIkiRtl1GrZh+V5LtJbk/y8SSrk0wDbwfOWJyIkiRJ8xt1ZuZdDFak3pvBgo9fAT5WVU+rqgv6DidJkjTKqDkzVVVfHL7/2yRbq+p9PWeSJEnqbFSZ2SvJy2ZsZ+a2Z2ckSdK4jSozXwJeMs92AZYZSZI0VqMWmnzNYgWRJEm6L7b71mxJkqRJYpmRJElNs8xIkqSmdSozSV6RZI/h+7cmuSDJU/uNJkmSNFrXMzN/XFW3JXk28ALgI/gEYEmSNAG6lpl7hq8vAs6oqk8Du/UTSZIkqbuuZeZHST4I/B5wUZIHbcdnJUmSetO1kPwecDFwdFX9FHgE8Ae9pZIkSepowYfmJXnEjM0vzth3BzDdXyxJkqRuRi1nsIHBsgUBVgI/Gb7fC/ghcECv6SRJkkZY8DJTVR1QVQcyuMT0kqpaXlV7Ay/GdZkkSdIE6Dpn5ulVddG2jar6HHBEP5EkSZK6G3WZaZubkrwV+DiDy06vAn7cWypJkqSOup6ZOQFYAfwN8LfAI4f7JEmSxqrTmZmquhk4uecskiRJ223UrdmfZXBZaU5V9dIdnkiSJGk7jDoz887h68uAfRjMmYHBJaYf9JRJkiSpswXLTFV9CSDJaVX1nBlDn01yWa/JJEmSOug6AXhFkgO3bSQ5gMGEYEmSpLHqemv2G4EvJrl2uL0KWNNLIkmSpO3Q9W6mzyc5CDh4uOvvq+qO/mJJkiR1M7LMJHkkcCLwBAZ3Nn0b+ACwpd9okiRJoy04ZybJ4cD64eZH+ae7mb46HJMkSRqrUWdm3gUcV1Vfm7Hv00n+Bvgg8IzekkmSJHUw6m6mZbOKDABV9XVgj34iSZIkdTeqzCTJw+fY+YgOn5UkSerdqELyHuCSJEck2WP4cyTwueGYJEnSWI16AvC6JNcDp/GrdzP9eVV9dhHySZIkLWjkrdlVdSFw4SJkUVd/uue4E+w4f3rLuBNIkho3atXstwKnV9XN84wfBTxkWHhmj50NvBjYUlVPnGP8YODDwFOB/1xV75w1vgSYBn5UVS/u+PfsHCwAkiT90qgzM99isKjkL4CrgK3A7sBBwJOB/w38xTyfPQd4P4Pn08zlZuAk4Lh5xk8GNgHLRmSUJEk7sQUnAFfVp6vqcOB1wEZgCXArg4fnHVpVb6yqrfN89jIGhWW+795SVeuBu2aPJdkPeBFwVtc/RJIk7Zy6rs30PeB7PWeZ6b3Am/FZNpIkaYSJe1ZMkm3zbDZ0PH5Nkukk01u3znmSSJIkPYBNXJkBDgdemuQHwHnAUUk+Pt/BVbWuqqaqamrFihWLlVGSJE2IiSszVfWWqtqvqlYBxwNfqKpXjTmWJEmaUJ3mzCT5TeAM4FFV9cQkhwAvrao/X+Az5wJHAsuTbAbWAksBqurMJPswuPV6GXBvklOA1VV16/35gyRJ0s4lVTX6oORLwB8AH6yqpwz3XT3X82PGaWpqqqanp8cdQ5Ik7QBJNlTV1Kjjul5mekhVfXXWvru3P5YkSdKO1bXM3JTkcQzWZiLJy4EbekslSZLUUac5M8CJwDrg4CQ/Ar4POClXkiSNXdeH5l0LPC/JQ4Fdquq2fmNJkiR10+kyU5J7krwd+Nm2IpPkql6TSZIkddB1zszG4bGXJHnEcF/6iSRJktRd1zJzd1W9GfhL4O+SPI3hZGBJkqRx6joBOABV9ckkG4FzgZW9pZIkSeqoa5n5d9veVNXGJM8GjusnkiRJUncLlpkkR1XVF4DHJnnsrOHb+4slSZLUzagzM0cAXwBeMsdYARfs8ESSJEnbYcEyU1Vrh6+vWZw4kiRJ26frc2ZOTrIsA2cluSrJ8/sOJ0mSNErXW7P/bVXdCjwfeCTwGuDtvaWSJEnqqGuZ2faAvGOAD1fVN/CheZIkaQJ0LTMbklzCoMxcnGQP4N7+YkmSJHXT9TkzrwWeDFxbVT9LsjeDS02SJElj1XXV7HuBq2Zs/xj4cV+hJEmSuup6mUmSJGkiWWYkSVLTuj5n5mNd9kmSJC22rmdmnjBzI8kS4Gk7Po4kSdL2WbDMJHlLktuAQ5LcOvy5DdgCfHpREkqSJC1gwTJTVW+rqj2Ad1TVsuHPHlW1d1W9ZZEySpIkzavrZaavJtlz20aSvZIc11MmSZKkzrqWmbVVdcu2jar6KbC2n0iSJEnddS0zcx3X9enBkiRJvelaZqaTvDvJ45IcmOQ9wIY+g0mSJHXRtcy8AbgT+Cvgk8DPgRP7CiVJktRV17WZ/hE4NcnDqur2njNJkiR11vUJwIcl+Tbw7eH2k5Kc3msySZKkDrpeZnoP8AKGK2VX1TeA5/QVSpIkqavOC01W1XWzdt2zg7NIkiRtt663V1+X5DCgkuwGnARs6i+WJElSN13PzLyOwd1L+wKbgSfj3UySJGkCLHhmJsl/rao/BH67ql65SJkkSZI6G3Vm5pgkSwEXlZQkSRNp1JyZzwM3AQ9NcisQoLa9VtWynvNJkiQtaNSZmbdW1Z7A/6yqZVW1x8zXxQgoSZK0kFFl5orh6619B5EkSbovRl1m2i3JvwYOS/Ky2YNVdUE/sSRJkroZVWZeB7wS2At4yayxAiwzkiRprBYsM1V1OXB5kumq+tAiZZIkSepswTkzSd4MUFUfSvKKWWN/MeKzZyfZkuTqecYPTnJFkjuSvGnG/v2TXJpkU5KNSU7u/udIkqSdzagJwMfPeD/7WTNHj/jsOSOOuZnBsgjvnLX/buA/VdXjgWcCJyZZPeJ3SZKkndSoMpN53s+1/Suq6jIGhWW+8S1VtR64a9b+G6rqquH72xisAbXviJySJGknNarM1Dzv59re4ZKsAp4CXNn375IkSW0adTfTk2Y8+ffBw/cMt3fvM1iShwHnA6dU1bzPuUmyBlgDsHLlyj4jSZKkCTTqbqYlixVkpuF6UOcDnxj1LJuqWgesA5iamur9bJEkSZosoy4zLbokAT4EbKqqd487jyRJmmyjLjPdZ0nOBY4ElifZDKwFlgJU1ZlJ9gGmgWXAvUlOAVYDhwCvBr6V5OvDr/ujqrqor6ySJKldvZWZqjphxPiNwH5zDF3OiDulJEmStpm4y0ySJEnbwzIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTeutzCQ5O8mWJFfPM35wkiuS3JHkTbPGjk7ynSTXJDm1r4ySJKl9fZ6ZOQc4eoHxm4GTgHfO3JlkCfAB4IXAauCEJKt7yihJkhrXW5mpqssYFJb5xrdU1XrgrllDhwLXVNW1VXUncB5wbF85JUlS2yZxzsy+wHUztjcP90mSJP2aSSwzmWNfzXtwsibJdJLprVu39hhLkiRNokksM5uB/Wds7wdcP9/BVbWuqqaqamrFihW9h5MkSZNlEsvMeuCgJAck2Q04HvjMmDNJkqQJtWtfX5zkXOBIYHmSzcBaYClAVZ2ZZB9gGlgG3JvkFGB1Vd2a5PXAxcAS4Oyq2thXTkmS1LbeykxVnTBi/EYGl5DmGrsIuKiPXJIk6YFlEi8zSZIkdWaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKZZZiRJUtMsM5IkqWmWGUmS1DTLjCRJapplRpIkNc0yI0mSmmaZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElqmmVGkiQ1zTIjSZKaZpmRJElNs8xIkqSmWWYkSVLTLDOSJKlplhlJktQ0y4wkSWqaZUaSJDXNMiNJkppmmZEkSU2zzEiSpKb1WmaSnJ1kS5Kr5xlPkv+e5Jok30zy1Blj/y3JxiSbhsekz6ySJKlNfZ+ZOQc4eoHxFwIHDX/WAGcAJDkMOBw4BHgi8HTgiD6DSpKkNvVaZqrqMuDmBQ45FvhoDXwF2CvJo4ECdgd2Ax4ELAX+X59ZJUlSm8Y9Z2Zf4LoZ25uBfavqCuBS4Ibhz8VVtWkM+SRJ0oQbd5mZax5MJfkN4PHAfgwKz1FJnjPnFyRrkkwnmd66dWuPUSVJ0iQad5nZDOw/Y3s/4HrgXwBfqarbq+p24HPAM+f6gqpaV1VTVTW1YsWK3gNLkqTJsuuYf/9ngNcnOQ94BnBLVd2Q5IfAv0/yNgZnb44A3jvqyzZs2HBTkn/oNfHOYzlw07hDSPPw36cmlf82d6zHdjmo1zKT5FzgSGB5ks3AWgaTeamqM4GLgGOAa4CfAa8ZfvRTwFHAtxhMBv58VX121O+rKk/N7CBJpqtqatw5pLn471OTyn+b49FrmamqE0aMF3DiHPvvAX6/r1ySJOmBY9xzZiRJku4Xy4zms27cAaQF+O9Tk8p/m2OQwZUeSZKkNnlmRpIkNc0yo1+T5Ogk3xkuAHrquPNI24xavFYalyT7J7l0uDjyxiQnjzvTzsTLTPoVSZYA3wV+h8FDDdcDJ1TVt8caTAKGTwK/ncGabk8cdx5pm+G6go+uqquS7AFsAI7z/87F4ZkZzXYocE1VXVtVdwLnMVgQVBq7DovXSmNRVTdU1VXD97cBmxgsx6NFYJnRbHMu/jmmLJLUnCSrgKcAV443yc7DMqPZ5lz8c9FTSFKDkjwMOB84papuHXeenYVlRrPNt/inJGkBSZYyKDKfqKoLxp1nZ2KZ0WzrgYOSHJBkN+B4BguCSpLmkSTAh4BNVfXucefZ2Vhm9Cuq6m7g9cDFDCawfbKqNo43lTQwXLz2CuC3kmxO8tpxZ5KGDgdeDRyV5OvDn2PGHWpn4a3ZkiSpaZ6ZkSRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUtF3HHUDSziPJPcC3gKXA3cBHgPdW1b1jDSapaZYZSYvp51X1ZIAkjwT+B7AnsPb+fnGSJVV1z/39Hknt8TKTpLGoqi3AGuD1GViS5B1J1if5ZpLfB0iyS5LTk2xMcmGSi5K8fDj2gyR/kuRy4BVJHpfk80k2JPm7JAcPj1uR5Pzhd69PcvjY/nBJO5xnZiSNTVVdm2QX4JHAscAtVfX0JA8C/m+SS4CnAauAfzY8bhNw9oyv+UVVPRsgyf8BXldV30vyDOB04CjgfcB7quryJCsZPOH68YvyR0rqnWVG0rhtW6n9+cAh2866MLj8dBDwbOCvh/Nqbkxy6azP/xX8crXiw4C/HiyTA8CDhq/PA1bP2L8syR5VdduO/mMkLT7LjKSxSXIgcA+whUGpeUNVXTzrmBeN+Jp/HL7uAvx025ycWXYBnlVVP7+fkSVNIOfMSBqLJCuAM4H312CRuIuB/5Bk6XD8N5M8FLgc+N3h3JlHAUfO9X1VdSvw/SSvGH4+SZ40HL6EwQKq2373XIVHUqM8MyNpMT04ydf5p1uzPwa8ezh2FoO5MVdlcD1oK3AccD7wXOBq4LvAlcAt83z/K4Ezkrx1+DvOA74BnAR8IMk3Gfy/dxnwuh39x0kaD1fNljTxkjysqm5PsjfwVeDwqrpx3LkkTQbPzEhqwYVJ9gJ2A06zyEiayTMzkiSpaU4AliRJTbPMSJKkpllmJElS0ywzkiSpaZYZSZLUNMuMJElq2v8HfFH+AifeZf4AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 648x432 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_to_plot = [degreesEff[0], degreesEff[1], degreesEff[2]]\n",
    "print(np.median(degreesEff[0]),np.median(degreesEff[1]),np.median(degreesEff[2]))\n",
    "print(np.mean(degreesEff[0]),np.mean(degreesEff[1]),np.mean(degreesEff[2]))\n",
    "fig = plt.figure(1, figsize=(9, 6))\n",
    "ax = fig.add_subplot(111)\n",
    "bp = ax.boxplot(data_to_plot)\n",
    "ax.set_xticklabels(['0', '1', '2'])\n",
    "ax.set_xlabel(\"Degree\")\n",
    "ax.set_ylabel(\"Effect size (Odds Ratio)\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "from biomart import BiomartServer\n",
    "server = BiomartServer( \"http://www.ensembl.org/biomart/martservice\" )\n",
    "# set verbose to True to get some messages\n",
    "server.verbose = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[BiomartServer:'http://www.ensembl.org/biomart/martservice'] Fetching databases\n",
      "{'ENSEMBL_MART_ENSEMBL': Ensembl Genes 98,\n",
      " 'ENSEMBL_MART_FUNCGEN': Ensembl Regulation 98,\n",
      " 'ENSEMBL_MART_GENOMIC': Genomic features 98,\n",
      " 'ENSEMBL_MART_MOUSE': Mouse strains 98,\n",
      " 'ENSEMBL_MART_ONTOLOGY': Ontology,\n",
      " 'ENSEMBL_MART_SEQUENCE': Sequence,\n",
      " 'ENSEMBL_MART_SNP': Ensembl Variation 98}\n"
     ]
    }
   ],
   "source": [
    "server.show_databases()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[BiomartDataset:'hsapiens_gene_ensembl'] Fetching attributes\n",
      "{'3_utr_end': '3' UTR end' (default: False),\n",
      " '3_utr_start': '3' UTR start' (default: False),\n",
      " '3utr': '3' UTR' (default: False),\n",
      " '5_utr_end': '5' UTR end' (default: False),\n",
      " '5_utr_start': '5' UTR start' (default: False),\n",
      " '5utr': '5' UTR' (default: False),\n",
      " 'abrachyrhynchus_homolog_associated_gene_name': 'Pink-footed goose gene name' (default: False),\n",
      " 'abrachyrhynchus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_chrom_end': 'Pink-footed goose chromosome/scaffold end (bp)' (default: False),\n",
      " 'abrachyrhynchus_homolog_chrom_start': 'Pink-footed goose chromosome/scaffold start (bp)' (default: False),\n",
      " 'abrachyrhynchus_homolog_chromosome': 'Pink-footed goose chromosome/scaffold name' (default: False),\n",
      " 'abrachyrhynchus_homolog_dn': 'dN with Pink-footed goose' (default: False),\n",
      " 'abrachyrhynchus_homolog_ds': 'dS with Pink-footed goose' (default: False),\n",
      " 'abrachyrhynchus_homolog_ensembl_gene': 'Pink-footed goose gene stable ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_ensembl_peptide': 'Pink-footed goose protein or transcript stable ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_goc_score': 'Pink-footed goose Gene-order conservation score' (default: False),\n",
      " 'abrachyrhynchus_homolog_orthology_confidence': 'Pink-footed goose orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'abrachyrhynchus_homolog_orthology_type': 'Pink-footed goose homology type' (default: False),\n",
      " 'abrachyrhynchus_homolog_perc_id': '%id. target Pink-footed goose gene identical to query gene' (default: False),\n",
      " 'abrachyrhynchus_homolog_perc_id_r1': '%id. query gene identical to target Pink-footed goose gene' (default: False),\n",
      " 'abrachyrhynchus_homolog_subtype': 'Last common ancestor with Pink-footed goose' (default: False),\n",
      " 'abrachyrhynchus_homolog_wga_coverage': 'Pink-footed goose Whole-genome alignment coverage' (default: False),\n",
      " 'acalliptera_homolog_associated_gene_name': 'Eastern happy gene name' (default: False),\n",
      " 'acalliptera_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acalliptera_homolog_chrom_end': 'Eastern happy chromosome/scaffold end (bp)' (default: False),\n",
      " 'acalliptera_homolog_chrom_start': 'Eastern happy chromosome/scaffold start (bp)' (default: False),\n",
      " 'acalliptera_homolog_chromosome': 'Eastern happy chromosome/scaffold name' (default: False),\n",
      " 'acalliptera_homolog_dn': 'dN with Eastern happy' (default: False),\n",
      " 'acalliptera_homolog_ds': 'dS with Eastern happy' (default: False),\n",
      " 'acalliptera_homolog_ensembl_gene': 'Eastern happy gene stable ID' (default: False),\n",
      " 'acalliptera_homolog_ensembl_peptide': 'Eastern happy protein or transcript stable ID' (default: False),\n",
      " 'acalliptera_homolog_goc_score': 'Eastern happy Gene-order conservation score' (default: False),\n",
      " 'acalliptera_homolog_orthology_confidence': 'Eastern happy orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acalliptera_homolog_orthology_type': 'Eastern happy homology type' (default: False),\n",
      " 'acalliptera_homolog_perc_id': '%id. target Eastern happy gene identical to query gene' (default: False),\n",
      " 'acalliptera_homolog_perc_id_r1': '%id. query gene identical to target Eastern happy gene' (default: False),\n",
      " 'acalliptera_homolog_subtype': 'Last common ancestor with Eastern happy' (default: False),\n",
      " 'acalliptera_homolog_wga_coverage': 'Eastern happy Whole-genome alignment coverage' (default: False),\n",
      " 'acarolinensis_homolog_associated_gene_name': 'Anole lizard gene name' (default: False),\n",
      " 'acarolinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acarolinensis_homolog_chrom_end': 'Anole lizard chromosome/scaffold end (bp)' (default: False),\n",
      " 'acarolinensis_homolog_chrom_start': 'Anole lizard chromosome/scaffold start (bp)' (default: False),\n",
      " 'acarolinensis_homolog_chromosome': 'Anole lizard chromosome/scaffold name' (default: False),\n",
      " 'acarolinensis_homolog_dn': 'dN with Anole lizard' (default: False),\n",
      " 'acarolinensis_homolog_ds': 'dS with Anole lizard' (default: False),\n",
      " 'acarolinensis_homolog_ensembl_gene': 'Anole lizard gene stable ID' (default: False),\n",
      " 'acarolinensis_homolog_ensembl_peptide': 'Anole lizard protein or transcript stable ID' (default: False),\n",
      " 'acarolinensis_homolog_goc_score': 'Anole lizard Gene-order conservation score' (default: False),\n",
      " 'acarolinensis_homolog_orthology_confidence': 'Anole lizard orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acarolinensis_homolog_orthology_type': 'Anole lizard homology type' (default: False),\n",
      " 'acarolinensis_homolog_perc_id': '%id. target Anole lizard gene identical to query gene' (default: False),\n",
      " 'acarolinensis_homolog_perc_id_r1': '%id. query gene identical to target Anole lizard gene' (default: False),\n",
      " 'acarolinensis_homolog_subtype': 'Last common ancestor with Anole lizard' (default: False),\n",
      " 'acarolinensis_homolog_wga_coverage': 'Anole lizard Whole-genome alignment coverage' (default: False),\n",
      " 'acitrinellus_homolog_associated_gene_name': 'Midas cichlid gene name' (default: False),\n",
      " 'acitrinellus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acitrinellus_homolog_chrom_end': 'Midas cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'acitrinellus_homolog_chrom_start': 'Midas cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'acitrinellus_homolog_chromosome': 'Midas cichlid chromosome/scaffold name' (default: False),\n",
      " 'acitrinellus_homolog_dn': 'dN with Midas cichlid' (default: False),\n",
      " 'acitrinellus_homolog_ds': 'dS with Midas cichlid' (default: False),\n",
      " 'acitrinellus_homolog_ensembl_gene': 'Midas cichlid gene stable ID' (default: False),\n",
      " 'acitrinellus_homolog_ensembl_peptide': 'Midas cichlid protein or transcript stable ID' (default: False),\n",
      " 'acitrinellus_homolog_goc_score': 'Midas cichlid Gene-order conservation score' (default: False),\n",
      " 'acitrinellus_homolog_orthology_confidence': 'Midas cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acitrinellus_homolog_orthology_type': 'Midas cichlid homology type' (default: False),\n",
      " 'acitrinellus_homolog_perc_id': '%id. target Midas cichlid gene identical to query gene' (default: False),\n",
      " 'acitrinellus_homolog_perc_id_r1': '%id. query gene identical to target Midas cichlid gene' (default: False),\n",
      " 'acitrinellus_homolog_subtype': 'Last common ancestor with Midas cichlid' (default: False),\n",
      " 'acitrinellus_homolog_wga_coverage': 'Midas cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'affy_hc_g110': 'AFFY HC G110 probe' (default: False),\n",
      " 'affy_hg_focus': 'AFFY HG Focus probe' (default: False),\n",
      " 'affy_hg_u133_plus_2': 'AFFY HG U133 Plus 2 probe' (default: False),\n",
      " 'affy_hg_u133a': 'AFFY HG U133A probe' (default: False),\n",
      " 'affy_hg_u133a_2': 'AFFY HG U133A 2 probe' (default: False),\n",
      " 'affy_hg_u133b': 'AFFY HG U133B probe' (default: False),\n",
      " 'affy_hg_u95a': 'AFFY HG U95A probe' (default: False),\n",
      " 'affy_hg_u95av2': 'AFFY HG U95Av2 probe' (default: False),\n",
      " 'affy_hg_u95b': 'AFFY HG U95B probe' (default: False),\n",
      " 'affy_hg_u95c': 'AFFY HG U95C probe' (default: False),\n",
      " 'affy_hg_u95d': 'AFFY HG U95D probe' (default: False),\n",
      " 'affy_hg_u95e': 'AFFY HG U95E probe' (default: False),\n",
      " 'affy_hta_2_0': 'AFFY HTA 2 0 probe' (default: False),\n",
      " 'affy_huex_1_0_st_v2': 'AFFY HuEx 1 0 st v2 probe' (default: False),\n",
      " 'affy_hugene_1_0_st_v1': 'AFFY HuGene 1 0 st v1 probe' (default: False),\n",
      " 'affy_hugene_2_0_st_v1': 'AFFY HuGene 2 0 st v1 probe' (default: False),\n",
      " 'affy_hugenefl': 'AFFY HuGeneFL probe' (default: False),\n",
      " 'affy_primeview': 'AFFY PrimeView probe' (default: False),\n",
      " 'affy_u133_x3p': 'AFFY U133 X3P probe' (default: False),\n",
      " 'agilent_cgh_44b': 'AGILENT CGH 44b probe' (default: False),\n",
      " 'agilent_gpl6848': 'AGILENT GPL6848 probe' (default: False),\n",
      " 'agilent_sureprint_g3_ge_8x60k': 'AGILENT SurePrint G3 GE 8x60k probe' (default: False),\n",
      " 'agilent_sureprint_g3_ge_8x60k_v2': 'AGILENT SurePrint G3 GE 8x60k v2 probe' (default: False),\n",
      " 'agilent_wholegenome': 'AGILENT WholeGenome probe' (default: False),\n",
      " 'agilent_wholegenome_4x44k_v1': 'AGILENT WholeGenome 4x44k v1 probe' (default: False),\n",
      " 'agilent_wholegenome_4x44k_v2': 'AGILENT WholeGenome 4x44k v2 probe' (default: False),\n",
      " 'ahaastii_homolog_associated_gene_name': 'Great spotted kiwi gene name' (default: False),\n",
      " 'ahaastii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ahaastii_homolog_chrom_end': 'Great spotted kiwi chromosome/scaffold end (bp)' (default: False),\n",
      " 'ahaastii_homolog_chrom_start': 'Great spotted kiwi chromosome/scaffold start (bp)' (default: False),\n",
      " 'ahaastii_homolog_chromosome': 'Great spotted kiwi chromosome/scaffold name' (default: False),\n",
      " 'ahaastii_homolog_dn': 'dN with Great spotted kiwi' (default: False),\n",
      " 'ahaastii_homolog_ds': 'dS with Great spotted kiwi' (default: False),\n",
      " 'ahaastii_homolog_ensembl_gene': 'Great spotted kiwi gene stable ID' (default: False),\n",
      " 'ahaastii_homolog_ensembl_peptide': 'Great spotted kiwi protein or transcript stable ID' (default: False),\n",
      " 'ahaastii_homolog_goc_score': 'Great spotted kiwi Gene-order conservation score' (default: False),\n",
      " 'ahaastii_homolog_orthology_confidence': 'Great spotted kiwi orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ahaastii_homolog_orthology_type': 'Great spotted kiwi homology type' (default: False),\n",
      " 'ahaastii_homolog_perc_id': '%id. target Great spotted kiwi gene identical to query gene' (default: False),\n",
      " 'ahaastii_homolog_perc_id_r1': '%id. query gene identical to target Great spotted kiwi gene' (default: False),\n",
      " 'ahaastii_homolog_subtype': 'Last common ancestor with Great spotted kiwi' (default: False),\n",
      " 'ahaastii_homolog_wga_coverage': 'Great spotted kiwi Whole-genome alignment coverage' (default: False),\n",
      " 'allele': 'Variant alleles' (default: False),\n",
      " 'allele_string_2076': 'Consequence specific allele' (default: False),\n",
      " 'amelanoleuca_homolog_associated_gene_name': 'Panda gene name' (default: False),\n",
      " 'amelanoleuca_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'amelanoleuca_homolog_chrom_end': 'Panda chromosome/scaffold end (bp)' (default: False),\n",
      " 'amelanoleuca_homolog_chrom_start': 'Panda chromosome/scaffold start (bp)' (default: False),\n",
      " 'amelanoleuca_homolog_chromosome': 'Panda chromosome/scaffold name' (default: False),\n",
      " 'amelanoleuca_homolog_dn': 'dN with Panda' (default: False),\n",
      " 'amelanoleuca_homolog_ds': 'dS with Panda' (default: False),\n",
      " 'amelanoleuca_homolog_ensembl_gene': 'Panda gene stable ID' (default: False),\n",
      " 'amelanoleuca_homolog_ensembl_peptide': 'Panda protein or transcript stable ID' (default: False),\n",
      " 'amelanoleuca_homolog_goc_score': 'Panda Gene-order conservation score' (default: False),\n",
      " 'amelanoleuca_homolog_orthology_confidence': 'Panda orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'amelanoleuca_homolog_orthology_type': 'Panda homology type' (default: False),\n",
      " 'amelanoleuca_homolog_perc_id': '%id. target Panda gene identical to query gene' (default: False),\n",
      " 'amelanoleuca_homolog_perc_id_r1': '%id. query gene identical to target Panda gene' (default: False),\n",
      " 'amelanoleuca_homolog_subtype': 'Last common ancestor with Panda' (default: False),\n",
      " 'amelanoleuca_homolog_wga_coverage': 'Panda Whole-genome alignment coverage' (default: False),\n",
      " 'amexicanus_homolog_associated_gene_name': 'Mexican tetra gene name' (default: False),\n",
      " 'amexicanus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'amexicanus_homolog_chrom_end': 'Mexican tetra chromosome/scaffold end (bp)' (default: False),\n",
      " 'amexicanus_homolog_chrom_start': 'Mexican tetra chromosome/scaffold start (bp)' (default: False),\n",
      " 'amexicanus_homolog_chromosome': 'Mexican tetra chromosome/scaffold name' (default: False),\n",
      " 'amexicanus_homolog_dn': 'dN with Mexican tetra' (default: False),\n",
      " 'amexicanus_homolog_ds': 'dS with Mexican tetra' (default: False),\n",
      " 'amexicanus_homolog_ensembl_gene': 'Mexican tetra gene stable ID' (default: False),\n",
      " 'amexicanus_homolog_ensembl_peptide': 'Mexican tetra protein or transcript stable ID' (default: False),\n",
      " 'amexicanus_homolog_goc_score': 'Mexican tetra Gene-order conservation score' (default: False),\n",
      " 'amexicanus_homolog_orthology_confidence': 'Mexican tetra orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'amexicanus_homolog_orthology_type': 'Mexican tetra homology type' (default: False),\n",
      " 'amexicanus_homolog_perc_id': '%id. target Mexican tetra gene identical to query gene' (default: False),\n",
      " 'amexicanus_homolog_perc_id_r1': '%id. query gene identical to target Mexican tetra gene' (default: False),\n",
      " 'amexicanus_homolog_subtype': 'Last common ancestor with Mexican tetra' (default: False),\n",
      " 'amexicanus_homolog_wga_coverage': 'Mexican tetra Whole-genome alignment coverage' (default: False),\n",
      " 'ampachon_homolog_associated_gene_name': 'Pachon cavefish gene name' (default: False),\n",
      " 'ampachon_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ampachon_homolog_chrom_end': 'Pachon cavefish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ampachon_homolog_chrom_start': 'Pachon cavefish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ampachon_homolog_chromosome': 'Pachon cavefish chromosome/scaffold name' (default: False),\n",
      " 'ampachon_homolog_dn': 'dN with Pachon cavefish' (default: False),\n",
      " 'ampachon_homolog_ds': 'dS with Pachon cavefish' (default: False),\n",
      " 'ampachon_homolog_ensembl_gene': 'Pachon cavefish gene stable ID' (default: False),\n",
      " 'ampachon_homolog_ensembl_peptide': 'Pachon cavefish protein or transcript stable ID' (default: False),\n",
      " 'ampachon_homolog_goc_score': 'Pachon cavefish Gene-order conservation score' (default: False),\n",
      " 'ampachon_homolog_orthology_confidence': 'Pachon cavefish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ampachon_homolog_orthology_type': 'Pachon cavefish homology type' (default: False),\n",
      " 'ampachon_homolog_perc_id': '%id. target Pachon cavefish gene identical to query gene' (default: False),\n",
      " 'ampachon_homolog_perc_id_r1': '%id. query gene identical to target Pachon cavefish gene' (default: False),\n",
      " 'ampachon_homolog_subtype': 'Last common ancestor with Pachon cavefish' (default: False),\n",
      " 'ampachon_homolog_wga_coverage': 'Pachon cavefish Whole-genome alignment coverage' (default: False),\n",
      " 'anancymaae_homolog_associated_gene_name': 'Ma's night monkey gene name' (default: False),\n",
      " 'anancymaae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'anancymaae_homolog_chrom_end': 'Ma's night monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'anancymaae_homolog_chrom_start': 'Ma's night monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'anancymaae_homolog_chromosome': 'Ma's night monkey chromosome/scaffold name' (default: False),\n",
      " 'anancymaae_homolog_dn': 'dN with Ma's night monkey' (default: False),\n",
      " 'anancymaae_homolog_ds': 'dS with Ma's night monkey' (default: False),\n",
      " 'anancymaae_homolog_ensembl_gene': 'Ma's night monkey gene stable ID' (default: False),\n",
      " 'anancymaae_homolog_ensembl_peptide': 'Ma's night monkey protein or transcript stable ID' (default: False),\n",
      " 'anancymaae_homolog_goc_score': 'Ma's night monkey Gene-order conservation score' (default: False),\n",
      " 'anancymaae_homolog_orthology_confidence': 'Ma's night monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'anancymaae_homolog_orthology_type': 'Ma's night monkey homology type' (default: False),\n",
      " 'anancymaae_homolog_perc_id': '%id. target Ma's night monkey gene identical to query gene' (default: False),\n",
      " 'anancymaae_homolog_perc_id_r1': '%id. query gene identical to target Ma's night monkey gene' (default: False),\n",
      " 'anancymaae_homolog_subtype': 'Last common ancestor with Ma's night monkey' (default: False),\n",
      " 'anancymaae_homolog_wga_coverage': 'Ma's night monkey Whole-genome alignment coverage' (default: False),\n",
      " 'aocellaris_homolog_associated_gene_name': 'Clown anemonefish gene name' (default: False),\n",
      " 'aocellaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'aocellaris_homolog_chrom_end': 'Clown anemonefish chromosome/scaffold end (bp)' (default: False),\n",
      " 'aocellaris_homolog_chrom_start': 'Clown anemonefish chromosome/scaffold start (bp)' (default: False),\n",
      " 'aocellaris_homolog_chromosome': 'Clown anemonefish chromosome/scaffold name' (default: False),\n",
      " 'aocellaris_homolog_dn': 'dN with Clown anemonefish' (default: False),\n",
      " 'aocellaris_homolog_ds': 'dS with Clown anemonefish' (default: False),\n",
      " 'aocellaris_homolog_ensembl_gene': 'Clown anemonefish gene stable ID' (default: False),\n",
      " 'aocellaris_homolog_ensembl_peptide': 'Clown anemonefish protein or transcript stable ID' (default: False),\n",
      " 'aocellaris_homolog_goc_score': 'Clown anemonefish Gene-order conservation score' (default: False),\n",
      " 'aocellaris_homolog_orthology_confidence': 'Clown anemonefish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'aocellaris_homolog_orthology_type': 'Clown anemonefish homology type' (default: False),\n",
      " 'aocellaris_homolog_perc_id': '%id. target Clown anemonefish gene identical to query gene' (default: False),\n",
      " 'aocellaris_homolog_perc_id_r1': '%id. query gene identical to target Clown anemonefish gene' (default: False),\n",
      " 'aocellaris_homolog_subtype': 'Last common ancestor with Clown anemonefish' (default: False),\n",
      " 'aocellaris_homolog_wga_coverage': 'Clown anemonefish Whole-genome alignment coverage' (default: False),\n",
      " 'aowenii_homolog_associated_gene_name': 'Little spotted kiwi gene name' (default: False),\n",
      " 'aowenii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'aowenii_homolog_chrom_end': 'Little spotted kiwi chromosome/scaffold end (bp)' (default: False),\n",
      " 'aowenii_homolog_chrom_start': 'Little spotted kiwi chromosome/scaffold start (bp)' (default: False),\n",
      " 'aowenii_homolog_chromosome': 'Little spotted kiwi chromosome/scaffold name' (default: False),\n",
      " 'aowenii_homolog_dn': 'dN with Little spotted kiwi' (default: False),\n",
      " 'aowenii_homolog_ds': 'dS with Little spotted kiwi' (default: False),\n",
      " 'aowenii_homolog_ensembl_gene': 'Little spotted kiwi gene stable ID' (default: False),\n",
      " 'aowenii_homolog_ensembl_peptide': 'Little spotted kiwi protein or transcript stable ID' (default: False),\n",
      " 'aowenii_homolog_goc_score': 'Little spotted kiwi Gene-order conservation score' (default: False),\n",
      " 'aowenii_homolog_orthology_confidence': 'Little spotted kiwi orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'aowenii_homolog_orthology_type': 'Little spotted kiwi homology type' (default: False),\n",
      " 'aowenii_homolog_perc_id': '%id. target Little spotted kiwi gene identical to query gene' (default: False),\n",
      " 'aowenii_homolog_perc_id_r1': '%id. query gene identical to target Little spotted kiwi gene' (default: False),\n",
      " 'aowenii_homolog_subtype': 'Last common ancestor with Little spotted kiwi' (default: False),\n",
      " 'aowenii_homolog_wga_coverage': 'Little spotted kiwi Whole-genome alignment coverage' (default: False),\n",
      " 'apercula_homolog_associated_gene_name': 'Orange clownfish gene name' (default: False),\n",
      " 'apercula_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'apercula_homolog_chrom_end': 'Orange clownfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'apercula_homolog_chrom_start': 'Orange clownfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'apercula_homolog_chromosome': 'Orange clownfish chromosome/scaffold name' (default: False),\n",
      " 'apercula_homolog_dn': 'dN with Orange clownfish' (default: False),\n",
      " 'apercula_homolog_ds': 'dS with Orange clownfish' (default: False),\n",
      " 'apercula_homolog_ensembl_gene': 'Orange clownfish gene stable ID' (default: False),\n",
      " 'apercula_homolog_ensembl_peptide': 'Orange clownfish protein or transcript stable ID' (default: False),\n",
      " 'apercula_homolog_goc_score': 'Orange clownfish Gene-order conservation score' (default: False),\n",
      " 'apercula_homolog_orthology_confidence': 'Orange clownfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'apercula_homolog_orthology_type': 'Orange clownfish homology type' (default: False),\n",
      " 'apercula_homolog_perc_id': '%id. target Orange clownfish gene identical to query gene' (default: False),\n",
      " 'apercula_homolog_perc_id_r1': '%id. query gene identical to target Orange clownfish gene' (default: False),\n",
      " 'apercula_homolog_subtype': 'Last common ancestor with Orange clownfish' (default: False),\n",
      " 'apercula_homolog_wga_coverage': 'Orange clownfish Whole-genome alignment coverage' (default: False),\n",
      " 'apolyacanthus_homolog_associated_gene_name': 'Spiny chromis gene name' (default: False),\n",
      " 'apolyacanthus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'apolyacanthus_homolog_chrom_end': 'Spiny chromis chromosome/scaffold end (bp)' (default: False),\n",
      " 'apolyacanthus_homolog_chrom_start': 'Spiny chromis chromosome/scaffold start (bp)' (default: False),\n",
      " 'apolyacanthus_homolog_chromosome': 'Spiny chromis chromosome/scaffold name' (default: False),\n",
      " 'apolyacanthus_homolog_dn': 'dN with Spiny chromis' (default: False),\n",
      " 'apolyacanthus_homolog_ds': 'dS with Spiny chromis' (default: False),\n",
      " 'apolyacanthus_homolog_ensembl_gene': 'Spiny chromis gene stable ID' (default: False),\n",
      " 'apolyacanthus_homolog_ensembl_peptide': 'Spiny chromis protein or transcript stable ID' (default: False),\n",
      " 'apolyacanthus_homolog_goc_score': 'Spiny chromis Gene-order conservation score' (default: False),\n",
      " 'apolyacanthus_homolog_orthology_confidence': 'Spiny chromis orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'apolyacanthus_homolog_orthology_type': 'Spiny chromis homology type' (default: False),\n",
      " 'apolyacanthus_homolog_perc_id': '%id. target Spiny chromis gene identical to query gene' (default: False),\n",
      " 'apolyacanthus_homolog_perc_id_r1': '%id. query gene identical to target Spiny chromis gene' (default: False),\n",
      " 'apolyacanthus_homolog_subtype': 'Last common ancestor with Spiny chromis' (default: False),\n",
      " 'apolyacanthus_homolog_wga_coverage': 'Spiny chromis Whole-genome alignment coverage' (default: False),\n",
      " 'applatyrhynchos_homolog_associated_gene_name': 'Duck gene name' (default: False),\n",
      " 'applatyrhynchos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'applatyrhynchos_homolog_chrom_end': 'Duck chromosome/scaffold end (bp)' (default: False),\n",
      " 'applatyrhynchos_homolog_chrom_start': 'Duck chromosome/scaffold start (bp)' (default: False),\n",
      " 'applatyrhynchos_homolog_chromosome': 'Duck chromosome/scaffold name' (default: False),\n",
      " 'applatyrhynchos_homolog_dn': 'dN with Duck' (default: False),\n",
      " 'applatyrhynchos_homolog_ds': 'dS with Duck' (default: False),\n",
      " 'applatyrhynchos_homolog_ensembl_gene': 'Duck gene stable ID' (default: False),\n",
      " 'applatyrhynchos_homolog_ensembl_peptide': 'Duck protein or transcript stable ID' (default: False),\n",
      " 'applatyrhynchos_homolog_goc_score': 'Duck Gene-order conservation score' (default: False),\n",
      " 'applatyrhynchos_homolog_orthology_confidence': 'Duck orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'applatyrhynchos_homolog_orthology_type': 'Duck homology type' (default: False),\n",
      " 'applatyrhynchos_homolog_perc_id': '%id. target Duck gene identical to query gene' (default: False),\n",
      " 'applatyrhynchos_homolog_perc_id_r1': '%id. query gene identical to target Duck gene' (default: False),\n",
      " 'applatyrhynchos_homolog_subtype': 'Last common ancestor with Duck' (default: False),\n",
      " 'applatyrhynchos_homolog_wga_coverage': 'Duck Whole-genome alignment coverage' (default: False),\n",
      " 'arowi_homolog_associated_gene_name': 'Okarito brown kiwi gene name' (default: False),\n",
      " 'arowi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'arowi_homolog_chrom_end': 'Okarito brown kiwi chromosome/scaffold end (bp)' (default: False),\n",
      " 'arowi_homolog_chrom_start': 'Okarito brown kiwi chromosome/scaffold start (bp)' (default: False),\n",
      " 'arowi_homolog_chromosome': 'Okarito brown kiwi chromosome/scaffold name' (default: False),\n",
      " 'arowi_homolog_dn': 'dN with Okarito brown kiwi' (default: False),\n",
      " 'arowi_homolog_ds': 'dS with Okarito brown kiwi' (default: False),\n",
      " 'arowi_homolog_ensembl_gene': 'Okarito brown kiwi gene stable ID' (default: False),\n",
      " 'arowi_homolog_ensembl_peptide': 'Okarito brown kiwi protein or transcript stable ID' (default: False),\n",
      " 'arowi_homolog_goc_score': 'Okarito brown kiwi Gene-order conservation score' (default: False),\n",
      " 'arowi_homolog_orthology_confidence': 'Okarito brown kiwi orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'arowi_homolog_orthology_type': 'Okarito brown kiwi homology type' (default: False),\n",
      " 'arowi_homolog_perc_id': '%id. target Okarito brown kiwi gene identical to query gene' (default: False),\n",
      " 'arowi_homolog_perc_id_r1': '%id. query gene identical to target Okarito brown kiwi gene' (default: False),\n",
      " 'arowi_homolog_subtype': 'Last common ancestor with Okarito brown kiwi' (default: False),\n",
      " 'arowi_homolog_wga_coverage': 'Okarito brown kiwi Whole-genome alignment coverage' (default: False),\n",
      " 'arrayexpress': 'Expression Atlas ID' (default: False),\n",
      " 'atestudineus_homolog_associated_gene_name': 'Climbing perch gene name' (default: False),\n",
      " 'atestudineus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'atestudineus_homolog_chrom_end': 'Climbing perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'atestudineus_homolog_chrom_start': 'Climbing perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'atestudineus_homolog_chromosome': 'Climbing perch chromosome/scaffold name' (default: False),\n",
      " 'atestudineus_homolog_dn': 'dN with Climbing perch' (default: False),\n",
      " 'atestudineus_homolog_ds': 'dS with Climbing perch' (default: False),\n",
      " 'atestudineus_homolog_ensembl_gene': 'Climbing perch gene stable ID' (default: False),\n",
      " 'atestudineus_homolog_ensembl_peptide': 'Climbing perch protein or transcript stable ID' (default: False),\n",
      " 'atestudineus_homolog_goc_score': 'Climbing perch Gene-order conservation score' (default: False),\n",
      " 'atestudineus_homolog_orthology_confidence': 'Climbing perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'atestudineus_homolog_orthology_type': 'Climbing perch homology type' (default: False),\n",
      " 'atestudineus_homolog_perc_id': '%id. target Climbing perch gene identical to query gene' (default: False),\n",
      " 'atestudineus_homolog_perc_id_r1': '%id. query gene identical to target Climbing perch gene' (default: False),\n",
      " 'atestudineus_homolog_subtype': 'Last common ancestor with Climbing perch' (default: False),\n",
      " 'atestudineus_homolog_wga_coverage': 'Climbing perch Whole-genome alignment coverage' (default: False),\n",
      " 'band': 'Karyotype band' (default: False),\n",
      " 'bbbison_homolog_associated_gene_name': 'American bison gene name' (default: False),\n",
      " 'bbbison_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bbbison_homolog_chrom_end': 'American bison chromosome/scaffold end (bp)' (default: False),\n",
      " 'bbbison_homolog_chrom_start': 'American bison chromosome/scaffold start (bp)' (default: False),\n",
      " 'bbbison_homolog_chromosome': 'American bison chromosome/scaffold name' (default: False),\n",
      " 'bbbison_homolog_dn': 'dN with American bison' (default: False),\n",
      " 'bbbison_homolog_ds': 'dS with American bison' (default: False),\n",
      " 'bbbison_homolog_ensembl_gene': 'American bison gene stable ID' (default: False),\n",
      " 'bbbison_homolog_ensembl_peptide': 'American bison protein or transcript stable ID' (default: False),\n",
      " 'bbbison_homolog_goc_score': 'American bison Gene-order conservation score' (default: False),\n",
      " 'bbbison_homolog_orthology_confidence': 'American bison orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bbbison_homolog_orthology_type': 'American bison homology type' (default: False),\n",
      " 'bbbison_homolog_perc_id': '%id. target American bison gene identical to query gene' (default: False),\n",
      " 'bbbison_homolog_perc_id_r1': '%id. query gene identical to target American bison gene' (default: False),\n",
      " 'bbbison_homolog_subtype': 'Last common ancestor with American bison' (default: False),\n",
      " 'bbbison_homolog_wga_coverage': 'American bison Whole-genome alignment coverage' (default: False),\n",
      " 'bihybrid_homolog_associated_gene_name': 'Hybrid - Bos Indicus gene name' (default: False),\n",
      " 'bihybrid_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bihybrid_homolog_chrom_end': 'Hybrid - Bos Indicus chromosome/scaffold end (bp)' (default: False),\n",
      " 'bihybrid_homolog_chrom_start': 'Hybrid - Bos Indicus chromosome/scaffold start (bp)' (default: False),\n",
      " 'bihybrid_homolog_chromosome': 'Hybrid - Bos Indicus chromosome/scaffold name' (default: False),\n",
      " 'bihybrid_homolog_dn': 'dN with Hybrid - Bos Indicus' (default: False),\n",
      " 'bihybrid_homolog_ds': 'dS with Hybrid - Bos Indicus' (default: False),\n",
      " 'bihybrid_homolog_ensembl_gene': 'Hybrid - Bos Indicus gene stable ID' (default: False),\n",
      " 'bihybrid_homolog_ensembl_peptide': 'Hybrid - Bos Indicus protein or transcript stable ID' (default: False),\n",
      " 'bihybrid_homolog_goc_score': 'Hybrid - Bos Indicus Gene-order conservation score' (default: False),\n",
      " 'bihybrid_homolog_orthology_confidence': 'Hybrid - Bos Indicus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bihybrid_homolog_orthology_type': 'Hybrid - Bos Indicus homology type' (default: False),\n",
      " 'bihybrid_homolog_perc_id': '%id. target Hybrid - Bos Indicus gene identical to query gene' (default: False),\n",
      " 'bihybrid_homolog_perc_id_r1': '%id. query gene identical to target Hybrid - Bos Indicus gene' (default: False),\n",
      " 'bihybrid_homolog_subtype': 'Last common ancestor with Hybrid - Bos Indicus' (default: False),\n",
      " 'bihybrid_homolog_wga_coverage': 'Hybrid - Bos Indicus Whole-genome alignment coverage' (default: False),\n",
      " 'bmutus_homolog_associated_gene_name': 'Wild yak gene name' (default: False),\n",
      " 'bmutus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bmutus_homolog_chrom_end': 'Wild yak chromosome/scaffold end (bp)' (default: False),\n",
      " 'bmutus_homolog_chrom_start': 'Wild yak chromosome/scaffold start (bp)' (default: False),\n",
      " 'bmutus_homolog_chromosome': 'Wild yak chromosome/scaffold name' (default: False),\n",
      " 'bmutus_homolog_dn': 'dN with Wild yak' (default: False),\n",
      " 'bmutus_homolog_ds': 'dS with Wild yak' (default: False),\n",
      " 'bmutus_homolog_ensembl_gene': 'Wild yak gene stable ID' (default: False),\n",
      " 'bmutus_homolog_ensembl_peptide': 'Wild yak protein or transcript stable ID' (default: False),\n",
      " 'bmutus_homolog_goc_score': 'Wild yak Gene-order conservation score' (default: False),\n",
      " 'bmutus_homolog_orthology_confidence': 'Wild yak orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bmutus_homolog_orthology_type': 'Wild yak homology type' (default: False),\n",
      " 'bmutus_homolog_perc_id': '%id. target Wild yak gene identical to query gene' (default: False),\n",
      " 'bmutus_homolog_perc_id_r1': '%id. query gene identical to target Wild yak gene' (default: False),\n",
      " 'bmutus_homolog_subtype': 'Last common ancestor with Wild yak' (default: False),\n",
      " 'bmutus_homolog_wga_coverage': 'Wild yak Whole-genome alignment coverage' (default: False),\n",
      " 'bsplendens_homolog_associated_gene_name': 'Siamese fighting fish gene name' (default: False),\n",
      " 'bsplendens_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bsplendens_homolog_chrom_end': 'Siamese fighting fish chromosome/scaffold end (bp)' (default: False),\n",
      " 'bsplendens_homolog_chrom_start': 'Siamese fighting fish chromosome/scaffold start (bp)' (default: False),\n",
      " 'bsplendens_homolog_chromosome': 'Siamese fighting fish chromosome/scaffold name' (default: False),\n",
      " 'bsplendens_homolog_dn': 'dN with Siamese fighting fish' (default: False),\n",
      " 'bsplendens_homolog_ds': 'dS with Siamese fighting fish' (default: False),\n",
      " 'bsplendens_homolog_ensembl_gene': 'Siamese fighting fish gene stable ID' (default: False),\n",
      " 'bsplendens_homolog_ensembl_peptide': 'Siamese fighting fish protein or transcript stable ID' (default: False),\n",
      " 'bsplendens_homolog_goc_score': 'Siamese fighting fish Gene-order conservation score' (default: False),\n",
      " 'bsplendens_homolog_orthology_confidence': 'Siamese fighting fish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bsplendens_homolog_orthology_type': 'Siamese fighting fish homology type' (default: False),\n",
      " 'bsplendens_homolog_perc_id': '%id. target Siamese fighting fish gene identical to query gene' (default: False),\n",
      " 'bsplendens_homolog_perc_id_r1': '%id. query gene identical to target Siamese fighting fish gene' (default: False),\n",
      " 'bsplendens_homolog_subtype': 'Last common ancestor with Siamese fighting fish' (default: False),\n",
      " 'bsplendens_homolog_wga_coverage': 'Siamese fighting fish Whole-genome alignment coverage' (default: False),\n",
      " 'btaurus_homolog_associated_gene_name': 'Cow gene name' (default: False),\n",
      " 'btaurus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'btaurus_homolog_chrom_end': 'Cow chromosome/scaffold end (bp)' (default: False),\n",
      " 'btaurus_homolog_chrom_start': 'Cow chromosome/scaffold start (bp)' (default: False),\n",
      " 'btaurus_homolog_chromosome': 'Cow chromosome/scaffold name' (default: False),\n",
      " 'btaurus_homolog_dn': 'dN with Cow' (default: False),\n",
      " 'btaurus_homolog_ds': 'dS with Cow' (default: False),\n",
      " 'btaurus_homolog_ensembl_gene': 'Cow gene stable ID' (default: False),\n",
      " 'btaurus_homolog_ensembl_peptide': 'Cow protein or transcript stable ID' (default: False),\n",
      " 'btaurus_homolog_goc_score': 'Cow Gene-order conservation score' (default: False),\n",
      " 'btaurus_homolog_orthology_confidence': 'Cow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'btaurus_homolog_orthology_type': 'Cow homology type' (default: False),\n",
      " 'btaurus_homolog_perc_id': '%id. target Cow gene identical to query gene' (default: False),\n",
      " 'btaurus_homolog_perc_id_r1': '%id. query gene identical to target Cow gene' (default: False),\n",
      " 'btaurus_homolog_subtype': 'Last common ancestor with Cow' (default: False),\n",
      " 'btaurus_homolog_wga_coverage': 'Cow Whole-genome alignment coverage' (default: False),\n",
      " 'bthybrid_homolog_associated_gene_name': 'Hybrid - Bos Taurus gene name' (default: False),\n",
      " 'bthybrid_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bthybrid_homolog_chrom_end': 'Hybrid - Bos Taurus chromosome/scaffold end (bp)' (default: False),\n",
      " 'bthybrid_homolog_chrom_start': 'Hybrid - Bos Taurus chromosome/scaffold start (bp)' (default: False),\n",
      " 'bthybrid_homolog_chromosome': 'Hybrid - Bos Taurus chromosome/scaffold name' (default: False),\n",
      " 'bthybrid_homolog_dn': 'dN with Hybrid - Bos Taurus' (default: False),\n",
      " 'bthybrid_homolog_ds': 'dS with Hybrid - Bos Taurus' (default: False),\n",
      " 'bthybrid_homolog_ensembl_gene': 'Hybrid - Bos Taurus gene stable ID' (default: False),\n",
      " 'bthybrid_homolog_ensembl_peptide': 'Hybrid - Bos Taurus protein or transcript stable ID' (default: False),\n",
      " 'bthybrid_homolog_goc_score': 'Hybrid - Bos Taurus Gene-order conservation score' (default: False),\n",
      " 'bthybrid_homolog_orthology_confidence': 'Hybrid - Bos Taurus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bthybrid_homolog_orthology_type': 'Hybrid - Bos Taurus homology type' (default: False),\n",
      " 'bthybrid_homolog_perc_id': '%id. target Hybrid - Bos Taurus gene identical to query gene' (default: False),\n",
      " 'bthybrid_homolog_perc_id_r1': '%id. query gene identical to target Hybrid - Bos Taurus gene' (default: False),\n",
      " 'bthybrid_homolog_subtype': 'Last common ancestor with Hybrid - Bos Taurus' (default: False),\n",
      " 'bthybrid_homolog_wga_coverage': 'Hybrid - Bos Taurus Whole-genome alignment coverage' (default: False),\n",
      " 'cabingdonii_homolog_associated_gene_name': 'Abingdon island giant tortoise gene name' (default: False),\n",
      " 'cabingdonii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cabingdonii_homolog_chrom_end': 'Abingdon island giant tortoise chromosome/scaffold end (bp)' (default: False),\n",
      " 'cabingdonii_homolog_chrom_start': 'Abingdon island giant tortoise chromosome/scaffold start (bp)' (default: False),\n",
      " 'cabingdonii_homolog_chromosome': 'Abingdon island giant tortoise chromosome/scaffold name' (default: False),\n",
      " 'cabingdonii_homolog_dn': 'dN with Abingdon island giant tortoise' (default: False),\n",
      " 'cabingdonii_homolog_ds': 'dS with Abingdon island giant tortoise' (default: False),\n",
      " 'cabingdonii_homolog_ensembl_gene': 'Abingdon island giant tortoise gene stable ID' (default: False),\n",
      " 'cabingdonii_homolog_ensembl_peptide': 'Abingdon island giant tortoise protein or transcript stable ID' (default: False),\n",
      " 'cabingdonii_homolog_goc_score': 'Abingdon island giant tortoise Gene-order conservation score' (default: False),\n",
      " 'cabingdonii_homolog_orthology_confidence': 'Abingdon island giant tortoise orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cabingdonii_homolog_orthology_type': 'Abingdon island giant tortoise homology type' (default: False),\n",
      " 'cabingdonii_homolog_perc_id': '%id. target Abingdon island giant tortoise gene identical to query gene' (default: False),\n",
      " 'cabingdonii_homolog_perc_id_r1': '%id. query gene identical to target Abingdon island giant tortoise gene' (default: False),\n",
      " 'cabingdonii_homolog_subtype': 'Last common ancestor with Abingdon island giant tortoise' (default: False),\n",
      " 'cabingdonii_homolog_wga_coverage': 'Abingdon island giant tortoise Whole-genome alignment coverage' (default: False),\n",
      " 'capalliatus_homolog_associated_gene_name': 'Angola colobus gene name' (default: False),\n",
      " 'capalliatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'capalliatus_homolog_chrom_end': 'Angola colobus chromosome/scaffold end (bp)' (default: False),\n",
      " 'capalliatus_homolog_chrom_start': 'Angola colobus chromosome/scaffold start (bp)' (default: False),\n",
      " 'capalliatus_homolog_chromosome': 'Angola colobus chromosome/scaffold name' (default: False),\n",
      " 'capalliatus_homolog_dn': 'dN with Angola colobus' (default: False),\n",
      " 'capalliatus_homolog_ds': 'dS with Angola colobus' (default: False),\n",
      " 'capalliatus_homolog_ensembl_gene': 'Angola colobus gene stable ID' (default: False),\n",
      " 'capalliatus_homolog_ensembl_peptide': 'Angola colobus protein or transcript stable ID' (default: False),\n",
      " 'capalliatus_homolog_goc_score': 'Angola colobus Gene-order conservation score' (default: False),\n",
      " 'capalliatus_homolog_orthology_confidence': 'Angola colobus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'capalliatus_homolog_orthology_type': 'Angola colobus homology type' (default: False),\n",
      " 'capalliatus_homolog_perc_id': '%id. target Angola colobus gene identical to query gene' (default: False),\n",
      " 'capalliatus_homolog_perc_id_r1': '%id. query gene identical to target Angola colobus gene' (default: False),\n",
      " 'capalliatus_homolog_subtype': 'Last common ancestor with Angola colobus' (default: False),\n",
      " 'capalliatus_homolog_wga_coverage': 'Angola colobus Whole-genome alignment coverage' (default: False),\n",
      " 'caperea_homolog_associated_gene_name': 'Brazilian guinea pig gene name' (default: False),\n",
      " 'caperea_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'caperea_homolog_chrom_end': 'Brazilian guinea pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'caperea_homolog_chrom_start': 'Brazilian guinea pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'caperea_homolog_chromosome': 'Brazilian guinea pig chromosome/scaffold name' (default: False),\n",
      " 'caperea_homolog_dn': 'dN with Brazilian guinea pig' (default: False),\n",
      " 'caperea_homolog_ds': 'dS with Brazilian guinea pig' (default: False),\n",
      " 'caperea_homolog_ensembl_gene': 'Brazilian guinea pig gene stable ID' (default: False),\n",
      " 'caperea_homolog_ensembl_peptide': 'Brazilian guinea pig protein or transcript stable ID' (default: False),\n",
      " 'caperea_homolog_goc_score': 'Brazilian guinea pig Gene-order conservation score' (default: False),\n",
      " 'caperea_homolog_orthology_confidence': 'Brazilian guinea pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'caperea_homolog_orthology_type': 'Brazilian guinea pig homology type' (default: False),\n",
      " 'caperea_homolog_perc_id': '%id. target Brazilian guinea pig gene identical to query gene' (default: False),\n",
      " 'caperea_homolog_perc_id_r1': '%id. query gene identical to target Brazilian guinea pig gene' (default: False),\n",
      " 'caperea_homolog_subtype': 'Last common ancestor with Brazilian guinea pig' (default: False),\n",
      " 'caperea_homolog_wga_coverage': 'Brazilian guinea pig Whole-genome alignment coverage' (default: False),\n",
      " 'catys_homolog_associated_gene_name': 'Sooty mangabey gene name' (default: False),\n",
      " 'catys_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'catys_homolog_chrom_end': 'Sooty mangabey chromosome/scaffold end (bp)' (default: False),\n",
      " 'catys_homolog_chrom_start': 'Sooty mangabey chromosome/scaffold start (bp)' (default: False),\n",
      " 'catys_homolog_chromosome': 'Sooty mangabey chromosome/scaffold name' (default: False),\n",
      " 'catys_homolog_dn': 'dN with Sooty mangabey' (default: False),\n",
      " 'catys_homolog_ds': 'dS with Sooty mangabey' (default: False),\n",
      " 'catys_homolog_ensembl_gene': 'Sooty mangabey gene stable ID' (default: False),\n",
      " 'catys_homolog_ensembl_peptide': 'Sooty mangabey protein or transcript stable ID' (default: False),\n",
      " 'catys_homolog_goc_score': 'Sooty mangabey Gene-order conservation score' (default: False),\n",
      " 'catys_homolog_orthology_confidence': 'Sooty mangabey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'catys_homolog_orthology_type': 'Sooty mangabey homology type' (default: False),\n",
      " 'catys_homolog_perc_id': '%id. target Sooty mangabey gene identical to query gene' (default: False),\n",
      " 'catys_homolog_perc_id_r1': '%id. query gene identical to target Sooty mangabey gene' (default: False),\n",
      " 'catys_homolog_subtype': 'Last common ancestor with Sooty mangabey' (default: False),\n",
      " 'catys_homolog_wga_coverage': 'Sooty mangabey Whole-genome alignment coverage' (default: False),\n",
      " 'ccaeruleus_homolog_associated_gene_name': 'Blue tit gene name' (default: False),\n",
      " 'ccaeruleus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ccaeruleus_homolog_chrom_end': 'Blue tit chromosome/scaffold end (bp)' (default: False),\n",
      " 'ccaeruleus_homolog_chrom_start': 'Blue tit chromosome/scaffold start (bp)' (default: False),\n",
      " 'ccaeruleus_homolog_chromosome': 'Blue tit chromosome/scaffold name' (default: False),\n",
      " 'ccaeruleus_homolog_dn': 'dN with Blue tit' (default: False),\n",
      " 'ccaeruleus_homolog_ds': 'dS with Blue tit' (default: False),\n",
      " 'ccaeruleus_homolog_ensembl_gene': 'Blue tit gene stable ID' (default: False),\n",
      " 'ccaeruleus_homolog_ensembl_peptide': 'Blue tit protein or transcript stable ID' (default: False),\n",
      " 'ccaeruleus_homolog_goc_score': 'Blue tit Gene-order conservation score' (default: False),\n",
      " 'ccaeruleus_homolog_orthology_confidence': 'Blue tit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ccaeruleus_homolog_orthology_type': 'Blue tit homology type' (default: False),\n",
      " 'ccaeruleus_homolog_perc_id': '%id. target Blue tit gene identical to query gene' (default: False),\n",
      " 'ccaeruleus_homolog_perc_id_r1': '%id. query gene identical to target Blue tit gene' (default: False),\n",
      " 'ccaeruleus_homolog_subtype': 'Last common ancestor with Blue tit' (default: False),\n",
      " 'ccaeruleus_homolog_wga_coverage': 'Blue tit Whole-genome alignment coverage' (default: False),\n",
      " 'ccanadensis_homolog_associated_gene_name': 'American beaver gene name' (default: False),\n",
      " 'ccanadensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ccanadensis_homolog_chrom_end': 'American beaver chromosome/scaffold end (bp)' (default: False),\n",
      " 'ccanadensis_homolog_chrom_start': 'American beaver chromosome/scaffold start (bp)' (default: False),\n",
      " 'ccanadensis_homolog_chromosome': 'American beaver chromosome/scaffold name' (default: False),\n",
      " 'ccanadensis_homolog_dn': 'dN with American beaver' (default: False),\n",
      " 'ccanadensis_homolog_ds': 'dS with American beaver' (default: False),\n",
      " 'ccanadensis_homolog_ensembl_gene': 'American beaver gene stable ID' (default: False),\n",
      " 'ccanadensis_homolog_ensembl_peptide': 'American beaver protein or transcript stable ID' (default: False),\n",
      " 'ccanadensis_homolog_goc_score': 'American beaver Gene-order conservation score' (default: False),\n",
      " 'ccanadensis_homolog_orthology_confidence': 'American beaver orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ccanadensis_homolog_orthology_type': 'American beaver homology type' (default: False),\n",
      " 'ccanadensis_homolog_perc_id': '%id. target American beaver gene identical to query gene' (default: False),\n",
      " 'ccanadensis_homolog_perc_id_r1': '%id. query gene identical to target American beaver gene' (default: False),\n",
      " 'ccanadensis_homolog_subtype': 'Last common ancestor with American beaver' (default: False),\n",
      " 'ccanadensis_homolog_wga_coverage': 'American beaver Whole-genome alignment coverage' (default: False),\n",
      " 'ccapucinus_homolog_associated_gene_name': 'Capuchin gene name' (default: False),\n",
      " 'ccapucinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ccapucinus_homolog_chrom_end': 'Capuchin chromosome/scaffold end (bp)' (default: False),\n",
      " 'ccapucinus_homolog_chrom_start': 'Capuchin chromosome/scaffold start (bp)' (default: False),\n",
      " 'ccapucinus_homolog_chromosome': 'Capuchin chromosome/scaffold name' (default: False),\n",
      " 'ccapucinus_homolog_dn': 'dN with Capuchin' (default: False),\n",
      " 'ccapucinus_homolog_ds': 'dS with Capuchin' (default: False),\n",
      " 'ccapucinus_homolog_ensembl_gene': 'Capuchin gene stable ID' (default: False),\n",
      " 'ccapucinus_homolog_ensembl_peptide': 'Capuchin protein or transcript stable ID' (default: False),\n",
      " 'ccapucinus_homolog_goc_score': 'Capuchin Gene-order conservation score' (default: False),\n",
      " 'ccapucinus_homolog_orthology_confidence': 'Capuchin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ccapucinus_homolog_orthology_type': 'Capuchin homology type' (default: False),\n",
      " 'ccapucinus_homolog_perc_id': '%id. target Capuchin gene identical to query gene' (default: False),\n",
      " 'ccapucinus_homolog_perc_id_r1': '%id. query gene identical to target Capuchin gene' (default: False),\n",
      " 'ccapucinus_homolog_subtype': 'Last common ancestor with Capuchin' (default: False),\n",
      " 'ccapucinus_homolog_wga_coverage': 'Capuchin Whole-genome alignment coverage' (default: False),\n",
      " 'ccds': 'CCDS ID' (default: False),\n",
      " 'cdd': 'CDD ID' (default: False),\n",
      " 'cdd_end': 'CDD end' (default: False),\n",
      " 'cdd_start': 'CDD start' (default: False),\n",
      " 'cdna': 'cDNA sequences' (default: False),\n",
      " 'cdna_coding_end': 'cDNA coding end' (default: False),\n",
      " 'cdna_coding_start': 'cDNA coding start' (default: False),\n",
      " 'cds_end': 'CDS end' (default: False),\n",
      " 'cds_end_2076': 'CDS end' (default: False),\n",
      " 'cds_length': 'CDS Length' (default: False),\n",
      " 'cds_start': 'CDS start' (default: False),\n",
      " 'cds_start_2076': 'CDS start' (default: False),\n",
      " 'celegans_homolog_associated_gene_name': 'Caenorhabditis elegans gene name' (default: False),\n",
      " 'celegans_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'celegans_homolog_chrom_end': 'Caenorhabditis elegans chromosome/scaffold end (bp)' (default: False),\n",
      " 'celegans_homolog_chrom_start': 'Caenorhabditis elegans chromosome/scaffold start (bp)' (default: False),\n",
      " 'celegans_homolog_chromosome': 'Caenorhabditis elegans chromosome/scaffold name' (default: False),\n",
      " 'celegans_homolog_dn': 'dN with Caenorhabditis elegans' (default: False),\n",
      " 'celegans_homolog_ds': 'dS with Caenorhabditis elegans' (default: False),\n",
      " 'celegans_homolog_ensembl_gene': 'Caenorhabditis elegans gene stable ID' (default: False),\n",
      " 'celegans_homolog_ensembl_peptide': 'Caenorhabditis elegans protein or transcript stable ID' (default: False),\n",
      " 'celegans_homolog_goc_score': 'Caenorhabditis elegans Gene-order conservation score' (default: False),\n",
      " 'celegans_homolog_orthology_confidence': 'Caenorhabditis elegans orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'celegans_homolog_orthology_type': 'Caenorhabditis elegans homology type' (default: False),\n",
      " 'celegans_homolog_perc_id': '%id. target Caenorhabditis elegans gene identical to query gene' (default: False),\n",
      " 'celegans_homolog_perc_id_r1': '%id. query gene identical to target Caenorhabditis elegans gene' (default: False),\n",
      " 'celegans_homolog_subtype': 'Last common ancestor with Caenorhabditis elegans' (default: False),\n",
      " 'celegans_homolog_wga_coverage': 'Caenorhabditis elegans Whole-genome alignment coverage' (default: False),\n",
      " 'cfamiliaris_homolog_associated_gene_name': 'Dog gene name' (default: False),\n",
      " 'cfamiliaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cfamiliaris_homolog_chrom_end': 'Dog chromosome/scaffold end (bp)' (default: False),\n",
      " 'cfamiliaris_homolog_chrom_start': 'Dog chromosome/scaffold start (bp)' (default: False),\n",
      " 'cfamiliaris_homolog_chromosome': 'Dog chromosome/scaffold name' (default: False),\n",
      " 'cfamiliaris_homolog_dn': 'dN with Dog' (default: False),\n",
      " 'cfamiliaris_homolog_ds': 'dS with Dog' (default: False),\n",
      " 'cfamiliaris_homolog_ensembl_gene': 'Dog gene stable ID' (default: False),\n",
      " 'cfamiliaris_homolog_ensembl_peptide': 'Dog protein or transcript stable ID' (default: False),\n",
      " 'cfamiliaris_homolog_goc_score': 'Dog Gene-order conservation score' (default: False),\n",
      " 'cfamiliaris_homolog_orthology_confidence': 'Dog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cfamiliaris_homolog_orthology_type': 'Dog homology type' (default: False),\n",
      " 'cfamiliaris_homolog_perc_id': '%id. target Dog gene identical to query gene' (default: False),\n",
      " 'cfamiliaris_homolog_perc_id_r1': '%id. query gene identical to target Dog gene' (default: False),\n",
      " 'cfamiliaris_homolog_subtype': 'Last common ancestor with Dog' (default: False),\n",
      " 'cfamiliaris_homolog_wga_coverage': 'Dog Whole-genome alignment coverage' (default: False),\n",
      " 'cgchok1gshd_homolog_associated_gene_name': 'Chinese hamster CHOK1GS gene name' (default: False),\n",
      " 'cgchok1gshd_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgchok1gshd_homolog_chrom_end': 'Chinese hamster CHOK1GS chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgchok1gshd_homolog_chrom_start': 'Chinese hamster CHOK1GS chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgchok1gshd_homolog_chromosome': 'Chinese hamster CHOK1GS chromosome/scaffold name' (default: False),\n",
      " 'cgchok1gshd_homolog_dn': 'dN with Chinese hamster CHOK1GS' (default: False),\n",
      " 'cgchok1gshd_homolog_ds': 'dS with Chinese hamster CHOK1GS' (default: False),\n",
      " 'cgchok1gshd_homolog_ensembl_gene': 'Chinese hamster CHOK1GS gene stable ID' (default: False),\n",
      " 'cgchok1gshd_homolog_ensembl_peptide': 'Chinese hamster CHOK1GS protein or transcript stable ID' (default: False),\n",
      " 'cgchok1gshd_homolog_goc_score': 'Chinese hamster CHOK1GS Gene-order conservation score' (default: False),\n",
      " 'cgchok1gshd_homolog_orthology_confidence': 'Chinese hamster CHOK1GS orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgchok1gshd_homolog_orthology_type': 'Chinese hamster CHOK1GS homology type' (default: False),\n",
      " 'cgchok1gshd_homolog_perc_id': '%id. target Chinese hamster CHOK1GS gene identical to query gene' (default: False),\n",
      " 'cgchok1gshd_homolog_perc_id_r1': '%id. query gene identical to target Chinese hamster CHOK1GS gene' (default: False),\n",
      " 'cgchok1gshd_homolog_subtype': 'Last common ancestor with Chinese hamster CHOK1GS' (default: False),\n",
      " 'cgchok1gshd_homolog_wga_coverage': 'Chinese hamster CHOK1GS Whole-genome alignment coverage' (default: False),\n",
      " 'cgcrigri_homolog_associated_gene_name': 'Chinese hamster CriGri gene name' (default: False),\n",
      " 'cgcrigri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgcrigri_homolog_chrom_end': 'Chinese hamster CriGri chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgcrigri_homolog_chrom_start': 'Chinese hamster CriGri chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgcrigri_homolog_chromosome': 'Chinese hamster CriGri chromosome/scaffold name' (default: False),\n",
      " 'cgcrigri_homolog_dn': 'dN with Chinese hamster CriGri' (default: False),\n",
      " 'cgcrigri_homolog_ds': 'dS with Chinese hamster CriGri' (default: False),\n",
      " 'cgcrigri_homolog_ensembl_gene': 'Chinese hamster CriGri gene stable ID' (default: False),\n",
      " 'cgcrigri_homolog_ensembl_peptide': 'Chinese hamster CriGri protein or transcript stable ID' (default: False),\n",
      " 'cgcrigri_homolog_goc_score': 'Chinese hamster CriGri Gene-order conservation score' (default: False),\n",
      " 'cgcrigri_homolog_orthology_confidence': 'Chinese hamster CriGri orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgcrigri_homolog_orthology_type': 'Chinese hamster CriGri homology type' (default: False),\n",
      " 'cgcrigri_homolog_perc_id': '%id. target Chinese hamster CriGri gene identical to query gene' (default: False),\n",
      " 'cgcrigri_homolog_perc_id_r1': '%id. query gene identical to target Chinese hamster CriGri gene' (default: False),\n",
      " 'cgcrigri_homolog_subtype': 'Last common ancestor with Chinese hamster CriGri' (default: False),\n",
      " 'cgcrigri_homolog_wga_coverage': 'Chinese hamster CriGri Whole-genome alignment coverage' (default: False),\n",
      " 'cgobio_homolog_associated_gene_name': 'Channel bull blenny gene name' (default: False),\n",
      " 'cgobio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgobio_homolog_chrom_end': 'Channel bull blenny chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgobio_homolog_chrom_start': 'Channel bull blenny chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgobio_homolog_chromosome': 'Channel bull blenny chromosome/scaffold name' (default: False),\n",
      " 'cgobio_homolog_dn': 'dN with Channel bull blenny' (default: False),\n",
      " 'cgobio_homolog_ds': 'dS with Channel bull blenny' (default: False),\n",
      " 'cgobio_homolog_ensembl_gene': 'Channel bull blenny gene stable ID' (default: False),\n",
      " 'cgobio_homolog_ensembl_peptide': 'Channel bull blenny protein or transcript stable ID' (default: False),\n",
      " 'cgobio_homolog_goc_score': 'Channel bull blenny Gene-order conservation score' (default: False),\n",
      " 'cgobio_homolog_orthology_confidence': 'Channel bull blenny orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgobio_homolog_orthology_type': 'Channel bull blenny homology type' (default: False),\n",
      " 'cgobio_homolog_perc_id': '%id. target Channel bull blenny gene identical to query gene' (default: False),\n",
      " 'cgobio_homolog_perc_id_r1': '%id. query gene identical to target Channel bull blenny gene' (default: False),\n",
      " 'cgobio_homolog_subtype': 'Last common ancestor with Channel bull blenny' (default: False),\n",
      " 'cgobio_homolog_wga_coverage': 'Channel bull blenny Whole-genome alignment coverage' (default: False),\n",
      " 'cgpicr_homolog_associated_gene_name': 'Chinese hamster PICR gene name' (default: False),\n",
      " 'cgpicr_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgpicr_homolog_chrom_end': 'Chinese hamster PICR chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgpicr_homolog_chrom_start': 'Chinese hamster PICR chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgpicr_homolog_chromosome': 'Chinese hamster PICR chromosome/scaffold name' (default: False),\n",
      " 'cgpicr_homolog_dn': 'dN with Chinese hamster PICR' (default: False),\n",
      " 'cgpicr_homolog_ds': 'dS with Chinese hamster PICR' (default: False),\n",
      " 'cgpicr_homolog_ensembl_gene': 'Chinese hamster PICR gene stable ID' (default: False),\n",
      " 'cgpicr_homolog_ensembl_peptide': 'Chinese hamster PICR protein or transcript stable ID' (default: False),\n",
      " 'cgpicr_homolog_goc_score': 'Chinese hamster PICR Gene-order conservation score' (default: False),\n",
      " 'cgpicr_homolog_orthology_confidence': 'Chinese hamster PICR orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgpicr_homolog_orthology_type': 'Chinese hamster PICR homology type' (default: False),\n",
      " 'cgpicr_homolog_perc_id': '%id. target Chinese hamster PICR gene identical to query gene' (default: False),\n",
      " 'cgpicr_homolog_perc_id_r1': '%id. query gene identical to target Chinese hamster PICR gene' (default: False),\n",
      " 'cgpicr_homolog_subtype': 'Last common ancestor with Chinese hamster PICR' (default: False),\n",
      " 'cgpicr_homolog_wga_coverage': 'Chinese hamster PICR Whole-genome alignment coverage' (default: False),\n",
      " 'charengus_homolog_associated_gene_name': 'Atlantic herring gene name' (default: False),\n",
      " 'charengus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'charengus_homolog_chrom_end': 'Atlantic herring chromosome/scaffold end (bp)' (default: False),\n",
      " 'charengus_homolog_chrom_start': 'Atlantic herring chromosome/scaffold start (bp)' (default: False),\n",
      " 'charengus_homolog_chromosome': 'Atlantic herring chromosome/scaffold name' (default: False),\n",
      " 'charengus_homolog_dn': 'dN with Atlantic herring' (default: False),\n",
      " 'charengus_homolog_ds': 'dS with Atlantic herring' (default: False),\n",
      " 'charengus_homolog_ensembl_gene': 'Atlantic herring gene stable ID' (default: False),\n",
      " 'charengus_homolog_ensembl_peptide': 'Atlantic herring protein or transcript stable ID' (default: False),\n",
      " 'charengus_homolog_goc_score': 'Atlantic herring Gene-order conservation score' (default: False),\n",
      " 'charengus_homolog_orthology_confidence': 'Atlantic herring orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'charengus_homolog_orthology_type': 'Atlantic herring homology type' (default: False),\n",
      " 'charengus_homolog_perc_id': '%id. target Atlantic herring gene identical to query gene' (default: False),\n",
      " 'charengus_homolog_perc_id_r1': '%id. query gene identical to target Atlantic herring gene' (default: False),\n",
      " 'charengus_homolog_subtype': 'Last common ancestor with Atlantic herring' (default: False),\n",
      " 'charengus_homolog_wga_coverage': 'Atlantic herring Whole-genome alignment coverage' (default: False),\n",
      " 'chembl': 'ChEMBL ID' (default: False),\n",
      " 'chircus_homolog_associated_gene_name': 'Goat gene name' (default: False),\n",
      " 'chircus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'chircus_homolog_chrom_end': 'Goat chromosome/scaffold end (bp)' (default: False),\n",
      " 'chircus_homolog_chrom_start': 'Goat chromosome/scaffold start (bp)' (default: False),\n",
      " 'chircus_homolog_chromosome': 'Goat chromosome/scaffold name' (default: False),\n",
      " 'chircus_homolog_dn': 'dN with Goat' (default: False),\n",
      " 'chircus_homolog_ds': 'dS with Goat' (default: False),\n",
      " 'chircus_homolog_ensembl_gene': 'Goat gene stable ID' (default: False),\n",
      " 'chircus_homolog_ensembl_peptide': 'Goat protein or transcript stable ID' (default: False),\n",
      " 'chircus_homolog_goc_score': 'Goat Gene-order conservation score' (default: False),\n",
      " 'chircus_homolog_orthology_confidence': 'Goat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'chircus_homolog_orthology_type': 'Goat homology type' (default: False),\n",
      " 'chircus_homolog_perc_id': '%id. target Goat gene identical to query gene' (default: False),\n",
      " 'chircus_homolog_perc_id_r1': '%id. query gene identical to target Goat gene' (default: False),\n",
      " 'chircus_homolog_subtype': 'Last common ancestor with Goat' (default: False),\n",
      " 'chircus_homolog_wga_coverage': 'Goat Whole-genome alignment coverage' (default: False),\n",
      " 'choffmanni_homolog_associated_gene_name': 'Sloth gene name' (default: False),\n",
      " 'choffmanni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'choffmanni_homolog_chrom_end': 'Sloth chromosome/scaffold end (bp)' (default: False),\n",
      " 'choffmanni_homolog_chrom_start': 'Sloth chromosome/scaffold start (bp)' (default: False),\n",
      " 'choffmanni_homolog_chromosome': 'Sloth chromosome/scaffold name' (default: False),\n",
      " 'choffmanni_homolog_dn': 'dN with Sloth' (default: False),\n",
      " 'choffmanni_homolog_ds': 'dS with Sloth' (default: False),\n",
      " 'choffmanni_homolog_ensembl_gene': 'Sloth gene stable ID' (default: False),\n",
      " 'choffmanni_homolog_ensembl_peptide': 'Sloth protein or transcript stable ID' (default: False),\n",
      " 'choffmanni_homolog_goc_score': 'Sloth Gene-order conservation score' (default: False),\n",
      " 'choffmanni_homolog_orthology_confidence': 'Sloth orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'choffmanni_homolog_orthology_type': 'Sloth homology type' (default: False),\n",
      " 'choffmanni_homolog_perc_id': '%id. target Sloth gene identical to query gene' (default: False),\n",
      " 'choffmanni_homolog_perc_id_r1': '%id. query gene identical to target Sloth gene' (default: False),\n",
      " 'choffmanni_homolog_subtype': 'Last common ancestor with Sloth' (default: False),\n",
      " 'choffmanni_homolog_wga_coverage': 'Sloth Whole-genome alignment coverage' (default: False),\n",
      " 'chromosome_end': 'Chromosome/scaffold position end (bp)' (default: False),\n",
      " 'chromosome_name': 'Chromosome/scaffold name' (default: False),\n",
      " 'chromosome_start': 'chromosome/scaffold position start (bp)' (default: False),\n",
      " 'cintestinalis_homolog_associated_gene_name': 'C.intestinalis gene name' (default: False),\n",
      " 'cintestinalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cintestinalis_homolog_chrom_end': 'C.intestinalis chromosome/scaffold end (bp)' (default: False),\n",
      " 'cintestinalis_homolog_chrom_start': 'C.intestinalis chromosome/scaffold start (bp)' (default: False),\n",
      " 'cintestinalis_homolog_chromosome': 'C.intestinalis chromosome/scaffold name' (default: False),\n",
      " 'cintestinalis_homolog_dn': 'dN with C.intestinalis' (default: False),\n",
      " 'cintestinalis_homolog_ds': 'dS with C.intestinalis' (default: False),\n",
      " 'cintestinalis_homolog_ensembl_gene': 'C.intestinalis gene stable ID' (default: False),\n",
      " 'cintestinalis_homolog_ensembl_peptide': 'C.intestinalis protein or transcript stable ID' (default: False),\n",
      " 'cintestinalis_homolog_goc_score': 'C.intestinalis Gene-order conservation score' (default: False),\n",
      " 'cintestinalis_homolog_orthology_confidence': 'C.intestinalis orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cintestinalis_homolog_orthology_type': 'C.intestinalis homology type' (default: False),\n",
      " 'cintestinalis_homolog_perc_id': '%id. target C.intestinalis gene identical to query gene' (default: False),\n",
      " 'cintestinalis_homolog_perc_id_r1': '%id. query gene identical to target C.intestinalis gene' (default: False),\n",
      " 'cintestinalis_homolog_subtype': 'Last common ancestor with C.intestinalis' (default: False),\n",
      " 'cintestinalis_homolog_wga_coverage': 'C.intestinalis Whole-genome alignment coverage' (default: False),\n",
      " 'cjacchus_homolog_associated_gene_name': 'Marmoset gene name' (default: False),\n",
      " 'cjacchus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cjacchus_homolog_chrom_end': 'Marmoset chromosome/scaffold end (bp)' (default: False),\n",
      " 'cjacchus_homolog_chrom_start': 'Marmoset chromosome/scaffold start (bp)' (default: False),\n",
      " 'cjacchus_homolog_chromosome': 'Marmoset chromosome/scaffold name' (default: False),\n",
      " 'cjacchus_homolog_dn': 'dN with Marmoset' (default: False),\n",
      " 'cjacchus_homolog_ds': 'dS with Marmoset' (default: False),\n",
      " 'cjacchus_homolog_ensembl_gene': 'Marmoset gene stable ID' (default: False),\n",
      " 'cjacchus_homolog_ensembl_peptide': 'Marmoset protein or transcript stable ID' (default: False),\n",
      " 'cjacchus_homolog_goc_score': 'Marmoset Gene-order conservation score' (default: False),\n",
      " 'cjacchus_homolog_orthology_confidence': 'Marmoset orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cjacchus_homolog_orthology_type': 'Marmoset homology type' (default: False),\n",
      " 'cjacchus_homolog_perc_id': '%id. target Marmoset gene identical to query gene' (default: False),\n",
      " 'cjacchus_homolog_perc_id_r1': '%id. query gene identical to target Marmoset gene' (default: False),\n",
      " 'cjacchus_homolog_subtype': 'Last common ancestor with Marmoset' (default: False),\n",
      " 'cjacchus_homolog_wga_coverage': 'Marmoset Whole-genome alignment coverage' (default: False),\n",
      " 'cjaponica_homolog_associated_gene_name': 'Japanese quail gene name' (default: False),\n",
      " 'cjaponica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cjaponica_homolog_chrom_end': 'Japanese quail chromosome/scaffold end (bp)' (default: False),\n",
      " 'cjaponica_homolog_chrom_start': 'Japanese quail chromosome/scaffold start (bp)' (default: False),\n",
      " 'cjaponica_homolog_chromosome': 'Japanese quail chromosome/scaffold name' (default: False),\n",
      " 'cjaponica_homolog_dn': 'dN with Japanese quail' (default: False),\n",
      " 'cjaponica_homolog_ds': 'dS with Japanese quail' (default: False),\n",
      " 'cjaponica_homolog_ensembl_gene': 'Japanese quail gene stable ID' (default: False),\n",
      " 'cjaponica_homolog_ensembl_peptide': 'Japanese quail protein or transcript stable ID' (default: False),\n",
      " 'cjaponica_homolog_goc_score': 'Japanese quail Gene-order conservation score' (default: False),\n",
      " 'cjaponica_homolog_orthology_confidence': 'Japanese quail orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cjaponica_homolog_orthology_type': 'Japanese quail homology type' (default: False),\n",
      " 'cjaponica_homolog_perc_id': '%id. target Japanese quail gene identical to query gene' (default: False),\n",
      " 'cjaponica_homolog_perc_id_r1': '%id. query gene identical to target Japanese quail gene' (default: False),\n",
      " 'cjaponica_homolog_subtype': 'Last common ancestor with Japanese quail' (default: False),\n",
      " 'cjaponica_homolog_wga_coverage': 'Japanese quail Whole-genome alignment coverage' (default: False),\n",
      " 'clanigera_homolog_associated_gene_name': 'Long-tailed chinchilla gene name' (default: False),\n",
      " 'clanigera_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clanigera_homolog_chrom_end': 'Long-tailed chinchilla chromosome/scaffold end (bp)' (default: False),\n",
      " 'clanigera_homolog_chrom_start': 'Long-tailed chinchilla chromosome/scaffold start (bp)' (default: False),\n",
      " 'clanigera_homolog_chromosome': 'Long-tailed chinchilla chromosome/scaffold name' (default: False),\n",
      " 'clanigera_homolog_dn': 'dN with Long-tailed chinchilla' (default: False),\n",
      " 'clanigera_homolog_ds': 'dS with Long-tailed chinchilla' (default: False),\n",
      " 'clanigera_homolog_ensembl_gene': 'Long-tailed chinchilla gene stable ID' (default: False),\n",
      " 'clanigera_homolog_ensembl_peptide': 'Long-tailed chinchilla protein or transcript stable ID' (default: False),\n",
      " 'clanigera_homolog_goc_score': 'Long-tailed chinchilla Gene-order conservation score' (default: False),\n",
      " 'clanigera_homolog_orthology_confidence': 'Long-tailed chinchilla orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clanigera_homolog_orthology_type': 'Long-tailed chinchilla homology type' (default: False),\n",
      " 'clanigera_homolog_perc_id': '%id. target Long-tailed chinchilla gene identical to query gene' (default: False),\n",
      " 'clanigera_homolog_perc_id_r1': '%id. query gene identical to target Long-tailed chinchilla gene' (default: False),\n",
      " 'clanigera_homolog_subtype': 'Last common ancestor with Long-tailed chinchilla' (default: False),\n",
      " 'clanigera_homolog_wga_coverage': 'Long-tailed chinchilla Whole-genome alignment coverage' (default: False),\n",
      " 'cldingo_homolog_associated_gene_name': 'Dingo gene name' (default: False),\n",
      " 'cldingo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cldingo_homolog_chrom_end': 'Dingo chromosome/scaffold end (bp)' (default: False),\n",
      " 'cldingo_homolog_chrom_start': 'Dingo chromosome/scaffold start (bp)' (default: False),\n",
      " 'cldingo_homolog_chromosome': 'Dingo chromosome/scaffold name' (default: False),\n",
      " 'cldingo_homolog_dn': 'dN with Dingo' (default: False),\n",
      " 'cldingo_homolog_ds': 'dS with Dingo' (default: False),\n",
      " 'cldingo_homolog_ensembl_gene': 'Dingo gene stable ID' (default: False),\n",
      " 'cldingo_homolog_ensembl_peptide': 'Dingo protein or transcript stable ID' (default: False),\n",
      " 'cldingo_homolog_goc_score': 'Dingo Gene-order conservation score' (default: False),\n",
      " 'cldingo_homolog_orthology_confidence': 'Dingo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cldingo_homolog_orthology_type': 'Dingo homology type' (default: False),\n",
      " 'cldingo_homolog_perc_id': '%id. target Dingo gene identical to query gene' (default: False),\n",
      " 'cldingo_homolog_perc_id_r1': '%id. query gene identical to target Dingo gene' (default: False),\n",
      " 'cldingo_homolog_subtype': 'Last common ancestor with Dingo' (default: False),\n",
      " 'cldingo_homolog_wga_coverage': 'Dingo Whole-genome alignment coverage' (default: False),\n",
      " 'clinical_significance': 'Clinical significance' (default: False),\n",
      " 'clone_based_ensembl_gene': 'Clone-based (Ensembl) gene ID' (default: False),\n",
      " 'clone_based_ensembl_transcript': 'Clone-based (Ensembl) transcript ID' (default: False),\n",
      " 'cmilii_homolog_associated_gene_name': 'Elephant shark gene name' (default: False),\n",
      " 'cmilii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cmilii_homolog_chrom_end': 'Elephant shark chromosome/scaffold end (bp)' (default: False),\n",
      " 'cmilii_homolog_chrom_start': 'Elephant shark chromosome/scaffold start (bp)' (default: False),\n",
      " 'cmilii_homolog_chromosome': 'Elephant shark chromosome/scaffold name' (default: False),\n",
      " 'cmilii_homolog_dn': 'dN with Elephant shark' (default: False),\n",
      " 'cmilii_homolog_ds': 'dS with Elephant shark' (default: False),\n",
      " 'cmilii_homolog_ensembl_gene': 'Elephant shark gene stable ID' (default: False),\n",
      " 'cmilii_homolog_ensembl_peptide': 'Elephant shark protein or transcript stable ID' (default: False),\n",
      " 'cmilii_homolog_goc_score': 'Elephant shark Gene-order conservation score' (default: False),\n",
      " 'cmilii_homolog_orthology_confidence': 'Elephant shark orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cmilii_homolog_orthology_type': 'Elephant shark homology type' (default: False),\n",
      " 'cmilii_homolog_perc_id': '%id. target Elephant shark gene identical to query gene' (default: False),\n",
      " 'cmilii_homolog_perc_id_r1': '%id. query gene identical to target Elephant shark gene' (default: False),\n",
      " 'cmilii_homolog_subtype': 'Last common ancestor with Elephant shark' (default: False),\n",
      " 'cmilii_homolog_wga_coverage': 'Elephant shark Whole-genome alignment coverage' (default: False),\n",
      " 'codelink_codelink': 'CODELINK CODELINK probe' (default: False),\n",
      " 'coding': 'Coding sequence' (default: False),\n",
      " 'coding_gene_flank': 'Flank-coding region (Gene)' (default: False),\n",
      " 'coding_transcript_flank': 'Flank-coding region (Transcript)' (default: False),\n",
      " 'cpbellii_homolog_associated_gene_name': 'Painted turtle gene name' (default: False),\n",
      " 'cpbellii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cpbellii_homolog_chrom_end': 'Painted turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'cpbellii_homolog_chrom_start': 'Painted turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'cpbellii_homolog_chromosome': 'Painted turtle chromosome/scaffold name' (default: False),\n",
      " 'cpbellii_homolog_dn': 'dN with Painted turtle' (default: False),\n",
      " 'cpbellii_homolog_ds': 'dS with Painted turtle' (default: False),\n",
      " 'cpbellii_homolog_ensembl_gene': 'Painted turtle gene stable ID' (default: False),\n",
      " 'cpbellii_homolog_ensembl_peptide': 'Painted turtle protein or transcript stable ID' (default: False),\n",
      " 'cpbellii_homolog_goc_score': 'Painted turtle Gene-order conservation score' (default: False),\n",
      " 'cpbellii_homolog_orthology_confidence': 'Painted turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cpbellii_homolog_orthology_type': 'Painted turtle homology type' (default: False),\n",
      " 'cpbellii_homolog_perc_id': '%id. target Painted turtle gene identical to query gene' (default: False),\n",
      " 'cpbellii_homolog_perc_id_r1': '%id. query gene identical to target Painted turtle gene' (default: False),\n",
      " 'cpbellii_homolog_subtype': 'Last common ancestor with Painted turtle' (default: False),\n",
      " 'cpbellii_homolog_wga_coverage': 'Painted turtle Whole-genome alignment coverage' (default: False),\n",
      " 'cporcellus_homolog_associated_gene_name': 'Guinea Pig gene name' (default: False),\n",
      " 'cporcellus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cporcellus_homolog_chrom_end': 'Guinea Pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'cporcellus_homolog_chrom_start': 'Guinea Pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'cporcellus_homolog_chromosome': 'Guinea Pig chromosome/scaffold name' (default: False),\n",
      " 'cporcellus_homolog_dn': 'dN with Guinea Pig' (default: False),\n",
      " 'cporcellus_homolog_ds': 'dS with Guinea Pig' (default: False),\n",
      " 'cporcellus_homolog_ensembl_gene': 'Guinea Pig gene stable ID' (default: False),\n",
      " 'cporcellus_homolog_ensembl_peptide': 'Guinea Pig protein or transcript stable ID' (default: False),\n",
      " 'cporcellus_homolog_goc_score': 'Guinea Pig Gene-order conservation score' (default: False),\n",
      " 'cporcellus_homolog_orthology_confidence': 'Guinea Pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cporcellus_homolog_orthology_type': 'Guinea Pig homology type' (default: False),\n",
      " 'cporcellus_homolog_perc_id': '%id. target Guinea Pig gene identical to query gene' (default: False),\n",
      " 'cporcellus_homolog_perc_id_r1': '%id. query gene identical to target Guinea Pig gene' (default: False),\n",
      " 'cporcellus_homolog_subtype': 'Last common ancestor with Guinea Pig' (default: False),\n",
      " 'cporcellus_homolog_wga_coverage': 'Guinea Pig Whole-genome alignment coverage' (default: False),\n",
      " 'cporosus_homolog_associated_gene_name': 'Australian saltwater crocodile gene name' (default: False),\n",
      " 'cporosus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cporosus_homolog_chrom_end': 'Australian saltwater crocodile chromosome/scaffold end (bp)' (default: False),\n",
      " 'cporosus_homolog_chrom_start': 'Australian saltwater crocodile chromosome/scaffold start (bp)' (default: False),\n",
      " 'cporosus_homolog_chromosome': 'Australian saltwater crocodile chromosome/scaffold name' (default: False),\n",
      " 'cporosus_homolog_dn': 'dN with Australian saltwater crocodile' (default: False),\n",
      " 'cporosus_homolog_ds': 'dS with Australian saltwater crocodile' (default: False),\n",
      " 'cporosus_homolog_ensembl_gene': 'Australian saltwater crocodile gene stable ID' (default: False),\n",
      " 'cporosus_homolog_ensembl_peptide': 'Australian saltwater crocodile protein or transcript stable ID' (default: False),\n",
      " 'cporosus_homolog_goc_score': 'Australian saltwater crocodile Gene-order conservation score' (default: False),\n",
      " 'cporosus_homolog_orthology_confidence': 'Australian saltwater crocodile orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cporosus_homolog_orthology_type': 'Australian saltwater crocodile homology type' (default: False),\n",
      " 'cporosus_homolog_perc_id': '%id. target Australian saltwater crocodile gene identical to query gene' (default: False),\n",
      " 'cporosus_homolog_perc_id_r1': '%id. query gene identical to target Australian saltwater crocodile gene' (default: False),\n",
      " 'cporosus_homolog_subtype': 'Last common ancestor with Australian saltwater crocodile' (default: False),\n",
      " 'cporosus_homolog_wga_coverage': 'Australian saltwater crocodile Whole-genome alignment coverage' (default: False),\n",
      " 'cpugnax_homolog_associated_gene_name': 'Ruff gene name' (default: False),\n",
      " 'cpugnax_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cpugnax_homolog_chrom_end': 'Ruff chromosome/scaffold end (bp)' (default: False),\n",
      " 'cpugnax_homolog_chrom_start': 'Ruff chromosome/scaffold start (bp)' (default: False),\n",
      " 'cpugnax_homolog_chromosome': 'Ruff chromosome/scaffold name' (default: False),\n",
      " 'cpugnax_homolog_dn': 'dN with Ruff' (default: False),\n",
      " 'cpugnax_homolog_ds': 'dS with Ruff' (default: False),\n",
      " 'cpugnax_homolog_ensembl_gene': 'Ruff gene stable ID' (default: False),\n",
      " 'cpugnax_homolog_ensembl_peptide': 'Ruff protein or transcript stable ID' (default: False),\n",
      " 'cpugnax_homolog_goc_score': 'Ruff Gene-order conservation score' (default: False),\n",
      " 'cpugnax_homolog_orthology_confidence': 'Ruff orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cpugnax_homolog_orthology_type': 'Ruff homology type' (default: False),\n",
      " 'cpugnax_homolog_perc_id': '%id. target Ruff gene identical to query gene' (default: False),\n",
      " 'cpugnax_homolog_perc_id_r1': '%id. query gene identical to target Ruff gene' (default: False),\n",
      " 'cpugnax_homolog_subtype': 'Last common ancestor with Ruff' (default: False),\n",
      " 'cpugnax_homolog_wga_coverage': 'Ruff Whole-genome alignment coverage' (default: False),\n",
      " 'cpygmaea_homolog_associated_gene_name': 'Spoon-billed sandpiper gene name' (default: False),\n",
      " 'cpygmaea_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cpygmaea_homolog_chrom_end': 'Spoon-billed sandpiper chromosome/scaffold end (bp)' (default: False),\n",
      " 'cpygmaea_homolog_chrom_start': 'Spoon-billed sandpiper chromosome/scaffold start (bp)' (default: False),\n",
      " 'cpygmaea_homolog_chromosome': 'Spoon-billed sandpiper chromosome/scaffold name' (default: False),\n",
      " 'cpygmaea_homolog_dn': 'dN with Spoon-billed sandpiper' (default: False),\n",
      " 'cpygmaea_homolog_ds': 'dS with Spoon-billed sandpiper' (default: False),\n",
      " 'cpygmaea_homolog_ensembl_gene': 'Spoon-billed sandpiper gene stable ID' (default: False),\n",
      " 'cpygmaea_homolog_ensembl_peptide': 'Spoon-billed sandpiper protein or transcript stable ID' (default: False),\n",
      " 'cpygmaea_homolog_goc_score': 'Spoon-billed sandpiper Gene-order conservation score' (default: False),\n",
      " 'cpygmaea_homolog_orthology_confidence': 'Spoon-billed sandpiper orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cpygmaea_homolog_orthology_type': 'Spoon-billed sandpiper homology type' (default: False),\n",
      " 'cpygmaea_homolog_perc_id': '%id. target Spoon-billed sandpiper gene identical to query gene' (default: False),\n",
      " 'cpygmaea_homolog_perc_id_r1': '%id. query gene identical to target Spoon-billed sandpiper gene' (default: False),\n",
      " 'cpygmaea_homolog_subtype': 'Last common ancestor with Spoon-billed sandpiper' (default: False),\n",
      " 'cpygmaea_homolog_wga_coverage': 'Spoon-billed sandpiper Whole-genome alignment coverage' (default: False),\n",
      " 'csabaeus_homolog_associated_gene_name': 'Vervet-AGM gene name' (default: False),\n",
      " 'csabaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csabaeus_homolog_chrom_end': 'Vervet-AGM chromosome/scaffold end (bp)' (default: False),\n",
      " 'csabaeus_homolog_chrom_start': 'Vervet-AGM chromosome/scaffold start (bp)' (default: False),\n",
      " 'csabaeus_homolog_chromosome': 'Vervet-AGM chromosome/scaffold name' (default: False),\n",
      " 'csabaeus_homolog_dn': 'dN with Vervet-AGM' (default: False),\n",
      " 'csabaeus_homolog_ds': 'dS with Vervet-AGM' (default: False),\n",
      " 'csabaeus_homolog_ensembl_gene': 'Vervet-AGM gene stable ID' (default: False),\n",
      " 'csabaeus_homolog_ensembl_peptide': 'Vervet-AGM protein or transcript stable ID' (default: False),\n",
      " 'csabaeus_homolog_goc_score': 'Vervet-AGM Gene-order conservation score' (default: False),\n",
      " 'csabaeus_homolog_orthology_confidence': 'Vervet-AGM orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csabaeus_homolog_orthology_type': 'Vervet-AGM homology type' (default: False),\n",
      " 'csabaeus_homolog_perc_id': '%id. target Vervet-AGM gene identical to query gene' (default: False),\n",
      " 'csabaeus_homolog_perc_id_r1': '%id. query gene identical to target Vervet-AGM gene' (default: False),\n",
      " 'csabaeus_homolog_subtype': 'Last common ancestor with Vervet-AGM' (default: False),\n",
      " 'csabaeus_homolog_wga_coverage': 'Vervet-AGM Whole-genome alignment coverage' (default: False),\n",
      " 'csavignyi_homolog_associated_gene_name': 'C.savignyi gene name' (default: False),\n",
      " 'csavignyi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csavignyi_homolog_chrom_end': 'C.savignyi chromosome/scaffold end (bp)' (default: False),\n",
      " 'csavignyi_homolog_chrom_start': 'C.savignyi chromosome/scaffold start (bp)' (default: False),\n",
      " 'csavignyi_homolog_chromosome': 'C.savignyi chromosome/scaffold name' (default: False),\n",
      " 'csavignyi_homolog_dn': 'dN with C.savignyi' (default: False),\n",
      " 'csavignyi_homolog_ds': 'dS with C.savignyi' (default: False),\n",
      " 'csavignyi_homolog_ensembl_gene': 'C.savignyi gene stable ID' (default: False),\n",
      " 'csavignyi_homolog_ensembl_peptide': 'C.savignyi protein or transcript stable ID' (default: False),\n",
      " 'csavignyi_homolog_goc_score': 'C.savignyi Gene-order conservation score' (default: False),\n",
      " 'csavignyi_homolog_orthology_confidence': 'C.savignyi orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csavignyi_homolog_orthology_type': 'C.savignyi homology type' (default: False),\n",
      " 'csavignyi_homolog_perc_id': '%id. target C.savignyi gene identical to query gene' (default: False),\n",
      " 'csavignyi_homolog_perc_id_r1': '%id. query gene identical to target C.savignyi gene' (default: False),\n",
      " 'csavignyi_homolog_subtype': 'Last common ancestor with C.savignyi' (default: False),\n",
      " 'csavignyi_homolog_wga_coverage': 'C.savignyi Whole-genome alignment coverage' (default: False),\n",
      " 'csemilaevis_homolog_associated_gene_name': 'Tongue sole gene name' (default: False),\n",
      " 'csemilaevis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csemilaevis_homolog_chrom_end': 'Tongue sole chromosome/scaffold end (bp)' (default: False),\n",
      " 'csemilaevis_homolog_chrom_start': 'Tongue sole chromosome/scaffold start (bp)' (default: False),\n",
      " 'csemilaevis_homolog_chromosome': 'Tongue sole chromosome/scaffold name' (default: False),\n",
      " 'csemilaevis_homolog_dn': 'dN with Tongue sole' (default: False),\n",
      " 'csemilaevis_homolog_ds': 'dS with Tongue sole' (default: False),\n",
      " 'csemilaevis_homolog_ensembl_gene': 'Tongue sole gene stable ID' (default: False),\n",
      " 'csemilaevis_homolog_ensembl_peptide': 'Tongue sole protein or transcript stable ID' (default: False),\n",
      " 'csemilaevis_homolog_goc_score': 'Tongue sole Gene-order conservation score' (default: False),\n",
      " 'csemilaevis_homolog_orthology_confidence': 'Tongue sole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csemilaevis_homolog_orthology_type': 'Tongue sole homology type' (default: False),\n",
      " 'csemilaevis_homolog_perc_id': '%id. target Tongue sole gene identical to query gene' (default: False),\n",
      " 'csemilaevis_homolog_perc_id_r1': '%id. query gene identical to target Tongue sole gene' (default: False),\n",
      " 'csemilaevis_homolog_subtype': 'Last common ancestor with Tongue sole' (default: False),\n",
      " 'csemilaevis_homolog_wga_coverage': 'Tongue sole Whole-genome alignment coverage' (default: False),\n",
      " 'csyrichta_homolog_associated_gene_name': 'Tarsier gene name' (default: False),\n",
      " 'csyrichta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csyrichta_homolog_chrom_end': 'Tarsier chromosome/scaffold end (bp)' (default: False),\n",
      " 'csyrichta_homolog_chrom_start': 'Tarsier chromosome/scaffold start (bp)' (default: False),\n",
      " 'csyrichta_homolog_chromosome': 'Tarsier chromosome/scaffold name' (default: False),\n",
      " 'csyrichta_homolog_dn': 'dN with Tarsier' (default: False),\n",
      " 'csyrichta_homolog_ds': 'dS with Tarsier' (default: False),\n",
      " 'csyrichta_homolog_ensembl_gene': 'Tarsier gene stable ID' (default: False),\n",
      " 'csyrichta_homolog_ensembl_peptide': 'Tarsier protein or transcript stable ID' (default: False),\n",
      " 'csyrichta_homolog_goc_score': 'Tarsier Gene-order conservation score' (default: False),\n",
      " 'csyrichta_homolog_orthology_confidence': 'Tarsier orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csyrichta_homolog_orthology_type': 'Tarsier homology type' (default: False),\n",
      " 'csyrichta_homolog_perc_id': '%id. target Tarsier gene identical to query gene' (default: False),\n",
      " 'csyrichta_homolog_perc_id_r1': '%id. query gene identical to target Tarsier gene' (default: False),\n",
      " 'csyrichta_homolog_subtype': 'Last common ancestor with Tarsier' (default: False),\n",
      " 'csyrichta_homolog_wga_coverage': 'Tarsier Whole-genome alignment coverage' (default: False),\n",
      " 'cvariegatus_homolog_associated_gene_name': 'Sheepshead minnow gene name' (default: False),\n",
      " 'cvariegatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cvariegatus_homolog_chrom_end': 'Sheepshead minnow chromosome/scaffold end (bp)' (default: False),\n",
      " 'cvariegatus_homolog_chrom_start': 'Sheepshead minnow chromosome/scaffold start (bp)' (default: False),\n",
      " 'cvariegatus_homolog_chromosome': 'Sheepshead minnow chromosome/scaffold name' (default: False),\n",
      " 'cvariegatus_homolog_dn': 'dN with Sheepshead minnow' (default: False),\n",
      " 'cvariegatus_homolog_ds': 'dS with Sheepshead minnow' (default: False),\n",
      " 'cvariegatus_homolog_ensembl_gene': 'Sheepshead minnow gene stable ID' (default: False),\n",
      " 'cvariegatus_homolog_ensembl_peptide': 'Sheepshead minnow protein or transcript stable ID' (default: False),\n",
      " 'cvariegatus_homolog_goc_score': 'Sheepshead minnow Gene-order conservation score' (default: False),\n",
      " 'cvariegatus_homolog_orthology_confidence': 'Sheepshead minnow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cvariegatus_homolog_orthology_type': 'Sheepshead minnow homology type' (default: False),\n",
      " 'cvariegatus_homolog_perc_id': '%id. target Sheepshead minnow gene identical to query gene' (default: False),\n",
      " 'cvariegatus_homolog_perc_id_r1': '%id. query gene identical to target Sheepshead minnow gene' (default: False),\n",
      " 'cvariegatus_homolog_subtype': 'Last common ancestor with Sheepshead minnow' (default: False),\n",
      " 'cvariegatus_homolog_wga_coverage': 'Sheepshead minnow Whole-genome alignment coverage' (default: False),\n",
      " 'dbass3_id': 'DataBase of Aberrant 3' Splice Sites ID' (default: False),\n",
      " 'dbass3_name': 'DataBase of Aberrant 3' Splice Sites name' (default: False),\n",
      " 'dbass5_id': 'DataBase of Aberrant 5' Splice Sites ID' (default: False),\n",
      " 'dbass5_name': 'DataBase of Aberrant 5' Splice Sites name' (default: False),\n",
      " 'dclupeoides_homolog_associated_gene_name': 'Denticle herring gene name' (default: False),\n",
      " 'dclupeoides_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dclupeoides_homolog_chrom_end': 'Denticle herring chromosome/scaffold end (bp)' (default: False),\n",
      " 'dclupeoides_homolog_chrom_start': 'Denticle herring chromosome/scaffold start (bp)' (default: False),\n",
      " 'dclupeoides_homolog_chromosome': 'Denticle herring chromosome/scaffold name' (default: False),\n",
      " 'dclupeoides_homolog_dn': 'dN with Denticle herring' (default: False),\n",
      " 'dclupeoides_homolog_ds': 'dS with Denticle herring' (default: False),\n",
      " 'dclupeoides_homolog_ensembl_gene': 'Denticle herring gene stable ID' (default: False),\n",
      " 'dclupeoides_homolog_ensembl_peptide': 'Denticle herring protein or transcript stable ID' (default: False),\n",
      " 'dclupeoides_homolog_goc_score': 'Denticle herring Gene-order conservation score' (default: False),\n",
      " 'dclupeoides_homolog_orthology_confidence': 'Denticle herring orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dclupeoides_homolog_orthology_type': 'Denticle herring homology type' (default: False),\n",
      " 'dclupeoides_homolog_perc_id': '%id. target Denticle herring gene identical to query gene' (default: False),\n",
      " 'dclupeoides_homolog_perc_id_r1': '%id. query gene identical to target Denticle herring gene' (default: False),\n",
      " 'dclupeoides_homolog_subtype': 'Last common ancestor with Denticle herring' (default: False),\n",
      " 'dclupeoides_homolog_wga_coverage': 'Denticle herring Whole-genome alignment coverage' (default: False),\n",
      " 'definition_1006': 'GO term definition' (default: False),\n",
      " 'description': 'Gene description' (default: False),\n",
      " 'distance_to_transcript_2076': 'Distance to transcript' (default: False),\n",
      " 'dmelanogaster_homolog_associated_gene_name': 'Drosophila melanogaster gene name' (default: False),\n",
      " 'dmelanogaster_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dmelanogaster_homolog_chrom_end': 'Drosophila melanogaster chromosome/scaffold end (bp)' (default: False),\n",
      " 'dmelanogaster_homolog_chrom_start': 'Drosophila melanogaster chromosome/scaffold start (bp)' (default: False),\n",
      " 'dmelanogaster_homolog_chromosome': 'Drosophila melanogaster chromosome/scaffold name' (default: False),\n",
      " 'dmelanogaster_homolog_dn': 'dN with Drosophila melanogaster' (default: False),\n",
      " 'dmelanogaster_homolog_ds': 'dS with Drosophila melanogaster' (default: False),\n",
      " 'dmelanogaster_homolog_ensembl_gene': 'Drosophila melanogaster gene stable ID' (default: False),\n",
      " 'dmelanogaster_homolog_ensembl_peptide': 'Drosophila melanogaster protein or transcript stable ID' (default: False),\n",
      " 'dmelanogaster_homolog_goc_score': 'Drosophila melanogaster Gene-order conservation score' (default: False),\n",
      " 'dmelanogaster_homolog_orthology_confidence': 'Drosophila melanogaster orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dmelanogaster_homolog_orthology_type': 'Drosophila melanogaster homology type' (default: False),\n",
      " 'dmelanogaster_homolog_perc_id': '%id. target Drosophila melanogaster gene identical to query gene' (default: False),\n",
      " 'dmelanogaster_homolog_perc_id_r1': '%id. query gene identical to target Drosophila melanogaster gene' (default: False),\n",
      " 'dmelanogaster_homolog_subtype': 'Last common ancestor with Drosophila melanogaster' (default: False),\n",
      " 'dmelanogaster_homolog_wga_coverage': 'Drosophila melanogaster Whole-genome alignment coverage' (default: False),\n",
      " 'dnovaehollandiae_homolog_associated_gene_name': 'Emu gene name' (default: False),\n",
      " 'dnovaehollandiae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dnovaehollandiae_homolog_chrom_end': 'Emu chromosome/scaffold end (bp)' (default: False),\n",
      " 'dnovaehollandiae_homolog_chrom_start': 'Emu chromosome/scaffold start (bp)' (default: False),\n",
      " 'dnovaehollandiae_homolog_chromosome': 'Emu chromosome/scaffold name' (default: False),\n",
      " 'dnovaehollandiae_homolog_dn': 'dN with Emu' (default: False),\n",
      " 'dnovaehollandiae_homolog_ds': 'dS with Emu' (default: False),\n",
      " 'dnovaehollandiae_homolog_ensembl_gene': 'Emu gene stable ID' (default: False),\n",
      " 'dnovaehollandiae_homolog_ensembl_peptide': 'Emu protein or transcript stable ID' (default: False),\n",
      " 'dnovaehollandiae_homolog_goc_score': 'Emu Gene-order conservation score' (default: False),\n",
      " 'dnovaehollandiae_homolog_orthology_confidence': 'Emu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dnovaehollandiae_homolog_orthology_type': 'Emu homology type' (default: False),\n",
      " 'dnovaehollandiae_homolog_perc_id': '%id. target Emu gene identical to query gene' (default: False),\n",
      " 'dnovaehollandiae_homolog_perc_id_r1': '%id. query gene identical to target Emu gene' (default: False),\n",
      " 'dnovaehollandiae_homolog_subtype': 'Last common ancestor with Emu' (default: False),\n",
      " 'dnovaehollandiae_homolog_wga_coverage': 'Emu Whole-genome alignment coverage' (default: False),\n",
      " 'dnovemcinctus_homolog_associated_gene_name': 'Armadillo gene name' (default: False),\n",
      " 'dnovemcinctus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dnovemcinctus_homolog_chrom_end': 'Armadillo chromosome/scaffold end (bp)' (default: False),\n",
      " 'dnovemcinctus_homolog_chrom_start': 'Armadillo chromosome/scaffold start (bp)' (default: False),\n",
      " 'dnovemcinctus_homolog_chromosome': 'Armadillo chromosome/scaffold name' (default: False),\n",
      " 'dnovemcinctus_homolog_dn': 'dN with Armadillo' (default: False),\n",
      " 'dnovemcinctus_homolog_ds': 'dS with Armadillo' (default: False),\n",
      " 'dnovemcinctus_homolog_ensembl_gene': 'Armadillo gene stable ID' (default: False),\n",
      " 'dnovemcinctus_homolog_ensembl_peptide': 'Armadillo protein or transcript stable ID' (default: False),\n",
      " 'dnovemcinctus_homolog_goc_score': 'Armadillo Gene-order conservation score' (default: False),\n",
      " 'dnovemcinctus_homolog_orthology_confidence': 'Armadillo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dnovemcinctus_homolog_orthology_type': 'Armadillo homology type' (default: False),\n",
      " 'dnovemcinctus_homolog_perc_id': '%id. target Armadillo gene identical to query gene' (default: False),\n",
      " 'dnovemcinctus_homolog_perc_id_r1': '%id. query gene identical to target Armadillo gene' (default: False),\n",
      " 'dnovemcinctus_homolog_subtype': 'Last common ancestor with Armadillo' (default: False),\n",
      " 'dnovemcinctus_homolog_wga_coverage': 'Armadillo Whole-genome alignment coverage' (default: False),\n",
      " 'dordii_homolog_associated_gene_name': 'Kangaroo rat gene name' (default: False),\n",
      " 'dordii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dordii_homolog_chrom_end': 'Kangaroo rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'dordii_homolog_chrom_start': 'Kangaroo rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'dordii_homolog_chromosome': 'Kangaroo rat chromosome/scaffold name' (default: False),\n",
      " 'dordii_homolog_dn': 'dN with Kangaroo rat' (default: False),\n",
      " 'dordii_homolog_ds': 'dS with Kangaroo rat' (default: False),\n",
      " 'dordii_homolog_ensembl_gene': 'Kangaroo rat gene stable ID' (default: False),\n",
      " 'dordii_homolog_ensembl_peptide': 'Kangaroo rat protein or transcript stable ID' (default: False),\n",
      " 'dordii_homolog_goc_score': 'Kangaroo rat Gene-order conservation score' (default: False),\n",
      " 'dordii_homolog_orthology_confidence': 'Kangaroo rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dordii_homolog_orthology_type': 'Kangaroo rat homology type' (default: False),\n",
      " 'dordii_homolog_perc_id': '%id. target Kangaroo rat gene identical to query gene' (default: False),\n",
      " 'dordii_homolog_perc_id_r1': '%id. query gene identical to target Kangaroo rat gene' (default: False),\n",
      " 'dordii_homolog_subtype': 'Last common ancestor with Kangaroo rat' (default: False),\n",
      " 'dordii_homolog_wga_coverage': 'Kangaroo rat Whole-genome alignment coverage' (default: False),\n",
      " 'downstream_flank': 'downstream_flank' (default: False),\n",
      " 'drerio_homolog_associated_gene_name': 'Zebrafish gene name' (default: False),\n",
      " 'drerio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'drerio_homolog_chrom_end': 'Zebrafish chromosome/scaffold end (bp)' (default: False),\n",
      " 'drerio_homolog_chrom_start': 'Zebrafish chromosome/scaffold start (bp)' (default: False),\n",
      " 'drerio_homolog_chromosome': 'Zebrafish chromosome/scaffold name' (default: False),\n",
      " 'drerio_homolog_dn': 'dN with Zebrafish' (default: False),\n",
      " 'drerio_homolog_ds': 'dS with Zebrafish' (default: False),\n",
      " 'drerio_homolog_ensembl_gene': 'Zebrafish gene stable ID' (default: False),\n",
      " 'drerio_homolog_ensembl_peptide': 'Zebrafish protein or transcript stable ID' (default: False),\n",
      " 'drerio_homolog_goc_score': 'Zebrafish Gene-order conservation score' (default: False),\n",
      " 'drerio_homolog_orthology_confidence': 'Zebrafish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'drerio_homolog_orthology_type': 'Zebrafish homology type' (default: False),\n",
      " 'drerio_homolog_perc_id': '%id. target Zebrafish gene identical to query gene' (default: False),\n",
      " 'drerio_homolog_perc_id_r1': '%id. query gene identical to target Zebrafish gene' (default: False),\n",
      " 'drerio_homolog_subtype': 'Last common ancestor with Zebrafish' (default: False),\n",
      " 'drerio_homolog_wga_coverage': 'Zebrafish Whole-genome alignment coverage' (default: False),\n",
      " 'eaasinus_homolog_associated_gene_name': 'Donkey gene name' (default: False),\n",
      " 'eaasinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eaasinus_homolog_chrom_end': 'Donkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'eaasinus_homolog_chrom_start': 'Donkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'eaasinus_homolog_chromosome': 'Donkey chromosome/scaffold name' (default: False),\n",
      " 'eaasinus_homolog_dn': 'dN with Donkey' (default: False),\n",
      " 'eaasinus_homolog_ds': 'dS with Donkey' (default: False),\n",
      " 'eaasinus_homolog_ensembl_gene': 'Donkey gene stable ID' (default: False),\n",
      " 'eaasinus_homolog_ensembl_peptide': 'Donkey protein or transcript stable ID' (default: False),\n",
      " 'eaasinus_homolog_goc_score': 'Donkey Gene-order conservation score' (default: False),\n",
      " 'eaasinus_homolog_orthology_confidence': 'Donkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eaasinus_homolog_orthology_type': 'Donkey homology type' (default: False),\n",
      " 'eaasinus_homolog_perc_id': '%id. target Donkey gene identical to query gene' (default: False),\n",
      " 'eaasinus_homolog_perc_id_r1': '%id. query gene identical to target Donkey gene' (default: False),\n",
      " 'eaasinus_homolog_subtype': 'Last common ancestor with Donkey' (default: False),\n",
      " 'eaasinus_homolog_wga_coverage': 'Donkey Whole-genome alignment coverage' (default: False),\n",
      " 'eburgeri_homolog_associated_gene_name': 'Hagfish gene name' (default: False),\n",
      " 'eburgeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eburgeri_homolog_chrom_end': 'Hagfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'eburgeri_homolog_chrom_start': 'Hagfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'eburgeri_homolog_chromosome': 'Hagfish chromosome/scaffold name' (default: False),\n",
      " 'eburgeri_homolog_dn': 'dN with Hagfish' (default: False),\n",
      " 'eburgeri_homolog_ds': 'dS with Hagfish' (default: False),\n",
      " 'eburgeri_homolog_ensembl_gene': 'Hagfish gene stable ID' (default: False),\n",
      " 'eburgeri_homolog_ensembl_peptide': 'Hagfish protein or transcript stable ID' (default: False),\n",
      " 'eburgeri_homolog_goc_score': 'Hagfish Gene-order conservation score' (default: False),\n",
      " 'eburgeri_homolog_orthology_confidence': 'Hagfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eburgeri_homolog_orthology_type': 'Hagfish homology type' (default: False),\n",
      " 'eburgeri_homolog_perc_id': '%id. target Hagfish gene identical to query gene' (default: False),\n",
      " 'eburgeri_homolog_perc_id_r1': '%id. query gene identical to target Hagfish gene' (default: False),\n",
      " 'eburgeri_homolog_subtype': 'Last common ancestor with Hagfish' (default: False),\n",
      " 'eburgeri_homolog_wga_coverage': 'Hagfish Whole-genome alignment coverage' (default: False),\n",
      " 'ecaballus_homolog_associated_gene_name': 'Horse gene name' (default: False),\n",
      " 'ecaballus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ecaballus_homolog_chrom_end': 'Horse chromosome/scaffold end (bp)' (default: False),\n",
      " 'ecaballus_homolog_chrom_start': 'Horse chromosome/scaffold start (bp)' (default: False),\n",
      " 'ecaballus_homolog_chromosome': 'Horse chromosome/scaffold name' (default: False),\n",
      " 'ecaballus_homolog_dn': 'dN with Horse' (default: False),\n",
      " 'ecaballus_homolog_ds': 'dS with Horse' (default: False),\n",
      " 'ecaballus_homolog_ensembl_gene': 'Horse gene stable ID' (default: False),\n",
      " 'ecaballus_homolog_ensembl_peptide': 'Horse protein or transcript stable ID' (default: False),\n",
      " 'ecaballus_homolog_goc_score': 'Horse Gene-order conservation score' (default: False),\n",
      " 'ecaballus_homolog_orthology_confidence': 'Horse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ecaballus_homolog_orthology_type': 'Horse homology type' (default: False),\n",
      " 'ecaballus_homolog_perc_id': '%id. target Horse gene identical to query gene' (default: False),\n",
      " 'ecaballus_homolog_perc_id_r1': '%id. query gene identical to target Horse gene' (default: False),\n",
      " 'ecaballus_homolog_subtype': 'Last common ancestor with Horse' (default: False),\n",
      " 'ecaballus_homolog_wga_coverage': 'Horse Whole-genome alignment coverage' (default: False),\n",
      " 'ecalabaricus_homolog_associated_gene_name': 'Reedfish gene name' (default: False),\n",
      " 'ecalabaricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ecalabaricus_homolog_chrom_end': 'Reedfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ecalabaricus_homolog_chrom_start': 'Reedfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ecalabaricus_homolog_chromosome': 'Reedfish chromosome/scaffold name' (default: False),\n",
      " 'ecalabaricus_homolog_dn': 'dN with Reedfish' (default: False),\n",
      " 'ecalabaricus_homolog_ds': 'dS with Reedfish' (default: False),\n",
      " 'ecalabaricus_homolog_ensembl_gene': 'Reedfish gene stable ID' (default: False),\n",
      " 'ecalabaricus_homolog_ensembl_peptide': 'Reedfish protein or transcript stable ID' (default: False),\n",
      " 'ecalabaricus_homolog_goc_score': 'Reedfish Gene-order conservation score' (default: False),\n",
      " 'ecalabaricus_homolog_orthology_confidence': 'Reedfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ecalabaricus_homolog_orthology_type': 'Reedfish homology type' (default: False),\n",
      " 'ecalabaricus_homolog_perc_id': '%id. target Reedfish gene identical to query gene' (default: False),\n",
      " 'ecalabaricus_homolog_perc_id_r1': '%id. query gene identical to target Reedfish gene' (default: False),\n",
      " 'ecalabaricus_homolog_subtype': 'Last common ancestor with Reedfish' (default: False),\n",
      " 'ecalabaricus_homolog_wga_coverage': 'Reedfish Whole-genome alignment coverage' (default: False),\n",
      " 'eelectricus_homolog_associated_gene_name': 'Electric eel gene name' (default: False),\n",
      " 'eelectricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eelectricus_homolog_chrom_end': 'Electric eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'eelectricus_homolog_chrom_start': 'Electric eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'eelectricus_homolog_chromosome': 'Electric eel chromosome/scaffold name' (default: False),\n",
      " 'eelectricus_homolog_dn': 'dN with Electric eel' (default: False),\n",
      " 'eelectricus_homolog_ds': 'dS with Electric eel' (default: False),\n",
      " 'eelectricus_homolog_ensembl_gene': 'Electric eel gene stable ID' (default: False),\n",
      " 'eelectricus_homolog_ensembl_peptide': 'Electric eel protein or transcript stable ID' (default: False),\n",
      " 'eelectricus_homolog_goc_score': 'Electric eel Gene-order conservation score' (default: False),\n",
      " 'eelectricus_homolog_orthology_confidence': 'Electric eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eelectricus_homolog_orthology_type': 'Electric eel homology type' (default: False),\n",
      " 'eelectricus_homolog_perc_id': '%id. target Electric eel gene identical to query gene' (default: False),\n",
      " 'eelectricus_homolog_perc_id_r1': '%id. query gene identical to target Electric eel gene' (default: False),\n",
      " 'eelectricus_homolog_subtype': 'Last common ancestor with Electric eel' (default: False),\n",
      " 'eelectricus_homolog_wga_coverage': 'Electric eel Whole-genome alignment coverage' (default: False),\n",
      " 'eeuropaeus_homolog_associated_gene_name': 'Hedgehog gene name' (default: False),\n",
      " 'eeuropaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eeuropaeus_homolog_chrom_end': 'Hedgehog chromosome/scaffold end (bp)' (default: False),\n",
      " 'eeuropaeus_homolog_chrom_start': 'Hedgehog chromosome/scaffold start (bp)' (default: False),\n",
      " 'eeuropaeus_homolog_chromosome': 'Hedgehog chromosome/scaffold name' (default: False),\n",
      " 'eeuropaeus_homolog_dn': 'dN with Hedgehog' (default: False),\n",
      " 'eeuropaeus_homolog_ds': 'dS with Hedgehog' (default: False),\n",
      " 'eeuropaeus_homolog_ensembl_gene': 'Hedgehog gene stable ID' (default: False),\n",
      " 'eeuropaeus_homolog_ensembl_peptide': 'Hedgehog protein or transcript stable ID' (default: False),\n",
      " 'eeuropaeus_homolog_goc_score': 'Hedgehog Gene-order conservation score' (default: False),\n",
      " 'eeuropaeus_homolog_orthology_confidence': 'Hedgehog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eeuropaeus_homolog_orthology_type': 'Hedgehog homology type' (default: False),\n",
      " 'eeuropaeus_homolog_perc_id': '%id. target Hedgehog gene identical to query gene' (default: False),\n",
      " 'eeuropaeus_homolog_perc_id_r1': '%id. query gene identical to target Hedgehog gene' (default: False),\n",
      " 'eeuropaeus_homolog_subtype': 'Last common ancestor with Hedgehog' (default: False),\n",
      " 'eeuropaeus_homolog_wga_coverage': 'Hedgehog Whole-genome alignment coverage' (default: False),\n",
      " 'elucius_homolog_associated_gene_name': 'Northern pike gene name' (default: False),\n",
      " 'elucius_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'elucius_homolog_chrom_end': 'Northern pike chromosome/scaffold end (bp)' (default: False),\n",
      " 'elucius_homolog_chrom_start': 'Northern pike chromosome/scaffold start (bp)' (default: False),\n",
      " 'elucius_homolog_chromosome': 'Northern pike chromosome/scaffold name' (default: False),\n",
      " 'elucius_homolog_dn': 'dN with Northern pike' (default: False),\n",
      " 'elucius_homolog_ds': 'dS with Northern pike' (default: False),\n",
      " 'elucius_homolog_ensembl_gene': 'Northern pike gene stable ID' (default: False),\n",
      " 'elucius_homolog_ensembl_peptide': 'Northern pike protein or transcript stable ID' (default: False),\n",
      " 'elucius_homolog_goc_score': 'Northern pike Gene-order conservation score' (default: False),\n",
      " 'elucius_homolog_orthology_confidence': 'Northern pike orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'elucius_homolog_orthology_type': 'Northern pike homology type' (default: False),\n",
      " 'elucius_homolog_perc_id': '%id. target Northern pike gene identical to query gene' (default: False),\n",
      " 'elucius_homolog_perc_id_r1': '%id. query gene identical to target Northern pike gene' (default: False),\n",
      " 'elucius_homolog_subtype': 'Last common ancestor with Northern pike' (default: False),\n",
      " 'elucius_homolog_wga_coverage': 'Northern pike Whole-genome alignment coverage' (default: False),\n",
      " 'embl': 'European Nucleotide Archive ID' (default: False),\n",
      " 'end_phase': 'End phase' (default: False),\n",
      " 'end_position': 'Gene end (bp)' (default: False),\n",
      " 'ens_hs_transcript': 'Ensembl Human Transcript ID' (default: False),\n",
      " 'ens_hs_translation': 'Ensembl Human Translation ID' (default: False),\n",
      " 'ens_lrg_gene': 'LRG display in Ensembl gene ID' (default: False),\n",
      " 'ens_lrg_transcript': 'LRG display in Ensembl transcript ID' (default: False),\n",
      " 'ensembl_exon_id': 'Exon stable ID' (default: False),\n",
      " 'ensembl_gene_id': 'Gene stable ID' (default: False),\n",
      " 'ensembl_gene_id_version': 'Gene stable ID version' (default: False),\n",
      " 'ensembl_peptide_id': 'Protein stable ID' (default: False),\n",
      " 'ensembl_peptide_id_version': 'Protein stable ID version' (default: False),\n",
      " 'ensembl_transcript_id': 'Transcript stable ID' (default: False),\n",
      " 'ensembl_transcript_id_version': 'Transcript stable ID version' (default: False),\n",
      " 'entrezgene_accession': 'NCBI gene accession' (default: False),\n",
      " 'entrezgene_description': 'NCBI gene description' (default: False),\n",
      " 'entrezgene_id': 'NCBI gene ID' (default: False),\n",
      " 'entrezgene_trans_name': 'EntrezGene transcript name ID' (default: False),\n",
      " 'etelfairi_homolog_associated_gene_name': 'Lesser hedgehog tenrec gene name' (default: False),\n",
      " 'etelfairi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'etelfairi_homolog_chrom_end': 'Lesser hedgehog tenrec chromosome/scaffold end (bp)' (default: False),\n",
      " 'etelfairi_homolog_chrom_start': 'Lesser hedgehog tenrec chromosome/scaffold start (bp)' (default: False),\n",
      " 'etelfairi_homolog_chromosome': 'Lesser hedgehog tenrec chromosome/scaffold name' (default: False),\n",
      " 'etelfairi_homolog_dn': 'dN with Lesser hedgehog tenrec' (default: False),\n",
      " 'etelfairi_homolog_ds': 'dS with Lesser hedgehog tenrec' (default: False),\n",
      " 'etelfairi_homolog_ensembl_gene': 'Lesser hedgehog tenrec gene stable ID' (default: False),\n",
      " 'etelfairi_homolog_ensembl_peptide': 'Lesser hedgehog tenrec protein or transcript stable ID' (default: False),\n",
      " 'etelfairi_homolog_goc_score': 'Lesser hedgehog tenrec Gene-order conservation score' (default: False),\n",
      " 'etelfairi_homolog_orthology_confidence': 'Lesser hedgehog tenrec orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'etelfairi_homolog_orthology_type': 'Lesser hedgehog tenrec homology type' (default: False),\n",
      " 'etelfairi_homolog_perc_id': '%id. target Lesser hedgehog tenrec gene identical to query gene' (default: False),\n",
      " 'etelfairi_homolog_perc_id_r1': '%id. query gene identical to target Lesser hedgehog tenrec gene' (default: False),\n",
      " 'etelfairi_homolog_subtype': 'Last common ancestor with Lesser hedgehog tenrec' (default: False),\n",
      " 'etelfairi_homolog_wga_coverage': 'Lesser hedgehog tenrec Whole-genome alignment coverage' (default: False),\n",
      " 'exon_chrom_end': 'Exon region end (bp)' (default: False),\n",
      " 'exon_chrom_start': 'Exon region start (bp)' (default: False),\n",
      " 'external_gene_name': 'Gene name' (default: False),\n",
      " 'external_gene_source': 'Source of gene name' (default: False),\n",
      " 'external_transcript_name': 'Transcript name' (default: False),\n",
      " 'external_transcript_source_name': 'Source of transcript name' (default: False),\n",
      " 'falbicollis_homolog_associated_gene_name': 'Flycatcher gene name' (default: False),\n",
      " 'falbicollis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'falbicollis_homolog_chrom_end': 'Flycatcher chromosome/scaffold end (bp)' (default: False),\n",
      " 'falbicollis_homolog_chrom_start': 'Flycatcher chromosome/scaffold start (bp)' (default: False),\n",
      " 'falbicollis_homolog_chromosome': 'Flycatcher chromosome/scaffold name' (default: False),\n",
      " 'falbicollis_homolog_dn': 'dN with Flycatcher' (default: False),\n",
      " 'falbicollis_homolog_ds': 'dS with Flycatcher' (default: False),\n",
      " 'falbicollis_homolog_ensembl_gene': 'Flycatcher gene stable ID' (default: False),\n",
      " 'falbicollis_homolog_ensembl_peptide': 'Flycatcher protein or transcript stable ID' (default: False),\n",
      " 'falbicollis_homolog_goc_score': 'Flycatcher Gene-order conservation score' (default: False),\n",
      " 'falbicollis_homolog_orthology_confidence': 'Flycatcher orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'falbicollis_homolog_orthology_type': 'Flycatcher homology type' (default: False),\n",
      " 'falbicollis_homolog_perc_id': '%id. target Flycatcher gene identical to query gene' (default: False),\n",
      " 'falbicollis_homolog_perc_id_r1': '%id. query gene identical to target Flycatcher gene' (default: False),\n",
      " 'falbicollis_homolog_subtype': 'Last common ancestor with Flycatcher' (default: False),\n",
      " 'falbicollis_homolog_wga_coverage': 'Flycatcher Whole-genome alignment coverage' (default: False),\n",
      " 'family': 'Ensembl Protein Family ID(s)' (default: False),\n",
      " 'family_description': 'Ensembl Family Description' (default: False),\n",
      " 'fcatus_homolog_associated_gene_name': 'Cat gene name' (default: False),\n",
      " 'fcatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fcatus_homolog_chrom_end': 'Cat chromosome/scaffold end (bp)' (default: False),\n",
      " 'fcatus_homolog_chrom_start': 'Cat chromosome/scaffold start (bp)' (default: False),\n",
      " 'fcatus_homolog_chromosome': 'Cat chromosome/scaffold name' (default: False),\n",
      " 'fcatus_homolog_dn': 'dN with Cat' (default: False),\n",
      " 'fcatus_homolog_ds': 'dS with Cat' (default: False),\n",
      " 'fcatus_homolog_ensembl_gene': 'Cat gene stable ID' (default: False),\n",
      " 'fcatus_homolog_ensembl_peptide': 'Cat protein or transcript stable ID' (default: False),\n",
      " 'fcatus_homolog_goc_score': 'Cat Gene-order conservation score' (default: False),\n",
      " 'fcatus_homolog_orthology_confidence': 'Cat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fcatus_homolog_orthology_type': 'Cat homology type' (default: False),\n",
      " 'fcatus_homolog_perc_id': '%id. target Cat gene identical to query gene' (default: False),\n",
      " 'fcatus_homolog_perc_id_r1': '%id. query gene identical to target Cat gene' (default: False),\n",
      " 'fcatus_homolog_subtype': 'Last common ancestor with Cat' (default: False),\n",
      " 'fcatus_homolog_wga_coverage': 'Cat Whole-genome alignment coverage' (default: False),\n",
      " 'fdamarensis_homolog_associated_gene_name': 'Damara mole rat gene name' (default: False),\n",
      " 'fdamarensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fdamarensis_homolog_chrom_end': 'Damara mole rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'fdamarensis_homolog_chrom_start': 'Damara mole rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'fdamarensis_homolog_chromosome': 'Damara mole rat chromosome/scaffold name' (default: False),\n",
      " 'fdamarensis_homolog_dn': 'dN with Damara mole rat' (default: False),\n",
      " 'fdamarensis_homolog_ds': 'dS with Damara mole rat' (default: False),\n",
      " 'fdamarensis_homolog_ensembl_gene': 'Damara mole rat gene stable ID' (default: False),\n",
      " 'fdamarensis_homolog_ensembl_peptide': 'Damara mole rat protein or transcript stable ID' (default: False),\n",
      " 'fdamarensis_homolog_goc_score': 'Damara mole rat Gene-order conservation score' (default: False),\n",
      " 'fdamarensis_homolog_orthology_confidence': 'Damara mole rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fdamarensis_homolog_orthology_type': 'Damara mole rat homology type' (default: False),\n",
      " 'fdamarensis_homolog_perc_id': '%id. target Damara mole rat gene identical to query gene' (default: False),\n",
      " 'fdamarensis_homolog_perc_id_r1': '%id. query gene identical to target Damara mole rat gene' (default: False),\n",
      " 'fdamarensis_homolog_subtype': 'Last common ancestor with Damara mole rat' (default: False),\n",
      " 'fdamarensis_homolog_wga_coverage': 'Damara mole rat Whole-genome alignment coverage' (default: False),\n",
      " 'fheteroclitus_homolog_associated_gene_name': 'Mummichog gene name' (default: False),\n",
      " 'fheteroclitus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fheteroclitus_homolog_chrom_end': 'Mummichog chromosome/scaffold end (bp)' (default: False),\n",
      " 'fheteroclitus_homolog_chrom_start': 'Mummichog chromosome/scaffold start (bp)' (default: False),\n",
      " 'fheteroclitus_homolog_chromosome': 'Mummichog chromosome/scaffold name' (default: False),\n",
      " 'fheteroclitus_homolog_dn': 'dN with Mummichog' (default: False),\n",
      " 'fheteroclitus_homolog_ds': 'dS with Mummichog' (default: False),\n",
      " 'fheteroclitus_homolog_ensembl_gene': 'Mummichog gene stable ID' (default: False),\n",
      " 'fheteroclitus_homolog_ensembl_peptide': 'Mummichog protein or transcript stable ID' (default: False),\n",
      " 'fheteroclitus_homolog_goc_score': 'Mummichog Gene-order conservation score' (default: False),\n",
      " 'fheteroclitus_homolog_orthology_confidence': 'Mummichog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fheteroclitus_homolog_orthology_type': 'Mummichog homology type' (default: False),\n",
      " 'fheteroclitus_homolog_perc_id': '%id. target Mummichog gene identical to query gene' (default: False),\n",
      " 'fheteroclitus_homolog_perc_id_r1': '%id. query gene identical to target Mummichog gene' (default: False),\n",
      " 'fheteroclitus_homolog_subtype': 'Last common ancestor with Mummichog' (default: False),\n",
      " 'fheteroclitus_homolog_wga_coverage': 'Mummichog Whole-genome alignment coverage' (default: False),\n",
      " 'gaculeatus_homolog_associated_gene_name': 'Stickleback gene name' (default: False),\n",
      " 'gaculeatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gaculeatus_homolog_chrom_end': 'Stickleback chromosome/scaffold end (bp)' (default: False),\n",
      " 'gaculeatus_homolog_chrom_start': 'Stickleback chromosome/scaffold start (bp)' (default: False),\n",
      " 'gaculeatus_homolog_chromosome': 'Stickleback chromosome/scaffold name' (default: False),\n",
      " 'gaculeatus_homolog_dn': 'dN with Stickleback' (default: False),\n",
      " 'gaculeatus_homolog_ds': 'dS with Stickleback' (default: False),\n",
      " 'gaculeatus_homolog_ensembl_gene': 'Stickleback gene stable ID' (default: False),\n",
      " 'gaculeatus_homolog_ensembl_peptide': 'Stickleback protein or transcript stable ID' (default: False),\n",
      " 'gaculeatus_homolog_goc_score': 'Stickleback Gene-order conservation score' (default: False),\n",
      " 'gaculeatus_homolog_orthology_confidence': 'Stickleback orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gaculeatus_homolog_orthology_type': 'Stickleback homology type' (default: False),\n",
      " 'gaculeatus_homolog_perc_id': '%id. target Stickleback gene identical to query gene' (default: False),\n",
      " 'gaculeatus_homolog_perc_id_r1': '%id. query gene identical to target Stickleback gene' (default: False),\n",
      " 'gaculeatus_homolog_subtype': 'Last common ancestor with Stickleback' (default: False),\n",
      " 'gaculeatus_homolog_wga_coverage': 'Stickleback Whole-genome alignment coverage' (default: False),\n",
      " 'gaffinis_homolog_associated_gene_name': 'Western mosquitofish gene name' (default: False),\n",
      " 'gaffinis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gaffinis_homolog_chrom_end': 'Western mosquitofish chromosome/scaffold end (bp)' (default: False),\n",
      " 'gaffinis_homolog_chrom_start': 'Western mosquitofish chromosome/scaffold start (bp)' (default: False),\n",
      " 'gaffinis_homolog_chromosome': 'Western mosquitofish chromosome/scaffold name' (default: False),\n",
      " 'gaffinis_homolog_dn': 'dN with Western mosquitofish' (default: False),\n",
      " 'gaffinis_homolog_ds': 'dS with Western mosquitofish' (default: False),\n",
      " 'gaffinis_homolog_ensembl_gene': 'Western mosquitofish gene stable ID' (default: False),\n",
      " 'gaffinis_homolog_ensembl_peptide': 'Western mosquitofish protein or transcript stable ID' (default: False),\n",
      " 'gaffinis_homolog_goc_score': 'Western mosquitofish Gene-order conservation score' (default: False),\n",
      " 'gaffinis_homolog_orthology_confidence': 'Western mosquitofish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gaffinis_homolog_orthology_type': 'Western mosquitofish homology type' (default: False),\n",
      " 'gaffinis_homolog_perc_id': '%id. target Western mosquitofish gene identical to query gene' (default: False),\n",
      " 'gaffinis_homolog_perc_id_r1': '%id. query gene identical to target Western mosquitofish gene' (default: False),\n",
      " 'gaffinis_homolog_subtype': 'Last common ancestor with Western mosquitofish' (default: False),\n",
      " 'gaffinis_homolog_wga_coverage': 'Western mosquitofish Whole-genome alignment coverage' (default: False),\n",
      " 'gagassizii_homolog_associated_gene_name': 'Agassiz's desert tortoise gene name' (default: False),\n",
      " 'gagassizii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gagassizii_homolog_chrom_end': 'Agassiz's desert tortoise chromosome/scaffold end (bp)' (default: False),\n",
      " 'gagassizii_homolog_chrom_start': 'Agassiz's desert tortoise chromosome/scaffold start (bp)' (default: False),\n",
      " 'gagassizii_homolog_chromosome': 'Agassiz's desert tortoise chromosome/scaffold name' (default: False),\n",
      " 'gagassizii_homolog_dn': 'dN with Agassiz's desert tortoise' (default: False),\n",
      " 'gagassizii_homolog_ds': 'dS with Agassiz's desert tortoise' (default: False),\n",
      " 'gagassizii_homolog_ensembl_gene': 'Agassiz's desert tortoise gene stable ID' (default: False),\n",
      " 'gagassizii_homolog_ensembl_peptide': 'Agassiz's desert tortoise protein or transcript stable ID' (default: False),\n",
      " 'gagassizii_homolog_goc_score': 'Agassiz's desert tortoise Gene-order conservation score' (default: False),\n",
      " 'gagassizii_homolog_orthology_confidence': 'Agassiz's desert tortoise orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gagassizii_homolog_orthology_type': 'Agassiz's desert tortoise homology type' (default: False),\n",
      " 'gagassizii_homolog_perc_id': '%id. target Agassiz's desert tortoise gene identical to query gene' (default: False),\n",
      " 'gagassizii_homolog_perc_id_r1': '%id. query gene identical to target Agassiz's desert tortoise gene' (default: False),\n",
      " 'gagassizii_homolog_subtype': 'Last common ancestor with Agassiz's desert tortoise' (default: False),\n",
      " 'gagassizii_homolog_wga_coverage': 'Agassiz's desert tortoise Whole-genome alignment coverage' (default: False),\n",
      " 'gene3d': 'Gene3D ID' (default: False),\n",
      " 'gene3d_end': 'Gene3D end' (default: False),\n",
      " 'gene3d_start': 'Gene3D start' (default: False),\n",
      " 'gene_biotype': 'Gene type' (default: False),\n",
      " 'gene_exon': 'Exon sequences' (default: False),\n",
      " 'gene_exon_intron': 'Unspliced (Gene)' (default: False),\n",
      " 'gene_flank': 'Flank (Gene)' (default: False),\n",
      " 'genedb': 'GeneDB ID' (default: False),\n",
      " 'genomic_coding_end': 'Genomic coding end' (default: False),\n",
      " 'genomic_coding_start': 'Genomic coding start' (default: False),\n",
      " 'germ_line_variation_source': 'Variant source' (default: False),\n",
      " 'ggallus_homolog_associated_gene_name': 'Chicken gene name' (default: False),\n",
      " 'ggallus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ggallus_homolog_chrom_end': 'Chicken chromosome/scaffold end (bp)' (default: False),\n",
      " 'ggallus_homolog_chrom_start': 'Chicken chromosome/scaffold start (bp)' (default: False),\n",
      " 'ggallus_homolog_chromosome': 'Chicken chromosome/scaffold name' (default: False),\n",
      " 'ggallus_homolog_dn': 'dN with Chicken' (default: False),\n",
      " 'ggallus_homolog_ds': 'dS with Chicken' (default: False),\n",
      " 'ggallus_homolog_ensembl_gene': 'Chicken gene stable ID' (default: False),\n",
      " 'ggallus_homolog_ensembl_peptide': 'Chicken protein or transcript stable ID' (default: False),\n",
      " 'ggallus_homolog_goc_score': 'Chicken Gene-order conservation score' (default: False),\n",
      " 'ggallus_homolog_orthology_confidence': 'Chicken orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ggallus_homolog_orthology_type': 'Chicken homology type' (default: False),\n",
      " 'ggallus_homolog_perc_id': '%id. target Chicken gene identical to query gene' (default: False),\n",
      " 'ggallus_homolog_perc_id_r1': '%id. query gene identical to target Chicken gene' (default: False),\n",
      " 'ggallus_homolog_subtype': 'Last common ancestor with Chicken' (default: False),\n",
      " 'ggallus_homolog_wga_coverage': 'Chicken Whole-genome alignment coverage' (default: False),\n",
      " 'ggorilla_homolog_associated_gene_name': 'Gorilla gene name' (default: False),\n",
      " 'ggorilla_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ggorilla_homolog_chrom_end': 'Gorilla chromosome/scaffold end (bp)' (default: False),\n",
      " 'ggorilla_homolog_chrom_start': 'Gorilla chromosome/scaffold start (bp)' (default: False),\n",
      " 'ggorilla_homolog_chromosome': 'Gorilla chromosome/scaffold name' (default: False),\n",
      " 'ggorilla_homolog_dn': 'dN with Gorilla' (default: False),\n",
      " 'ggorilla_homolog_ds': 'dS with Gorilla' (default: False),\n",
      " 'ggorilla_homolog_ensembl_gene': 'Gorilla gene stable ID' (default: False),\n",
      " 'ggorilla_homolog_ensembl_peptide': 'Gorilla protein or transcript stable ID' (default: False),\n",
      " 'ggorilla_homolog_goc_score': 'Gorilla Gene-order conservation score' (default: False),\n",
      " 'ggorilla_homolog_orthology_confidence': 'Gorilla orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ggorilla_homolog_orthology_type': 'Gorilla homology type' (default: False),\n",
      " 'ggorilla_homolog_perc_id': '%id. target Gorilla gene identical to query gene' (default: False),\n",
      " 'ggorilla_homolog_perc_id_r1': '%id. query gene identical to target Gorilla gene' (default: False),\n",
      " 'ggorilla_homolog_subtype': 'Last common ancestor with Gorilla' (default: False),\n",
      " 'ggorilla_homolog_wga_coverage': 'Gorilla Whole-genome alignment coverage' (default: False),\n",
      " 'gmorhua_homolog_associated_gene_name': 'Cod gene name' (default: False),\n",
      " 'gmorhua_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gmorhua_homolog_chrom_end': 'Cod chromosome/scaffold end (bp)' (default: False),\n",
      " 'gmorhua_homolog_chrom_start': 'Cod chromosome/scaffold start (bp)' (default: False),\n",
      " 'gmorhua_homolog_chromosome': 'Cod chromosome/scaffold name' (default: False),\n",
      " 'gmorhua_homolog_dn': 'dN with Cod' (default: False),\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 'gmorhua_homolog_ds': 'dS with Cod' (default: False),\n",
      " 'gmorhua_homolog_ensembl_gene': 'Cod gene stable ID' (default: False),\n",
      " 'gmorhua_homolog_ensembl_peptide': 'Cod protein or transcript stable ID' (default: False),\n",
      " 'gmorhua_homolog_goc_score': 'Cod Gene-order conservation score' (default: False),\n",
      " 'gmorhua_homolog_orthology_confidence': 'Cod orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gmorhua_homolog_orthology_type': 'Cod homology type' (default: False),\n",
      " 'gmorhua_homolog_perc_id': '%id. target Cod gene identical to query gene' (default: False),\n",
      " 'gmorhua_homolog_perc_id_r1': '%id. query gene identical to target Cod gene' (default: False),\n",
      " 'gmorhua_homolog_subtype': 'Last common ancestor with Cod' (default: False),\n",
      " 'gmorhua_homolog_wga_coverage': 'Cod Whole-genome alignment coverage' (default: False),\n",
      " 'go_id': 'GO term accession' (default: False),\n",
      " 'go_linkage_type': 'GO term evidence code' (default: False),\n",
      " 'goslim_goa_accession': 'GOSlim GOA Accession(s)' (default: False),\n",
      " 'goslim_goa_description': 'GOSlim GOA Description' (default: False),\n",
      " 'gwilldenowi_homolog_associated_gene_name': 'Blunt-snouted clingfish gene name' (default: False),\n",
      " 'gwilldenowi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gwilldenowi_homolog_chrom_end': 'Blunt-snouted clingfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'gwilldenowi_homolog_chrom_start': 'Blunt-snouted clingfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'gwilldenowi_homolog_chromosome': 'Blunt-snouted clingfish chromosome/scaffold name' (default: False),\n",
      " 'gwilldenowi_homolog_dn': 'dN with Blunt-snouted clingfish' (default: False),\n",
      " 'gwilldenowi_homolog_ds': 'dS with Blunt-snouted clingfish' (default: False),\n",
      " 'gwilldenowi_homolog_ensembl_gene': 'Blunt-snouted clingfish gene stable ID' (default: False),\n",
      " 'gwilldenowi_homolog_ensembl_peptide': 'Blunt-snouted clingfish protein or transcript stable ID' (default: False),\n",
      " 'gwilldenowi_homolog_goc_score': 'Blunt-snouted clingfish Gene-order conservation score' (default: False),\n",
      " 'gwilldenowi_homolog_orthology_confidence': 'Blunt-snouted clingfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gwilldenowi_homolog_orthology_type': 'Blunt-snouted clingfish homology type' (default: False),\n",
      " 'gwilldenowi_homolog_perc_id': '%id. target Blunt-snouted clingfish gene identical to query gene' (default: False),\n",
      " 'gwilldenowi_homolog_perc_id_r1': '%id. query gene identical to target Blunt-snouted clingfish gene' (default: False),\n",
      " 'gwilldenowi_homolog_subtype': 'Last common ancestor with Blunt-snouted clingfish' (default: False),\n",
      " 'gwilldenowi_homolog_wga_coverage': 'Blunt-snouted clingfish Whole-genome alignment coverage' (default: False),\n",
      " 'hamap': 'HAMAP ID' (default: False),\n",
      " 'hamap_end': 'HAMAP end' (default: False),\n",
      " 'hamap_start': 'HAMAP start' (default: False),\n",
      " 'hburtoni_homolog_associated_gene_name': 'Burton's mouthbrooder gene name' (default: False),\n",
      " 'hburtoni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hburtoni_homolog_chrom_end': 'Burton's mouthbrooder chromosome/scaffold end (bp)' (default: False),\n",
      " 'hburtoni_homolog_chrom_start': 'Burton's mouthbrooder chromosome/scaffold start (bp)' (default: False),\n",
      " 'hburtoni_homolog_chromosome': 'Burton's mouthbrooder chromosome/scaffold name' (default: False),\n",
      " 'hburtoni_homolog_dn': 'dN with Burton's mouthbrooder' (default: False),\n",
      " 'hburtoni_homolog_ds': 'dS with Burton's mouthbrooder' (default: False),\n",
      " 'hburtoni_homolog_ensembl_gene': 'Burton's mouthbrooder gene stable ID' (default: False),\n",
      " 'hburtoni_homolog_ensembl_peptide': 'Burton's mouthbrooder protein or transcript stable ID' (default: False),\n",
      " 'hburtoni_homolog_goc_score': 'Burton's mouthbrooder Gene-order conservation score' (default: False),\n",
      " 'hburtoni_homolog_orthology_confidence': 'Burton's mouthbrooder orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hburtoni_homolog_orthology_type': 'Burton's mouthbrooder homology type' (default: False),\n",
      " 'hburtoni_homolog_perc_id': '%id. target Burton's mouthbrooder gene identical to query gene' (default: False),\n",
      " 'hburtoni_homolog_perc_id_r1': '%id. query gene identical to target Burton's mouthbrooder gene' (default: False),\n",
      " 'hburtoni_homolog_subtype': 'Last common ancestor with Burton's mouthbrooder' (default: False),\n",
      " 'hburtoni_homolog_wga_coverage': 'Burton's mouthbrooder Whole-genome alignment coverage' (default: False),\n",
      " 'hcomes_homolog_associated_gene_name': 'Tiger tail seahorse gene name' (default: False),\n",
      " 'hcomes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hcomes_homolog_chrom_end': 'Tiger tail seahorse chromosome/scaffold end (bp)' (default: False),\n",
      " 'hcomes_homolog_chrom_start': 'Tiger tail seahorse chromosome/scaffold start (bp)' (default: False),\n",
      " 'hcomes_homolog_chromosome': 'Tiger tail seahorse chromosome/scaffold name' (default: False),\n",
      " 'hcomes_homolog_dn': 'dN with Tiger tail seahorse' (default: False),\n",
      " 'hcomes_homolog_ds': 'dS with Tiger tail seahorse' (default: False),\n",
      " 'hcomes_homolog_ensembl_gene': 'Tiger tail seahorse gene stable ID' (default: False),\n",
      " 'hcomes_homolog_ensembl_peptide': 'Tiger tail seahorse protein or transcript stable ID' (default: False),\n",
      " 'hcomes_homolog_goc_score': 'Tiger tail seahorse Gene-order conservation score' (default: False),\n",
      " 'hcomes_homolog_orthology_confidence': 'Tiger tail seahorse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hcomes_homolog_orthology_type': 'Tiger tail seahorse homology type' (default: False),\n",
      " 'hcomes_homolog_perc_id': '%id. target Tiger tail seahorse gene identical to query gene' (default: False),\n",
      " 'hcomes_homolog_perc_id_r1': '%id. query gene identical to target Tiger tail seahorse gene' (default: False),\n",
      " 'hcomes_homolog_subtype': 'Last common ancestor with Tiger tail seahorse' (default: False),\n",
      " 'hcomes_homolog_wga_coverage': 'Tiger tail seahorse Whole-genome alignment coverage' (default: False),\n",
      " 'hgfemale_homolog_associated_gene_name': 'Naked mole-rat female gene name' (default: False),\n",
      " 'hgfemale_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hgfemale_homolog_chrom_end': 'Naked mole-rat female chromosome/scaffold end (bp)' (default: False),\n",
      " 'hgfemale_homolog_chrom_start': 'Naked mole-rat female chromosome/scaffold start (bp)' (default: False),\n",
      " 'hgfemale_homolog_chromosome': 'Naked mole-rat female chromosome/scaffold name' (default: False),\n",
      " 'hgfemale_homolog_dn': 'dN with Naked mole-rat female' (default: False),\n",
      " 'hgfemale_homolog_ds': 'dS with Naked mole-rat female' (default: False),\n",
      " 'hgfemale_homolog_ensembl_gene': 'Naked mole-rat female gene stable ID' (default: False),\n",
      " 'hgfemale_homolog_ensembl_peptide': 'Naked mole-rat female protein or transcript stable ID' (default: False),\n",
      " 'hgfemale_homolog_goc_score': 'Naked mole-rat female Gene-order conservation score' (default: False),\n",
      " 'hgfemale_homolog_orthology_confidence': 'Naked mole-rat female orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hgfemale_homolog_orthology_type': 'Naked mole-rat female homology type' (default: False),\n",
      " 'hgfemale_homolog_perc_id': '%id. target Naked mole-rat female gene identical to query gene' (default: False),\n",
      " 'hgfemale_homolog_perc_id_r1': '%id. query gene identical to target Naked mole-rat female gene' (default: False),\n",
      " 'hgfemale_homolog_subtype': 'Last common ancestor with Naked mole-rat female' (default: False),\n",
      " 'hgfemale_homolog_wga_coverage': 'Naked mole-rat female Whole-genome alignment coverage' (default: False),\n",
      " 'hgmale_homolog_associated_gene_name': 'Naked mole-rat male gene name' (default: False),\n",
      " 'hgmale_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hgmale_homolog_chrom_end': 'Naked mole-rat male chromosome/scaffold end (bp)' (default: False),\n",
      " 'hgmale_homolog_chrom_start': 'Naked mole-rat male chromosome/scaffold start (bp)' (default: False),\n",
      " 'hgmale_homolog_chromosome': 'Naked mole-rat male chromosome/scaffold name' (default: False),\n",
      " 'hgmale_homolog_dn': 'dN with Naked mole-rat male' (default: False),\n",
      " 'hgmale_homolog_ds': 'dS with Naked mole-rat male' (default: False),\n",
      " 'hgmale_homolog_ensembl_gene': 'Naked mole-rat male gene stable ID' (default: False),\n",
      " 'hgmale_homolog_ensembl_peptide': 'Naked mole-rat male protein or transcript stable ID' (default: False),\n",
      " 'hgmale_homolog_goc_score': 'Naked mole-rat male Gene-order conservation score' (default: False),\n",
      " 'hgmale_homolog_orthology_confidence': 'Naked mole-rat male orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hgmale_homolog_orthology_type': 'Naked mole-rat male homology type' (default: False),\n",
      " 'hgmale_homolog_perc_id': '%id. target Naked mole-rat male gene identical to query gene' (default: False),\n",
      " 'hgmale_homolog_perc_id_r1': '%id. query gene identical to target Naked mole-rat male gene' (default: False),\n",
      " 'hgmale_homolog_subtype': 'Last common ancestor with Naked mole-rat male' (default: False),\n",
      " 'hgmale_homolog_wga_coverage': 'Naked mole-rat male Whole-genome alignment coverage' (default: False),\n",
      " 'hgnc_id': 'HGNC ID' (default: False),\n",
      " 'hgnc_symbol': 'HGNC symbol' (default: False),\n",
      " 'hgnc_trans_name': 'Transcript name ID' (default: False),\n",
      " 'hhucho_homolog_associated_gene_name': 'Huchen gene name' (default: False),\n",
      " 'hhucho_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hhucho_homolog_chrom_end': 'Huchen chromosome/scaffold end (bp)' (default: False),\n",
      " 'hhucho_homolog_chrom_start': 'Huchen chromosome/scaffold start (bp)' (default: False),\n",
      " 'hhucho_homolog_chromosome': 'Huchen chromosome/scaffold name' (default: False),\n",
      " 'hhucho_homolog_dn': 'dN with Huchen' (default: False),\n",
      " 'hhucho_homolog_ds': 'dS with Huchen' (default: False),\n",
      " 'hhucho_homolog_ensembl_gene': 'Huchen gene stable ID' (default: False),\n",
      " 'hhucho_homolog_ensembl_peptide': 'Huchen protein or transcript stable ID' (default: False),\n",
      " 'hhucho_homolog_goc_score': 'Huchen Gene-order conservation score' (default: False),\n",
      " 'hhucho_homolog_orthology_confidence': 'Huchen orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hhucho_homolog_orthology_type': 'Huchen homology type' (default: False),\n",
      " 'hhucho_homolog_perc_id': '%id. target Huchen gene identical to query gene' (default: False),\n",
      " 'hhucho_homolog_perc_id_r1': '%id. query gene identical to target Huchen gene' (default: False),\n",
      " 'hhucho_homolog_subtype': 'Last common ancestor with Huchen' (default: False),\n",
      " 'hhucho_homolog_wga_coverage': 'Huchen Whole-genome alignment coverage' (default: False),\n",
      " 'hmmpanther': 'PANTHER ID' (default: False),\n",
      " 'hmmpanther_end': 'PANTHER end' (default: False),\n",
      " 'hmmpanther_start': 'PANTHER start' (default: False),\n",
      " 'hpa_accession': 'Human Protein Atlas accession' (default: False),\n",
      " 'hpa_id': 'Human Protein Atlas ID' (default: False),\n",
      " 'hsapiens_paralog_associated_gene_name': 'Human paralogue associated gene name' (default: False),\n",
      " 'hsapiens_paralog_canonical_transcript_protein': 'Paralogue query protein or transcript ID' (default: False),\n",
      " 'hsapiens_paralog_chrom_end': 'Human paralogue chromosome/scaffold end (bp)' (default: False),\n",
      " 'hsapiens_paralog_chrom_start': 'Human paralogue chromosome/scaffold start (bp)' (default: False),\n",
      " 'hsapiens_paralog_chromosome': 'Human paralogue chromosome/scaffold name' (default: False),\n",
      " 'hsapiens_paralog_dn': 'Paralogue dN with Human' (default: False),\n",
      " 'hsapiens_paralog_ds': 'Paralogue dS with Human' (default: False),\n",
      " 'hsapiens_paralog_ensembl_gene': 'Human paralogue gene stable ID' (default: False),\n",
      " 'hsapiens_paralog_ensembl_peptide': 'Human paralogue protein or transcript ID' (default: False),\n",
      " 'hsapiens_paralog_orthology_type': 'Human paralogue homology type' (default: False),\n",
      " 'hsapiens_paralog_paralogy_confidence': 'Human paralogy confidence [0 low, 1 high]' (default: False),\n",
      " 'hsapiens_paralog_perc_id': 'Paralogue %id. target Human gene identical to query gene' (default: False),\n",
      " 'hsapiens_paralog_perc_id_r1': 'Paralogue %id. query gene identical to target Human gene' (default: False),\n",
      " 'hsapiens_paralog_subtype': 'Paralogue last common ancestor with Human' (default: False),\n",
      " 'illumina_humanht_12_v3': 'ILLUMINA HumanHT 12 V3 probe' (default: False),\n",
      " 'illumina_humanht_12_v4': 'ILLUMINA HumanHT 12 V4 probe' (default: False),\n",
      " 'illumina_humanref_8_v3': 'ILLUMINA HumanRef 8 V3 probe' (default: False),\n",
      " 'illumina_humanwg_6_v1': 'ILLUMINA HumanWG 6 V1 probe' (default: False),\n",
      " 'illumina_humanwg_6_v2': 'ILLUMINA HumanWG 6 V2 probe' (default: False),\n",
      " 'illumina_humanwg_6_v3': 'ILLUMINA HumanWG 6 V3 probe' (default: False),\n",
      " 'interpro': 'Interpro ID' (default: False),\n",
      " 'interpro_description': 'Interpro Description' (default: False),\n",
      " 'interpro_end': 'Interpro end' (default: False),\n",
      " 'interpro_short_description': 'Interpro Short Description' (default: False),\n",
      " 'interpro_start': 'Interpro start' (default: False),\n",
      " 'ipunctatus_homolog_associated_gene_name': 'Channel catfish gene name' (default: False),\n",
      " 'ipunctatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ipunctatus_homolog_chrom_end': 'Channel catfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ipunctatus_homolog_chrom_start': 'Channel catfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ipunctatus_homolog_chromosome': 'Channel catfish chromosome/scaffold name' (default: False),\n",
      " 'ipunctatus_homolog_dn': 'dN with Channel catfish' (default: False),\n",
      " 'ipunctatus_homolog_ds': 'dS with Channel catfish' (default: False),\n",
      " 'ipunctatus_homolog_ensembl_gene': 'Channel catfish gene stable ID' (default: False),\n",
      " 'ipunctatus_homolog_ensembl_peptide': 'Channel catfish protein or transcript stable ID' (default: False),\n",
      " 'ipunctatus_homolog_goc_score': 'Channel catfish Gene-order conservation score' (default: False),\n",
      " 'ipunctatus_homolog_orthology_confidence': 'Channel catfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ipunctatus_homolog_orthology_type': 'Channel catfish homology type' (default: False),\n",
      " 'ipunctatus_homolog_perc_id': '%id. target Channel catfish gene identical to query gene' (default: False),\n",
      " 'ipunctatus_homolog_perc_id_r1': '%id. query gene identical to target Channel catfish gene' (default: False),\n",
      " 'ipunctatus_homolog_subtype': 'Last common ancestor with Channel catfish' (default: False),\n",
      " 'ipunctatus_homolog_wga_coverage': 'Channel catfish Whole-genome alignment coverage' (default: False),\n",
      " 'is_constitutive': 'Constitutive exon' (default: False),\n",
      " 'itridecemlineatus_homolog_associated_gene_name': 'Squirrel gene name' (default: False),\n",
      " 'itridecemlineatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'itridecemlineatus_homolog_chrom_end': 'Squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'itridecemlineatus_homolog_chrom_start': 'Squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'itridecemlineatus_homolog_chromosome': 'Squirrel chromosome/scaffold name' (default: False),\n",
      " 'itridecemlineatus_homolog_dn': 'dN with Squirrel' (default: False),\n",
      " 'itridecemlineatus_homolog_ds': 'dS with Squirrel' (default: False),\n",
      " 'itridecemlineatus_homolog_ensembl_gene': 'Squirrel gene stable ID' (default: False),\n",
      " 'itridecemlineatus_homolog_ensembl_peptide': 'Squirrel protein or transcript stable ID' (default: False),\n",
      " 'itridecemlineatus_homolog_goc_score': 'Squirrel Gene-order conservation score' (default: False),\n",
      " 'itridecemlineatus_homolog_orthology_confidence': 'Squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'itridecemlineatus_homolog_orthology_type': 'Squirrel homology type' (default: False),\n",
      " 'itridecemlineatus_homolog_perc_id': '%id. target Squirrel gene identical to query gene' (default: False),\n",
      " 'itridecemlineatus_homolog_perc_id_r1': '%id. query gene identical to target Squirrel gene' (default: False),\n",
      " 'itridecemlineatus_homolog_subtype': 'Last common ancestor with Squirrel' (default: False),\n",
      " 'itridecemlineatus_homolog_wga_coverage': 'Squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'jhyemalis_homolog_associated_gene_name': 'Dark-eyed junco gene name' (default: False),\n",
      " 'jhyemalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'jhyemalis_homolog_chrom_end': 'Dark-eyed junco chromosome/scaffold end (bp)' (default: False),\n",
      " 'jhyemalis_homolog_chrom_start': 'Dark-eyed junco chromosome/scaffold start (bp)' (default: False),\n",
      " 'jhyemalis_homolog_chromosome': 'Dark-eyed junco chromosome/scaffold name' (default: False),\n",
      " 'jhyemalis_homolog_dn': 'dN with Dark-eyed junco' (default: False),\n",
      " 'jhyemalis_homolog_ds': 'dS with Dark-eyed junco' (default: False),\n",
      " 'jhyemalis_homolog_ensembl_gene': 'Dark-eyed junco gene stable ID' (default: False),\n",
      " 'jhyemalis_homolog_ensembl_peptide': 'Dark-eyed junco protein or transcript stable ID' (default: False),\n",
      " 'jhyemalis_homolog_goc_score': 'Dark-eyed junco Gene-order conservation score' (default: False),\n",
      " 'jhyemalis_homolog_orthology_confidence': 'Dark-eyed junco orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'jhyemalis_homolog_orthology_type': 'Dark-eyed junco homology type' (default: False),\n",
      " 'jhyemalis_homolog_perc_id': '%id. target Dark-eyed junco gene identical to query gene' (default: False),\n",
      " 'jhyemalis_homolog_perc_id_r1': '%id. query gene identical to target Dark-eyed junco gene' (default: False),\n",
      " 'jhyemalis_homolog_subtype': 'Last common ancestor with Dark-eyed junco' (default: False),\n",
      " 'jhyemalis_homolog_wga_coverage': 'Dark-eyed junco Whole-genome alignment coverage' (default: False),\n",
      " 'jjaculus_homolog_associated_gene_name': 'Lesser Egyptian jerboa gene name' (default: False),\n",
      " 'jjaculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'jjaculus_homolog_chrom_end': 'Lesser Egyptian jerboa chromosome/scaffold end (bp)' (default: False),\n",
      " 'jjaculus_homolog_chrom_start': 'Lesser Egyptian jerboa chromosome/scaffold start (bp)' (default: False),\n",
      " 'jjaculus_homolog_chromosome': 'Lesser Egyptian jerboa chromosome/scaffold name' (default: False),\n",
      " 'jjaculus_homolog_dn': 'dN with Lesser Egyptian jerboa' (default: False),\n",
      " 'jjaculus_homolog_ds': 'dS with Lesser Egyptian jerboa' (default: False),\n",
      " 'jjaculus_homolog_ensembl_gene': 'Lesser Egyptian jerboa gene stable ID' (default: False),\n",
      " 'jjaculus_homolog_ensembl_peptide': 'Lesser Egyptian jerboa protein or transcript stable ID' (default: False),\n",
      " 'jjaculus_homolog_goc_score': 'Lesser Egyptian jerboa Gene-order conservation score' (default: False),\n",
      " 'jjaculus_homolog_orthology_confidence': 'Lesser Egyptian jerboa orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'jjaculus_homolog_orthology_type': 'Lesser Egyptian jerboa homology type' (default: False),\n",
      " 'jjaculus_homolog_perc_id': '%id. target Lesser Egyptian jerboa gene identical to query gene' (default: False),\n",
      " 'jjaculus_homolog_perc_id_r1': '%id. query gene identical to target Lesser Egyptian jerboa gene' (default: False),\n",
      " 'jjaculus_homolog_subtype': 'Last common ancestor with Lesser Egyptian jerboa' (default: False),\n",
      " 'jjaculus_homolog_wga_coverage': 'Lesser Egyptian jerboa Whole-genome alignment coverage' (default: False),\n",
      " 'kegg_enzyme': 'KEGG Pathway and Enzyme ID' (default: False),\n",
      " 'kmarmoratus_homolog_associated_gene_name': 'Mangrove rivulus gene name' (default: False),\n",
      " 'kmarmoratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'kmarmoratus_homolog_chrom_end': 'Mangrove rivulus chromosome/scaffold end (bp)' (default: False),\n",
      " 'kmarmoratus_homolog_chrom_start': 'Mangrove rivulus chromosome/scaffold start (bp)' (default: False),\n",
      " 'kmarmoratus_homolog_chromosome': 'Mangrove rivulus chromosome/scaffold name' (default: False),\n",
      " 'kmarmoratus_homolog_dn': 'dN with Mangrove rivulus' (default: False),\n",
      " 'kmarmoratus_homolog_ds': 'dS with Mangrove rivulus' (default: False),\n",
      " 'kmarmoratus_homolog_ensembl_gene': 'Mangrove rivulus gene stable ID' (default: False),\n",
      " 'kmarmoratus_homolog_ensembl_peptide': 'Mangrove rivulus protein or transcript stable ID' (default: False),\n",
      " 'kmarmoratus_homolog_goc_score': 'Mangrove rivulus Gene-order conservation score' (default: False),\n",
      " 'kmarmoratus_homolog_orthology_confidence': 'Mangrove rivulus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'kmarmoratus_homolog_orthology_type': 'Mangrove rivulus homology type' (default: False),\n",
      " 'kmarmoratus_homolog_perc_id': '%id. target Mangrove rivulus gene identical to query gene' (default: False),\n",
      " 'kmarmoratus_homolog_perc_id_r1': '%id. query gene identical to target Mangrove rivulus gene' (default: False),\n",
      " 'kmarmoratus_homolog_subtype': 'Last common ancestor with Mangrove rivulus' (default: False),\n",
      " 'kmarmoratus_homolog_wga_coverage': 'Mangrove rivulus Whole-genome alignment coverage' (default: False),\n",
      " 'lafricana_homolog_associated_gene_name': 'Elephant gene name' (default: False),\n",
      " 'lafricana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lafricana_homolog_chrom_end': 'Elephant chromosome/scaffold end (bp)' (default: False),\n",
      " 'lafricana_homolog_chrom_start': 'Elephant chromosome/scaffold start (bp)' (default: False),\n",
      " 'lafricana_homolog_chromosome': 'Elephant chromosome/scaffold name' (default: False),\n",
      " 'lafricana_homolog_dn': 'dN with Elephant' (default: False),\n",
      " 'lafricana_homolog_ds': 'dS with Elephant' (default: False),\n",
      " 'lafricana_homolog_ensembl_gene': 'Elephant gene stable ID' (default: False),\n",
      " 'lafricana_homolog_ensembl_peptide': 'Elephant protein or transcript stable ID' (default: False),\n",
      " 'lafricana_homolog_goc_score': 'Elephant Gene-order conservation score' (default: False),\n",
      " 'lafricana_homolog_orthology_confidence': 'Elephant orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lafricana_homolog_orthology_type': 'Elephant homology type' (default: False),\n",
      " 'lafricana_homolog_perc_id': '%id. target Elephant gene identical to query gene' (default: False),\n",
      " 'lafricana_homolog_perc_id_r1': '%id. query gene identical to target Elephant gene' (default: False),\n",
      " 'lafricana_homolog_subtype': 'Last common ancestor with Elephant' (default: False),\n",
      " 'lafricana_homolog_wga_coverage': 'Elephant Whole-genome alignment coverage' (default: False),\n",
      " 'lbergylta_homolog_associated_gene_name': 'Ballan wrasse gene name' (default: False),\n",
      " 'lbergylta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lbergylta_homolog_chrom_end': 'Ballan wrasse chromosome/scaffold end (bp)' (default: False),\n",
      " 'lbergylta_homolog_chrom_start': 'Ballan wrasse chromosome/scaffold start (bp)' (default: False),\n",
      " 'lbergylta_homolog_chromosome': 'Ballan wrasse chromosome/scaffold name' (default: False),\n",
      " 'lbergylta_homolog_dn': 'dN with Ballan wrasse' (default: False),\n",
      " 'lbergylta_homolog_ds': 'dS with Ballan wrasse' (default: False),\n",
      " 'lbergylta_homolog_ensembl_gene': 'Ballan wrasse gene stable ID' (default: False),\n",
      " 'lbergylta_homolog_ensembl_peptide': 'Ballan wrasse protein or transcript stable ID' (default: False),\n",
      " 'lbergylta_homolog_goc_score': 'Ballan wrasse Gene-order conservation score' (default: False),\n",
      " 'lbergylta_homolog_orthology_confidence': 'Ballan wrasse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lbergylta_homolog_orthology_type': 'Ballan wrasse homology type' (default: False),\n",
      " 'lbergylta_homolog_perc_id': '%id. target Ballan wrasse gene identical to query gene' (default: False),\n",
      " 'lbergylta_homolog_perc_id_r1': '%id. query gene identical to target Ballan wrasse gene' (default: False),\n",
      " 'lbergylta_homolog_subtype': 'Last common ancestor with Ballan wrasse' (default: False),\n",
      " 'lbergylta_homolog_wga_coverage': 'Ballan wrasse Whole-genome alignment coverage' (default: False),\n",
      " 'lcalcarifer_homolog_associated_gene_name': 'Barramundi perch gene name' (default: False),\n",
      " 'lcalcarifer_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcalcarifer_homolog_chrom_end': 'Barramundi perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcalcarifer_homolog_chrom_start': 'Barramundi perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcalcarifer_homolog_chromosome': 'Barramundi perch chromosome/scaffold name' (default: False),\n",
      " 'lcalcarifer_homolog_dn': 'dN with Barramundi perch' (default: False),\n",
      " 'lcalcarifer_homolog_ds': 'dS with Barramundi perch' (default: False),\n",
      " 'lcalcarifer_homolog_ensembl_gene': 'Barramundi perch gene stable ID' (default: False),\n",
      " 'lcalcarifer_homolog_ensembl_peptide': 'Barramundi perch protein or transcript stable ID' (default: False),\n",
      " 'lcalcarifer_homolog_goc_score': 'Barramundi perch Gene-order conservation score' (default: False),\n",
      " 'lcalcarifer_homolog_orthology_confidence': 'Barramundi perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcalcarifer_homolog_orthology_type': 'Barramundi perch homology type' (default: False),\n",
      " 'lcalcarifer_homolog_perc_id': '%id. target Barramundi perch gene identical to query gene' (default: False),\n",
      " 'lcalcarifer_homolog_perc_id_r1': '%id. query gene identical to target Barramundi perch gene' (default: False),\n",
      " 'lcalcarifer_homolog_subtype': 'Last common ancestor with Barramundi perch' (default: False),\n",
      " 'lcalcarifer_homolog_wga_coverage': 'Barramundi perch Whole-genome alignment coverage' (default: False),\n",
      " 'lchalumnae_homolog_associated_gene_name': 'Coelacanth gene name' (default: False),\n",
      " 'lchalumnae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lchalumnae_homolog_chrom_end': 'Coelacanth chromosome/scaffold end (bp)' (default: False),\n",
      " 'lchalumnae_homolog_chrom_start': 'Coelacanth chromosome/scaffold start (bp)' (default: False),\n",
      " 'lchalumnae_homolog_chromosome': 'Coelacanth chromosome/scaffold name' (default: False),\n",
      " 'lchalumnae_homolog_dn': 'dN with Coelacanth' (default: False),\n",
      " 'lchalumnae_homolog_ds': 'dS with Coelacanth' (default: False),\n",
      " 'lchalumnae_homolog_ensembl_gene': 'Coelacanth gene stable ID' (default: False),\n",
      " 'lchalumnae_homolog_ensembl_peptide': 'Coelacanth protein or transcript stable ID' (default: False),\n",
      " 'lchalumnae_homolog_goc_score': 'Coelacanth Gene-order conservation score' (default: False),\n",
      " 'lchalumnae_homolog_orthology_confidence': 'Coelacanth orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lchalumnae_homolog_orthology_type': 'Coelacanth homology type' (default: False),\n",
      " 'lchalumnae_homolog_perc_id': '%id. target Coelacanth gene identical to query gene' (default: False),\n",
      " 'lchalumnae_homolog_perc_id_r1': '%id. query gene identical to target Coelacanth gene' (default: False),\n",
      " 'lchalumnae_homolog_subtype': 'Last common ancestor with Coelacanth' (default: False),\n",
      " 'lchalumnae_homolog_wga_coverage': 'Coelacanth Whole-genome alignment coverage' (default: False),\n",
      " 'lcoronata_homolog_associated_gene_name': 'Blue-crowned manakin gene name' (default: False),\n",
      " 'lcoronata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcoronata_homolog_chrom_end': 'Blue-crowned manakin chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcoronata_homolog_chrom_start': 'Blue-crowned manakin chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcoronata_homolog_chromosome': 'Blue-crowned manakin chromosome/scaffold name' (default: False),\n",
      " 'lcoronata_homolog_dn': 'dN with Blue-crowned manakin' (default: False),\n",
      " 'lcoronata_homolog_ds': 'dS with Blue-crowned manakin' (default: False),\n",
      " 'lcoronata_homolog_ensembl_gene': 'Blue-crowned manakin gene stable ID' (default: False),\n",
      " 'lcoronata_homolog_ensembl_peptide': 'Blue-crowned manakin protein or transcript stable ID' (default: False),\n",
      " 'lcoronata_homolog_goc_score': 'Blue-crowned manakin Gene-order conservation score' (default: False),\n",
      " 'lcoronata_homolog_orthology_confidence': 'Blue-crowned manakin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcoronata_homolog_orthology_type': 'Blue-crowned manakin homology type' (default: False),\n",
      " 'lcoronata_homolog_perc_id': '%id. target Blue-crowned manakin gene identical to query gene' (default: False),\n",
      " 'lcoronata_homolog_perc_id_r1': '%id. query gene identical to target Blue-crowned manakin gene' (default: False),\n",
      " 'lcoronata_homolog_subtype': 'Last common ancestor with Blue-crowned manakin' (default: False),\n",
      " 'lcoronata_homolog_wga_coverage': 'Blue-crowned manakin Whole-genome alignment coverage' (default: False),\n",
      " 'lcrocea_homolog_associated_gene_name': 'Large yellow croaker gene name' (default: False),\n",
      " 'lcrocea_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcrocea_homolog_chrom_end': 'Large yellow croaker chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcrocea_homolog_chrom_start': 'Large yellow croaker chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcrocea_homolog_chromosome': 'Large yellow croaker chromosome/scaffold name' (default: False),\n",
      " 'lcrocea_homolog_dn': 'dN with Large yellow croaker' (default: False),\n",
      " 'lcrocea_homolog_ds': 'dS with Large yellow croaker' (default: False),\n",
      " 'lcrocea_homolog_ensembl_gene': 'Large yellow croaker gene stable ID' (default: False),\n",
      " 'lcrocea_homolog_ensembl_peptide': 'Large yellow croaker protein or transcript stable ID' (default: False),\n",
      " 'lcrocea_homolog_goc_score': 'Large yellow croaker Gene-order conservation score' (default: False),\n",
      " 'lcrocea_homolog_orthology_confidence': 'Large yellow croaker orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcrocea_homolog_orthology_type': 'Large yellow croaker homology type' (default: False),\n",
      " 'lcrocea_homolog_perc_id': '%id. target Large yellow croaker gene identical to query gene' (default: False),\n",
      " 'lcrocea_homolog_perc_id_r1': '%id. query gene identical to target Large yellow croaker gene' (default: False),\n",
      " 'lcrocea_homolog_subtype': 'Last common ancestor with Large yellow croaker' (default: False),\n",
      " 'lcrocea_homolog_wga_coverage': 'Large yellow croaker Whole-genome alignment coverage' (default: False),\n",
      " 'loculatus_homolog_associated_gene_name': 'Spotted gar gene name' (default: False),\n",
      " 'loculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'loculatus_homolog_chrom_end': 'Spotted gar chromosome/scaffold end (bp)' (default: False),\n",
      " 'loculatus_homolog_chrom_start': 'Spotted gar chromosome/scaffold start (bp)' (default: False),\n",
      " 'loculatus_homolog_chromosome': 'Spotted gar chromosome/scaffold name' (default: False),\n",
      " 'loculatus_homolog_dn': 'dN with Spotted gar' (default: False),\n",
      " 'loculatus_homolog_ds': 'dS with Spotted gar' (default: False),\n",
      " 'loculatus_homolog_ensembl_gene': 'Spotted gar gene stable ID' (default: False),\n",
      " 'loculatus_homolog_ensembl_peptide': 'Spotted gar protein or transcript stable ID' (default: False),\n",
      " 'loculatus_homolog_goc_score': 'Spotted gar Gene-order conservation score' (default: False),\n",
      " 'loculatus_homolog_orthology_confidence': 'Spotted gar orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'loculatus_homolog_orthology_type': 'Spotted gar homology type' (default: False),\n",
      " 'loculatus_homolog_perc_id': '%id. target Spotted gar gene identical to query gene' (default: False),\n",
      " 'loculatus_homolog_perc_id_r1': '%id. query gene identical to target Spotted gar gene' (default: False),\n",
      " 'loculatus_homolog_subtype': 'Last common ancestor with Spotted gar' (default: False),\n",
      " 'loculatus_homolog_wga_coverage': 'Spotted gar Whole-genome alignment coverage' (default: False),\n",
      " 'lsdomestica_homolog_associated_gene_name': 'Bengalese finch gene name' (default: False),\n",
      " 'lsdomestica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lsdomestica_homolog_chrom_end': 'Bengalese finch chromosome/scaffold end (bp)' (default: False),\n",
      " 'lsdomestica_homolog_chrom_start': 'Bengalese finch chromosome/scaffold start (bp)' (default: False),\n",
      " 'lsdomestica_homolog_chromosome': 'Bengalese finch chromosome/scaffold name' (default: False),\n",
      " 'lsdomestica_homolog_dn': 'dN with Bengalese finch' (default: False),\n",
      " 'lsdomestica_homolog_ds': 'dS with Bengalese finch' (default: False),\n",
      " 'lsdomestica_homolog_ensembl_gene': 'Bengalese finch gene stable ID' (default: False),\n",
      " 'lsdomestica_homolog_ensembl_peptide': 'Bengalese finch protein or transcript stable ID' (default: False),\n",
      " 'lsdomestica_homolog_goc_score': 'Bengalese finch Gene-order conservation score' (default: False),\n",
      " 'lsdomestica_homolog_orthology_confidence': 'Bengalese finch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lsdomestica_homolog_orthology_type': 'Bengalese finch homology type' (default: False),\n",
      " 'lsdomestica_homolog_perc_id': '%id. target Bengalese finch gene identical to query gene' (default: False),\n",
      " 'lsdomestica_homolog_perc_id_r1': '%id. query gene identical to target Bengalese finch gene' (default: False),\n",
      " 'lsdomestica_homolog_subtype': 'Last common ancestor with Bengalese finch' (default: False),\n",
      " 'lsdomestica_homolog_wga_coverage': 'Bengalese finch Whole-genome alignment coverage' (default: False),\n",
      " 'malbus_homolog_associated_gene_name': 'Swamp eel gene name' (default: False),\n",
      " 'malbus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'malbus_homolog_chrom_end': 'Swamp eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'malbus_homolog_chrom_start': 'Swamp eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'malbus_homolog_chromosome': 'Swamp eel chromosome/scaffold name' (default: False),\n",
      " 'malbus_homolog_dn': 'dN with Swamp eel' (default: False),\n",
      " 'malbus_homolog_ds': 'dS with Swamp eel' (default: False),\n",
      " 'malbus_homolog_ensembl_gene': 'Swamp eel gene stable ID' (default: False),\n",
      " 'malbus_homolog_ensembl_peptide': 'Swamp eel protein or transcript stable ID' (default: False),\n",
      " 'malbus_homolog_goc_score': 'Swamp eel Gene-order conservation score' (default: False),\n",
      " 'malbus_homolog_orthology_confidence': 'Swamp eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'malbus_homolog_orthology_type': 'Swamp eel homology type' (default: False),\n",
      " 'malbus_homolog_perc_id': '%id. target Swamp eel gene identical to query gene' (default: False),\n",
      " 'malbus_homolog_perc_id_r1': '%id. query gene identical to target Swamp eel gene' (default: False),\n",
      " 'malbus_homolog_subtype': 'Last common ancestor with Swamp eel' (default: False),\n",
      " 'malbus_homolog_wga_coverage': 'Swamp eel Whole-genome alignment coverage' (default: False),\n",
      " 'mapweight': 'Mapweight' (default: False),\n",
      " 'marmatus_homolog_associated_gene_name': 'Zig-zag eel gene name' (default: False),\n",
      " 'marmatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'marmatus_homolog_chrom_end': 'Zig-zag eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'marmatus_homolog_chrom_start': 'Zig-zag eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'marmatus_homolog_chromosome': 'Zig-zag eel chromosome/scaffold name' (default: False),\n",
      " 'marmatus_homolog_dn': 'dN with Zig-zag eel' (default: False),\n",
      " 'marmatus_homolog_ds': 'dS with Zig-zag eel' (default: False),\n",
      " 'marmatus_homolog_ensembl_gene': 'Zig-zag eel gene stable ID' (default: False),\n",
      " 'marmatus_homolog_ensembl_peptide': 'Zig-zag eel protein or transcript stable ID' (default: False),\n",
      " 'marmatus_homolog_goc_score': 'Zig-zag eel Gene-order conservation score' (default: False),\n",
      " 'marmatus_homolog_orthology_confidence': 'Zig-zag eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'marmatus_homolog_orthology_type': 'Zig-zag eel homology type' (default: False),\n",
      " 'marmatus_homolog_perc_id': '%id. target Zig-zag eel gene identical to query gene' (default: False),\n",
      " 'marmatus_homolog_perc_id_r1': '%id. query gene identical to target Zig-zag eel gene' (default: False),\n",
      " 'marmatus_homolog_subtype': 'Last common ancestor with Zig-zag eel' (default: False),\n",
      " 'marmatus_homolog_wga_coverage': 'Zig-zag eel Whole-genome alignment coverage' (default: False),\n",
      " 'mart_transcript_variation_som__dm_allele_string_2076': 'Consequence specific allele' (default: False),\n",
      " 'mart_transcript_variation_som__dm_distance_to_transcript_2076': 'Distance to transcript' (default: False),\n",
      " 'mart_transcript_variation_som__dm_polyphen_prediction_2076': 'PolyPhen prediction' (default: False),\n",
      " 'mart_transcript_variation_som__dm_polyphen_score_2076': 'PolyPhen score' (default: False),\n",
      " 'mart_transcript_variation_som__dm_sift_prediction_2076': 'SIFT prediction' (default: False),\n",
      " 'mart_transcript_variation_som__dm_sift_score_2076': 'SIFT score' (default: False),\n",
      " 'mauratus_homolog_associated_gene_name': 'Golden Hamster gene name' (default: False),\n",
      " 'mauratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mauratus_homolog_chrom_end': 'Golden Hamster chromosome/scaffold end (bp)' (default: False),\n",
      " 'mauratus_homolog_chrom_start': 'Golden Hamster chromosome/scaffold start (bp)' (default: False),\n",
      " 'mauratus_homolog_chromosome': 'Golden Hamster chromosome/scaffold name' (default: False),\n",
      " 'mauratus_homolog_dn': 'dN with Golden Hamster' (default: False),\n",
      " 'mauratus_homolog_ds': 'dS with Golden Hamster' (default: False),\n",
      " 'mauratus_homolog_ensembl_gene': 'Golden Hamster gene stable ID' (default: False),\n",
      " 'mauratus_homolog_ensembl_peptide': 'Golden Hamster protein or transcript stable ID' (default: False),\n",
      " 'mauratus_homolog_goc_score': 'Golden Hamster Gene-order conservation score' (default: False),\n",
      " 'mauratus_homolog_orthology_confidence': 'Golden Hamster orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mauratus_homolog_orthology_type': 'Golden Hamster homology type' (default: False),\n",
      " 'mauratus_homolog_perc_id': '%id. target Golden Hamster gene identical to query gene' (default: False),\n",
      " 'mauratus_homolog_perc_id_r1': '%id. query gene identical to target Golden Hamster gene' (default: False),\n",
      " 'mauratus_homolog_subtype': 'Last common ancestor with Golden Hamster' (default: False),\n",
      " 'mauratus_homolog_wga_coverage': 'Golden Hamster Whole-genome alignment coverage' (default: False),\n",
      " 'mcaroli_homolog_associated_gene_name': 'Ryukyu mouse gene name' (default: False),\n",
      " 'mcaroli_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mcaroli_homolog_chrom_end': 'Ryukyu mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mcaroli_homolog_chrom_start': 'Ryukyu mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mcaroli_homolog_chromosome': 'Ryukyu mouse chromosome/scaffold name' (default: False),\n",
      " 'mcaroli_homolog_dn': 'dN with Ryukyu mouse' (default: False),\n",
      " 'mcaroli_homolog_ds': 'dS with Ryukyu mouse' (default: False),\n",
      " 'mcaroli_homolog_ensembl_gene': 'Ryukyu mouse gene stable ID' (default: False),\n",
      " 'mcaroli_homolog_ensembl_peptide': 'Ryukyu mouse protein or transcript stable ID' (default: False),\n",
      " 'mcaroli_homolog_goc_score': 'Ryukyu mouse Gene-order conservation score' (default: False),\n",
      " 'mcaroli_homolog_orthology_confidence': 'Ryukyu mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mcaroli_homolog_orthology_type': 'Ryukyu mouse homology type' (default: False),\n",
      " 'mcaroli_homolog_perc_id': '%id. target Ryukyu mouse gene identical to query gene' (default: False),\n",
      " 'mcaroli_homolog_perc_id_r1': '%id. query gene identical to target Ryukyu mouse gene' (default: False),\n",
      " 'mcaroli_homolog_subtype': 'Last common ancestor with Ryukyu mouse' (default: False),\n",
      " 'mcaroli_homolog_wga_coverage': 'Ryukyu mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mdomestica_homolog_associated_gene_name': 'Opossum gene name' (default: False),\n",
      " 'mdomestica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mdomestica_homolog_chrom_end': 'Opossum chromosome/scaffold end (bp)' (default: False),\n",
      " 'mdomestica_homolog_chrom_start': 'Opossum chromosome/scaffold start (bp)' (default: False),\n",
      " 'mdomestica_homolog_chromosome': 'Opossum chromosome/scaffold name' (default: False),\n",
      " 'mdomestica_homolog_dn': 'dN with Opossum' (default: False),\n",
      " 'mdomestica_homolog_ds': 'dS with Opossum' (default: False),\n",
      " 'mdomestica_homolog_ensembl_gene': 'Opossum gene stable ID' (default: False),\n",
      " 'mdomestica_homolog_ensembl_peptide': 'Opossum protein or transcript stable ID' (default: False),\n",
      " 'mdomestica_homolog_goc_score': 'Opossum Gene-order conservation score' (default: False),\n",
      " 'mdomestica_homolog_orthology_confidence': 'Opossum orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mdomestica_homolog_orthology_type': 'Opossum homology type' (default: False),\n",
      " 'mdomestica_homolog_perc_id': '%id. target Opossum gene identical to query gene' (default: False),\n",
      " 'mdomestica_homolog_perc_id_r1': '%id. query gene identical to target Opossum gene' (default: False),\n",
      " 'mdomestica_homolog_subtype': 'Last common ancestor with Opossum' (default: False),\n",
      " 'mdomestica_homolog_wga_coverage': 'Opossum Whole-genome alignment coverage' (default: False),\n",
      " 'merops': 'MEROPS - the Peptidase Database ID' (default: False),\n",
      " 'mfascicularis_homolog_associated_gene_name': 'Crab-eating macaque gene name' (default: False),\n",
      " 'mfascicularis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mfascicularis_homolog_chrom_end': 'Crab-eating macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mfascicularis_homolog_chrom_start': 'Crab-eating macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mfascicularis_homolog_chromosome': 'Crab-eating macaque chromosome/scaffold name' (default: False),\n",
      " 'mfascicularis_homolog_dn': 'dN with Crab-eating macaque' (default: False),\n",
      " 'mfascicularis_homolog_ds': 'dS with Crab-eating macaque' (default: False),\n",
      " 'mfascicularis_homolog_ensembl_gene': 'Crab-eating macaque gene stable ID' (default: False),\n",
      " 'mfascicularis_homolog_ensembl_peptide': 'Crab-eating macaque protein or transcript stable ID' (default: False),\n",
      " 'mfascicularis_homolog_goc_score': 'Crab-eating macaque Gene-order conservation score' (default: False),\n",
      " 'mfascicularis_homolog_orthology_confidence': 'Crab-eating macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mfascicularis_homolog_orthology_type': 'Crab-eating macaque homology type' (default: False),\n",
      " 'mfascicularis_homolog_perc_id': '%id. target Crab-eating macaque gene identical to query gene' (default: False),\n",
      " 'mfascicularis_homolog_perc_id_r1': '%id. query gene identical to target Crab-eating macaque gene' (default: False),\n",
      " 'mfascicularis_homolog_subtype': 'Last common ancestor with Crab-eating macaque' (default: False),\n",
      " 'mfascicularis_homolog_wga_coverage': 'Crab-eating macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mgallopavo_homolog_associated_gene_name': 'Turkey gene name' (default: False),\n",
      " 'mgallopavo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mgallopavo_homolog_chrom_end': 'Turkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'mgallopavo_homolog_chrom_start': 'Turkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'mgallopavo_homolog_chromosome': 'Turkey chromosome/scaffold name' (default: False),\n",
      " 'mgallopavo_homolog_dn': 'dN with Turkey' (default: False),\n",
      " 'mgallopavo_homolog_ds': 'dS with Turkey' (default: False),\n",
      " 'mgallopavo_homolog_ensembl_gene': 'Turkey gene stable ID' (default: False),\n",
      " 'mgallopavo_homolog_ensembl_peptide': 'Turkey protein or transcript stable ID' (default: False),\n",
      " 'mgallopavo_homolog_goc_score': 'Turkey Gene-order conservation score' (default: False),\n",
      " 'mgallopavo_homolog_orthology_confidence': 'Turkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mgallopavo_homolog_orthology_type': 'Turkey homology type' (default: False),\n",
      " 'mgallopavo_homolog_perc_id': '%id. target Turkey gene identical to query gene' (default: False),\n",
      " 'mgallopavo_homolog_perc_id_r1': '%id. query gene identical to target Turkey gene' (default: False),\n",
      " 'mgallopavo_homolog_subtype': 'Last common ancestor with Turkey' (default: False),\n",
      " 'mgallopavo_homolog_wga_coverage': 'Turkey Whole-genome alignment coverage' (default: False),\n",
      " 'mim_gene_accession': 'MIM gene accession' (default: False),\n",
      " 'mim_gene_description': 'MIM gene description' (default: False),\n",
      " 'mim_morbid_accession': 'MIM morbid accession' (default: False),\n",
      " 'mim_morbid_description': 'MIM morbid description' (default: False),\n",
      " 'minor_allele': 'Minor allele' (default: False),\n",
      " 'minor_allele_count': 'Minor allele count' (default: False),\n",
      " 'minor_allele_freq': 'Minor allele frequency' (default: False),\n",
      " 'mirbase_accession': 'miRBase accession' (default: False),\n",
      " 'mirbase_id': 'miRBase ID' (default: False),\n",
      " 'mirbase_trans_name': 'miRBase transcript name ID' (default: False),\n",
      " 'mleucophaeus_homolog_associated_gene_name': 'Drill gene name' (default: False),\n",
      " 'mleucophaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mleucophaeus_homolog_chrom_end': 'Drill chromosome/scaffold end (bp)' (default: False),\n",
      " 'mleucophaeus_homolog_chrom_start': 'Drill chromosome/scaffold start (bp)' (default: False),\n",
      " 'mleucophaeus_homolog_chromosome': 'Drill chromosome/scaffold name' (default: False),\n",
      " 'mleucophaeus_homolog_dn': 'dN with Drill' (default: False),\n",
      " 'mleucophaeus_homolog_ds': 'dS with Drill' (default: False),\n",
      " 'mleucophaeus_homolog_ensembl_gene': 'Drill gene stable ID' (default: False),\n",
      " 'mleucophaeus_homolog_ensembl_peptide': 'Drill protein or transcript stable ID' (default: False),\n",
      " 'mleucophaeus_homolog_goc_score': 'Drill Gene-order conservation score' (default: False),\n",
      " 'mleucophaeus_homolog_orthology_confidence': 'Drill orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mleucophaeus_homolog_orthology_type': 'Drill homology type' (default: False),\n",
      " 'mleucophaeus_homolog_perc_id': '%id. target Drill gene identical to query gene' (default: False),\n",
      " 'mleucophaeus_homolog_perc_id_r1': '%id. query gene identical to target Drill gene' (default: False),\n",
      " 'mleucophaeus_homolog_subtype': 'Last common ancestor with Drill' (default: False),\n",
      " 'mleucophaeus_homolog_wga_coverage': 'Drill Whole-genome alignment coverage' (default: False),\n",
      " 'mlucifugus_homolog_associated_gene_name': 'Microbat gene name' (default: False),\n",
      " 'mlucifugus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mlucifugus_homolog_chrom_end': 'Microbat chromosome/scaffold end (bp)' (default: False),\n",
      " 'mlucifugus_homolog_chrom_start': 'Microbat chromosome/scaffold start (bp)' (default: False),\n",
      " 'mlucifugus_homolog_chromosome': 'Microbat chromosome/scaffold name' (default: False),\n",
      " 'mlucifugus_homolog_dn': 'dN with Microbat' (default: False),\n",
      " 'mlucifugus_homolog_ds': 'dS with Microbat' (default: False),\n",
      " 'mlucifugus_homolog_ensembl_gene': 'Microbat gene stable ID' (default: False),\n",
      " 'mlucifugus_homolog_ensembl_peptide': 'Microbat protein or transcript stable ID' (default: False),\n",
      " 'mlucifugus_homolog_goc_score': 'Microbat Gene-order conservation score' (default: False),\n",
      " 'mlucifugus_homolog_orthology_confidence': 'Microbat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mlucifugus_homolog_orthology_type': 'Microbat homology type' (default: False),\n",
      " 'mlucifugus_homolog_perc_id': '%id. target Microbat gene identical to query gene' (default: False),\n",
      " 'mlucifugus_homolog_perc_id_r1': '%id. query gene identical to target Microbat gene' (default: False),\n",
      " 'mlucifugus_homolog_subtype': 'Last common ancestor with Microbat' (default: False),\n",
      " 'mlucifugus_homolog_wga_coverage': 'Microbat Whole-genome alignment coverage' (default: False),\n",
      " 'mmmarmota_homolog_associated_gene_name': 'Alpine marmot gene name' (default: False),\n",
      " 'mmmarmota_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmmarmota_homolog_chrom_end': 'Alpine marmot chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmmarmota_homolog_chrom_start': 'Alpine marmot chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmmarmota_homolog_chromosome': 'Alpine marmot chromosome/scaffold name' (default: False),\n",
      " 'mmmarmota_homolog_dn': 'dN with Alpine marmot' (default: False),\n",
      " 'mmmarmota_homolog_ds': 'dS with Alpine marmot' (default: False),\n",
      " 'mmmarmota_homolog_ensembl_gene': 'Alpine marmot gene stable ID' (default: False),\n",
      " 'mmmarmota_homolog_ensembl_peptide': 'Alpine marmot protein or transcript stable ID' (default: False),\n",
      " 'mmmarmota_homolog_goc_score': 'Alpine marmot Gene-order conservation score' (default: False),\n",
      " 'mmmarmota_homolog_orthology_confidence': 'Alpine marmot orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmmarmota_homolog_orthology_type': 'Alpine marmot homology type' (default: False),\n",
      " 'mmmarmota_homolog_perc_id': '%id. target Alpine marmot gene identical to query gene' (default: False),\n",
      " 'mmmarmota_homolog_perc_id_r1': '%id. query gene identical to target Alpine marmot gene' (default: False),\n",
      " 'mmmarmota_homolog_subtype': 'Last common ancestor with Alpine marmot' (default: False),\n",
      " 'mmmarmota_homolog_wga_coverage': 'Alpine marmot Whole-genome alignment coverage' (default: False),\n",
      " 'mmola_homolog_associated_gene_name': 'Ocean sunfish gene name' (default: False),\n",
      " 'mmola_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmola_homolog_chrom_end': 'Ocean sunfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmola_homolog_chrom_start': 'Ocean sunfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmola_homolog_chromosome': 'Ocean sunfish chromosome/scaffold name' (default: False),\n",
      " 'mmola_homolog_dn': 'dN with Ocean sunfish' (default: False),\n",
      " 'mmola_homolog_ds': 'dS with Ocean sunfish' (default: False),\n",
      " 'mmola_homolog_ensembl_gene': 'Ocean sunfish gene stable ID' (default: False),\n",
      " 'mmola_homolog_ensembl_peptide': 'Ocean sunfish protein or transcript stable ID' (default: False),\n",
      " 'mmola_homolog_goc_score': 'Ocean sunfish Gene-order conservation score' (default: False),\n",
      " 'mmola_homolog_orthology_confidence': 'Ocean sunfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmola_homolog_orthology_type': 'Ocean sunfish homology type' (default: False),\n",
      " 'mmola_homolog_perc_id': '%id. target Ocean sunfish gene identical to query gene' (default: False),\n",
      " 'mmola_homolog_perc_id_r1': '%id. query gene identical to target Ocean sunfish gene' (default: False),\n",
      " 'mmola_homolog_subtype': 'Last common ancestor with Ocean sunfish' (default: False),\n",
      " 'mmola_homolog_wga_coverage': 'Ocean sunfish Whole-genome alignment coverage' (default: False),\n",
      " 'mmulatta_homolog_associated_gene_name': 'Macaque gene name' (default: False),\n",
      " 'mmulatta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmulatta_homolog_chrom_end': 'Macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmulatta_homolog_chrom_start': 'Macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmulatta_homolog_chromosome': 'Macaque chromosome/scaffold name' (default: False),\n",
      " 'mmulatta_homolog_dn': 'dN with Macaque' (default: False),\n",
      " 'mmulatta_homolog_ds': 'dS with Macaque' (default: False),\n",
      " 'mmulatta_homolog_ensembl_gene': 'Macaque gene stable ID' (default: False),\n",
      " 'mmulatta_homolog_ensembl_peptide': 'Macaque protein or transcript stable ID' (default: False),\n",
      " 'mmulatta_homolog_goc_score': 'Macaque Gene-order conservation score' (default: False),\n",
      " 'mmulatta_homolog_orthology_confidence': 'Macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmulatta_homolog_orthology_type': 'Macaque homology type' (default: False),\n",
      " 'mmulatta_homolog_perc_id': '%id. target Macaque gene identical to query gene' (default: False),\n",
      " 'mmulatta_homolog_perc_id_r1': '%id. query gene identical to target Macaque gene' (default: False),\n",
      " 'mmulatta_homolog_subtype': 'Last common ancestor with Macaque' (default: False),\n",
      " 'mmulatta_homolog_wga_coverage': 'Macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mmurinus_homolog_associated_gene_name': 'Mouse Lemur gene name' (default: False),\n",
      " 'mmurinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmurinus_homolog_chrom_end': 'Mouse Lemur chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmurinus_homolog_chrom_start': 'Mouse Lemur chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmurinus_homolog_chromosome': 'Mouse Lemur chromosome/scaffold name' (default: False),\n",
      " 'mmurinus_homolog_dn': 'dN with Mouse Lemur' (default: False),\n",
      " 'mmurinus_homolog_ds': 'dS with Mouse Lemur' (default: False),\n",
      " 'mmurinus_homolog_ensembl_gene': 'Mouse Lemur gene stable ID' (default: False),\n",
      " 'mmurinus_homolog_ensembl_peptide': 'Mouse Lemur protein or transcript stable ID' (default: False),\n",
      " 'mmurinus_homolog_goc_score': 'Mouse Lemur Gene-order conservation score' (default: False),\n",
      " 'mmurinus_homolog_orthology_confidence': 'Mouse Lemur orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmurinus_homolog_orthology_type': 'Mouse Lemur homology type' (default: False),\n",
      " 'mmurinus_homolog_perc_id': '%id. target Mouse Lemur gene identical to query gene' (default: False),\n",
      " 'mmurinus_homolog_perc_id_r1': '%id. query gene identical to target Mouse Lemur gene' (default: False),\n",
      " 'mmurinus_homolog_subtype': 'Last common ancestor with Mouse Lemur' (default: False),\n",
      " 'mmurinus_homolog_wga_coverage': 'Mouse Lemur Whole-genome alignment coverage' (default: False),\n",
      " 'mmusculus_homolog_associated_gene_name': 'Mouse gene name' (default: False),\n",
      " 'mmusculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmusculus_homolog_chrom_end': 'Mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmusculus_homolog_chrom_start': 'Mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmusculus_homolog_chromosome': 'Mouse chromosome/scaffold name' (default: False),\n",
      " 'mmusculus_homolog_dn': 'dN with Mouse' (default: False),\n",
      " 'mmusculus_homolog_ds': 'dS with Mouse' (default: False),\n",
      " 'mmusculus_homolog_ensembl_gene': 'Mouse gene stable ID' (default: False),\n",
      " 'mmusculus_homolog_ensembl_peptide': 'Mouse protein or transcript stable ID' (default: False),\n",
      " 'mmusculus_homolog_goc_score': 'Mouse Gene-order conservation score' (default: False),\n",
      " 'mmusculus_homolog_orthology_confidence': 'Mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmusculus_homolog_orthology_type': 'Mouse homology type' (default: False),\n",
      " 'mmusculus_homolog_perc_id': '%id. target Mouse gene identical to query gene' (default: False),\n",
      " 'mmusculus_homolog_perc_id_r1': '%id. query gene identical to target Mouse gene' (default: False),\n",
      " 'mmusculus_homolog_subtype': 'Last common ancestor with Mouse' (default: False),\n",
      " 'mmusculus_homolog_wga_coverage': 'Mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mnemestrina_homolog_associated_gene_name': 'Pig-tailed macaque gene name' (default: False),\n",
      " 'mnemestrina_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mnemestrina_homolog_chrom_end': 'Pig-tailed macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mnemestrina_homolog_chrom_start': 'Pig-tailed macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mnemestrina_homolog_chromosome': 'Pig-tailed macaque chromosome/scaffold name' (default: False),\n",
      " 'mnemestrina_homolog_dn': 'dN with Pig-tailed macaque' (default: False),\n",
      " 'mnemestrina_homolog_ds': 'dS with Pig-tailed macaque' (default: False),\n",
      " 'mnemestrina_homolog_ensembl_gene': 'Pig-tailed macaque gene stable ID' (default: False),\n",
      " 'mnemestrina_homolog_ensembl_peptide': 'Pig-tailed macaque protein or transcript stable ID' (default: False),\n",
      " 'mnemestrina_homolog_goc_score': 'Pig-tailed macaque Gene-order conservation score' (default: False),\n",
      " 'mnemestrina_homolog_orthology_confidence': 'Pig-tailed macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mnemestrina_homolog_orthology_type': 'Pig-tailed macaque homology type' (default: False),\n",
      " 'mnemestrina_homolog_perc_id': '%id. target Pig-tailed macaque gene identical to query gene' (default: False),\n",
      " 'mnemestrina_homolog_perc_id_r1': '%id. query gene identical to target Pig-tailed macaque gene' (default: False),\n",
      " 'mnemestrina_homolog_subtype': 'Last common ancestor with Pig-tailed macaque' (default: False),\n",
      " 'mnemestrina_homolog_wga_coverage': 'Pig-tailed macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mobidblite': 'MobiDB lite' (default: False),\n",
      " 'mobidblite_end': 'MobiDB lite end' (default: False),\n",
      " 'mobidblite_start': 'MobiDB lite start' (default: False),\n",
      " 'mochrogaster_homolog_associated_gene_name': 'Prairie vole gene name' (default: False),\n",
      " 'mochrogaster_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mochrogaster_homolog_chrom_end': 'Prairie vole chromosome/scaffold end (bp)' (default: False),\n",
      " 'mochrogaster_homolog_chrom_start': 'Prairie vole chromosome/scaffold start (bp)' (default: False),\n",
      " 'mochrogaster_homolog_chromosome': 'Prairie vole chromosome/scaffold name' (default: False),\n",
      " 'mochrogaster_homolog_dn': 'dN with Prairie vole' (default: False),\n",
      " 'mochrogaster_homolog_ds': 'dS with Prairie vole' (default: False),\n",
      " 'mochrogaster_homolog_ensembl_gene': 'Prairie vole gene stable ID' (default: False),\n",
      " 'mochrogaster_homolog_ensembl_peptide': 'Prairie vole protein or transcript stable ID' (default: False),\n",
      " 'mochrogaster_homolog_goc_score': 'Prairie vole Gene-order conservation score' (default: False),\n",
      " 'mochrogaster_homolog_orthology_confidence': 'Prairie vole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mochrogaster_homolog_orthology_type': 'Prairie vole homology type' (default: False),\n",
      " 'mochrogaster_homolog_perc_id': '%id. target Prairie vole gene identical to query gene' (default: False),\n",
      " 'mochrogaster_homolog_perc_id_r1': '%id. query gene identical to target Prairie vole gene' (default: False),\n",
      " 'mochrogaster_homolog_subtype': 'Last common ancestor with Prairie vole' (default: False),\n",
      " 'mochrogaster_homolog_wga_coverage': 'Prairie vole Whole-genome alignment coverage' (default: False),\n",
      " 'mpahari_homolog_associated_gene_name': 'Shrew mouse gene name' (default: False),\n",
      " 'mpahari_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mpahari_homolog_chrom_end': 'Shrew mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mpahari_homolog_chrom_start': 'Shrew mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mpahari_homolog_chromosome': 'Shrew mouse chromosome/scaffold name' (default: False),\n",
      " 'mpahari_homolog_dn': 'dN with Shrew mouse' (default: False),\n",
      " 'mpahari_homolog_ds': 'dS with Shrew mouse' (default: False),\n",
      " 'mpahari_homolog_ensembl_gene': 'Shrew mouse gene stable ID' (default: False),\n",
      " 'mpahari_homolog_ensembl_peptide': 'Shrew mouse protein or transcript stable ID' (default: False),\n",
      " 'mpahari_homolog_goc_score': 'Shrew mouse Gene-order conservation score' (default: False),\n",
      " 'mpahari_homolog_orthology_confidence': 'Shrew mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mpahari_homolog_orthology_type': 'Shrew mouse homology type' (default: False),\n",
      " 'mpahari_homolog_perc_id': '%id. target Shrew mouse gene identical to query gene' (default: False),\n",
      " 'mpahari_homolog_perc_id_r1': '%id. query gene identical to target Shrew mouse gene' (default: False),\n",
      " 'mpahari_homolog_subtype': 'Last common ancestor with Shrew mouse' (default: False),\n",
      " 'mpahari_homolog_wga_coverage': 'Shrew mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mpfuro_homolog_associated_gene_name': 'Ferret gene name' (default: False),\n",
      " 'mpfuro_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mpfuro_homolog_chrom_end': 'Ferret chromosome/scaffold end (bp)' (default: False),\n",
      " 'mpfuro_homolog_chrom_start': 'Ferret chromosome/scaffold start (bp)' (default: False),\n",
      " 'mpfuro_homolog_chromosome': 'Ferret chromosome/scaffold name' (default: False),\n",
      " 'mpfuro_homolog_dn': 'dN with Ferret' (default: False),\n",
      " 'mpfuro_homolog_ds': 'dS with Ferret' (default: False),\n",
      " 'mpfuro_homolog_ensembl_gene': 'Ferret gene stable ID' (default: False),\n",
      " 'mpfuro_homolog_ensembl_peptide': 'Ferret protein or transcript stable ID' (default: False),\n",
      " 'mpfuro_homolog_goc_score': 'Ferret Gene-order conservation score' (default: False),\n",
      " 'mpfuro_homolog_orthology_confidence': 'Ferret orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mpfuro_homolog_orthology_type': 'Ferret homology type' (default: False),\n",
      " 'mpfuro_homolog_perc_id': '%id. target Ferret gene identical to query gene' (default: False),\n",
      " 'mpfuro_homolog_perc_id_r1': '%id. query gene identical to target Ferret gene' (default: False),\n",
      " 'mpfuro_homolog_subtype': 'Last common ancestor with Ferret' (default: False),\n",
      " 'mpfuro_homolog_wga_coverage': 'Ferret Whole-genome alignment coverage' (default: False),\n",
      " 'mspicilegus_homolog_associated_gene_name': 'Steppe mouse gene name' (default: False),\n",
      " 'mspicilegus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mspicilegus_homolog_chrom_end': 'Steppe mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mspicilegus_homolog_chrom_start': 'Steppe mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mspicilegus_homolog_chromosome': 'Steppe mouse chromosome/scaffold name' (default: False),\n",
      " 'mspicilegus_homolog_dn': 'dN with Steppe mouse' (default: False),\n",
      " 'mspicilegus_homolog_ds': 'dS with Steppe mouse' (default: False),\n",
      " 'mspicilegus_homolog_ensembl_gene': 'Steppe mouse gene stable ID' (default: False),\n",
      " 'mspicilegus_homolog_ensembl_peptide': 'Steppe mouse protein or transcript stable ID' (default: False),\n",
      " 'mspicilegus_homolog_goc_score': 'Steppe mouse Gene-order conservation score' (default: False),\n",
      " 'mspicilegus_homolog_orthology_confidence': 'Steppe mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mspicilegus_homolog_orthology_type': 'Steppe mouse homology type' (default: False),\n",
      " 'mspicilegus_homolog_perc_id': '%id. target Steppe mouse gene identical to query gene' (default: False),\n",
      " 'mspicilegus_homolog_perc_id_r1': '%id. query gene identical to target Steppe mouse gene' (default: False),\n",
      " 'mspicilegus_homolog_subtype': 'Last common ancestor with Steppe mouse' (default: False),\n",
      " 'mspicilegus_homolog_wga_coverage': 'Steppe mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mspretus_homolog_associated_gene_name': 'Algerian mouse gene name' (default: False),\n",
      " 'mspretus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mspretus_homolog_chrom_end': 'Algerian mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mspretus_homolog_chrom_start': 'Algerian mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mspretus_homolog_chromosome': 'Algerian mouse chromosome/scaffold name' (default: False),\n",
      " 'mspretus_homolog_dn': 'dN with Algerian mouse' (default: False),\n",
      " 'mspretus_homolog_ds': 'dS with Algerian mouse' (default: False),\n",
      " 'mspretus_homolog_ensembl_gene': 'Algerian mouse gene stable ID' (default: False),\n",
      " 'mspretus_homolog_ensembl_peptide': 'Algerian mouse protein or transcript stable ID' (default: False),\n",
      " 'mspretus_homolog_goc_score': 'Algerian mouse Gene-order conservation score' (default: False),\n",
      " 'mspretus_homolog_orthology_confidence': 'Algerian mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mspretus_homolog_orthology_type': 'Algerian mouse homology type' (default: False),\n",
      " 'mspretus_homolog_perc_id': '%id. target Algerian mouse gene identical to query gene' (default: False),\n",
      " 'mspretus_homolog_perc_id_r1': '%id. query gene identical to target Algerian mouse gene' (default: False),\n",
      " 'mspretus_homolog_subtype': 'Last common ancestor with Algerian mouse' (default: False),\n",
      " 'mspretus_homolog_wga_coverage': 'Algerian mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mundulatus_homolog_associated_gene_name': 'Budgerigar gene name' (default: False),\n",
      " 'mundulatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mundulatus_homolog_chrom_end': 'Budgerigar chromosome/scaffold end (bp)' (default: False),\n",
      " 'mundulatus_homolog_chrom_start': 'Budgerigar chromosome/scaffold start (bp)' (default: False),\n",
      " 'mundulatus_homolog_chromosome': 'Budgerigar chromosome/scaffold name' (default: False),\n",
      " 'mundulatus_homolog_dn': 'dN with Budgerigar' (default: False),\n",
      " 'mundulatus_homolog_ds': 'dS with Budgerigar' (default: False),\n",
      " 'mundulatus_homolog_ensembl_gene': 'Budgerigar gene stable ID' (default: False),\n",
      " 'mundulatus_homolog_ensembl_peptide': 'Budgerigar protein or transcript stable ID' (default: False),\n",
      " 'mundulatus_homolog_goc_score': 'Budgerigar Gene-order conservation score' (default: False),\n",
      " 'mundulatus_homolog_orthology_confidence': 'Budgerigar orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mundulatus_homolog_orthology_type': 'Budgerigar homology type' (default: False),\n",
      " 'mundulatus_homolog_perc_id': '%id. target Budgerigar gene identical to query gene' (default: False),\n",
      " 'mundulatus_homolog_perc_id_r1': '%id. query gene identical to target Budgerigar gene' (default: False),\n",
      " 'mundulatus_homolog_subtype': 'Last common ancestor with Budgerigar' (default: False),\n",
      " 'mundulatus_homolog_wga_coverage': 'Budgerigar Whole-genome alignment coverage' (default: False),\n",
      " 'munguiculatus_homolog_associated_gene_name': 'Mongolian gerbil gene name' (default: False),\n",
      " 'munguiculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'munguiculatus_homolog_chrom_end': 'Mongolian gerbil chromosome/scaffold end (bp)' (default: False),\n",
      " 'munguiculatus_homolog_chrom_start': 'Mongolian gerbil chromosome/scaffold start (bp)' (default: False),\n",
      " 'munguiculatus_homolog_chromosome': 'Mongolian gerbil chromosome/scaffold name' (default: False),\n",
      " 'munguiculatus_homolog_dn': 'dN with Mongolian gerbil' (default: False),\n",
      " 'munguiculatus_homolog_ds': 'dS with Mongolian gerbil' (default: False),\n",
      " 'munguiculatus_homolog_ensembl_gene': 'Mongolian gerbil gene stable ID' (default: False),\n",
      " 'munguiculatus_homolog_ensembl_peptide': 'Mongolian gerbil protein or transcript stable ID' (default: False),\n",
      " 'munguiculatus_homolog_goc_score': 'Mongolian gerbil Gene-order conservation score' (default: False),\n",
      " 'munguiculatus_homolog_orthology_confidence': 'Mongolian gerbil orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'munguiculatus_homolog_orthology_type': 'Mongolian gerbil homology type' (default: False),\n",
      " 'munguiculatus_homolog_perc_id': '%id. target Mongolian gerbil gene identical to query gene' (default: False),\n",
      " 'munguiculatus_homolog_perc_id_r1': '%id. query gene identical to target Mongolian gerbil gene' (default: False),\n",
      " 'munguiculatus_homolog_subtype': 'Last common ancestor with Mongolian gerbil' (default: False),\n",
      " 'munguiculatus_homolog_wga_coverage': 'Mongolian gerbil Whole-genome alignment coverage' (default: False),\n",
      " 'mvitellinus_homolog_associated_gene_name': 'Golden-collared manakin gene name' (default: False),\n",
      " 'mvitellinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mvitellinus_homolog_chrom_end': 'Golden-collared manakin chromosome/scaffold end (bp)' (default: False),\n",
      " 'mvitellinus_homolog_chrom_start': 'Golden-collared manakin chromosome/scaffold start (bp)' (default: False),\n",
      " 'mvitellinus_homolog_chromosome': 'Golden-collared manakin chromosome/scaffold name' (default: False),\n",
      " 'mvitellinus_homolog_dn': 'dN with Golden-collared manakin' (default: False),\n",
      " 'mvitellinus_homolog_ds': 'dS with Golden-collared manakin' (default: False),\n",
      " 'mvitellinus_homolog_ensembl_gene': 'Golden-collared manakin gene stable ID' (default: False),\n",
      " 'mvitellinus_homolog_ensembl_peptide': 'Golden-collared manakin protein or transcript stable ID' (default: False),\n",
      " 'mvitellinus_homolog_goc_score': 'Golden-collared manakin Gene-order conservation score' (default: False),\n",
      " 'mvitellinus_homolog_orthology_confidence': 'Golden-collared manakin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mvitellinus_homolog_orthology_type': 'Golden-collared manakin homology type' (default: False),\n",
      " 'mvitellinus_homolog_perc_id': '%id. target Golden-collared manakin gene identical to query gene' (default: False),\n",
      " 'mvitellinus_homolog_perc_id_r1': '%id. query gene identical to target Golden-collared manakin gene' (default: False),\n",
      " 'mvitellinus_homolog_subtype': 'Last common ancestor with Golden-collared manakin' (default: False),\n",
      " 'mvitellinus_homolog_wga_coverage': 'Golden-collared manakin Whole-genome alignment coverage' (default: False),\n",
      " 'mzebra_homolog_associated_gene_name': 'Zebra mbuna gene name' (default: False),\n",
      " 'mzebra_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mzebra_homolog_chrom_end': 'Zebra mbuna chromosome/scaffold end (bp)' (default: False),\n",
      " 'mzebra_homolog_chrom_start': 'Zebra mbuna chromosome/scaffold start (bp)' (default: False),\n",
      " 'mzebra_homolog_chromosome': 'Zebra mbuna chromosome/scaffold name' (default: False),\n",
      " 'mzebra_homolog_dn': 'dN with Zebra mbuna' (default: False),\n",
      " 'mzebra_homolog_ds': 'dS with Zebra mbuna' (default: False),\n",
      " 'mzebra_homolog_ensembl_gene': 'Zebra mbuna gene stable ID' (default: False),\n",
      " 'mzebra_homolog_ensembl_peptide': 'Zebra mbuna protein or transcript stable ID' (default: False),\n",
      " 'mzebra_homolog_goc_score': 'Zebra mbuna Gene-order conservation score' (default: False),\n",
      " 'mzebra_homolog_orthology_confidence': 'Zebra mbuna orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mzebra_homolog_orthology_type': 'Zebra mbuna homology type' (default: False),\n",
      " 'mzebra_homolog_perc_id': '%id. target Zebra mbuna gene identical to query gene' (default: False),\n",
      " 'mzebra_homolog_perc_id_r1': '%id. query gene identical to target Zebra mbuna gene' (default: False),\n",
      " 'mzebra_homolog_subtype': 'Last common ancestor with Zebra mbuna' (default: False),\n",
      " 'mzebra_homolog_wga_coverage': 'Zebra mbuna Whole-genome alignment coverage' (default: False),\n",
      " 'name_1006': 'GO term name' (default: False),\n",
      " 'namespace_1003': 'GO domain' (default: False),\n",
      " 'nbrichardi_homolog_associated_gene_name': 'Lyretail cichlid gene name' (default: False),\n",
      " 'nbrichardi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nbrichardi_homolog_chrom_end': 'Lyretail cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'nbrichardi_homolog_chrom_start': 'Lyretail cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'nbrichardi_homolog_chromosome': 'Lyretail cichlid chromosome/scaffold name' (default: False),\n",
      " 'nbrichardi_homolog_dn': 'dN with Lyretail cichlid' (default: False),\n",
      " 'nbrichardi_homolog_ds': 'dS with Lyretail cichlid' (default: False),\n",
      " 'nbrichardi_homolog_ensembl_gene': 'Lyretail cichlid gene stable ID' (default: False),\n",
      " 'nbrichardi_homolog_ensembl_peptide': 'Lyretail cichlid protein or transcript stable ID' (default: False),\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 'nbrichardi_homolog_goc_score': 'Lyretail cichlid Gene-order conservation score' (default: False),\n",
      " 'nbrichardi_homolog_orthology_confidence': 'Lyretail cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nbrichardi_homolog_orthology_type': 'Lyretail cichlid homology type' (default: False),\n",
      " 'nbrichardi_homolog_perc_id': '%id. target Lyretail cichlid gene identical to query gene' (default: False),\n",
      " 'nbrichardi_homolog_perc_id_r1': '%id. query gene identical to target Lyretail cichlid gene' (default: False),\n",
      " 'nbrichardi_homolog_subtype': 'Last common ancestor with Lyretail cichlid' (default: False),\n",
      " 'nbrichardi_homolog_wga_coverage': 'Lyretail cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'ncoils': 'Coiled-coils (Ncoils)' (default: False),\n",
      " 'ncoils_end': 'Coiled-coils (Ncoils) end' (default: False),\n",
      " 'ncoils_start': 'Coiled-coils (Ncoils) start' (default: False),\n",
      " 'neugenii_homolog_associated_gene_name': 'Wallaby gene name' (default: False),\n",
      " 'neugenii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'neugenii_homolog_chrom_end': 'Wallaby chromosome/scaffold end (bp)' (default: False),\n",
      " 'neugenii_homolog_chrom_start': 'Wallaby chromosome/scaffold start (bp)' (default: False),\n",
      " 'neugenii_homolog_chromosome': 'Wallaby chromosome/scaffold name' (default: False),\n",
      " 'neugenii_homolog_dn': 'dN with Wallaby' (default: False),\n",
      " 'neugenii_homolog_ds': 'dS with Wallaby' (default: False),\n",
      " 'neugenii_homolog_ensembl_gene': 'Wallaby gene stable ID' (default: False),\n",
      " 'neugenii_homolog_ensembl_peptide': 'Wallaby protein or transcript stable ID' (default: False),\n",
      " 'neugenii_homolog_goc_score': 'Wallaby Gene-order conservation score' (default: False),\n",
      " 'neugenii_homolog_orthology_confidence': 'Wallaby orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'neugenii_homolog_orthology_type': 'Wallaby homology type' (default: False),\n",
      " 'neugenii_homolog_perc_id': '%id. target Wallaby gene identical to query gene' (default: False),\n",
      " 'neugenii_homolog_perc_id_r1': '%id. query gene identical to target Wallaby gene' (default: False),\n",
      " 'neugenii_homolog_subtype': 'Last common ancestor with Wallaby' (default: False),\n",
      " 'neugenii_homolog_wga_coverage': 'Wallaby Whole-genome alignment coverage' (default: False),\n",
      " 'ngalili_homolog_associated_gene_name': 'Upper Galilee mountains blind mole rat gene name' (default: False),\n",
      " 'ngalili_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ngalili_homolog_chrom_end': 'Upper Galilee mountains blind mole rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'ngalili_homolog_chrom_start': 'Upper Galilee mountains blind mole rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'ngalili_homolog_chromosome': 'Upper Galilee mountains blind mole rat chromosome/scaffold name' (default: False),\n",
      " 'ngalili_homolog_dn': 'dN with Upper Galilee mountains blind mole rat' (default: False),\n",
      " 'ngalili_homolog_ds': 'dS with Upper Galilee mountains blind mole rat' (default: False),\n",
      " 'ngalili_homolog_ensembl_gene': 'Upper Galilee mountains blind mole rat gene stable ID' (default: False),\n",
      " 'ngalili_homolog_ensembl_peptide': 'Upper Galilee mountains blind mole rat protein or transcript stable ID' (default: False),\n",
      " 'ngalili_homolog_goc_score': 'Upper Galilee mountains blind mole rat Gene-order conservation score' (default: False),\n",
      " 'ngalili_homolog_orthology_confidence': 'Upper Galilee mountains blind mole rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ngalili_homolog_orthology_type': 'Upper Galilee mountains blind mole rat homology type' (default: False),\n",
      " 'ngalili_homolog_perc_id': '%id. target Upper Galilee mountains blind mole rat gene identical to query gene' (default: False),\n",
      " 'ngalili_homolog_perc_id_r1': '%id. query gene identical to target Upper Galilee mountains blind mole rat gene' (default: False),\n",
      " 'ngalili_homolog_subtype': 'Last common ancestor with Upper Galilee mountains blind mole rat' (default: False),\n",
      " 'ngalili_homolog_wga_coverage': 'Upper Galilee mountains blind mole rat Whole-genome alignment coverage' (default: False),\n",
      " 'nleucogenys_homolog_associated_gene_name': 'Gibbon gene name' (default: False),\n",
      " 'nleucogenys_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nleucogenys_homolog_chrom_end': 'Gibbon chromosome/scaffold end (bp)' (default: False),\n",
      " 'nleucogenys_homolog_chrom_start': 'Gibbon chromosome/scaffold start (bp)' (default: False),\n",
      " 'nleucogenys_homolog_chromosome': 'Gibbon chromosome/scaffold name' (default: False),\n",
      " 'nleucogenys_homolog_dn': 'dN with Gibbon' (default: False),\n",
      " 'nleucogenys_homolog_ds': 'dS with Gibbon' (default: False),\n",
      " 'nleucogenys_homolog_ensembl_gene': 'Gibbon gene stable ID' (default: False),\n",
      " 'nleucogenys_homolog_ensembl_peptide': 'Gibbon protein or transcript stable ID' (default: False),\n",
      " 'nleucogenys_homolog_goc_score': 'Gibbon Gene-order conservation score' (default: False),\n",
      " 'nleucogenys_homolog_orthology_confidence': 'Gibbon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nleucogenys_homolog_orthology_type': 'Gibbon homology type' (default: False),\n",
      " 'nleucogenys_homolog_perc_id': '%id. target Gibbon gene identical to query gene' (default: False),\n",
      " 'nleucogenys_homolog_perc_id_r1': '%id. query gene identical to target Gibbon gene' (default: False),\n",
      " 'nleucogenys_homolog_subtype': 'Last common ancestor with Gibbon' (default: False),\n",
      " 'nleucogenys_homolog_wga_coverage': 'Gibbon Whole-genome alignment coverage' (default: False),\n",
      " 'nmeleagris_homolog_associated_gene_name': 'Helmeted guineafowl gene name' (default: False),\n",
      " 'nmeleagris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nmeleagris_homolog_chrom_end': 'Helmeted guineafowl chromosome/scaffold end (bp)' (default: False),\n",
      " 'nmeleagris_homolog_chrom_start': 'Helmeted guineafowl chromosome/scaffold start (bp)' (default: False),\n",
      " 'nmeleagris_homolog_chromosome': 'Helmeted guineafowl chromosome/scaffold name' (default: False),\n",
      " 'nmeleagris_homolog_dn': 'dN with Helmeted guineafowl' (default: False),\n",
      " 'nmeleagris_homolog_ds': 'dS with Helmeted guineafowl' (default: False),\n",
      " 'nmeleagris_homolog_ensembl_gene': 'Helmeted guineafowl gene stable ID' (default: False),\n",
      " 'nmeleagris_homolog_ensembl_peptide': 'Helmeted guineafowl protein or transcript stable ID' (default: False),\n",
      " 'nmeleagris_homolog_goc_score': 'Helmeted guineafowl Gene-order conservation score' (default: False),\n",
      " 'nmeleagris_homolog_orthology_confidence': 'Helmeted guineafowl orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nmeleagris_homolog_orthology_type': 'Helmeted guineafowl homology type' (default: False),\n",
      " 'nmeleagris_homolog_perc_id': '%id. target Helmeted guineafowl gene identical to query gene' (default: False),\n",
      " 'nmeleagris_homolog_perc_id_r1': '%id. query gene identical to target Helmeted guineafowl gene' (default: False),\n",
      " 'nmeleagris_homolog_subtype': 'Last common ancestor with Helmeted guineafowl' (default: False),\n",
      " 'nmeleagris_homolog_wga_coverage': 'Helmeted guineafowl Whole-genome alignment coverage' (default: False),\n",
      " 'nperdicaria_homolog_associated_gene_name': 'Chilean tinamou gene name' (default: False),\n",
      " 'nperdicaria_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nperdicaria_homolog_chrom_end': 'Chilean tinamou chromosome/scaffold end (bp)' (default: False),\n",
      " 'nperdicaria_homolog_chrom_start': 'Chilean tinamou chromosome/scaffold start (bp)' (default: False),\n",
      " 'nperdicaria_homolog_chromosome': 'Chilean tinamou chromosome/scaffold name' (default: False),\n",
      " 'nperdicaria_homolog_dn': 'dN with Chilean tinamou' (default: False),\n",
      " 'nperdicaria_homolog_ds': 'dS with Chilean tinamou' (default: False),\n",
      " 'nperdicaria_homolog_ensembl_gene': 'Chilean tinamou gene stable ID' (default: False),\n",
      " 'nperdicaria_homolog_ensembl_peptide': 'Chilean tinamou protein or transcript stable ID' (default: False),\n",
      " 'nperdicaria_homolog_goc_score': 'Chilean tinamou Gene-order conservation score' (default: False),\n",
      " 'nperdicaria_homolog_orthology_confidence': 'Chilean tinamou orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nperdicaria_homolog_orthology_type': 'Chilean tinamou homology type' (default: False),\n",
      " 'nperdicaria_homolog_perc_id': '%id. target Chilean tinamou gene identical to query gene' (default: False),\n",
      " 'nperdicaria_homolog_perc_id_r1': '%id. query gene identical to target Chilean tinamou gene' (default: False),\n",
      " 'nperdicaria_homolog_subtype': 'Last common ancestor with Chilean tinamou' (default: False),\n",
      " 'nperdicaria_homolog_wga_coverage': 'Chilean tinamou Whole-genome alignment coverage' (default: False),\n",
      " 'nscutatus_homolog_associated_gene_name': 'Mainland tiger snake gene name' (default: False),\n",
      " 'nscutatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nscutatus_homolog_chrom_end': 'Mainland tiger snake chromosome/scaffold end (bp)' (default: False),\n",
      " 'nscutatus_homolog_chrom_start': 'Mainland tiger snake chromosome/scaffold start (bp)' (default: False),\n",
      " 'nscutatus_homolog_chromosome': 'Mainland tiger snake chromosome/scaffold name' (default: False),\n",
      " 'nscutatus_homolog_dn': 'dN with Mainland tiger snake' (default: False),\n",
      " 'nscutatus_homolog_ds': 'dS with Mainland tiger snake' (default: False),\n",
      " 'nscutatus_homolog_ensembl_gene': 'Mainland tiger snake gene stable ID' (default: False),\n",
      " 'nscutatus_homolog_ensembl_peptide': 'Mainland tiger snake protein or transcript stable ID' (default: False),\n",
      " 'nscutatus_homolog_goc_score': 'Mainland tiger snake Gene-order conservation score' (default: False),\n",
      " 'nscutatus_homolog_orthology_confidence': 'Mainland tiger snake orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nscutatus_homolog_orthology_type': 'Mainland tiger snake homology type' (default: False),\n",
      " 'nscutatus_homolog_perc_id': '%id. target Mainland tiger snake gene identical to query gene' (default: False),\n",
      " 'nscutatus_homolog_perc_id_r1': '%id. query gene identical to target Mainland tiger snake gene' (default: False),\n",
      " 'nscutatus_homolog_subtype': 'Last common ancestor with Mainland tiger snake' (default: False),\n",
      " 'nscutatus_homolog_wga_coverage': 'Mainland tiger snake Whole-genome alignment coverage' (default: False),\n",
      " 'nvison_homolog_associated_gene_name': 'American mink gene name' (default: False),\n",
      " 'nvison_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nvison_homolog_chrom_end': 'American mink chromosome/scaffold end (bp)' (default: False),\n",
      " 'nvison_homolog_chrom_start': 'American mink chromosome/scaffold start (bp)' (default: False),\n",
      " 'nvison_homolog_chromosome': 'American mink chromosome/scaffold name' (default: False),\n",
      " 'nvison_homolog_dn': 'dN with American mink' (default: False),\n",
      " 'nvison_homolog_ds': 'dS with American mink' (default: False),\n",
      " 'nvison_homolog_ensembl_gene': 'American mink gene stable ID' (default: False),\n",
      " 'nvison_homolog_ensembl_peptide': 'American mink protein or transcript stable ID' (default: False),\n",
      " 'nvison_homolog_goc_score': 'American mink Gene-order conservation score' (default: False),\n",
      " 'nvison_homolog_orthology_confidence': 'American mink orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nvison_homolog_orthology_type': 'American mink homology type' (default: False),\n",
      " 'nvison_homolog_perc_id': '%id. target American mink gene identical to query gene' (default: False),\n",
      " 'nvison_homolog_perc_id_r1': '%id. query gene identical to target American mink gene' (default: False),\n",
      " 'nvison_homolog_subtype': 'Last common ancestor with American mink' (default: False),\n",
      " 'nvison_homolog_wga_coverage': 'American mink Whole-genome alignment coverage' (default: False),\n",
      " 'oanatinus_homolog_associated_gene_name': 'Platypus gene name' (default: False),\n",
      " 'oanatinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oanatinus_homolog_chrom_end': 'Platypus chromosome/scaffold end (bp)' (default: False),\n",
      " 'oanatinus_homolog_chrom_start': 'Platypus chromosome/scaffold start (bp)' (default: False),\n",
      " 'oanatinus_homolog_chromosome': 'Platypus chromosome/scaffold name' (default: False),\n",
      " 'oanatinus_homolog_dn': 'dN with Platypus' (default: False),\n",
      " 'oanatinus_homolog_ds': 'dS with Platypus' (default: False),\n",
      " 'oanatinus_homolog_ensembl_gene': 'Platypus gene stable ID' (default: False),\n",
      " 'oanatinus_homolog_ensembl_peptide': 'Platypus protein or transcript stable ID' (default: False),\n",
      " 'oanatinus_homolog_goc_score': 'Platypus Gene-order conservation score' (default: False),\n",
      " 'oanatinus_homolog_orthology_confidence': 'Platypus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oanatinus_homolog_orthology_type': 'Platypus homology type' (default: False),\n",
      " 'oanatinus_homolog_perc_id': '%id. target Platypus gene identical to query gene' (default: False),\n",
      " 'oanatinus_homolog_perc_id_r1': '%id. query gene identical to target Platypus gene' (default: False),\n",
      " 'oanatinus_homolog_subtype': 'Last common ancestor with Platypus' (default: False),\n",
      " 'oanatinus_homolog_wga_coverage': 'Platypus Whole-genome alignment coverage' (default: False),\n",
      " 'oaries_homolog_associated_gene_name': 'Sheep gene name' (default: False),\n",
      " 'oaries_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oaries_homolog_chrom_end': 'Sheep chromosome/scaffold end (bp)' (default: False),\n",
      " 'oaries_homolog_chrom_start': 'Sheep chromosome/scaffold start (bp)' (default: False),\n",
      " 'oaries_homolog_chromosome': 'Sheep chromosome/scaffold name' (default: False),\n",
      " 'oaries_homolog_dn': 'dN with Sheep' (default: False),\n",
      " 'oaries_homolog_ds': 'dS with Sheep' (default: False),\n",
      " 'oaries_homolog_ensembl_gene': 'Sheep gene stable ID' (default: False),\n",
      " 'oaries_homolog_ensembl_peptide': 'Sheep protein or transcript stable ID' (default: False),\n",
      " 'oaries_homolog_goc_score': 'Sheep Gene-order conservation score' (default: False),\n",
      " 'oaries_homolog_orthology_confidence': 'Sheep orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oaries_homolog_orthology_type': 'Sheep homology type' (default: False),\n",
      " 'oaries_homolog_perc_id': '%id. target Sheep gene identical to query gene' (default: False),\n",
      " 'oaries_homolog_perc_id_r1': '%id. query gene identical to target Sheep gene' (default: False),\n",
      " 'oaries_homolog_subtype': 'Last common ancestor with Sheep' (default: False),\n",
      " 'oaries_homolog_wga_coverage': 'Sheep Whole-genome alignment coverage' (default: False),\n",
      " 'ocuniculus_homolog_associated_gene_name': 'Rabbit gene name' (default: False),\n",
      " 'ocuniculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ocuniculus_homolog_chrom_end': 'Rabbit chromosome/scaffold end (bp)' (default: False),\n",
      " 'ocuniculus_homolog_chrom_start': 'Rabbit chromosome/scaffold start (bp)' (default: False),\n",
      " 'ocuniculus_homolog_chromosome': 'Rabbit chromosome/scaffold name' (default: False),\n",
      " 'ocuniculus_homolog_dn': 'dN with Rabbit' (default: False),\n",
      " 'ocuniculus_homolog_ds': 'dS with Rabbit' (default: False),\n",
      " 'ocuniculus_homolog_ensembl_gene': 'Rabbit gene stable ID' (default: False),\n",
      " 'ocuniculus_homolog_ensembl_peptide': 'Rabbit protein or transcript stable ID' (default: False),\n",
      " 'ocuniculus_homolog_goc_score': 'Rabbit Gene-order conservation score' (default: False),\n",
      " 'ocuniculus_homolog_orthology_confidence': 'Rabbit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ocuniculus_homolog_orthology_type': 'Rabbit homology type' (default: False),\n",
      " 'ocuniculus_homolog_perc_id': '%id. target Rabbit gene identical to query gene' (default: False),\n",
      " 'ocuniculus_homolog_perc_id_r1': '%id. query gene identical to target Rabbit gene' (default: False),\n",
      " 'ocuniculus_homolog_subtype': 'Last common ancestor with Rabbit' (default: False),\n",
      " 'ocuniculus_homolog_wga_coverage': 'Rabbit Whole-genome alignment coverage' (default: False),\n",
      " 'odegus_homolog_associated_gene_name': 'Degu gene name' (default: False),\n",
      " 'odegus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'odegus_homolog_chrom_end': 'Degu chromosome/scaffold end (bp)' (default: False),\n",
      " 'odegus_homolog_chrom_start': 'Degu chromosome/scaffold start (bp)' (default: False),\n",
      " 'odegus_homolog_chromosome': 'Degu chromosome/scaffold name' (default: False),\n",
      " 'odegus_homolog_dn': 'dN with Degu' (default: False),\n",
      " 'odegus_homolog_ds': 'dS with Degu' (default: False),\n",
      " 'odegus_homolog_ensembl_gene': 'Degu gene stable ID' (default: False),\n",
      " 'odegus_homolog_ensembl_peptide': 'Degu protein or transcript stable ID' (default: False),\n",
      " 'odegus_homolog_goc_score': 'Degu Gene-order conservation score' (default: False),\n",
      " 'odegus_homolog_orthology_confidence': 'Degu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'odegus_homolog_orthology_type': 'Degu homology type' (default: False),\n",
      " 'odegus_homolog_perc_id': '%id. target Degu gene identical to query gene' (default: False),\n",
      " 'odegus_homolog_perc_id_r1': '%id. query gene identical to target Degu gene' (default: False),\n",
      " 'odegus_homolog_subtype': 'Last common ancestor with Degu' (default: False),\n",
      " 'odegus_homolog_wga_coverage': 'Degu Whole-genome alignment coverage' (default: False),\n",
      " 'ogarnettii_homolog_associated_gene_name': 'Bushbaby gene name' (default: False),\n",
      " 'ogarnettii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ogarnettii_homolog_chrom_end': 'Bushbaby chromosome/scaffold end (bp)' (default: False),\n",
      " 'ogarnettii_homolog_chrom_start': 'Bushbaby chromosome/scaffold start (bp)' (default: False),\n",
      " 'ogarnettii_homolog_chromosome': 'Bushbaby chromosome/scaffold name' (default: False),\n",
      " 'ogarnettii_homolog_dn': 'dN with Bushbaby' (default: False),\n",
      " 'ogarnettii_homolog_ds': 'dS with Bushbaby' (default: False),\n",
      " 'ogarnettii_homolog_ensembl_gene': 'Bushbaby gene stable ID' (default: False),\n",
      " 'ogarnettii_homolog_ensembl_peptide': 'Bushbaby protein or transcript stable ID' (default: False),\n",
      " 'ogarnettii_homolog_goc_score': 'Bushbaby Gene-order conservation score' (default: False),\n",
      " 'ogarnettii_homolog_orthology_confidence': 'Bushbaby orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ogarnettii_homolog_orthology_type': 'Bushbaby homology type' (default: False),\n",
      " 'ogarnettii_homolog_perc_id': '%id. target Bushbaby gene identical to query gene' (default: False),\n",
      " 'ogarnettii_homolog_perc_id_r1': '%id. query gene identical to target Bushbaby gene' (default: False),\n",
      " 'ogarnettii_homolog_subtype': 'Last common ancestor with Bushbaby' (default: False),\n",
      " 'ogarnettii_homolog_wga_coverage': 'Bushbaby Whole-genome alignment coverage' (default: False),\n",
      " 'olatipes_homolog_associated_gene_name': 'Japanese medaka HdrR gene name' (default: False),\n",
      " 'olatipes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'olatipes_homolog_chrom_end': 'Japanese medaka HdrR chromosome/scaffold end (bp)' (default: False),\n",
      " 'olatipes_homolog_chrom_start': 'Japanese medaka HdrR chromosome/scaffold start (bp)' (default: False),\n",
      " 'olatipes_homolog_chromosome': 'Japanese medaka HdrR chromosome/scaffold name' (default: False),\n",
      " 'olatipes_homolog_dn': 'dN with Japanese medaka HdrR' (default: False),\n",
      " 'olatipes_homolog_ds': 'dS with Japanese medaka HdrR' (default: False),\n",
      " 'olatipes_homolog_ensembl_gene': 'Japanese medaka HdrR gene stable ID' (default: False),\n",
      " 'olatipes_homolog_ensembl_peptide': 'Japanese medaka HdrR protein or transcript stable ID' (default: False),\n",
      " 'olatipes_homolog_goc_score': 'Japanese medaka HdrR Gene-order conservation score' (default: False),\n",
      " 'olatipes_homolog_orthology_confidence': 'Japanese medaka HdrR orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'olatipes_homolog_orthology_type': 'Japanese medaka HdrR homology type' (default: False),\n",
      " 'olatipes_homolog_perc_id': '%id. target Japanese medaka HdrR gene identical to query gene' (default: False),\n",
      " 'olatipes_homolog_perc_id_r1': '%id. query gene identical to target Japanese medaka HdrR gene' (default: False),\n",
      " 'olatipes_homolog_subtype': 'Last common ancestor with Japanese medaka HdrR' (default: False),\n",
      " 'olatipes_homolog_wga_coverage': 'Japanese medaka HdrR Whole-genome alignment coverage' (default: False),\n",
      " 'olhni_homolog_associated_gene_name': 'Japanese medaka HNI gene name' (default: False),\n",
      " 'olhni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'olhni_homolog_chrom_end': 'Japanese medaka HNI chromosome/scaffold end (bp)' (default: False),\n",
      " 'olhni_homolog_chrom_start': 'Japanese medaka HNI chromosome/scaffold start (bp)' (default: False),\n",
      " 'olhni_homolog_chromosome': 'Japanese medaka HNI chromosome/scaffold name' (default: False),\n",
      " 'olhni_homolog_dn': 'dN with Japanese medaka HNI' (default: False),\n",
      " 'olhni_homolog_ds': 'dS with Japanese medaka HNI' (default: False),\n",
      " 'olhni_homolog_ensembl_gene': 'Japanese medaka HNI gene stable ID' (default: False),\n",
      " 'olhni_homolog_ensembl_peptide': 'Japanese medaka HNI protein or transcript stable ID' (default: False),\n",
      " 'olhni_homolog_goc_score': 'Japanese medaka HNI Gene-order conservation score' (default: False),\n",
      " 'olhni_homolog_orthology_confidence': 'Japanese medaka HNI orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'olhni_homolog_orthology_type': 'Japanese medaka HNI homology type' (default: False),\n",
      " 'olhni_homolog_perc_id': '%id. target Japanese medaka HNI gene identical to query gene' (default: False),\n",
      " 'olhni_homolog_perc_id_r1': '%id. query gene identical to target Japanese medaka HNI gene' (default: False),\n",
      " 'olhni_homolog_subtype': 'Last common ancestor with Japanese medaka HNI' (default: False),\n",
      " 'olhni_homolog_wga_coverage': 'Japanese medaka HNI Whole-genome alignment coverage' (default: False),\n",
      " 'olhsok_homolog_associated_gene_name': 'Japanese medaka HSOK gene name' (default: False),\n",
      " 'olhsok_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'olhsok_homolog_chrom_end': 'Japanese medaka HSOK chromosome/scaffold end (bp)' (default: False),\n",
      " 'olhsok_homolog_chrom_start': 'Japanese medaka HSOK chromosome/scaffold start (bp)' (default: False),\n",
      " 'olhsok_homolog_chromosome': 'Japanese medaka HSOK chromosome/scaffold name' (default: False),\n",
      " 'olhsok_homolog_dn': 'dN with Japanese medaka HSOK' (default: False),\n",
      " 'olhsok_homolog_ds': 'dS with Japanese medaka HSOK' (default: False),\n",
      " 'olhsok_homolog_ensembl_gene': 'Japanese medaka HSOK gene stable ID' (default: False),\n",
      " 'olhsok_homolog_ensembl_peptide': 'Japanese medaka HSOK protein or transcript stable ID' (default: False),\n",
      " 'olhsok_homolog_goc_score': 'Japanese medaka HSOK Gene-order conservation score' (default: False),\n",
      " 'olhsok_homolog_orthology_confidence': 'Japanese medaka HSOK orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'olhsok_homolog_orthology_type': 'Japanese medaka HSOK homology type' (default: False),\n",
      " 'olhsok_homolog_perc_id': '%id. target Japanese medaka HSOK gene identical to query gene' (default: False),\n",
      " 'olhsok_homolog_perc_id_r1': '%id. query gene identical to target Japanese medaka HSOK gene' (default: False),\n",
      " 'olhsok_homolog_subtype': 'Last common ancestor with Japanese medaka HSOK' (default: False),\n",
      " 'olhsok_homolog_wga_coverage': 'Japanese medaka HSOK Whole-genome alignment coverage' (default: False),\n",
      " 'omelastigma_homolog_associated_gene_name': 'Indian medaka gene name' (default: False),\n",
      " 'omelastigma_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'omelastigma_homolog_chrom_end': 'Indian medaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'omelastigma_homolog_chrom_start': 'Indian medaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'omelastigma_homolog_chromosome': 'Indian medaka chromosome/scaffold name' (default: False),\n",
      " 'omelastigma_homolog_dn': 'dN with Indian medaka' (default: False),\n",
      " 'omelastigma_homolog_ds': 'dS with Indian medaka' (default: False),\n",
      " 'omelastigma_homolog_ensembl_gene': 'Indian medaka gene stable ID' (default: False),\n",
      " 'omelastigma_homolog_ensembl_peptide': 'Indian medaka protein or transcript stable ID' (default: False),\n",
      " 'omelastigma_homolog_goc_score': 'Indian medaka Gene-order conservation score' (default: False),\n",
      " 'omelastigma_homolog_orthology_confidence': 'Indian medaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'omelastigma_homolog_orthology_type': 'Indian medaka homology type' (default: False),\n",
      " 'omelastigma_homolog_perc_id': '%id. target Indian medaka gene identical to query gene' (default: False),\n",
      " 'omelastigma_homolog_perc_id_r1': '%id. query gene identical to target Indian medaka gene' (default: False),\n",
      " 'omelastigma_homolog_subtype': 'Last common ancestor with Indian medaka' (default: False),\n",
      " 'omelastigma_homolog_wga_coverage': 'Indian medaka Whole-genome alignment coverage' (default: False),\n",
      " 'oniloticus_homolog_associated_gene_name': 'Tilapia gene name' (default: False),\n",
      " 'oniloticus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oniloticus_homolog_chrom_end': 'Tilapia chromosome/scaffold end (bp)' (default: False),\n",
      " 'oniloticus_homolog_chrom_start': 'Tilapia chromosome/scaffold start (bp)' (default: False),\n",
      " 'oniloticus_homolog_chromosome': 'Tilapia chromosome/scaffold name' (default: False),\n",
      " 'oniloticus_homolog_dn': 'dN with Tilapia' (default: False),\n",
      " 'oniloticus_homolog_ds': 'dS with Tilapia' (default: False),\n",
      " 'oniloticus_homolog_ensembl_gene': 'Tilapia gene stable ID' (default: False),\n",
      " 'oniloticus_homolog_ensembl_peptide': 'Tilapia protein or transcript stable ID' (default: False),\n",
      " 'oniloticus_homolog_goc_score': 'Tilapia Gene-order conservation score' (default: False),\n",
      " 'oniloticus_homolog_orthology_confidence': 'Tilapia orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oniloticus_homolog_orthology_type': 'Tilapia homology type' (default: False),\n",
      " 'oniloticus_homolog_perc_id': '%id. target Tilapia gene identical to query gene' (default: False),\n",
      " 'oniloticus_homolog_perc_id_r1': '%id. query gene identical to target Tilapia gene' (default: False),\n",
      " 'oniloticus_homolog_subtype': 'Last common ancestor with Tilapia' (default: False),\n",
      " 'oniloticus_homolog_wga_coverage': 'Tilapia Whole-genome alignment coverage' (default: False),\n",
      " 'oprinceps_homolog_associated_gene_name': 'Pika gene name' (default: False),\n",
      " 'oprinceps_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oprinceps_homolog_chrom_end': 'Pika chromosome/scaffold end (bp)' (default: False),\n",
      " 'oprinceps_homolog_chrom_start': 'Pika chromosome/scaffold start (bp)' (default: False),\n",
      " 'oprinceps_homolog_chromosome': 'Pika chromosome/scaffold name' (default: False),\n",
      " 'oprinceps_homolog_dn': 'dN with Pika' (default: False),\n",
      " 'oprinceps_homolog_ds': 'dS with Pika' (default: False),\n",
      " 'oprinceps_homolog_ensembl_gene': 'Pika gene stable ID' (default: False),\n",
      " 'oprinceps_homolog_ensembl_peptide': 'Pika protein or transcript stable ID' (default: False),\n",
      " 'oprinceps_homolog_goc_score': 'Pika Gene-order conservation score' (default: False),\n",
      " 'oprinceps_homolog_orthology_confidence': 'Pika orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oprinceps_homolog_orthology_type': 'Pika homology type' (default: False),\n",
      " 'oprinceps_homolog_perc_id': '%id. target Pika gene identical to query gene' (default: False),\n",
      " 'oprinceps_homolog_perc_id_r1': '%id. query gene identical to target Pika gene' (default: False),\n",
      " 'oprinceps_homolog_subtype': 'Last common ancestor with Pika' (default: False),\n",
      " 'oprinceps_homolog_wga_coverage': 'Pika Whole-genome alignment coverage' (default: False),\n",
      " 'p_value': 'P value' (default: False),\n",
      " 'pabelii_homolog_associated_gene_name': 'Orangutan gene name' (default: False),\n",
      " 'pabelii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pabelii_homolog_chrom_end': 'Orangutan chromosome/scaffold end (bp)' (default: False),\n",
      " 'pabelii_homolog_chrom_start': 'Orangutan chromosome/scaffold start (bp)' (default: False),\n",
      " 'pabelii_homolog_chromosome': 'Orangutan chromosome/scaffold name' (default: False),\n",
      " 'pabelii_homolog_dn': 'dN with Orangutan' (default: False),\n",
      " 'pabelii_homolog_ds': 'dS with Orangutan' (default: False),\n",
      " 'pabelii_homolog_ensembl_gene': 'Orangutan gene stable ID' (default: False),\n",
      " 'pabelii_homolog_ensembl_peptide': 'Orangutan protein or transcript stable ID' (default: False),\n",
      " 'pabelii_homolog_goc_score': 'Orangutan Gene-order conservation score' (default: False),\n",
      " 'pabelii_homolog_orthology_confidence': 'Orangutan orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pabelii_homolog_orthology_type': 'Orangutan homology type' (default: False),\n",
      " 'pabelii_homolog_perc_id': '%id. target Orangutan gene identical to query gene' (default: False),\n",
      " 'pabelii_homolog_perc_id_r1': '%id. query gene identical to target Orangutan gene' (default: False),\n",
      " 'pabelii_homolog_subtype': 'Last common ancestor with Orangutan' (default: False),\n",
      " 'pabelii_homolog_wga_coverage': 'Orangutan Whole-genome alignment coverage' (default: False),\n",
      " 'panubis_homolog_associated_gene_name': 'Olive baboon gene name' (default: False),\n",
      " 'panubis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'panubis_homolog_chrom_end': 'Olive baboon chromosome/scaffold end (bp)' (default: False),\n",
      " 'panubis_homolog_chrom_start': 'Olive baboon chromosome/scaffold start (bp)' (default: False),\n",
      " 'panubis_homolog_chromosome': 'Olive baboon chromosome/scaffold name' (default: False),\n",
      " 'panubis_homolog_dn': 'dN with Olive baboon' (default: False),\n",
      " 'panubis_homolog_ds': 'dS with Olive baboon' (default: False),\n",
      " 'panubis_homolog_ensembl_gene': 'Olive baboon gene stable ID' (default: False),\n",
      " 'panubis_homolog_ensembl_peptide': 'Olive baboon protein or transcript stable ID' (default: False),\n",
      " 'panubis_homolog_goc_score': 'Olive baboon Gene-order conservation score' (default: False),\n",
      " 'panubis_homolog_orthology_confidence': 'Olive baboon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'panubis_homolog_orthology_type': 'Olive baboon homology type' (default: False),\n",
      " 'panubis_homolog_perc_id': '%id. target Olive baboon gene identical to query gene' (default: False),\n",
      " 'panubis_homolog_perc_id_r1': '%id. query gene identical to target Olive baboon gene' (default: False),\n",
      " 'panubis_homolog_subtype': 'Last common ancestor with Olive baboon' (default: False),\n",
      " 'panubis_homolog_wga_coverage': 'Olive baboon Whole-genome alignment coverage' (default: False),\n",
      " 'pcapensis_homolog_associated_gene_name': 'Hyrax gene name' (default: False),\n",
      " 'pcapensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcapensis_homolog_chrom_end': 'Hyrax chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcapensis_homolog_chrom_start': 'Hyrax chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcapensis_homolog_chromosome': 'Hyrax chromosome/scaffold name' (default: False),\n",
      " 'pcapensis_homolog_dn': 'dN with Hyrax' (default: False),\n",
      " 'pcapensis_homolog_ds': 'dS with Hyrax' (default: False),\n",
      " 'pcapensis_homolog_ensembl_gene': 'Hyrax gene stable ID' (default: False),\n",
      " 'pcapensis_homolog_ensembl_peptide': 'Hyrax protein or transcript stable ID' (default: False),\n",
      " 'pcapensis_homolog_goc_score': 'Hyrax Gene-order conservation score' (default: False),\n",
      " 'pcapensis_homolog_orthology_confidence': 'Hyrax orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcapensis_homolog_orthology_type': 'Hyrax homology type' (default: False),\n",
      " 'pcapensis_homolog_perc_id': '%id. target Hyrax gene identical to query gene' (default: False),\n",
      " 'pcapensis_homolog_perc_id_r1': '%id. query gene identical to target Hyrax gene' (default: False),\n",
      " 'pcapensis_homolog_subtype': 'Last common ancestor with Hyrax' (default: False),\n",
      " 'pcapensis_homolog_wga_coverage': 'Hyrax Whole-genome alignment coverage' (default: False),\n",
      " 'pcinereus_homolog_associated_gene_name': 'Koala gene name' (default: False),\n",
      " 'pcinereus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcinereus_homolog_chrom_end': 'Koala chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcinereus_homolog_chrom_start': 'Koala chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcinereus_homolog_chromosome': 'Koala chromosome/scaffold name' (default: False),\n",
      " 'pcinereus_homolog_dn': 'dN with Koala' (default: False),\n",
      " 'pcinereus_homolog_ds': 'dS with Koala' (default: False),\n",
      " 'pcinereus_homolog_ensembl_gene': 'Koala gene stable ID' (default: False),\n",
      " 'pcinereus_homolog_ensembl_peptide': 'Koala protein or transcript stable ID' (default: False),\n",
      " 'pcinereus_homolog_goc_score': 'Koala Gene-order conservation score' (default: False),\n",
      " 'pcinereus_homolog_orthology_confidence': 'Koala orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcinereus_homolog_orthology_type': 'Koala homology type' (default: False),\n",
      " 'pcinereus_homolog_perc_id': '%id. target Koala gene identical to query gene' (default: False),\n",
      " 'pcinereus_homolog_perc_id_r1': '%id. query gene identical to target Koala gene' (default: False),\n",
      " 'pcinereus_homolog_subtype': 'Last common ancestor with Koala' (default: False),\n",
      " 'pcinereus_homolog_wga_coverage': 'Koala Whole-genome alignment coverage' (default: False),\n",
      " 'pcoquereli_homolog_associated_gene_name': 'Coquerel's sifaka gene name' (default: False),\n",
      " 'pcoquereli_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcoquereli_homolog_chrom_end': 'Coquerel's sifaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcoquereli_homolog_chrom_start': 'Coquerel's sifaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcoquereli_homolog_chromosome': 'Coquerel's sifaka chromosome/scaffold name' (default: False),\n",
      " 'pcoquereli_homolog_dn': 'dN with Coquerel's sifaka' (default: False),\n",
      " 'pcoquereli_homolog_ds': 'dS with Coquerel's sifaka' (default: False),\n",
      " 'pcoquereli_homolog_ensembl_gene': 'Coquerel's sifaka gene stable ID' (default: False),\n",
      " 'pcoquereli_homolog_ensembl_peptide': 'Coquerel's sifaka protein or transcript stable ID' (default: False),\n",
      " 'pcoquereli_homolog_goc_score': 'Coquerel's sifaka Gene-order conservation score' (default: False),\n",
      " 'pcoquereli_homolog_orthology_confidence': 'Coquerel's sifaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcoquereli_homolog_orthology_type': 'Coquerel's sifaka homology type' (default: False),\n",
      " 'pcoquereli_homolog_perc_id': '%id. target Coquerel's sifaka gene identical to query gene' (default: False),\n",
      " 'pcoquereli_homolog_perc_id_r1': '%id. query gene identical to target Coquerel's sifaka gene' (default: False),\n",
      " 'pcoquereli_homolog_subtype': 'Last common ancestor with Coquerel's sifaka' (default: False),\n",
      " 'pcoquereli_homolog_wga_coverage': 'Coquerel's sifaka Whole-genome alignment coverage' (default: False),\n",
      " 'pdb': 'PDB ID' (default: False),\n",
      " 'peptide': 'Peptide' (default: False),\n",
      " 'peptide_location': 'Protein location (aa)' (default: False),\n",
      " 'peptide_shift': 'Protein allele' (default: False),\n",
      " 'peptide_version': 'Version (protein)' (default: False),\n",
      " 'percentage_gene_gc_content': 'Gene % GC content' (default: False),\n",
      " 'pfam': 'Pfam ID' (default: False),\n",
      " 'pfam_end': 'Pfam end' (default: False),\n",
      " 'pfam_start': 'Pfam start' (default: False),\n",
      " 'pformosa_homolog_associated_gene_name': 'Amazon molly gene name' (default: False),\n",
      " 'pformosa_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pformosa_homolog_chrom_end': 'Amazon molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'pformosa_homolog_chrom_start': 'Amazon molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'pformosa_homolog_chromosome': 'Amazon molly chromosome/scaffold name' (default: False),\n",
      " 'pformosa_homolog_dn': 'dN with Amazon molly' (default: False),\n",
      " 'pformosa_homolog_ds': 'dS with Amazon molly' (default: False),\n",
      " 'pformosa_homolog_ensembl_gene': 'Amazon molly gene stable ID' (default: False),\n",
      " 'pformosa_homolog_ensembl_peptide': 'Amazon molly protein or transcript stable ID' (default: False),\n",
      " 'pformosa_homolog_goc_score': 'Amazon molly Gene-order conservation score' (default: False),\n",
      " 'pformosa_homolog_orthology_confidence': 'Amazon molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pformosa_homolog_orthology_type': 'Amazon molly homology type' (default: False),\n",
      " 'pformosa_homolog_perc_id': '%id. target Amazon molly gene identical to query gene' (default: False),\n",
      " 'pformosa_homolog_perc_id_r1': '%id. query gene identical to target Amazon molly gene' (default: False),\n",
      " 'pformosa_homolog_subtype': 'Last common ancestor with Amazon molly' (default: False),\n",
      " 'pformosa_homolog_wga_coverage': 'Amazon molly Whole-genome alignment coverage' (default: False),\n",
      " 'pfscan': 'PROSITE profiles ID' (default: False),\n",
      " 'pfscan_end': 'PROSITE profiles end' (default: False),\n",
      " 'pfscan_start': 'PROSITE profiles start' (default: False),\n",
      " 'phalanx_onearray': 'PHALANX OneArray probe' (default: False),\n",
      " 'phase': 'Start phase' (default: False),\n",
      " 'phenotype_description': 'Phenotype description' (default: False),\n",
      " 'pirsf': 'PIRSF ID' (default: False),\n",
      " 'pirsf_end': 'PIRSF end' (default: False),\n",
      " 'pirsf_start': 'PIRSF start' (default: False),\n",
      " 'pkingsleyae_homolog_associated_gene_name': 'Paramormyrops kingsleyae gene name' (default: False),\n",
      " 'pkingsleyae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pkingsleyae_homolog_chrom_end': 'Paramormyrops kingsleyae chromosome/scaffold end (bp)' (default: False),\n",
      " 'pkingsleyae_homolog_chrom_start': 'Paramormyrops kingsleyae chromosome/scaffold start (bp)' (default: False),\n",
      " 'pkingsleyae_homolog_chromosome': 'Paramormyrops kingsleyae chromosome/scaffold name' (default: False),\n",
      " 'pkingsleyae_homolog_dn': 'dN with Paramormyrops kingsleyae' (default: False),\n",
      " 'pkingsleyae_homolog_ds': 'dS with Paramormyrops kingsleyae' (default: False),\n",
      " 'pkingsleyae_homolog_ensembl_gene': 'Paramormyrops kingsleyae gene stable ID' (default: False),\n",
      " 'pkingsleyae_homolog_ensembl_peptide': 'Paramormyrops kingsleyae protein or transcript stable ID' (default: False),\n",
      " 'pkingsleyae_homolog_goc_score': 'Paramormyrops kingsleyae Gene-order conservation score' (default: False),\n",
      " 'pkingsleyae_homolog_orthology_confidence': 'Paramormyrops kingsleyae orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pkingsleyae_homolog_orthology_type': 'Paramormyrops kingsleyae homology type' (default: False),\n",
      " 'pkingsleyae_homolog_perc_id': '%id. target Paramormyrops kingsleyae gene identical to query gene' (default: False),\n",
      " 'pkingsleyae_homolog_perc_id_r1': '%id. query gene identical to target Paramormyrops kingsleyae gene' (default: False),\n",
      " 'pkingsleyae_homolog_subtype': 'Last common ancestor with Paramormyrops kingsleyae' (default: False),\n",
      " 'pkingsleyae_homolog_wga_coverage': 'Paramormyrops kingsleyae Whole-genome alignment coverage' (default: False),\n",
      " 'platipinna_homolog_associated_gene_name': 'Sailfin molly gene name' (default: False),\n",
      " 'platipinna_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'platipinna_homolog_chrom_end': 'Sailfin molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'platipinna_homolog_chrom_start': 'Sailfin molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'platipinna_homolog_chromosome': 'Sailfin molly chromosome/scaffold name' (default: False),\n",
      " 'platipinna_homolog_dn': 'dN with Sailfin molly' (default: False),\n",
      " 'platipinna_homolog_ds': 'dS with Sailfin molly' (default: False),\n",
      " 'platipinna_homolog_ensembl_gene': 'Sailfin molly gene stable ID' (default: False),\n",
      " 'platipinna_homolog_ensembl_peptide': 'Sailfin molly protein or transcript stable ID' (default: False),\n",
      " 'platipinna_homolog_goc_score': 'Sailfin molly Gene-order conservation score' (default: False),\n",
      " 'platipinna_homolog_orthology_confidence': 'Sailfin molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'platipinna_homolog_orthology_type': 'Sailfin molly homology type' (default: False),\n",
      " 'platipinna_homolog_perc_id': '%id. target Sailfin molly gene identical to query gene' (default: False),\n",
      " 'platipinna_homolog_perc_id_r1': '%id. query gene identical to target Sailfin molly gene' (default: False),\n",
      " 'platipinna_homolog_subtype': 'Last common ancestor with Sailfin molly' (default: False),\n",
      " 'platipinna_homolog_wga_coverage': 'Sailfin molly Whole-genome alignment coverage' (default: False),\n",
      " 'pmagnuspinnatus_homolog_associated_gene_name': 'Periophthalmus magnuspinnatus gene name' (default: False),\n",
      " 'pmagnuspinnatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmagnuspinnatus_homolog_chrom_end': 'Periophthalmus magnuspinnatus chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmagnuspinnatus_homolog_chrom_start': 'Periophthalmus magnuspinnatus chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmagnuspinnatus_homolog_chromosome': 'Periophthalmus magnuspinnatus chromosome/scaffold name' (default: False),\n",
      " 'pmagnuspinnatus_homolog_dn': 'dN with Periophthalmus magnuspinnatus' (default: False),\n",
      " 'pmagnuspinnatus_homolog_ds': 'dS with Periophthalmus magnuspinnatus' (default: False),\n",
      " 'pmagnuspinnatus_homolog_ensembl_gene': 'Periophthalmus magnuspinnatus gene stable ID' (default: False),\n",
      " 'pmagnuspinnatus_homolog_ensembl_peptide': 'Periophthalmus magnuspinnatus protein or transcript stable ID' (default: False),\n",
      " 'pmagnuspinnatus_homolog_goc_score': 'Periophthalmus magnuspinnatus Gene-order conservation score' (default: False),\n",
      " 'pmagnuspinnatus_homolog_orthology_confidence': 'Periophthalmus magnuspinnatus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmagnuspinnatus_homolog_orthology_type': 'Periophthalmus magnuspinnatus homology type' (default: False),\n",
      " 'pmagnuspinnatus_homolog_perc_id': '%id. target Periophthalmus magnuspinnatus gene identical to query gene' (default: False),\n",
      " 'pmagnuspinnatus_homolog_perc_id_r1': '%id. query gene identical to target Periophthalmus magnuspinnatus gene' (default: False),\n",
      " 'pmagnuspinnatus_homolog_subtype': 'Last common ancestor with Periophthalmus magnuspinnatus' (default: False),\n",
      " 'pmagnuspinnatus_homolog_wga_coverage': 'Periophthalmus magnuspinnatus Whole-genome alignment coverage' (default: False),\n",
      " 'pmajor_homolog_associated_gene_name': 'Great Tit gene name' (default: False),\n",
      " 'pmajor_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmajor_homolog_chrom_end': 'Great Tit chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmajor_homolog_chrom_start': 'Great Tit chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmajor_homolog_chromosome': 'Great Tit chromosome/scaffold name' (default: False),\n",
      " 'pmajor_homolog_dn': 'dN with Great Tit' (default: False),\n",
      " 'pmajor_homolog_ds': 'dS with Great Tit' (default: False),\n",
      " 'pmajor_homolog_ensembl_gene': 'Great Tit gene stable ID' (default: False),\n",
      " 'pmajor_homolog_ensembl_peptide': 'Great Tit protein or transcript stable ID' (default: False),\n",
      " 'pmajor_homolog_goc_score': 'Great Tit Gene-order conservation score' (default: False),\n",
      " 'pmajor_homolog_orthology_confidence': 'Great Tit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmajor_homolog_orthology_type': 'Great Tit homology type' (default: False),\n",
      " 'pmajor_homolog_perc_id': '%id. target Great Tit gene identical to query gene' (default: False),\n",
      " 'pmajor_homolog_perc_id_r1': '%id. query gene identical to target Great Tit gene' (default: False),\n",
      " 'pmajor_homolog_subtype': 'Last common ancestor with Great Tit' (default: False),\n",
      " 'pmajor_homolog_wga_coverage': 'Great Tit Whole-genome alignment coverage' (default: False),\n",
      " 'pmarinus_homolog_associated_gene_name': 'Lamprey gene name' (default: False),\n",
      " 'pmarinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmarinus_homolog_chrom_end': 'Lamprey chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmarinus_homolog_chrom_start': 'Lamprey chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmarinus_homolog_chromosome': 'Lamprey chromosome/scaffold name' (default: False),\n",
      " 'pmarinus_homolog_dn': 'dN with Lamprey' (default: False),\n",
      " 'pmarinus_homolog_ds': 'dS with Lamprey' (default: False),\n",
      " 'pmarinus_homolog_ensembl_gene': 'Lamprey gene stable ID' (default: False),\n",
      " 'pmarinus_homolog_ensembl_peptide': 'Lamprey protein or transcript stable ID' (default: False),\n",
      " 'pmarinus_homolog_goc_score': 'Lamprey Gene-order conservation score' (default: False),\n",
      " 'pmarinus_homolog_orthology_confidence': 'Lamprey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmarinus_homolog_orthology_type': 'Lamprey homology type' (default: False),\n",
      " 'pmarinus_homolog_perc_id': '%id. target Lamprey gene identical to query gene' (default: False),\n",
      " 'pmarinus_homolog_perc_id_r1': '%id. query gene identical to target Lamprey gene' (default: False),\n",
      " 'pmarinus_homolog_subtype': 'Last common ancestor with Lamprey' (default: False),\n",
      " 'pmarinus_homolog_wga_coverage': 'Lamprey Whole-genome alignment coverage' (default: False),\n",
      " 'pmbairdii_homolog_associated_gene_name': 'Northern American deer mouse gene name' (default: False),\n",
      " 'pmbairdii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmbairdii_homolog_chrom_end': 'Northern American deer mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmbairdii_homolog_chrom_start': 'Northern American deer mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmbairdii_homolog_chromosome': 'Northern American deer mouse chromosome/scaffold name' (default: False),\n",
      " 'pmbairdii_homolog_dn': 'dN with Northern American deer mouse' (default: False),\n",
      " 'pmbairdii_homolog_ds': 'dS with Northern American deer mouse' (default: False),\n",
      " 'pmbairdii_homolog_ensembl_gene': 'Northern American deer mouse gene stable ID' (default: False),\n",
      " 'pmbairdii_homolog_ensembl_peptide': 'Northern American deer mouse protein or transcript stable ID' (default: False),\n",
      " 'pmbairdii_homolog_goc_score': 'Northern American deer mouse Gene-order conservation score' (default: False),\n",
      " 'pmbairdii_homolog_orthology_confidence': 'Northern American deer mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmbairdii_homolog_orthology_type': 'Northern American deer mouse homology type' (default: False),\n",
      " 'pmbairdii_homolog_perc_id': '%id. target Northern American deer mouse gene identical to query gene' (default: False),\n",
      " 'pmbairdii_homolog_perc_id_r1': '%id. query gene identical to target Northern American deer mouse gene' (default: False),\n",
      " 'pmbairdii_homolog_subtype': 'Last common ancestor with Northern American deer mouse' (default: False),\n",
      " 'pmbairdii_homolog_wga_coverage': 'Northern American deer mouse Whole-genome alignment coverage' (default: False),\n",
      " 'pmexicana_homolog_associated_gene_name': 'Shortfin molly gene name' (default: False),\n",
      " 'pmexicana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmexicana_homolog_chrom_end': 'Shortfin molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmexicana_homolog_chrom_start': 'Shortfin molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmexicana_homolog_chromosome': 'Shortfin molly chromosome/scaffold name' (default: False),\n",
      " 'pmexicana_homolog_dn': 'dN with Shortfin molly' (default: False),\n",
      " 'pmexicana_homolog_ds': 'dS with Shortfin molly' (default: False),\n",
      " 'pmexicana_homolog_ensembl_gene': 'Shortfin molly gene stable ID' (default: False),\n",
      " 'pmexicana_homolog_ensembl_peptide': 'Shortfin molly protein or transcript stable ID' (default: False),\n",
      " 'pmexicana_homolog_goc_score': 'Shortfin molly Gene-order conservation score' (default: False),\n",
      " 'pmexicana_homolog_orthology_confidence': 'Shortfin molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmexicana_homolog_orthology_type': 'Shortfin molly homology type' (default: False),\n",
      " 'pmexicana_homolog_perc_id': '%id. target Shortfin molly gene identical to query gene' (default: False),\n",
      " 'pmexicana_homolog_perc_id_r1': '%id. query gene identical to target Shortfin molly gene' (default: False),\n",
      " 'pmexicana_homolog_subtype': 'Last common ancestor with Shortfin molly' (default: False),\n",
      " 'pmexicana_homolog_wga_coverage': 'Shortfin molly Whole-genome alignment coverage' (default: False),\n",
      " 'pnattereri_homolog_associated_gene_name': 'Red-bellied piranha gene name' (default: False),\n",
      " 'pnattereri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pnattereri_homolog_chrom_end': 'Red-bellied piranha chromosome/scaffold end (bp)' (default: False),\n",
      " 'pnattereri_homolog_chrom_start': 'Red-bellied piranha chromosome/scaffold start (bp)' (default: False),\n",
      " 'pnattereri_homolog_chromosome': 'Red-bellied piranha chromosome/scaffold name' (default: False),\n",
      " 'pnattereri_homolog_dn': 'dN with Red-bellied piranha' (default: False),\n",
      " 'pnattereri_homolog_ds': 'dS with Red-bellied piranha' (default: False),\n",
      " 'pnattereri_homolog_ensembl_gene': 'Red-bellied piranha gene stable ID' (default: False),\n",
      " 'pnattereri_homolog_ensembl_peptide': 'Red-bellied piranha protein or transcript stable ID' (default: False),\n",
      " 'pnattereri_homolog_goc_score': 'Red-bellied piranha Gene-order conservation score' (default: False),\n",
      " 'pnattereri_homolog_orthology_confidence': 'Red-bellied piranha orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pnattereri_homolog_orthology_type': 'Red-bellied piranha homology type' (default: False),\n",
      " 'pnattereri_homolog_perc_id': '%id. target Red-bellied piranha gene identical to query gene' (default: False),\n",
      " 'pnattereri_homolog_perc_id_r1': '%id. query gene identical to target Red-bellied piranha gene' (default: False),\n",
      " 'pnattereri_homolog_subtype': 'Last common ancestor with Red-bellied piranha' (default: False),\n",
      " 'pnattereri_homolog_wga_coverage': 'Red-bellied piranha Whole-genome alignment coverage' (default: False),\n",
      " 'pnyererei_homolog_associated_gene_name': 'Makobe Island cichlid gene name' (default: False),\n",
      " 'pnyererei_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pnyererei_homolog_chrom_end': 'Makobe Island cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'pnyererei_homolog_chrom_start': 'Makobe Island cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'pnyererei_homolog_chromosome': 'Makobe Island cichlid chromosome/scaffold name' (default: False),\n",
      " 'pnyererei_homolog_dn': 'dN with Makobe Island cichlid' (default: False),\n",
      " 'pnyererei_homolog_ds': 'dS with Makobe Island cichlid' (default: False),\n",
      " 'pnyererei_homolog_ensembl_gene': 'Makobe Island cichlid gene stable ID' (default: False),\n",
      " 'pnyererei_homolog_ensembl_peptide': 'Makobe Island cichlid protein or transcript stable ID' (default: False),\n",
      " 'pnyererei_homolog_goc_score': 'Makobe Island cichlid Gene-order conservation score' (default: False),\n",
      " 'pnyererei_homolog_orthology_confidence': 'Makobe Island cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pnyererei_homolog_orthology_type': 'Makobe Island cichlid homology type' (default: False),\n",
      " 'pnyererei_homolog_perc_id': '%id. target Makobe Island cichlid gene identical to query gene' (default: False),\n",
      " 'pnyererei_homolog_perc_id_r1': '%id. query gene identical to target Makobe Island cichlid gene' (default: False),\n",
      " 'pnyererei_homolog_subtype': 'Last common ancestor with Makobe Island cichlid' (default: False),\n",
      " 'pnyererei_homolog_wga_coverage': 'Makobe Island cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'polyphen_prediction_2076': 'PolyPhen prediction' (default: False),\n",
      " 'polyphen_score_2076': 'PolyPhen score' (default: False),\n",
      " 'ppaniscus_homolog_associated_gene_name': 'Bonobo gene name' (default: False),\n",
      " 'ppaniscus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ppaniscus_homolog_chrom_end': 'Bonobo chromosome/scaffold end (bp)' (default: False),\n",
      " 'ppaniscus_homolog_chrom_start': 'Bonobo chromosome/scaffold start (bp)' (default: False),\n",
      " 'ppaniscus_homolog_chromosome': 'Bonobo chromosome/scaffold name' (default: False),\n",
      " 'ppaniscus_homolog_dn': 'dN with Bonobo' (default: False),\n",
      " 'ppaniscus_homolog_ds': 'dS with Bonobo' (default: False),\n",
      " 'ppaniscus_homolog_ensembl_gene': 'Bonobo gene stable ID' (default: False),\n",
      " 'ppaniscus_homolog_ensembl_peptide': 'Bonobo protein or transcript stable ID' (default: False),\n",
      " 'ppaniscus_homolog_goc_score': 'Bonobo Gene-order conservation score' (default: False),\n",
      " 'ppaniscus_homolog_orthology_confidence': 'Bonobo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ppaniscus_homolog_orthology_type': 'Bonobo homology type' (default: False),\n",
      " 'ppaniscus_homolog_perc_id': '%id. target Bonobo gene identical to query gene' (default: False),\n",
      " 'ppaniscus_homolog_perc_id_r1': '%id. query gene identical to target Bonobo gene' (default: False),\n",
      " 'ppaniscus_homolog_subtype': 'Last common ancestor with Bonobo' (default: False),\n",
      " 'ppaniscus_homolog_wga_coverage': 'Bonobo Whole-genome alignment coverage' (default: False),\n",
      " 'ppardus_homolog_associated_gene_name': 'Leopard gene name' (default: False),\n",
      " 'ppardus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ppardus_homolog_chrom_end': 'Leopard chromosome/scaffold end (bp)' (default: False),\n",
      " 'ppardus_homolog_chrom_start': 'Leopard chromosome/scaffold start (bp)' (default: False),\n",
      " 'ppardus_homolog_chromosome': 'Leopard chromosome/scaffold name' (default: False),\n",
      " 'ppardus_homolog_dn': 'dN with Leopard' (default: False),\n",
      " 'ppardus_homolog_ds': 'dS with Leopard' (default: False),\n",
      " 'ppardus_homolog_ensembl_gene': 'Leopard gene stable ID' (default: False),\n",
      " 'ppardus_homolog_ensembl_peptide': 'Leopard protein or transcript stable ID' (default: False),\n",
      " 'ppardus_homolog_goc_score': 'Leopard Gene-order conservation score' (default: False),\n",
      " 'ppardus_homolog_orthology_confidence': 'Leopard orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ppardus_homolog_orthology_type': 'Leopard homology type' (default: False),\n",
      " 'ppardus_homolog_perc_id': '%id. target Leopard gene identical to query gene' (default: False),\n",
      " 'ppardus_homolog_perc_id_r1': '%id. query gene identical to target Leopard gene' (default: False),\n",
      " 'ppardus_homolog_subtype': 'Last common ancestor with Leopard' (default: False),\n",
      " 'ppardus_homolog_wga_coverage': 'Leopard Whole-genome alignment coverage' (default: False),\n",
      " 'pranga_homolog_associated_gene_name': 'Indian glassy fish gene name' (default: False),\n",
      " 'pranga_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pranga_homolog_chrom_end': 'Indian glassy fish chromosome/scaffold end (bp)' (default: False),\n",
      " 'pranga_homolog_chrom_start': 'Indian glassy fish chromosome/scaffold start (bp)' (default: False),\n",
      " 'pranga_homolog_chromosome': 'Indian glassy fish chromosome/scaffold name' (default: False),\n",
      " 'pranga_homolog_dn': 'dN with Indian glassy fish' (default: False),\n",
      " 'pranga_homolog_ds': 'dS with Indian glassy fish' (default: False),\n",
      " 'pranga_homolog_ensembl_gene': 'Indian glassy fish gene stable ID' (default: False),\n",
      " 'pranga_homolog_ensembl_peptide': 'Indian glassy fish protein or transcript stable ID' (default: False),\n",
      " 'pranga_homolog_goc_score': 'Indian glassy fish Gene-order conservation score' (default: False),\n",
      " 'pranga_homolog_orthology_confidence': 'Indian glassy fish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pranga_homolog_orthology_type': 'Indian glassy fish homology type' (default: False),\n",
      " 'pranga_homolog_perc_id': '%id. target Indian glassy fish gene identical to query gene' (default: False),\n",
      " 'pranga_homolog_perc_id_r1': '%id. query gene identical to target Indian glassy fish gene' (default: False),\n",
      " 'pranga_homolog_subtype': 'Last common ancestor with Indian glassy fish' (default: False),\n",
      " 'pranga_homolog_wga_coverage': 'Indian glassy fish Whole-genome alignment coverage' (default: False),\n",
      " 'preticulata_homolog_associated_gene_name': 'Guppy gene name' (default: False),\n",
      " 'preticulata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'preticulata_homolog_chrom_end': 'Guppy chromosome/scaffold end (bp)' (default: False),\n",
      " 'preticulata_homolog_chrom_start': 'Guppy chromosome/scaffold start (bp)' (default: False),\n",
      " 'preticulata_homolog_chromosome': 'Guppy chromosome/scaffold name' (default: False),\n",
      " 'preticulata_homolog_dn': 'dN with Guppy' (default: False),\n",
      " 'preticulata_homolog_ds': 'dS with Guppy' (default: False),\n",
      " 'preticulata_homolog_ensembl_gene': 'Guppy gene stable ID' (default: False),\n",
      " 'preticulata_homolog_ensembl_peptide': 'Guppy protein or transcript stable ID' (default: False),\n",
      " 'preticulata_homolog_goc_score': 'Guppy Gene-order conservation score' (default: False),\n",
      " 'preticulata_homolog_orthology_confidence': 'Guppy orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'preticulata_homolog_orthology_type': 'Guppy homology type' (default: False),\n",
      " 'preticulata_homolog_perc_id': '%id. target Guppy gene identical to query gene' (default: False),\n",
      " 'preticulata_homolog_perc_id_r1': '%id. query gene identical to target Guppy gene' (default: False),\n",
      " 'preticulata_homolog_subtype': 'Last common ancestor with Guppy' (default: False),\n",
      " 'preticulata_homolog_wga_coverage': 'Guppy Whole-genome alignment coverage' (default: False),\n",
      " 'prints': 'Prints ID' (default: False),\n",
      " 'prints_end': 'Prints end' (default: False),\n",
      " 'prints_start': 'Prints start' (default: False),\n",
      " 'protein_id': 'INSDC protein ID' (default: False),\n",
      " 'psimus_homolog_associated_gene_name': 'Greater bamboo lemur gene name' (default: False),\n",
      " 'psimus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psimus_homolog_chrom_end': 'Greater bamboo lemur chromosome/scaffold end (bp)' (default: False),\n",
      " 'psimus_homolog_chrom_start': 'Greater bamboo lemur chromosome/scaffold start (bp)' (default: False),\n",
      " 'psimus_homolog_chromosome': 'Greater bamboo lemur chromosome/scaffold name' (default: False),\n",
      " 'psimus_homolog_dn': 'dN with Greater bamboo lemur' (default: False),\n",
      " 'psimus_homolog_ds': 'dS with Greater bamboo lemur' (default: False),\n",
      " 'psimus_homolog_ensembl_gene': 'Greater bamboo lemur gene stable ID' (default: False),\n",
      " 'psimus_homolog_ensembl_peptide': 'Greater bamboo lemur protein or transcript stable ID' (default: False),\n",
      " 'psimus_homolog_goc_score': 'Greater bamboo lemur Gene-order conservation score' (default: False),\n",
      " 'psimus_homolog_orthology_confidence': 'Greater bamboo lemur orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psimus_homolog_orthology_type': 'Greater bamboo lemur homology type' (default: False),\n",
      " 'psimus_homolog_perc_id': '%id. target Greater bamboo lemur gene identical to query gene' (default: False),\n",
      " 'psimus_homolog_perc_id_r1': '%id. query gene identical to target Greater bamboo lemur gene' (default: False),\n",
      " 'psimus_homolog_subtype': 'Last common ancestor with Greater bamboo lemur' (default: False),\n",
      " 'psimus_homolog_wga_coverage': 'Greater bamboo lemur Whole-genome alignment coverage' (default: False),\n",
      " 'psinensis_homolog_associated_gene_name': 'Chinese softshell turtle gene name' (default: False),\n",
      " 'psinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psinensis_homolog_chrom_end': 'Chinese softshell turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'psinensis_homolog_chrom_start': 'Chinese softshell turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'psinensis_homolog_chromosome': 'Chinese softshell turtle chromosome/scaffold name' (default: False),\n",
      " 'psinensis_homolog_dn': 'dN with Chinese softshell turtle' (default: False),\n",
      " 'psinensis_homolog_ds': 'dS with Chinese softshell turtle' (default: False),\n",
      " 'psinensis_homolog_ensembl_gene': 'Chinese softshell turtle gene stable ID' (default: False),\n",
      " 'psinensis_homolog_ensembl_peptide': 'Chinese softshell turtle protein or transcript stable ID' (default: False),\n",
      " 'psinensis_homolog_goc_score': 'Chinese softshell turtle Gene-order conservation score' (default: False),\n",
      " 'psinensis_homolog_orthology_confidence': 'Chinese softshell turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psinensis_homolog_orthology_type': 'Chinese softshell turtle homology type' (default: False),\n",
      " 'psinensis_homolog_perc_id': '%id. target Chinese softshell turtle gene identical to query gene' (default: False),\n",
      " 'psinensis_homolog_perc_id_r1': '%id. query gene identical to target Chinese softshell turtle gene' (default: False),\n",
      " 'psinensis_homolog_subtype': 'Last common ancestor with Chinese softshell turtle' (default: False),\n",
      " 'psinensis_homolog_wga_coverage': 'Chinese softshell turtle Whole-genome alignment coverage' (default: False),\n",
      " 'ptaltaica_homolog_associated_gene_name': 'Tiger gene name' (default: False),\n",
      " 'ptaltaica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptaltaica_homolog_chrom_end': 'Tiger chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptaltaica_homolog_chrom_start': 'Tiger chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptaltaica_homolog_chromosome': 'Tiger chromosome/scaffold name' (default: False),\n",
      " 'ptaltaica_homolog_dn': 'dN with Tiger' (default: False),\n",
      " 'ptaltaica_homolog_ds': 'dS with Tiger' (default: False),\n",
      " 'ptaltaica_homolog_ensembl_gene': 'Tiger gene stable ID' (default: False),\n",
      " 'ptaltaica_homolog_ensembl_peptide': 'Tiger protein or transcript stable ID' (default: False),\n",
      " 'ptaltaica_homolog_goc_score': 'Tiger Gene-order conservation score' (default: False),\n",
      " 'ptaltaica_homolog_orthology_confidence': 'Tiger orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptaltaica_homolog_orthology_type': 'Tiger homology type' (default: False),\n",
      " 'ptaltaica_homolog_perc_id': '%id. target Tiger gene identical to query gene' (default: False),\n",
      " 'ptaltaica_homolog_perc_id_r1': '%id. query gene identical to target Tiger gene' (default: False),\n",
      " 'ptaltaica_homolog_subtype': 'Last common ancestor with Tiger' (default: False),\n",
      " 'ptaltaica_homolog_wga_coverage': 'Tiger Whole-genome alignment coverage' (default: False),\n",
      " 'ptephrosceles_homolog_associated_gene_name': 'Ugandan red Colobus gene name' (default: False),\n",
      " 'ptephrosceles_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptephrosceles_homolog_chrom_end': 'Ugandan red Colobus chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptephrosceles_homolog_chrom_start': 'Ugandan red Colobus chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptephrosceles_homolog_chromosome': 'Ugandan red Colobus chromosome/scaffold name' (default: False),\n",
      " 'ptephrosceles_homolog_dn': 'dN with Ugandan red Colobus' (default: False),\n",
      " 'ptephrosceles_homolog_ds': 'dS with Ugandan red Colobus' (default: False),\n",
      " 'ptephrosceles_homolog_ensembl_gene': 'Ugandan red Colobus gene stable ID' (default: False),\n",
      " 'ptephrosceles_homolog_ensembl_peptide': 'Ugandan red Colobus protein or transcript stable ID' (default: False),\n",
      " 'ptephrosceles_homolog_goc_score': 'Ugandan red Colobus Gene-order conservation score' (default: False),\n",
      " 'ptephrosceles_homolog_orthology_confidence': 'Ugandan red Colobus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptephrosceles_homolog_orthology_type': 'Ugandan red Colobus homology type' (default: False),\n",
      " 'ptephrosceles_homolog_perc_id': '%id. target Ugandan red Colobus gene identical to query gene' (default: False),\n",
      " 'ptephrosceles_homolog_perc_id_r1': '%id. query gene identical to target Ugandan red Colobus gene' (default: False),\n",
      " 'ptephrosceles_homolog_subtype': 'Last common ancestor with Ugandan red Colobus' (default: False),\n",
      " 'ptephrosceles_homolog_wga_coverage': 'Ugandan red Colobus Whole-genome alignment coverage' (default: False),\n",
      " 'ptroglodytes_homolog_associated_gene_name': 'Chimpanzee gene name' (default: False),\n",
      " 'ptroglodytes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptroglodytes_homolog_chrom_end': 'Chimpanzee chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptroglodytes_homolog_chrom_start': 'Chimpanzee chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptroglodytes_homolog_chromosome': 'Chimpanzee chromosome/scaffold name' (default: False),\n",
      " 'ptroglodytes_homolog_dn': 'dN with Chimpanzee' (default: False),\n",
      " 'ptroglodytes_homolog_ds': 'dS with Chimpanzee' (default: False),\n",
      " 'ptroglodytes_homolog_ensembl_gene': 'Chimpanzee gene stable ID' (default: False),\n",
      " 'ptroglodytes_homolog_ensembl_peptide': 'Chimpanzee protein or transcript stable ID' (default: False),\n",
      " 'ptroglodytes_homolog_goc_score': 'Chimpanzee Gene-order conservation score' (default: False),\n",
      " 'ptroglodytes_homolog_orthology_confidence': 'Chimpanzee orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptroglodytes_homolog_orthology_type': 'Chimpanzee homology type' (default: False),\n",
      " 'ptroglodytes_homolog_perc_id': '%id. target Chimpanzee gene identical to query gene' (default: False),\n",
      " 'ptroglodytes_homolog_perc_id_r1': '%id. query gene identical to target Chimpanzee gene' (default: False),\n",
      " 'ptroglodytes_homolog_subtype': 'Last common ancestor with Chimpanzee' (default: False),\n",
      " 'ptroglodytes_homolog_wga_coverage': 'Chimpanzee Whole-genome alignment coverage' (default: False),\n",
      " 'pvampyrus_homolog_associated_gene_name': 'Megabat gene name' (default: False),\n",
      " 'pvampyrus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pvampyrus_homolog_chrom_end': 'Megabat chromosome/scaffold end (bp)' (default: False),\n",
      " 'pvampyrus_homolog_chrom_start': 'Megabat chromosome/scaffold start (bp)' (default: False),\n",
      " 'pvampyrus_homolog_chromosome': 'Megabat chromosome/scaffold name' (default: False),\n",
      " 'pvampyrus_homolog_dn': 'dN with Megabat' (default: False),\n",
      " 'pvampyrus_homolog_ds': 'dS with Megabat' (default: False),\n",
      " 'pvampyrus_homolog_ensembl_gene': 'Megabat gene stable ID' (default: False),\n",
      " 'pvampyrus_homolog_ensembl_peptide': 'Megabat protein or transcript stable ID' (default: False),\n",
      " 'pvampyrus_homolog_goc_score': 'Megabat Gene-order conservation score' (default: False),\n",
      " 'pvampyrus_homolog_orthology_confidence': 'Megabat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pvampyrus_homolog_orthology_type': 'Megabat homology type' (default: False),\n",
      " 'pvampyrus_homolog_perc_id': '%id. target Megabat gene identical to query gene' (default: False),\n",
      " 'pvampyrus_homolog_perc_id_r1': '%id. query gene identical to target Megabat gene' (default: False),\n",
      " 'pvampyrus_homolog_subtype': 'Last common ancestor with Megabat' (default: False),\n",
      " 'pvampyrus_homolog_wga_coverage': 'Megabat Whole-genome alignment coverage' (default: False),\n",
      " 'pvitticeps_homolog_associated_gene_name': 'Central bearded dragon gene name' (default: False),\n",
      " 'pvitticeps_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pvitticeps_homolog_chrom_end': 'Central bearded dragon chromosome/scaffold end (bp)' (default: False),\n",
      " 'pvitticeps_homolog_chrom_start': 'Central bearded dragon chromosome/scaffold start (bp)' (default: False),\n",
      " 'pvitticeps_homolog_chromosome': 'Central bearded dragon chromosome/scaffold name' (default: False),\n",
      " 'pvitticeps_homolog_dn': 'dN with Central bearded dragon' (default: False),\n",
      " 'pvitticeps_homolog_ds': 'dS with Central bearded dragon' (default: False),\n",
      " 'pvitticeps_homolog_ensembl_gene': 'Central bearded dragon gene stable ID' (default: False),\n",
      " 'pvitticeps_homolog_ensembl_peptide': 'Central bearded dragon protein or transcript stable ID' (default: False),\n",
      " 'pvitticeps_homolog_goc_score': 'Central bearded dragon Gene-order conservation score' (default: False),\n",
      " 'pvitticeps_homolog_orthology_confidence': 'Central bearded dragon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pvitticeps_homolog_orthology_type': 'Central bearded dragon homology type' (default: False),\n",
      " 'pvitticeps_homolog_perc_id': '%id. target Central bearded dragon gene identical to query gene' (default: False),\n",
      " 'pvitticeps_homolog_perc_id_r1': '%id. query gene identical to target Central bearded dragon gene' (default: False),\n",
      " 'pvitticeps_homolog_subtype': 'Last common ancestor with Central bearded dragon' (default: False),\n",
      " 'pvitticeps_homolog_wga_coverage': 'Central bearded dragon Whole-genome alignment coverage' (default: False),\n",
      " 'rank': 'Exon rank in transcript' (default: False),\n",
      " 'rbieti_homolog_associated_gene_name': 'Black snub-nosed monkey gene name' (default: False),\n",
      " 'rbieti_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rbieti_homolog_chrom_end': 'Black snub-nosed monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'rbieti_homolog_chrom_start': 'Black snub-nosed monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'rbieti_homolog_chromosome': 'Black snub-nosed monkey chromosome/scaffold name' (default: False),\n",
      " 'rbieti_homolog_dn': 'dN with Black snub-nosed monkey' (default: False),\n",
      " 'rbieti_homolog_ds': 'dS with Black snub-nosed monkey' (default: False),\n",
      " 'rbieti_homolog_ensembl_gene': 'Black snub-nosed monkey gene stable ID' (default: False),\n",
      " 'rbieti_homolog_ensembl_peptide': 'Black snub-nosed monkey protein or transcript stable ID' (default: False),\n",
      " 'rbieti_homolog_goc_score': 'Black snub-nosed monkey Gene-order conservation score' (default: False),\n",
      " 'rbieti_homolog_orthology_confidence': 'Black snub-nosed monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rbieti_homolog_orthology_type': 'Black snub-nosed monkey homology type' (default: False),\n",
      " 'rbieti_homolog_perc_id': '%id. target Black snub-nosed monkey gene identical to query gene' (default: False),\n",
      " 'rbieti_homolog_perc_id_r1': '%id. query gene identical to target Black snub-nosed monkey gene' (default: False),\n",
      " 'rbieti_homolog_subtype': 'Last common ancestor with Black snub-nosed monkey' (default: False),\n",
      " 'rbieti_homolog_wga_coverage': 'Black snub-nosed monkey Whole-genome alignment coverage' (default: False),\n",
      " 'reactome': 'Reactome ID' (default: False),\n",
      " 'reactome_gene': 'Reactome gene ID' (default: False),\n",
      " 'reactome_transcript': 'Reactome transcript ID' (default: False),\n",
      " 'refseq_mrna': 'RefSeq mRNA ID' (default: False),\n",
      " 'refseq_mrna_predicted': 'RefSeq mRNA predicted ID' (default: False),\n",
      " 'refseq_ncrna': 'RefSeq ncRNA ID' (default: False),\n",
      " 'refseq_ncrna_predicted': 'RefSeq ncRNA predicted ID' (default: False),\n",
      " 'refseq_peptide': 'RefSeq peptide ID' (default: False),\n",
      " 'refseq_peptide_predicted': 'RefSeq peptide predicted ID' (default: False),\n",
      " 'rfam': 'RFAM ID' (default: False),\n",
      " 'rfam_trans_name': 'RFAM transcript name ID' (default: False),\n",
      " 'rnacentral': 'RNAcentral ID' (default: False),\n",
      " 'rnorvegicus_homolog_associated_gene_name': 'Rat gene name' (default: False),\n",
      " 'rnorvegicus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rnorvegicus_homolog_chrom_end': 'Rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'rnorvegicus_homolog_chrom_start': 'Rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'rnorvegicus_homolog_chromosome': 'Rat chromosome/scaffold name' (default: False),\n",
      " 'rnorvegicus_homolog_dn': 'dN with Rat' (default: False),\n",
      " 'rnorvegicus_homolog_ds': 'dS with Rat' (default: False),\n",
      " 'rnorvegicus_homolog_ensembl_gene': 'Rat gene stable ID' (default: False),\n",
      " 'rnorvegicus_homolog_ensembl_peptide': 'Rat protein or transcript stable ID' (default: False),\n",
      " 'rnorvegicus_homolog_goc_score': 'Rat Gene-order conservation score' (default: False),\n",
      " 'rnorvegicus_homolog_orthology_confidence': 'Rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rnorvegicus_homolog_orthology_type': 'Rat homology type' (default: False),\n",
      " 'rnorvegicus_homolog_perc_id': '%id. target Rat gene identical to query gene' (default: False),\n",
      " 'rnorvegicus_homolog_perc_id_r1': '%id. query gene identical to target Rat gene' (default: False),\n",
      " 'rnorvegicus_homolog_subtype': 'Last common ancestor with Rat' (default: False),\n",
      " 'rnorvegicus_homolog_wga_coverage': 'Rat Whole-genome alignment coverage' (default: False),\n",
      " 'rroxellana_homolog_associated_gene_name': 'Golden snub-nosed monkey gene name' (default: False),\n",
      " 'rroxellana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rroxellana_homolog_chrom_end': 'Golden snub-nosed monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'rroxellana_homolog_chrom_start': 'Golden snub-nosed monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'rroxellana_homolog_chromosome': 'Golden snub-nosed monkey chromosome/scaffold name' (default: False),\n",
      " 'rroxellana_homolog_dn': 'dN with Golden snub-nosed monkey' (default: False),\n",
      " 'rroxellana_homolog_ds': 'dS with Golden snub-nosed monkey' (default: False),\n",
      " 'rroxellana_homolog_ensembl_gene': 'Golden snub-nosed monkey gene stable ID' (default: False),\n",
      " 'rroxellana_homolog_ensembl_peptide': 'Golden snub-nosed monkey protein or transcript stable ID' (default: False),\n",
      " 'rroxellana_homolog_goc_score': 'Golden snub-nosed monkey Gene-order conservation score' (default: False),\n",
      " 'rroxellana_homolog_orthology_confidence': 'Golden snub-nosed monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rroxellana_homolog_orthology_type': 'Golden snub-nosed monkey homology type' (default: False),\n",
      " 'rroxellana_homolog_perc_id': '%id. target Golden snub-nosed monkey gene identical to query gene' (default: False),\n",
      " 'rroxellana_homolog_perc_id_r1': '%id. query gene identical to target Golden snub-nosed monkey gene' (default: False),\n",
      " 'rroxellana_homolog_subtype': 'Last common ancestor with Golden snub-nosed monkey' (default: False),\n",
      " 'rroxellana_homolog_wga_coverage': 'Golden snub-nosed monkey Whole-genome alignment coverage' (default: False),\n",
      " 'saraneus_homolog_associated_gene_name': 'Shrew gene name' (default: False),\n",
      " 'saraneus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'saraneus_homolog_chrom_end': 'Shrew chromosome/scaffold end (bp)' (default: False),\n",
      " 'saraneus_homolog_chrom_start': 'Shrew chromosome/scaffold start (bp)' (default: False),\n",
      " 'saraneus_homolog_chromosome': 'Shrew chromosome/scaffold name' (default: False),\n",
      " 'saraneus_homolog_dn': 'dN with Shrew' (default: False),\n",
      " 'saraneus_homolog_ds': 'dS with Shrew' (default: False),\n",
      " 'saraneus_homolog_ensembl_gene': 'Shrew gene stable ID' (default: False),\n",
      " 'saraneus_homolog_ensembl_peptide': 'Shrew protein or transcript stable ID' (default: False),\n",
      " 'saraneus_homolog_goc_score': 'Shrew Gene-order conservation score' (default: False),\n",
      " 'saraneus_homolog_orthology_confidence': 'Shrew orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'saraneus_homolog_orthology_type': 'Shrew homology type' (default: False),\n",
      " 'saraneus_homolog_perc_id': '%id. target Shrew gene identical to query gene' (default: False),\n",
      " 'saraneus_homolog_perc_id_r1': '%id. query gene identical to target Shrew gene' (default: False),\n",
      " 'saraneus_homolog_subtype': 'Last common ancestor with Shrew' (default: False),\n",
      " 'saraneus_homolog_wga_coverage': 'Shrew Whole-genome alignment coverage' (default: False),\n",
      " 'sbboliviensis_homolog_associated_gene_name': 'Bolivian squirrel monkey gene name' (default: False),\n",
      " 'sbboliviensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sbboliviensis_homolog_chrom_end': 'Bolivian squirrel monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'sbboliviensis_homolog_chrom_start': 'Bolivian squirrel monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'sbboliviensis_homolog_chromosome': 'Bolivian squirrel monkey chromosome/scaffold name' (default: False),\n",
      " 'sbboliviensis_homolog_dn': 'dN with Bolivian squirrel monkey' (default: False),\n",
      " 'sbboliviensis_homolog_ds': 'dS with Bolivian squirrel monkey' (default: False),\n",
      " 'sbboliviensis_homolog_ensembl_gene': 'Bolivian squirrel monkey gene stable ID' (default: False),\n",
      " 'sbboliviensis_homolog_ensembl_peptide': 'Bolivian squirrel monkey protein or transcript stable ID' (default: False),\n",
      " 'sbboliviensis_homolog_goc_score': 'Bolivian squirrel monkey Gene-order conservation score' (default: False),\n",
      " 'sbboliviensis_homolog_orthology_confidence': 'Bolivian squirrel monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sbboliviensis_homolog_orthology_type': 'Bolivian squirrel monkey homology type' (default: False),\n",
      " 'sbboliviensis_homolog_perc_id': '%id. target Bolivian squirrel monkey gene identical to query gene' (default: False),\n",
      " 'sbboliviensis_homolog_perc_id_r1': '%id. query gene identical to target Bolivian squirrel monkey gene' (default: False),\n",
      " 'sbboliviensis_homolog_subtype': 'Last common ancestor with Bolivian squirrel monkey' (default: False),\n",
      " 'sbboliviensis_homolog_wga_coverage': 'Bolivian squirrel monkey Whole-genome alignment coverage' (default: False),\n",
      " 'scanaria_homolog_associated_gene_name': 'Common canary gene name' (default: False),\n",
      " 'scanaria_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scanaria_homolog_chrom_end': 'Common canary chromosome/scaffold end (bp)' (default: False),\n",
      " 'scanaria_homolog_chrom_start': 'Common canary chromosome/scaffold start (bp)' (default: False),\n",
      " 'scanaria_homolog_chromosome': 'Common canary chromosome/scaffold name' (default: False),\n",
      " 'scanaria_homolog_dn': 'dN with Common canary' (default: False),\n",
      " 'scanaria_homolog_ds': 'dS with Common canary' (default: False),\n",
      " 'scanaria_homolog_ensembl_gene': 'Common canary gene stable ID' (default: False),\n",
      " 'scanaria_homolog_ensembl_peptide': 'Common canary protein or transcript stable ID' (default: False),\n",
      " 'scanaria_homolog_goc_score': 'Common canary Gene-order conservation score' (default: False),\n",
      " 'scanaria_homolog_orthology_confidence': 'Common canary orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scanaria_homolog_orthology_type': 'Common canary homology type' (default: False),\n",
      " 'scanaria_homolog_perc_id': '%id. target Common canary gene identical to query gene' (default: False),\n",
      " 'scanaria_homolog_perc_id_r1': '%id. query gene identical to target Common canary gene' (default: False),\n",
      " 'scanaria_homolog_subtype': 'Last common ancestor with Common canary' (default: False),\n",
      " 'scanaria_homolog_wga_coverage': 'Common canary Whole-genome alignment coverage' (default: False),\n",
      " 'scanprosite': 'PROSITE patterns ID' (default: False),\n",
      " 'scanprosite_end': 'PROSITE patterns end' (default: False),\n",
      " 'scanprosite_start': 'PROSITE patterns start' (default: False),\n",
      " 'scerevisiae_homolog_associated_gene_name': 'Saccharomyces cerevisiae gene name' (default: False),\n",
      " 'scerevisiae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scerevisiae_homolog_chrom_end': 'Saccharomyces cerevisiae chromosome/scaffold end (bp)' (default: False),\n",
      " 'scerevisiae_homolog_chrom_start': 'Saccharomyces cerevisiae chromosome/scaffold start (bp)' (default: False),\n",
      " 'scerevisiae_homolog_chromosome': 'Saccharomyces cerevisiae chromosome/scaffold name' (default: False),\n",
      " 'scerevisiae_homolog_dn': 'dN with Saccharomyces cerevisiae' (default: False),\n",
      " 'scerevisiae_homolog_ds': 'dS with Saccharomyces cerevisiae' (default: False),\n",
      " 'scerevisiae_homolog_ensembl_gene': 'Saccharomyces cerevisiae gene stable ID' (default: False),\n",
      " 'scerevisiae_homolog_ensembl_peptide': 'Saccharomyces cerevisiae protein or transcript stable ID' (default: False),\n",
      " 'scerevisiae_homolog_goc_score': 'Saccharomyces cerevisiae Gene-order conservation score' (default: False),\n",
      " 'scerevisiae_homolog_orthology_confidence': 'Saccharomyces cerevisiae orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scerevisiae_homolog_orthology_type': 'Saccharomyces cerevisiae homology type' (default: False),\n",
      " 'scerevisiae_homolog_perc_id': '%id. target Saccharomyces cerevisiae gene identical to query gene' (default: False),\n",
      " 'scerevisiae_homolog_perc_id_r1': '%id. query gene identical to target Saccharomyces cerevisiae gene' (default: False),\n",
      " 'scerevisiae_homolog_subtype': 'Last common ancestor with Saccharomyces cerevisiae' (default: False),\n",
      " 'scerevisiae_homolog_wga_coverage': 'Saccharomyces cerevisiae Whole-genome alignment coverage' (default: False),\n",
      " 'sdauricus_homolog_associated_gene_name': 'Daurian ground squirrel gene name' (default: False),\n",
      " 'sdauricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sdauricus_homolog_chrom_end': 'Daurian ground squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'sdauricus_homolog_chrom_start': 'Daurian ground squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'sdauricus_homolog_chromosome': 'Daurian ground squirrel chromosome/scaffold name' (default: False),\n",
      " 'sdauricus_homolog_dn': 'dN with Daurian ground squirrel' (default: False),\n",
      " 'sdauricus_homolog_ds': 'dS with Daurian ground squirrel' (default: False),\n",
      " 'sdauricus_homolog_ensembl_gene': 'Daurian ground squirrel gene stable ID' (default: False),\n",
      " 'sdauricus_homolog_ensembl_peptide': 'Daurian ground squirrel protein or transcript stable ID' (default: False),\n",
      " 'sdauricus_homolog_goc_score': 'Daurian ground squirrel Gene-order conservation score' (default: False),\n",
      " 'sdauricus_homolog_orthology_confidence': 'Daurian ground squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sdauricus_homolog_orthology_type': 'Daurian ground squirrel homology type' (default: False),\n",
      " 'sdauricus_homolog_perc_id': '%id. target Daurian ground squirrel gene identical to query gene' (default: False),\n",
      " 'sdauricus_homolog_perc_id_r1': '%id. query gene identical to target Daurian ground squirrel gene' (default: False),\n",
      " 'sdauricus_homolog_subtype': 'Last common ancestor with Daurian ground squirrel' (default: False),\n",
      " 'sdauricus_homolog_wga_coverage': 'Daurian ground squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'sdumerili_homolog_associated_gene_name': 'Greater amberjack gene name' (default: False),\n",
      " 'sdumerili_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sdumerili_homolog_chrom_end': 'Greater amberjack chromosome/scaffold end (bp)' (default: False),\n",
      " 'sdumerili_homolog_chrom_start': 'Greater amberjack chromosome/scaffold start (bp)' (default: False),\n",
      " 'sdumerili_homolog_chromosome': 'Greater amberjack chromosome/scaffold name' (default: False),\n",
      " 'sdumerili_homolog_dn': 'dN with Greater amberjack' (default: False),\n",
      " 'sdumerili_homolog_ds': 'dS with Greater amberjack' (default: False),\n",
      " 'sdumerili_homolog_ensembl_gene': 'Greater amberjack gene stable ID' (default: False),\n",
      " 'sdumerili_homolog_ensembl_peptide': 'Greater amberjack protein or transcript stable ID' (default: False),\n",
      " 'sdumerili_homolog_goc_score': 'Greater amberjack Gene-order conservation score' (default: False),\n",
      " 'sdumerili_homolog_orthology_confidence': 'Greater amberjack orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sdumerili_homolog_orthology_type': 'Greater amberjack homology type' (default: False),\n",
      " 'sdumerili_homolog_perc_id': '%id. target Greater amberjack gene identical to query gene' (default: False),\n",
      " 'sdumerili_homolog_perc_id_r1': '%id. query gene identical to target Greater amberjack gene' (default: False),\n",
      " 'sdumerili_homolog_subtype': 'Last common ancestor with Greater amberjack' (default: False),\n",
      " 'sdumerili_homolog_wga_coverage': 'Greater amberjack Whole-genome alignment coverage' (default: False),\n",
      " 'seg': 'Low complexity (Seg)' (default: False),\n",
      " 'seg_end': 'Low complexity (Seg) end' (default: False),\n",
      " 'seg_start': 'Low complexity (Seg) start' (default: False),\n",
      " 'sfld': 'SFLD ID' (default: False),\n",
      " 'sfld_end': 'SFLD end' (default: False),\n",
      " 'sfld_start': 'SFLD start' (default: False),\n",
      " 'sformosus_homolog_associated_gene_name': 'Asian bonytongue gene name' (default: False),\n",
      " 'sformosus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sformosus_homolog_chrom_end': 'Asian bonytongue chromosome/scaffold end (bp)' (default: False),\n",
      " 'sformosus_homolog_chrom_start': 'Asian bonytongue chromosome/scaffold start (bp)' (default: False),\n",
      " 'sformosus_homolog_chromosome': 'Asian bonytongue chromosome/scaffold name' (default: False),\n",
      " 'sformosus_homolog_dn': 'dN with Asian bonytongue' (default: False),\n",
      " 'sformosus_homolog_ds': 'dS with Asian bonytongue' (default: False),\n",
      " 'sformosus_homolog_ensembl_gene': 'Asian bonytongue gene stable ID' (default: False),\n",
      " 'sformosus_homolog_ensembl_peptide': 'Asian bonytongue protein or transcript stable ID' (default: False),\n",
      " 'sformosus_homolog_goc_score': 'Asian bonytongue Gene-order conservation score' (default: False),\n",
      " 'sformosus_homolog_orthology_confidence': 'Asian bonytongue orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sformosus_homolog_orthology_type': 'Asian bonytongue homology type' (default: False),\n",
      " 'sformosus_homolog_perc_id': '%id. target Asian bonytongue gene identical to query gene' (default: False),\n",
      " 'sformosus_homolog_perc_id_r1': '%id. query gene identical to target Asian bonytongue gene' (default: False),\n",
      " 'sformosus_homolog_subtype': 'Last common ancestor with Asian bonytongue' (default: False),\n",
      " 'sformosus_homolog_wga_coverage': 'Asian bonytongue Whole-genome alignment coverage' (default: False),\n",
      " 'sharrisii_homolog_associated_gene_name': 'Tasmanian devil gene name' (default: False),\n",
      " 'sharrisii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sharrisii_homolog_chrom_end': 'Tasmanian devil chromosome/scaffold end (bp)' (default: False),\n",
      " 'sharrisii_homolog_chrom_start': 'Tasmanian devil chromosome/scaffold start (bp)' (default: False),\n",
      " 'sharrisii_homolog_chromosome': 'Tasmanian devil chromosome/scaffold name' (default: False),\n",
      " 'sharrisii_homolog_dn': 'dN with Tasmanian devil' (default: False),\n",
      " 'sharrisii_homolog_ds': 'dS with Tasmanian devil' (default: False),\n",
      " 'sharrisii_homolog_ensembl_gene': 'Tasmanian devil gene stable ID' (default: False),\n",
      " 'sharrisii_homolog_ensembl_peptide': 'Tasmanian devil protein or transcript stable ID' (default: False),\n",
      " 'sharrisii_homolog_goc_score': 'Tasmanian devil Gene-order conservation score' (default: False),\n",
      " 'sharrisii_homolog_orthology_confidence': 'Tasmanian devil orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sharrisii_homolog_orthology_type': 'Tasmanian devil homology type' (default: False),\n",
      " 'sharrisii_homolog_perc_id': '%id. target Tasmanian devil gene identical to query gene' (default: False),\n",
      " 'sharrisii_homolog_perc_id_r1': '%id. query gene identical to target Tasmanian devil gene' (default: False),\n",
      " 'sharrisii_homolog_subtype': 'Last common ancestor with Tasmanian devil' (default: False),\n",
      " 'sharrisii_homolog_wga_coverage': 'Tasmanian devil Whole-genome alignment coverage' (default: False),\n",
      " 'sift_prediction_2076': 'SIFT prediction' (default: False),\n",
      " 'sift_score_2076': 'SIFT score' (default: False),\n",
      " 'sifts_import': 'PDB-ENSP mappings' (default: False),\n",
      " 'sifts_import_end': 'PDB-ENSP mappings end' (default: False),\n",
      " 'sifts_import_start': 'PDB-ENSP mappings start' (default: False),\n",
      " 'signalp': 'Cleavage site (Signalp)' (default: False),\n",
      " 'signalp_end': 'Cleavage site (Signalp) end' (default: False),\n",
      " 'signalp_start': 'Cleavage site (Signalp) start' (default: False),\n",
      " 'sldorsalis_homolog_associated_gene_name': 'Yellowtail amberjack gene name' (default: False),\n",
      " 'sldorsalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sldorsalis_homolog_chrom_end': 'Yellowtail amberjack chromosome/scaffold end (bp)' (default: False),\n",
      " 'sldorsalis_homolog_chrom_start': 'Yellowtail amberjack chromosome/scaffold start (bp)' (default: False),\n",
      " 'sldorsalis_homolog_chromosome': 'Yellowtail amberjack chromosome/scaffold name' (default: False),\n",
      " 'sldorsalis_homolog_dn': 'dN with Yellowtail amberjack' (default: False),\n",
      " 'sldorsalis_homolog_ds': 'dS with Yellowtail amberjack' (default: False),\n",
      " 'sldorsalis_homolog_ensembl_gene': 'Yellowtail amberjack gene stable ID' (default: False),\n",
      " 'sldorsalis_homolog_ensembl_peptide': 'Yellowtail amberjack protein or transcript stable ID' (default: False),\n",
      " 'sldorsalis_homolog_goc_score': 'Yellowtail amberjack Gene-order conservation score' (default: False),\n",
      " 'sldorsalis_homolog_orthology_confidence': 'Yellowtail amberjack orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sldorsalis_homolog_orthology_type': 'Yellowtail amberjack homology type' (default: False),\n",
      " 'sldorsalis_homolog_perc_id': '%id. target Yellowtail amberjack gene identical to query gene' (default: False),\n",
      " 'sldorsalis_homolog_perc_id_r1': '%id. query gene identical to target Yellowtail amberjack gene' (default: False),\n",
      " 'sldorsalis_homolog_subtype': 'Last common ancestor with Yellowtail amberjack' (default: False),\n",
      " 'sldorsalis_homolog_wga_coverage': 'Yellowtail amberjack Whole-genome alignment coverage' (default: False),\n",
      " 'smart': 'SMART ID' (default: False),\n",
      " 'smart_end': 'SMART end' (default: False),\n",
      " 'smart_start': 'SMART start' (default: False),\n",
      " 'smaximus_homolog_associated_gene_name': 'Turbot gene name' (default: False),\n",
      " 'smaximus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'smaximus_homolog_chrom_end': 'Turbot chromosome/scaffold end (bp)' (default: False),\n",
      " 'smaximus_homolog_chrom_start': 'Turbot chromosome/scaffold start (bp)' (default: False),\n",
      " 'smaximus_homolog_chromosome': 'Turbot chromosome/scaffold name' (default: False),\n",
      " 'smaximus_homolog_dn': 'dN with Turbot' (default: False),\n",
      " 'smaximus_homolog_ds': 'dS with Turbot' (default: False),\n",
      " 'smaximus_homolog_ensembl_gene': 'Turbot gene stable ID' (default: False),\n",
      " 'smaximus_homolog_ensembl_peptide': 'Turbot protein or transcript stable ID' (default: False),\n",
      " 'smaximus_homolog_goc_score': 'Turbot Gene-order conservation score' (default: False),\n",
      " 'smaximus_homolog_orthology_confidence': 'Turbot orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'smaximus_homolog_orthology_type': 'Turbot homology type' (default: False),\n",
      " 'smaximus_homolog_perc_id': '%id. target Turbot gene identical to query gene' (default: False),\n",
      " 'smaximus_homolog_perc_id_r1': '%id. query gene identical to target Turbot gene' (default: False),\n",
      " 'smaximus_homolog_subtype': 'Last common ancestor with Turbot' (default: False),\n",
      " 'smaximus_homolog_wga_coverage': 'Turbot Whole-genome alignment coverage' (default: False),\n",
      " 'smerianae_homolog_associated_gene_name': 'Argentine black and white tegu gene name' (default: False),\n",
      " 'smerianae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'smerianae_homolog_chrom_end': 'Argentine black and white tegu chromosome/scaffold end (bp)' (default: False),\n",
      " 'smerianae_homolog_chrom_start': 'Argentine black and white tegu chromosome/scaffold start (bp)' (default: False),\n",
      " 'smerianae_homolog_chromosome': 'Argentine black and white tegu chromosome/scaffold name' (default: False),\n",
      " 'smerianae_homolog_dn': 'dN with Argentine black and white tegu' (default: False),\n",
      " 'smerianae_homolog_ds': 'dS with Argentine black and white tegu' (default: False),\n",
      " 'smerianae_homolog_ensembl_gene': 'Argentine black and white tegu gene stable ID' (default: False),\n",
      " 'smerianae_homolog_ensembl_peptide': 'Argentine black and white tegu protein or transcript stable ID' (default: False),\n",
      " 'smerianae_homolog_goc_score': 'Argentine black and white tegu Gene-order conservation score' (default: False),\n",
      " 'smerianae_homolog_orthology_confidence': 'Argentine black and white tegu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'smerianae_homolog_orthology_type': 'Argentine black and white tegu homology type' (default: False),\n",
      " 'smerianae_homolog_perc_id': '%id. target Argentine black and white tegu gene identical to query gene' (default: False),\n",
      " 'smerianae_homolog_perc_id_r1': '%id. query gene identical to target Argentine black and white tegu gene' (default: False),\n",
      " 'smerianae_homolog_subtype': 'Last common ancestor with Argentine black and white tegu' (default: False),\n",
      " 'smerianae_homolog_wga_coverage': 'Argentine black and white tegu Whole-genome alignment coverage' (default: False),\n",
      " 'snp_chromosome_strand': 'Variant chromosome Strand' (default: False),\n",
      " 'somatic_allele': 'Variant alleles' (default: False),\n",
      " 'somatic_cds_end_2076': 'CDS end' (default: False),\n",
      " 'somatic_cds_start_2076': 'CDS start' (default: False),\n",
      " 'somatic_chromosome_end': 'Chromosome/scaffold position end (bp)' (default: False),\n",
      " 'somatic_chromosome_start': 'Chromosome/scaffold position start (bp)' (default: False),\n",
      " 'somatic_clinical_significance': 'Clinical significance' (default: False),\n",
      " 'somatic_mapweight': 'Mapweight' (default: False),\n",
      " 'somatic_minor_allele': 'Minor allele' (default: False),\n",
      " 'somatic_minor_allele_count': 'Minor allele count' (default: False),\n",
      " 'somatic_minor_allele_freq': 'Minor allele frequency' (default: False),\n",
      " 'somatic_peptide_location': 'Protein location (aa)' (default: False),\n",
      " 'somatic_peptide_shift': 'Protein allele' (default: False),\n",
      " 'somatic_snp_chromosome_strand': 'Variant chromosome/scaffold strand' (default: False),\n",
      " 'somatic_source_description': 'Variant source description' (default: False),\n",
      " 'somatic_source_name': 'Variant source' (default: False),\n",
      " 'somatic_synonymous_status': 'Variant consequence' (default: False),\n",
      " 'somatic_transcript_location': 'Transcript location (bp)' (default: False),\n",
      " 'somatic_validated': 'Variant supporting evidence' (default: False),\n",
      " 'somatic_variation_name': 'Variant name' (default: True),\n",
      " 'source': 'Source (gene)' (default: False),\n",
      " 'source_description': 'Variant source description' (default: False),\n",
      " 'source_name': 'Source name' (default: False),\n",
      " 'spartitus_homolog_associated_gene_name': 'Bicolor damselfish gene name' (default: False),\n",
      " 'spartitus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'spartitus_homolog_chrom_end': 'Bicolor damselfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'spartitus_homolog_chrom_start': 'Bicolor damselfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'spartitus_homolog_chromosome': 'Bicolor damselfish chromosome/scaffold name' (default: False),\n",
      " 'spartitus_homolog_dn': 'dN with Bicolor damselfish' (default: False),\n",
      " 'spartitus_homolog_ds': 'dS with Bicolor damselfish' (default: False),\n",
      " 'spartitus_homolog_ensembl_gene': 'Bicolor damselfish gene stable ID' (default: False),\n",
      " 'spartitus_homolog_ensembl_peptide': 'Bicolor damselfish protein or transcript stable ID' (default: False),\n",
      " 'spartitus_homolog_goc_score': 'Bicolor damselfish Gene-order conservation score' (default: False),\n",
      " 'spartitus_homolog_orthology_confidence': 'Bicolor damselfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'spartitus_homolog_orthology_type': 'Bicolor damselfish homology type' (default: False),\n",
      " 'spartitus_homolog_perc_id': '%id. target Bicolor damselfish gene identical to query gene' (default: False),\n",
      " 'spartitus_homolog_perc_id_r1': '%id. query gene identical to target Bicolor damselfish gene' (default: False),\n",
      " 'spartitus_homolog_subtype': 'Last common ancestor with Bicolor damselfish' (default: False),\n",
      " 'spartitus_homolog_wga_coverage': 'Bicolor damselfish Whole-genome alignment coverage' (default: False),\n",
      " 'spunctatus_homolog_associated_gene_name': 'Tuatara gene name' (default: False),\n",
      " 'spunctatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'spunctatus_homolog_chrom_end': 'Tuatara chromosome/scaffold end (bp)' (default: False),\n",
      " 'spunctatus_homolog_chrom_start': 'Tuatara chromosome/scaffold start (bp)' (default: False),\n",
      " 'spunctatus_homolog_chromosome': 'Tuatara chromosome/scaffold name' (default: False),\n",
      " 'spunctatus_homolog_dn': 'dN with Tuatara' (default: False),\n",
      " 'spunctatus_homolog_ds': 'dS with Tuatara' (default: False),\n",
      " 'spunctatus_homolog_ensembl_gene': 'Tuatara gene stable ID' (default: False),\n",
      " 'spunctatus_homolog_ensembl_peptide': 'Tuatara protein or transcript stable ID' (default: False),\n",
      " 'spunctatus_homolog_goc_score': 'Tuatara Gene-order conservation score' (default: False),\n",
      " 'spunctatus_homolog_orthology_confidence': 'Tuatara orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'spunctatus_homolog_orthology_type': 'Tuatara homology type' (default: False),\n",
      " 'spunctatus_homolog_perc_id': '%id. target Tuatara gene identical to query gene' (default: False),\n",
      " 'spunctatus_homolog_perc_id_r1': '%id. query gene identical to target Tuatara gene' (default: False),\n",
      " 'spunctatus_homolog_subtype': 'Last common ancestor with Tuatara' (default: False),\n",
      " 'spunctatus_homolog_wga_coverage': 'Tuatara Whole-genome alignment coverage' (default: False),\n",
      " 'sscrofa_homolog_associated_gene_name': 'Pig gene name' (default: False),\n",
      " 'sscrofa_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sscrofa_homolog_chrom_end': 'Pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'sscrofa_homolog_chrom_start': 'Pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'sscrofa_homolog_chromosome': 'Pig chromosome/scaffold name' (default: False),\n",
      " 'sscrofa_homolog_dn': 'dN with Pig' (default: False),\n",
      " 'sscrofa_homolog_ds': 'dS with Pig' (default: False),\n",
      " 'sscrofa_homolog_ensembl_gene': 'Pig gene stable ID' (default: False),\n",
      " 'sscrofa_homolog_ensembl_peptide': 'Pig protein or transcript stable ID' (default: False),\n",
      " 'sscrofa_homolog_goc_score': 'Pig Gene-order conservation score' (default: False),\n",
      " 'sscrofa_homolog_orthology_confidence': 'Pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sscrofa_homolog_orthology_type': 'Pig homology type' (default: False),\n",
      " 'sscrofa_homolog_perc_id': '%id. target Pig gene identical to query gene' (default: False),\n",
      " 'sscrofa_homolog_perc_id_r1': '%id. query gene identical to target Pig gene' (default: False),\n",
      " 'sscrofa_homolog_subtype': 'Last common ancestor with Pig' (default: False),\n",
      " 'sscrofa_homolog_wga_coverage': 'Pig Whole-genome alignment coverage' (default: False),\n",
      " 'start_position': 'Gene start (bp)' (default: False),\n",
      " 'strain_gender': 'Strain gender' (default: False),\n",
      " 'strain_name': 'Strain name' (default: False),\n",
      " 'strand': 'Strand' (default: False),\n",
      " 'study_external_id': 'Study external reference' (default: False),\n",
      " 'superfamily': 'Superfamily ID' (default: False),\n",
      " 'superfamily_end': 'Superfamily end' (default: False),\n",
      " 'superfamily_start': 'Superfamily start' (default: False),\n",
      " 'synonymous_status': 'Variant consequence' (default: False),\n",
      " 'tbelangeri_homolog_associated_gene_name': 'Tree Shrew gene name' (default: False),\n",
      " 'tbelangeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tbelangeri_homolog_chrom_end': 'Tree Shrew chromosome/scaffold end (bp)' (default: False),\n",
      " 'tbelangeri_homolog_chrom_start': 'Tree Shrew chromosome/scaffold start (bp)' (default: False),\n",
      " 'tbelangeri_homolog_chromosome': 'Tree Shrew chromosome/scaffold name' (default: False),\n",
      " 'tbelangeri_homolog_dn': 'dN with Tree Shrew' (default: False),\n",
      " 'tbelangeri_homolog_ds': 'dS with Tree Shrew' (default: False),\n",
      " 'tbelangeri_homolog_ensembl_gene': 'Tree Shrew gene stable ID' (default: False),\n",
      " 'tbelangeri_homolog_ensembl_peptide': 'Tree Shrew protein or transcript stable ID' (default: False),\n",
      " 'tbelangeri_homolog_goc_score': 'Tree Shrew Gene-order conservation score' (default: False),\n",
      " 'tbelangeri_homolog_orthology_confidence': 'Tree Shrew orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tbelangeri_homolog_orthology_type': 'Tree Shrew homology type' (default: False),\n",
      " 'tbelangeri_homolog_perc_id': '%id. target Tree Shrew gene identical to query gene' (default: False),\n",
      " 'tbelangeri_homolog_perc_id_r1': '%id. query gene identical to target Tree Shrew gene' (default: False),\n",
      " 'tbelangeri_homolog_subtype': 'Last common ancestor with Tree Shrew' (default: False),\n",
      " 'tbelangeri_homolog_wga_coverage': 'Tree Shrew Whole-genome alignment coverage' (default: False),\n",
      " 'tgelada_homolog_associated_gene_name': 'Gelada gene name' (default: False),\n",
      " 'tgelada_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tgelada_homolog_chrom_end': 'Gelada chromosome/scaffold end (bp)' (default: False),\n",
      " 'tgelada_homolog_chrom_start': 'Gelada chromosome/scaffold start (bp)' (default: False),\n",
      " 'tgelada_homolog_chromosome': 'Gelada chromosome/scaffold name' (default: False),\n",
      " 'tgelada_homolog_dn': 'dN with Gelada' (default: False),\n",
      " 'tgelada_homolog_ds': 'dS with Gelada' (default: False),\n",
      " 'tgelada_homolog_ensembl_gene': 'Gelada gene stable ID' (default: False),\n",
      " 'tgelada_homolog_ensembl_peptide': 'Gelada protein or transcript stable ID' (default: False),\n",
      " 'tgelada_homolog_goc_score': 'Gelada Gene-order conservation score' (default: False),\n",
      " 'tgelada_homolog_orthology_confidence': 'Gelada orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tgelada_homolog_orthology_type': 'Gelada homology type' (default: False),\n",
      " 'tgelada_homolog_perc_id': '%id. target Gelada gene identical to query gene' (default: False),\n",
      " 'tgelada_homolog_perc_id_r1': '%id. query gene identical to target Gelada gene' (default: False),\n",
      " 'tgelada_homolog_subtype': 'Last common ancestor with Gelada' (default: False),\n",
      " 'tgelada_homolog_wga_coverage': 'Gelada Whole-genome alignment coverage' (default: False),\n",
      " 'tguttata_homolog_associated_gene_name': 'Zebra Finch gene name' (default: False),\n",
      " 'tguttata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tguttata_homolog_chrom_end': 'Zebra Finch chromosome/scaffold end (bp)' (default: False),\n",
      " 'tguttata_homolog_chrom_start': 'Zebra Finch chromosome/scaffold start (bp)' (default: False),\n",
      " 'tguttata_homolog_chromosome': 'Zebra Finch chromosome/scaffold name' (default: False),\n",
      " 'tguttata_homolog_dn': 'dN with Zebra Finch' (default: False),\n",
      " 'tguttata_homolog_ds': 'dS with Zebra Finch' (default: False),\n",
      " 'tguttata_homolog_ensembl_gene': 'Zebra Finch gene stable ID' (default: False),\n",
      " 'tguttata_homolog_ensembl_peptide': 'Zebra Finch protein or transcript stable ID' (default: False),\n",
      " 'tguttata_homolog_goc_score': 'Zebra Finch Gene-order conservation score' (default: False),\n",
      " 'tguttata_homolog_orthology_confidence': 'Zebra Finch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tguttata_homolog_orthology_type': 'Zebra Finch homology type' (default: False),\n",
      " 'tguttata_homolog_perc_id': '%id. target Zebra Finch gene identical to query gene' (default: False),\n",
      " 'tguttata_homolog_perc_id_r1': '%id. query gene identical to target Zebra Finch gene' (default: False),\n",
      " 'tguttata_homolog_subtype': 'Last common ancestor with Zebra Finch' (default: False),\n",
      " 'tguttata_homolog_wga_coverage': 'Zebra Finch Whole-genome alignment coverage' (default: False),\n",
      " 'tigrfam': 'TIGRFAM ID' (default: False),\n",
      " 'tigrfam_end': 'TIGRFAM end' (default: False),\n",
      " 'tigrfam_start': 'TIGRFAM start' (default: False),\n",
      " 'tmhmm': 'Transmembrane helices' (default: False),\n",
      " 'tmhmm_end': 'Transmembrane helices end' (default: False),\n",
      " 'tmhmm_start': 'Transmembrane helices start' (default: False),\n",
      " 'tnigroviridis_homolog_associated_gene_name': 'Tetraodon gene name' (default: False),\n",
      " 'tnigroviridis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tnigroviridis_homolog_chrom_end': 'Tetraodon chromosome/scaffold end (bp)' (default: False),\n",
      " 'tnigroviridis_homolog_chrom_start': 'Tetraodon chromosome/scaffold start (bp)' (default: False),\n",
      " 'tnigroviridis_homolog_chromosome': 'Tetraodon chromosome/scaffold name' (default: False),\n",
      " 'tnigroviridis_homolog_dn': 'dN with Tetraodon' (default: False),\n",
      " 'tnigroviridis_homolog_ds': 'dS with Tetraodon' (default: False),\n",
      " 'tnigroviridis_homolog_ensembl_gene': 'Tetraodon gene stable ID' (default: False),\n",
      " 'tnigroviridis_homolog_ensembl_peptide': 'Tetraodon protein or transcript stable ID' (default: False),\n",
      " 'tnigroviridis_homolog_goc_score': 'Tetraodon Gene-order conservation score' (default: False),\n",
      " 'tnigroviridis_homolog_orthology_confidence': 'Tetraodon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tnigroviridis_homolog_orthology_type': 'Tetraodon homology type' (default: False),\n",
      " 'tnigroviridis_homolog_perc_id': '%id. target Tetraodon gene identical to query gene' (default: False),\n",
      " 'tnigroviridis_homolog_perc_id_r1': '%id. query gene identical to target Tetraodon gene' (default: False),\n",
      " 'tnigroviridis_homolog_subtype': 'Last common ancestor with Tetraodon' (default: False),\n",
      " 'tnigroviridis_homolog_wga_coverage': 'Tetraodon Whole-genome alignment coverage' (default: False),\n",
      " 'transcript_appris': 'APPRIS annotation' (default: False),\n",
      " 'transcript_biotype': 'Transcript type' (default: False),\n",
      " 'transcript_count': 'Transcript count' (default: False),\n",
      " 'transcript_end': 'Transcript end (bp)' (default: False),\n",
      " 'transcript_exon_intron': 'Unspliced (Transcript)' (default: False),\n",
      " 'transcript_flank': 'Flank (Transcript)' (default: False),\n",
      " 'transcript_gencode_basic': 'GENCODE basic annotation' (default: False),\n",
      " 'transcript_length': 'Transcript length (including UTRs and CDS)' (default: False),\n",
      " 'transcript_location': 'Transcript location (bp)' (default: False),\n",
      " 'transcript_mane_select': 'RefSeq match transcript' (default: False),\n",
      " 'transcript_source': 'Source (transcript)' (default: False),\n",
      " 'transcript_start': 'Transcript start (bp)' (default: False),\n",
      " 'transcript_tsl': 'Transcript support level (TSL)' (default: False),\n",
      " 'transcript_version': 'Version (transcript)' (default: False),\n",
      " 'transcription_start_site': 'Transcription start site (TSS)' (default: False),\n",
      " 'trubripes_homolog_associated_gene_name': 'Fugu gene name' (default: False),\n",
      " 'trubripes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'trubripes_homolog_chrom_end': 'Fugu chromosome/scaffold end (bp)' (default: False),\n",
      " 'trubripes_homolog_chrom_start': 'Fugu chromosome/scaffold start (bp)' (default: False),\n",
      " 'trubripes_homolog_chromosome': 'Fugu chromosome/scaffold name' (default: False),\n",
      " 'trubripes_homolog_dn': 'dN with Fugu' (default: False),\n",
      " 'trubripes_homolog_ds': 'dS with Fugu' (default: False),\n",
      " 'trubripes_homolog_ensembl_gene': 'Fugu gene stable ID' (default: False),\n",
      " 'trubripes_homolog_ensembl_peptide': 'Fugu protein or transcript stable ID' (default: False),\n",
      " 'trubripes_homolog_goc_score': 'Fugu Gene-order conservation score' (default: False),\n",
      " 'trubripes_homolog_orthology_confidence': 'Fugu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'trubripes_homolog_orthology_type': 'Fugu homology type' (default: False),\n",
      " 'trubripes_homolog_perc_id': '%id. target Fugu gene identical to query gene' (default: False),\n",
      " 'trubripes_homolog_perc_id_r1': '%id. query gene identical to target Fugu gene' (default: False),\n",
      " 'trubripes_homolog_subtype': 'Last common ancestor with Fugu' (default: False),\n",
      " 'trubripes_homolog_wga_coverage': 'Fugu Whole-genome alignment coverage' (default: False),\n",
      " 'ttruncatus_homolog_associated_gene_name': 'Dolphin gene name' (default: False),\n",
      " 'ttruncatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ttruncatus_homolog_chrom_end': 'Dolphin chromosome/scaffold end (bp)' (default: False),\n",
      " 'ttruncatus_homolog_chrom_start': 'Dolphin chromosome/scaffold start (bp)' (default: False),\n",
      " 'ttruncatus_homolog_chromosome': 'Dolphin chromosome/scaffold name' (default: False),\n",
      " 'ttruncatus_homolog_dn': 'dN with Dolphin' (default: False),\n",
      " 'ttruncatus_homolog_ds': 'dS with Dolphin' (default: False),\n",
      " 'ttruncatus_homolog_ensembl_gene': 'Dolphin gene stable ID' (default: False),\n",
      " 'ttruncatus_homolog_ensembl_peptide': 'Dolphin protein or transcript stable ID' (default: False),\n",
      " 'ttruncatus_homolog_goc_score': 'Dolphin Gene-order conservation score' (default: False),\n",
      " 'ttruncatus_homolog_orthology_confidence': 'Dolphin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ttruncatus_homolog_orthology_type': 'Dolphin homology type' (default: False),\n",
      " 'ttruncatus_homolog_perc_id': '%id. target Dolphin gene identical to query gene' (default: False),\n",
      " 'ttruncatus_homolog_perc_id_r1': '%id. query gene identical to target Dolphin gene' (default: False),\n",
      " 'ttruncatus_homolog_subtype': 'Last common ancestor with Dolphin' (default: False),\n",
      " 'ttruncatus_homolog_wga_coverage': 'Dolphin Whole-genome alignment coverage' (default: False),\n",
      " 'uamericanus_homolog_associated_gene_name': 'American black bear gene name' (default: False),\n",
      " 'uamericanus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'uamericanus_homolog_chrom_end': 'American black bear chromosome/scaffold end (bp)' (default: False),\n",
      " 'uamericanus_homolog_chrom_start': 'American black bear chromosome/scaffold start (bp)' (default: False),\n",
      " 'uamericanus_homolog_chromosome': 'American black bear chromosome/scaffold name' (default: False),\n",
      " 'uamericanus_homolog_dn': 'dN with American black bear' (default: False),\n",
      " 'uamericanus_homolog_ds': 'dS with American black bear' (default: False),\n",
      " 'uamericanus_homolog_ensembl_gene': 'American black bear gene stable ID' (default: False),\n",
      " 'uamericanus_homolog_ensembl_peptide': 'American black bear protein or transcript stable ID' (default: False),\n",
      " 'uamericanus_homolog_goc_score': 'American black bear Gene-order conservation score' (default: False),\n",
      " 'uamericanus_homolog_orthology_confidence': 'American black bear orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'uamericanus_homolog_orthology_type': 'American black bear homology type' (default: False),\n",
      " 'uamericanus_homolog_perc_id': '%id. target American black bear gene identical to query gene' (default: False),\n",
      " 'uamericanus_homolog_perc_id_r1': '%id. query gene identical to target American black bear gene' (default: False),\n",
      " 'uamericanus_homolog_subtype': 'Last common ancestor with American black bear' (default: False),\n",
      " 'uamericanus_homolog_wga_coverage': 'American black bear Whole-genome alignment coverage' (default: False),\n",
      " 'ucsc': 'UCSC Stable ID' (default: False),\n",
      " 'umaritimus_homolog_associated_gene_name': 'Polar bear gene name' (default: False),\n",
      " 'umaritimus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'umaritimus_homolog_chrom_end': 'Polar bear chromosome/scaffold end (bp)' (default: False),\n",
      " 'umaritimus_homolog_chrom_start': 'Polar bear chromosome/scaffold start (bp)' (default: False),\n",
      " 'umaritimus_homolog_chromosome': 'Polar bear chromosome/scaffold name' (default: False),\n",
      " 'umaritimus_homolog_dn': 'dN with Polar bear' (default: False),\n",
      " 'umaritimus_homolog_ds': 'dS with Polar bear' (default: False),\n",
      " 'umaritimus_homolog_ensembl_gene': 'Polar bear gene stable ID' (default: False),\n",
      " 'umaritimus_homolog_ensembl_peptide': 'Polar bear protein or transcript stable ID' (default: False),\n",
      " 'umaritimus_homolog_goc_score': 'Polar bear Gene-order conservation score' (default: False),\n",
      " 'umaritimus_homolog_orthology_confidence': 'Polar bear orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'umaritimus_homolog_orthology_type': 'Polar bear homology type' (default: False),\n",
      " 'umaritimus_homolog_perc_id': '%id. target Polar bear gene identical to query gene' (default: False),\n",
      " 'umaritimus_homolog_perc_id_r1': '%id. query gene identical to target Polar bear gene' (default: False),\n",
      " 'umaritimus_homolog_subtype': 'Last common ancestor with Polar bear' (default: False),\n",
      " 'umaritimus_homolog_wga_coverage': 'Polar bear Whole-genome alignment coverage' (default: False),\n",
      " 'uniparc': 'UniParc ID' (default: False),\n",
      " 'uniprot_gn_id': 'UniProtKB Gene Name ID' (default: False),\n",
      " 'uniprot_gn_symbol': 'UniProtKB Gene Name symbol' (default: False),\n",
      " 'uniprotsptrembl': 'UniProtKB/TrEMBL ID' (default: False),\n",
      " 'uniprotswissprot': 'UniProtKB/Swiss-Prot ID' (default: False),\n",
      " 'uparryii_homolog_associated_gene_name': 'Arctic ground squirrel gene name' (default: False),\n",
      " 'uparryii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'uparryii_homolog_chrom_end': 'Arctic ground squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'uparryii_homolog_chrom_start': 'Arctic ground squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'uparryii_homolog_chromosome': 'Arctic ground squirrel chromosome/scaffold name' (default: False),\n",
      " 'uparryii_homolog_dn': 'dN with Arctic ground squirrel' (default: False),\n",
      " 'uparryii_homolog_ds': 'dS with Arctic ground squirrel' (default: False),\n",
      " 'uparryii_homolog_ensembl_gene': 'Arctic ground squirrel gene stable ID' (default: False),\n",
      " 'uparryii_homolog_ensembl_peptide': 'Arctic ground squirrel protein or transcript stable ID' (default: False),\n",
      " 'uparryii_homolog_goc_score': 'Arctic ground squirrel Gene-order conservation score' (default: False),\n",
      " 'uparryii_homolog_orthology_confidence': 'Arctic ground squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'uparryii_homolog_orthology_type': 'Arctic ground squirrel homology type' (default: False),\n",
      " 'uparryii_homolog_perc_id': '%id. target Arctic ground squirrel gene identical to query gene' (default: False),\n",
      " 'uparryii_homolog_perc_id_r1': '%id. query gene identical to target Arctic ground squirrel gene' (default: False),\n",
      " 'uparryii_homolog_subtype': 'Last common ancestor with Arctic ground squirrel' (default: False),\n",
      " 'uparryii_homolog_wga_coverage': 'Arctic ground squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'upstream_flank': 'upstream_flank' (default: False),\n",
      " 'validated': 'Variant supporting evidence' (default: False),\n",
      " 'variation_name': 'Variant name' (default: True),\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 'version': 'Version (gene)' (default: False),\n",
      " 'vpacos_homolog_associated_gene_name': 'Alpaca gene name' (default: False),\n",
      " 'vpacos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vpacos_homolog_chrom_end': 'Alpaca chromosome/scaffold end (bp)' (default: False),\n",
      " 'vpacos_homolog_chrom_start': 'Alpaca chromosome/scaffold start (bp)' (default: False),\n",
      " 'vpacos_homolog_chromosome': 'Alpaca chromosome/scaffold name' (default: False),\n",
      " 'vpacos_homolog_dn': 'dN with Alpaca' (default: False),\n",
      " 'vpacos_homolog_ds': 'dS with Alpaca' (default: False),\n",
      " 'vpacos_homolog_ensembl_gene': 'Alpaca gene stable ID' (default: False),\n",
      " 'vpacos_homolog_ensembl_peptide': 'Alpaca protein or transcript stable ID' (default: False),\n",
      " 'vpacos_homolog_goc_score': 'Alpaca Gene-order conservation score' (default: False),\n",
      " 'vpacos_homolog_orthology_confidence': 'Alpaca orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vpacos_homolog_orthology_type': 'Alpaca homology type' (default: False),\n",
      " 'vpacos_homolog_perc_id': '%id. target Alpaca gene identical to query gene' (default: False),\n",
      " 'vpacos_homolog_perc_id_r1': '%id. query gene identical to target Alpaca gene' (default: False),\n",
      " 'vpacos_homolog_subtype': 'Last common ancestor with Alpaca' (default: False),\n",
      " 'vpacos_homolog_wga_coverage': 'Alpaca Whole-genome alignment coverage' (default: False),\n",
      " 'vursinus_homolog_associated_gene_name': 'Common wombat gene name' (default: False),\n",
      " 'vursinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vursinus_homolog_chrom_end': 'Common wombat chromosome/scaffold end (bp)' (default: False),\n",
      " 'vursinus_homolog_chrom_start': 'Common wombat chromosome/scaffold start (bp)' (default: False),\n",
      " 'vursinus_homolog_chromosome': 'Common wombat chromosome/scaffold name' (default: False),\n",
      " 'vursinus_homolog_dn': 'dN with Common wombat' (default: False),\n",
      " 'vursinus_homolog_ds': 'dS with Common wombat' (default: False),\n",
      " 'vursinus_homolog_ensembl_gene': 'Common wombat gene stable ID' (default: False),\n",
      " 'vursinus_homolog_ensembl_peptide': 'Common wombat protein or transcript stable ID' (default: False),\n",
      " 'vursinus_homolog_goc_score': 'Common wombat Gene-order conservation score' (default: False),\n",
      " 'vursinus_homolog_orthology_confidence': 'Common wombat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vursinus_homolog_orthology_type': 'Common wombat homology type' (default: False),\n",
      " 'vursinus_homolog_perc_id': '%id. target Common wombat gene identical to query gene' (default: False),\n",
      " 'vursinus_homolog_perc_id_r1': '%id. query gene identical to target Common wombat gene' (default: False),\n",
      " 'vursinus_homolog_subtype': 'Last common ancestor with Common wombat' (default: False),\n",
      " 'vursinus_homolog_wga_coverage': 'Common wombat Whole-genome alignment coverage' (default: False),\n",
      " 'vvulpes_homolog_associated_gene_name': 'Red fox gene name' (default: False),\n",
      " 'vvulpes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vvulpes_homolog_chrom_end': 'Red fox chromosome/scaffold end (bp)' (default: False),\n",
      " 'vvulpes_homolog_chrom_start': 'Red fox chromosome/scaffold start (bp)' (default: False),\n",
      " 'vvulpes_homolog_chromosome': 'Red fox chromosome/scaffold name' (default: False),\n",
      " 'vvulpes_homolog_dn': 'dN with Red fox' (default: False),\n",
      " 'vvulpes_homolog_ds': 'dS with Red fox' (default: False),\n",
      " 'vvulpes_homolog_ensembl_gene': 'Red fox gene stable ID' (default: False),\n",
      " 'vvulpes_homolog_ensembl_peptide': 'Red fox protein or transcript stable ID' (default: False),\n",
      " 'vvulpes_homolog_goc_score': 'Red fox Gene-order conservation score' (default: False),\n",
      " 'vvulpes_homolog_orthology_confidence': 'Red fox orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vvulpes_homolog_orthology_type': 'Red fox homology type' (default: False),\n",
      " 'vvulpes_homolog_perc_id': '%id. target Red fox gene identical to query gene' (default: False),\n",
      " 'vvulpes_homolog_perc_id_r1': '%id. query gene identical to target Red fox gene' (default: False),\n",
      " 'vvulpes_homolog_subtype': 'Last common ancestor with Red fox' (default: False),\n",
      " 'vvulpes_homolog_wga_coverage': 'Red fox Whole-genome alignment coverage' (default: False),\n",
      " 'wikigene_description': 'WikiGene description' (default: False),\n",
      " 'wikigene_id': 'WikiGene ID' (default: False),\n",
      " 'wikigene_name': 'WikiGene name' (default: False),\n",
      " 'xcouchianus_homolog_associated_gene_name': 'Monterrey platyfish gene name' (default: False),\n",
      " 'xcouchianus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xcouchianus_homolog_chrom_end': 'Monterrey platyfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'xcouchianus_homolog_chrom_start': 'Monterrey platyfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'xcouchianus_homolog_chromosome': 'Monterrey platyfish chromosome/scaffold name' (default: False),\n",
      " 'xcouchianus_homolog_dn': 'dN with Monterrey platyfish' (default: False),\n",
      " 'xcouchianus_homolog_ds': 'dS with Monterrey platyfish' (default: False),\n",
      " 'xcouchianus_homolog_ensembl_gene': 'Monterrey platyfish gene stable ID' (default: False),\n",
      " 'xcouchianus_homolog_ensembl_peptide': 'Monterrey platyfish protein or transcript stable ID' (default: False),\n",
      " 'xcouchianus_homolog_goc_score': 'Monterrey platyfish Gene-order conservation score' (default: False),\n",
      " 'xcouchianus_homolog_orthology_confidence': 'Monterrey platyfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xcouchianus_homolog_orthology_type': 'Monterrey platyfish homology type' (default: False),\n",
      " 'xcouchianus_homolog_perc_id': '%id. target Monterrey platyfish gene identical to query gene' (default: False),\n",
      " 'xcouchianus_homolog_perc_id_r1': '%id. query gene identical to target Monterrey platyfish gene' (default: False),\n",
      " 'xcouchianus_homolog_subtype': 'Last common ancestor with Monterrey platyfish' (default: False),\n",
      " 'xcouchianus_homolog_wga_coverage': 'Monterrey platyfish Whole-genome alignment coverage' (default: False),\n",
      " 'xmaculatus_homolog_associated_gene_name': 'Platyfish gene name' (default: False),\n",
      " 'xmaculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xmaculatus_homolog_chrom_end': 'Platyfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'xmaculatus_homolog_chrom_start': 'Platyfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'xmaculatus_homolog_chromosome': 'Platyfish chromosome/scaffold name' (default: False),\n",
      " 'xmaculatus_homolog_dn': 'dN with Platyfish' (default: False),\n",
      " 'xmaculatus_homolog_ds': 'dS with Platyfish' (default: False),\n",
      " 'xmaculatus_homolog_ensembl_gene': 'Platyfish gene stable ID' (default: False),\n",
      " 'xmaculatus_homolog_ensembl_peptide': 'Platyfish protein or transcript stable ID' (default: False),\n",
      " 'xmaculatus_homolog_goc_score': 'Platyfish Gene-order conservation score' (default: False),\n",
      " 'xmaculatus_homolog_orthology_confidence': 'Platyfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xmaculatus_homolog_orthology_type': 'Platyfish homology type' (default: False),\n",
      " 'xmaculatus_homolog_perc_id': '%id. target Platyfish gene identical to query gene' (default: False),\n",
      " 'xmaculatus_homolog_perc_id_r1': '%id. query gene identical to target Platyfish gene' (default: False),\n",
      " 'xmaculatus_homolog_subtype': 'Last common ancestor with Platyfish' (default: False),\n",
      " 'xmaculatus_homolog_wga_coverage': 'Platyfish Whole-genome alignment coverage' (default: False),\n",
      " 'xtropicalis_homolog_associated_gene_name': 'Tropical clawed frog gene name' (default: False),\n",
      " 'xtropicalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xtropicalis_homolog_chrom_end': 'Tropical clawed frog chromosome/scaffold end (bp)' (default: False),\n",
      " 'xtropicalis_homolog_chrom_start': 'Tropical clawed frog chromosome/scaffold start (bp)' (default: False),\n",
      " 'xtropicalis_homolog_chromosome': 'Tropical clawed frog chromosome/scaffold name' (default: False),\n",
      " 'xtropicalis_homolog_dn': 'dN with Tropical clawed frog' (default: False),\n",
      " 'xtropicalis_homolog_ds': 'dS with Tropical clawed frog' (default: False),\n",
      " 'xtropicalis_homolog_ensembl_gene': 'Tropical clawed frog gene stable ID' (default: False),\n",
      " 'xtropicalis_homolog_ensembl_peptide': 'Tropical clawed frog protein or transcript stable ID' (default: False),\n",
      " 'xtropicalis_homolog_goc_score': 'Tropical clawed frog Gene-order conservation score' (default: False),\n",
      " 'xtropicalis_homolog_orthology_confidence': 'Tropical clawed frog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xtropicalis_homolog_orthology_type': 'Tropical clawed frog homology type' (default: False),\n",
      " 'xtropicalis_homolog_perc_id': '%id. target Tropical clawed frog gene identical to query gene' (default: False),\n",
      " 'xtropicalis_homolog_perc_id_r1': '%id. query gene identical to target Tropical clawed frog gene' (default: False),\n",
      " 'xtropicalis_homolog_subtype': 'Last common ancestor with Tropical clawed frog' (default: False),\n",
      " 'xtropicalis_homolog_wga_coverage': 'Tropical clawed frog Whole-genome alignment coverage' (default: False),\n",
      " 'zalbicollis_homolog_associated_gene_name': 'White-throated sparrow gene name' (default: False),\n",
      " 'zalbicollis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'zalbicollis_homolog_chrom_end': 'White-throated sparrow chromosome/scaffold end (bp)' (default: False),\n",
      " 'zalbicollis_homolog_chrom_start': 'White-throated sparrow chromosome/scaffold start (bp)' (default: False),\n",
      " 'zalbicollis_homolog_chromosome': 'White-throated sparrow chromosome/scaffold name' (default: False),\n",
      " 'zalbicollis_homolog_dn': 'dN with White-throated sparrow' (default: False),\n",
      " 'zalbicollis_homolog_ds': 'dS with White-throated sparrow' (default: False),\n",
      " 'zalbicollis_homolog_ensembl_gene': 'White-throated sparrow gene stable ID' (default: False),\n",
      " 'zalbicollis_homolog_ensembl_peptide': 'White-throated sparrow protein or transcript stable ID' (default: False),\n",
      " 'zalbicollis_homolog_goc_score': 'White-throated sparrow Gene-order conservation score' (default: False),\n",
      " 'zalbicollis_homolog_orthology_confidence': 'White-throated sparrow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'zalbicollis_homolog_orthology_type': 'White-throated sparrow homology type' (default: False),\n",
      " 'zalbicollis_homolog_perc_id': '%id. target White-throated sparrow gene identical to query gene' (default: False),\n",
      " 'zalbicollis_homolog_perc_id_r1': '%id. query gene identical to target White-throated sparrow gene' (default: False),\n",
      " 'zalbicollis_homolog_subtype': 'Last common ancestor with White-throated sparrow' (default: False),\n",
      " 'zalbicollis_homolog_wga_coverage': 'White-throated sparrow Whole-genome alignment coverage' (default: False)}\n"
     ]
    }
   ],
   "source": [
    "ENSEMBL = server.datasets['hsapiens_gene_ensembl']\n",
    "ENSEMBL.show_attributes()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46350"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#asd\n",
    "18381 + 27969"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "150064"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#scz\n",
    "36989 + 113075"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein1</th>\n",
       "      <th>protein2</th>\n",
       "      <th>combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000272298</td>\n",
       "      <td>490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000253401</td>\n",
       "      <td>198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000401445</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000418915</td>\n",
       "      <td>606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000327801</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               protein1              protein2  combined_score\n",
       "0  9606.ENSP00000000233  9606.ENSP00000272298             490\n",
       "1  9606.ENSP00000000233  9606.ENSP00000253401             198\n",
       "2  9606.ENSP00000000233  9606.ENSP00000401445             159\n",
       "3  9606.ENSP00000000233  9606.ENSP00000418915             606\n",
       "4  9606.ENSP00000000233  9606.ENSP00000327801             167"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [],
   "source": [
    "graph = {}\n",
    "count = 0\n",
    "for i, row in edges1.iterrows():\n",
    "    g1, g2 = row[\"protein1\"].split(\".\")[1], row[\"protein2\"].split(\".\")[1]\n",
    "    #print(g1, g2)\n",
    "    if g1 not in GeneDict or g2 not in GeneDict:\n",
    "        continue\n",
    "    if g1 not in graph:\n",
    "        graph[g1] = []\n",
    "    if g2 not in graph:\n",
    "        graph[g2] = []\n",
    "    graph[g1].append(g2)\n",
    "    graph[g2].append(g1)\n",
    "    if count % 1000 == 0:\n",
    "        sys.stdout.write(\"\\r%d\"%i)\n",
    "        count += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "print(len(graph.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
